<div class="paragraph">
<p>The use of a result data structure with file I/O improves performance because it moves the data in one large block from file to data structure (or vice versa) rather than field by field.</p>
</div>
<div class="paragraph">
<p>Additionally using a data structure can limit the problems caused by having bad data in a file. Without a data structure, the entire <code>READ</code> operation will fail at the first bad value. With one, the error comes only when the bad field is used.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>     F MyFile   IF   E             Disk
       * Noncompliant
     C                   READ      Record1</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>     F MyFile   IF   E             Disk

     D InputData       DS                  LikeRec(Record1)
     C                   READ      Record1       InputData</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use a result data structure for this file I/O.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_2_apr_2015_174259_ann_campbell_wrote">on 2 Apr 2015, 17:42:59 Ann Campbell wrote:</h3>
<div class="paragraph">
<p><a href="http://www.bmeyers.net/faqs/14-tips/32-rpg-iv-style?start=3" class="bare">http://www.bmeyers.net/faqs/14-tips/32-rpg-iv-style?start=3</a></p>
</div>
<div class="paragraph">
<p><a href="http://www.ibmsystemsmag.com/ibmi/developer/rpg/Ending-Those-Decimal-Data-Error-Blues/?page=1" class="bare">http://www.ibmsystemsmag.com/ibmi/developer/rpg/Ending-Those-Decimal-Data-Error-Blues/?page=1</a></p>
</div>
</div>
</div>
</div>