<div class="paragraph">
<p>Variables, Classes and functions should not be undefined, otherwise the code will fail with a NameError.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>my_var # Noncompliant (variable is never defined)

def noncompliant():
    foo()  # Noncompliant
    MyClass()  # Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>from mod import my_var

my_var

def compliant():
    foo = sum
    foo()

    class MyClass:
        pass
    MyClass()</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>XX is not defined. Change its name or define it before using it.</p>
</div>
</div>
</div>
</div>