<div class="paragraph">
<p>Query by the value of a non-existent column using a standard ANSI join, and you&#8217;ll get an error. Do the same thing in a sub-query, and you&#8217;ll silently get the wrong result because the condition will effectively be ignored.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>CREATE TABLE rule (
id integer(10),
.... );

CREATE TABLE issue (
id integer(10),
rule_id integer(10),
... );

SELECT id, status
FROM issue
WHERE rule_id in (
  SELECT rule_id   -- Noncompliant. Condition ignored and all issues selected
  FROM rule
  WHERE ...
)</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>"ccc" does not exist in "ttt".</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Non-existent column in <code>WHERE</code> clause</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_16_jun_2016_140026_ann_campbell_wrote">on 16 Jun 2016, 14:00:26 Ann Campbell wrote:</h3>
<div class="paragraph">
<p><a href="https://www.researchgate.net/publication/222432151_Semantic_errors_in_SQL_queries_A_quite_complete_list" class="bare">https://www.researchgate.net/publication/222432151_Semantic_errors_in_SQL_queries_A_quite_complete_list</a></p>
</div>
</div>
</div>
</div>