<div class="paragraph">
<p>Assigning between objects that have an overlap in their physical storage leads to undefined behaviour.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>struct s
{
  int16_t m1 [ 32 ];
};

struct t
{
  int32_t m2;
  struct s m3;
};

void fn ( )
{
  union
  {
    struct s u1;
    struct t u2;
  } a;
  a.u2.m3 = a.u1; // Noncompliant
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C:2004, 18.2</p>
</li>
<li>
<p>MISRA C&#43;&#43;:2008, 0-2-1</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this assignment between overlapping memory locations.</p>
</div>
</div>
</div>
</div>