<div class="paragraph">
<p>Debug statements are always useful during development. But include them in production code - particularly in code that runs client-side - and you run the risk of inadvertently exposing sensitive information, slowing down the browser, or even erroring-out the site for some users.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>console.log(password_entered); // Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://owasp.org/Top10/A09_2021-Security_Logging_and_Monitoring_Failures/">OWASP Top 10 2021 Category A9</a> - Security Logging and Monitoring Failures</p>
</li>
<li>
<p><a href="https://www.owasp.org/index.php/Top_10-2017_A3-Sensitive_Data_Exposure">OWASP Top 10 2017 Category A3</a> - Sensitive Data Exposure</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace this usage of console.{log|warn|error} by a logger.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_deprecates_s2228">deprecates: S2228</h3>

</div>
<div class="sect2">
<h3 id="_on_8_may_2013_095007_freddy_mallet_wrote">on 8 May 2013, 09:50:07 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>{panel:title=Java Title}</p>
</div>
<div class="paragraph">
<p>System.(out/err) should not be used to log anything</p>
</div>
<div class="paragraph">
<p>{panel}</p>
</div>
<div class="paragraph">
<p>{panel:title=C&#43;&#43; Title}</p>
</div>
<div class="paragraph">
<p>cout and cerr should not be used to log anything</p>
</div>
<div class="paragraph">
<p>{panel}</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_6_aug_2013_140558_dinesh_bolkensteyn_wrote">on 6 Aug 2013, 14:05:58 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>Implemented by https://jira.codehaus.org/browse/SONARJAVA-234</p>
</div>
</div>
</div>
</div>