<div class="paragraph">
<p>Chained assignments are confusing and hard to read, and should be avoided.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>x = y = 0; // Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>x = 0;
y = 0; // or y = x;</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_duplicates_s1121">duplicates: S1121</h3>

</div>
<div class="sect2">
<h3 id="_on_26_may_2015_181140_evgeny_mandrikov_wrote">on 26 May 2015, 18:11:40 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] RSPEC-1121 mentions "MISRA C:2012, 13.4 The result of an assignment operator should not be used", which supposed to cover this case. And C-Family implementation of RSPEC-1121 covers it. Hence question - this separate RSPEC redundant or not?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_27_may_2015_141756_ann_campbell_wrote">on 27 May 2015, 14:17:56 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>I wish I could remember why I wrote this rule [~evgeny.mandrikov].</p>
</div>
<div class="paragraph">
<p>Closed as a duplicate.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_27_may_2015_143523_evgeny_mandrikov_wrote">on 27 May 2015, 14:35:23 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] we can introduce notes for RSPECs, which could help to preserve original intention ;)</p>
</div>
</div>
</div>
</div>