<div class="paragraph">
<p>The functions <code>memcpy</code>, <code>memmove</code> and <code>memset</code> can only be used for objects of trivially copyable types. This includes scalar types, arrays, and trivially copyable classes.</p>
</div>
<div class="paragraph">
<p>A class type is trivially copyable if:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>One or more of the following special member functions is trivial and the rest are deleted: copy constructor, move constructor, copy assignment operator, and move assignment operator,</p>
</li>
<li>
<p>It has a trivial, non-deleted destructor,</p>
</li>
<li>
<p>It has trivially copyable members and base classes,</p>
</li>
<li>
<p>It has no virtual functions.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p><em>Note: a default implementation, both explicit (with <code>=default</code>) or implicit (if the special member function is omitted), is considered trivial.</em></p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class Shape {
public:
  int x;
  int y;
  virtual ~Shape(); // This makes the class non trivially copyable
};

void f(Shape *dest, Shape *source)
{
    memcpy(dest, source, sizeof Shape); // Noncompliant
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class Shape {
public:
  int x;
  int y;
  virtual ~Shape(); // This makes the class non trivially copyable
};

void f(Shape *dest, Shape *source)
{
    (*dest) = (*source);
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use constructors or assignment operators, XXX is not trivially copyable.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_6_nov_2018_203224_ann_campbell_wrote">on 6 Nov 2018, 20:32:24 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Double-check my changes please, [~loic.joly]</p>
</div>
</div>
</div>
</div>