<div class="paragraph">
<p>A cache is a long-lived object that holds references to shorter-lived objects. To prevent a cache from growing indefinitely, old objects the cache should be removed when they&#8217;re no longer used. This can be done with the use of soft references, which allow the garbage collector to remove unused cache entries when memory needs to be freed.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when a <code>static</code> collection does not use <code>SoftReference</code>.</p>
</div>
<div class="paragraph">
<p><strong>Note</strong> that this rule is automatically disabled when the project&#8217;s <code>sonar.java.source</code> is lower than 7.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public class MyClass {
  private static List&lt;MyClass&gt; cache = new ArrayList&lt;&gt;();  // Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public class MyClass {
  private static List&lt;SoftReference&lt;MyClass&gt;&gt; cache = new ArrayList&lt;&gt;();</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add a maximum size to this "xxx" or use soft references.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>cache declaration</p>
</div>
</div>
</div>
</div>