<div class="paragraph">
<p>It is often considered good practice at the end of an override to invoke <code>super</code>, but there are cases where according to the Apple developer documentation this should not be done.</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>updateLayer</code> - optimize the rendering of your view</p>
</li>
<li>
<p><code>loadView</code> - provide a <code>view</code> when <code>view</code> is <code>nil</code></p>
</li>
<li>
<p><code>providePlaceholder</code> - provide a placeholder for a document returned by the Document Picker but not yet stored locally</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>In all cases, these are actions that should happen once and only once. Subsequently invoking <code>super</code> would see your desired result replaced (at best) by less specialized results.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class VC: UIMyViewController {
  override func loadView() {
    // ...
    super.loadView()
  }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class VC: UIMyViewController {
  override func loadView() {
    // ...
  }
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this call to "super.xxx()".</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p><code>super.xxx()</code></p>
</div>
</div>
</div>
</div>