<div class="paragraph">
<p>Historically <code>/COPY</code> procedures contained all of the specifications required for each procedure, so for example <code>xxxxF /COPY</code> would be used to copy the F lines from another <code>F</code> spec. However, this practice results in code that is difficult to understand and maintain. Instead, the use of <code>/COPY</code> statements should be replaced with explicit declarations.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Given the default list of specs, H,F,I,D,C,O:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>F/COPY HRSILERPG,RSX36</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>FSSINRTP   IF   E           K DISK    INFSR(*PSSR)
F*
F*  Interest Rates File
F*</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace this <code>/COPY</code> directive with explicitly coded specifications</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="ulist">
<ul>
<li>
<p>key = specs
Description: Comma-delimited list of specs in which "/COPY" should not be used</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Default: H,F,I,D,C,O</p>
</div>
</div>
</div>
</div>