<div class="paragraph">
<p>While it is <em>possible</em> to access <code>static</code> members from a class instance, it&#8217;s bad form, and considered by most to be misleading because it implies to the readers of your code that there&#8217;s an instance of the member per class instance.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class MyClass {
public :
  static void Mymethod() {
    // ...
  }
};

MyClass* pmyclass = new MyClass();
pmyclass-&gt;Mymethod(); // Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class MyClass {
public :
  static Mymethod() {
    // ...
  }
};

Myclass::Mymethod();</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace "[.|&#8594;]" with "::" for access to "XXX".</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s1271">is related to: S1271</h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s3252">is related to: S3252</h3>

</div>
<div class="sect2">
<h3 id="_on_23_jul_2015_110057_evgeny_mandrikov_wrote">on 23 Jul 2015, 11:00:57 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>As discussed with [~nicolas.peru] - this is a very good practice, so should be active by default.</p>
</div>
</div>
</div>
</div>