<div class="paragraph">
<p>When a variable in a nested scope has the same name as one in an outer scope, it hides the outer scope&#8217;s variable, which can no longer be accessed in the inner scope. Thus all uses of that variable name in the inner scope will reference the variable declared in the inner scope. However, that may not be clear to everyone. particularly in large code blocks. Therefore identifiers in inner scopes should never hide those from outer scopes.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>void fn1 ( void )
{
  int16_t i;
  {
    int8_t i; /* Non-compliant - hides previous "i " */
    i = 3;       /* Could be confusing as to which "i" this is */
  }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>void fn1 ( void )
{
  int16_t i;
  {
    int8_t j;
    j = 3;
  }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C:2004, 5.2 - Identifiers in an inner scope shall not use the same name as an identifier in an outer scope, and therefore hide that identifier</p>
</li>
<li>
<p>MISRA C&#43;&#43;:2008, 2-10-2 - Identifiers declared in an inner scope shall not hide an identifier declared in an outer scope</p>
</li>
<li>
<p>MISRA C:2012, 5.3 - An identifier declared in an inner scope shall not hide an identifier declared in an outer scope</p>
</li>
<li>
<p><a href="https://www.securecoding.cert.org/confluence/x/VwE">CERT, DCL01-C</a> - Do not reuse variable names in subscopes</p>
</li>
<li>
<p><a href="https://www.securecoding.cert.org/confluence/x/cwAhAQ">CERT, DCL01-CPP</a> - Do not reuse variable names in subscopes</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_duplicates_s1524">duplicates: S1524</h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s1524">relates to: S1524</h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s1117">is related to: S1117</h3>

</div>
</div>
</div>