<div class="paragraph">
<p>Nested code blocks can be used to create a new scope and restrict the visibility of the variables defined inside it. Using this feature in a method typically indicates that the method has too many responsibilities, and should be refactored into smaller methods.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public void evaluate(int operator) {
  // Do some computation...
  {
    int a = stack.pop();
    int b = stack.pop();
    int result = a + b;
    stack.push(result);
  }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public void evaluate(int operator) {
  // Do some computation...
  evaluateAdd();
}

private void evaluateAdd() {
  int a = stack.pop();
  int b = stack.pop();
  int result = a + b;
  stack.push(result);
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Extract this nested code block into a separate method.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_20_aug_2013_081440_freddy_mallet_wrote">on 20 Aug 2013, 08:14:40 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARJAVA-298</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_24_aug_2013_185909_ann_campbell_wrote">on 24 Aug 2013, 18:59:09 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>This is a classic C trick. Should the rule be added there as well?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_25_aug_2013_184057_freddy_mallet_wrote">on 25 Aug 2013, 18:40:57 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>I&#8217;ve added C&#43;&#43; as targeted language</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_22_apr_2014_214041_evgeny_mandrikov_wrote">on 22 Apr 2014, 21:40:41 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] [~freddy.mallet] IMO title is a little bit misleading, because for some other rules we use "nested code blocks" to refer to compound-statement within if-statement.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_23_apr_2014_151812_ann_campbell_wrote">on 23 Apr 2014, 15:18:12 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~evgeny.mandrikov] what would you suggest?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_12_sep_2017_141330_pierre_yves_nicolas_wrote">on 12 Sep 2017, 14:13:30 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>PHP supports such blocks but that does not impact the scope of variables.</p>
</div>
</div>
</div>
</div>