<div class="paragraph">
<p>In addition to being difficult to use, too-long variable names can limit code portability. The ISO standard requires that variable, type, function and label names be no more than 31 characters long.</p>
</div>
<div class="paragraph">
<p>Note that 31 characters is an upper bound, rather than a length recommendation. Shorter names are better, as long as they&#8217;re still communicative.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>int this_is_a_very_long_identifier_that_definitely_should_be_renamed = 0;</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>int reasonable_identifier = 0;</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C:2004, 5.1 - Identifiers (internal and external) shall not rely on the significance of more than 31 character.</p>
</li>
<li>
<p><a href="https://wiki.sei.cmu.edu/confluence/x/FtcxBQ">CERT, DCL23-C.</a> - Guarantee that mutually visible identifiers are unique</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Give "xxx" a shorter name</p>
</div>
</div>
</div>
</div>