<div class="paragraph">
<p>It should be clear to a casual reader what code a test is testing and what results are expected. Unfortunately, that&#8217;s not usually the case with the <code>ExpectedException</code> attribute since an exception could be thrown from almost any line in the method.</p>
</div>
<div class="paragraph">
<p>This rule detects MSTest and NUnit <code>ExpectedException</code> attribute.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>[TestMethod]
[ExpectedException(typeof(ArgumentNullException))]  // Noncompliant
public void TestNullArg()
{
  //...
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>[TestMethod]
public void TestNullArg()
{
  bool callFailed = false;
  try
  {
    //...
  }
  catch (ArgumentNullException)
  {
    callFailed = true;
  }
  Assert.IsTrue(callFailed, "Expected call to MyMethod to fail with ArgumentNullException");
}</pre>
</div>
</div>
<div class="paragraph">
<p>or</p>
</div>
<div class="listingblock">
<div class="content">
<pre>[TestMethod]
public void TestNullArg()
{
  Assert.ThrowsException&lt;ArgumentNullException&gt;(() =&gt; /*...*/);
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_exceptions">Exceptions</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule ignores one-line test methods, since it is obvious in such methods where the exception is expected to be thrown.</p>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace the "ExpectedException" attribute with a throw assertion or a "try/catch" block.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>ExpectedException attribute</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s3658">relates to: S3658</h3>

</div>
<div class="sect2">
<h3 id="_on_1_dec_2015_143700_tamas_vajk_wrote">on 1 Dec 2015, 14:37:00 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I changed minor things. It looks good.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_9_aug_2017_115944_amaury_levé_wrote">on 9 Aug 2017, 11:59:44 Amaury Levé wrote:</h3>
<div class="paragraph">
<p>Please review</p>
</div>
</div>
</div>
</div>