<div class="paragraph">
<p>The use of wrapper objects for primitive types is gratuitous, confusing and dangerous. If you use a wrapper object constructor for type conversion, just remove the <code>new</code> keyword, and you&#8217;ll get a primitive value automatically. If you use a wrapper object as a way to add properties to a primitive, you should re-think the design. Such uses are considered bad practice, and should be refactored.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>let x = new Number("0");
if (x) {
  alert('hi');  // Shows 'hi'.
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>let x = Number("0");
if (x) {
  alert('hi');
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this use of 'xxx' constructor.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s2129">is related to: S2129</h3>

</div>
<div class="sect2">
<h3 id="_on_16_mar_2017_110510_elena_vilchik_wrote">on 16 Mar 2017, 11:05:10 Elena Vilchik wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2], I have changed description. Could you reword it in normal english? Please come back to me, if any questions about content. Thanks!</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_16_mar_2017_121453_ann_campbell_wrote">on 16 Mar 2017, 12:14:53 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Edited [~elena.vilchik]. Double-check me pls.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_16_mar_2017_143108_elena_vilchik_wrote">on 16 Mar 2017, 14:31:08 Elena Vilchik wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] thanks!</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_29_jan_2018_104326_tibor_blenessy_wrote">on 29 Jan 2018, 10:43:26 Tibor Blenessy wrote:</h3>
<div class="paragraph">
<p>RSPEC-2129 seems to be very similar, if not duplicate of this.</p>
</div>
</div>
</div>
</div>