<div class="paragraph">
<p>Include guards, wrapping around the entire content of a header file, are a best practice that ensure that no matter how many times the header is actually included in a translation unit, its content will only be seen once.</p>
</div>
<div class="paragraph">
<p>The include guard pattern is made up of four parts:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>#ifndef</code> at the top of the file, with a unique macro name (usually the name relates to the name of the file, to ensure uniqueness).</p>
</li>
<li>
<p><code>#define</code>Â with the same macro name.</p>
</li>
<li>
<p>The content of the file</p>
</li>
<li>
<p><code>#endif</code> at the end of the file</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>The rule raises an issue when the name in the second part differs from the name in the first (usually because of a typo or a copy/paste issue).</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>#ifndef MYFILE_H
#define MY_FILE_H // Noncompliant
//...
#endif</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>#ifndef MYFILE_H
#define MYFILE_H
//...
#endif</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace this macro definition with header guard macro.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>macro definition</p>
</div>
</div>
</div>
</div>