<div class="paragraph">
<p>When using a <code>PRINTER</code> defined file, an <code>INFDS</code> keyword should be used to read the file&#8217;s line count from <code>*PSSR</code> positions 367-368.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre> FCSO602R1  O    E             PRINTER USROPN
 or ...
 FCSO602R1  O    E             PRINTER USROPN
 F                                     INFDS(WSFD01)
 F                                     INFSR(*PSSR)

 D WSFD01          DS
 D  W1ST01           *STATUS
 D  W1SPNO               123    124B 0</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre> FCSO602R1  O    E             PRINTER USROPN
 F                                     INFDS(WSFD01)
 F                                     INFSR(*PSSR)

 D WSFD01          DS
 D  W1ST01           *STATUS
 D  W1SPNO               123    124B 0
 D  W1LNCT               367    368B 0
 D  W1PGNO               369    372B 0</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Include an <code>INFDS</code> keyword with a reference to position 367-368.</p>
</div>
</div>
</div>
</div>