<div class="paragraph">
<p>Correctly updating a <code>static</code> field from a non-static method is tricky to get right and could easily lead to bugs if there are multiple class instances  and/or multiple threads in play.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue each time a <code>static</code> field is updated from a non-static method or property.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public class MyClass
{
  private static int count = 0;

  public void DoSomething()
  {
    //...
    count++;  // Noncompliant
  }
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>Make the enclosing instance [method|property] "static" or remove this set on the "static" field.</p>
</li>
<li>
<p>Remove this set, which updates a "static" field from an instance [method|property].</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Primary: static field assignment</p>
</div>
<div class="paragraph">
<p>Secondary: static field declaration</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_24_jun_2015_070357_tamas_vajk_wrote">on 24 Jun 2015, 07:03:57 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I&#8217;ve created this subtask, because in C# this rule can also apply for properties. Furthermore, the <code>synchronized static</code> part doesn&#8217;t work for C#.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_24_jun_2015_140107_ann_campbell_wrote">on 24 Jun 2015, 14:01:07 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>looks good [~tamas.vajk]</p>
</div>
</div>
<div class="sect2">
<h3 id="_relates_to_s3010">relates to: S3010</h3>

</div>
</div>
</div>