<div class="paragraph">
<p>This variant of the <code>REFRESH</code> statement is deprecated and should be avoided.</p>
</div>
<div class="paragraph">
<p>This <code>REFRESH</code> statement initializes the internal table <code>itab</code>, reads several rows from the database table <code>dbtab</code>, and adds their contents to the internal table <code>itab</code>. A <code>SELECT</code> statement should be used instead.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>TABLES t100.
DATA itab TYPE STANDARD TABLE OF t100.

t100-sprsl = 'E'.
t100-arbgb = 'BC'.

REFRESH itab FROM TABLE t100.</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>DATA itab TYPE STANDARD TABLE OF t100.

SELECT *
       FROM t100
       INTO TABLE itab
       WHERE sprsl = 'E' AND
             arbgb LIKE 'BC%'.</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use a "SELECT" statement instead of "REFRESH itab FROM TABLE", which is obsolete.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s1491">relates to: S1491</h3>

</div>
<div class="sect2">
<h3 id="_on_11_jan_2014_194007_ann_campbell_wrote">on 11 Jan 2014, 19:40:07 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Hope this slight variant on the rule title standard is allowable. Also, I&#8217;ve tagged REFRESH as code in the Description. Not entirely sure whether it&#8217;s correct or not.</p>
</div>
</div>
</div>
</div>