<div class="paragraph">
<p>For better readability, and to prevent the <code>PLS-00312: a positional parameter association may not follow a named association</code> exception from being raised, do not mix named and positional argument invocations.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>SET SERVEROUTPUT ON

DECLARE
  PROCEDURE sub(op1 PLS_INTEGER, op2 PLS_INTEGER) AS
  BEGIN
    DBMS_OUTPUT.PUT_LINE('Sub = ' || (op1 - op2));
  END;
BEGIN

  sub(10, op2 =&gt; 2); -- Noncompliant
  sub(op1 =&gt; 10, 2); -- Noncompliant - raises PLS-00312: a positional parameter association may not follow a named association

END;
/</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>SET SERVEROUTPUT ON

DECLARE
  PROCEDURE sub(op1 PLS_INTEGER, op2 PLS_INTEGER) AS
  BEGIN
    DBMS_OUTPUT.PUT_LINE('Sub = ' || (op1 - op2));
  END;
BEGIN

  sub(10, 2); -- Compliant
  sub(op1 =&gt; 10, op2 =&gt; 2); -- Compliant

END;
/</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use only named or only positional arguments.</p>
</div>
</div>
</div>
</div>