<div class="paragraph">
<p>Restricting the number of exits from a loop is done in the interests of good structured programming. One <code>break</code> or <code>goto</code> statement is acceptable in a loop since this allows, for example, for dual-outcome loops or optimal coding.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="paragraph">
<p>With the default threshold of 1:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>for (int i = 0; i &lt; 10; i++) {
  if (...) {
    break;      //  Compliant
  }
  else if (...) {
    break;      //  Non-compliant - second jump from loop
  }
  else {
    ...
  }
}
while (...) {
  if (...) {
    break;      // Compliant
  }
  if (...) {
    break;      // Non-compliant - second jump from loop
  }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>for (int i = 0; i &lt; 10; i++) {
  if (...) {
    break;      //  Compliant
  }
}
while (...) {
  if (...) {
    break;    // Compliant
  }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C:2004, 14.6 - For any iteration statement there shall be at most one break statement used for loop termination.</p>
</li>
<li>
<p>MISRA C&#43;&#43;:2008, 6-6-4 - For any iteration statement there shall be no more than one break or goto statement used for loop termination.</p>
</li>
<li>
<p>MISRA C:2012, 15.4 - There should be no more than one break or goto statement used to terminate any iteration statement</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Reduce the number of nested break and goto statements from X to Y authorized.</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="ulist">
<ul>
<li>
<p>key = maxNumberOfTerminationStatements</p>
</li>
<li>
<p>Default value : 1</p>
</li>
<li>
<p>Description : Maximum authorized number of break/goto statements</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_duplicates_s135">duplicates: S135</h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s135">relates to: S135</h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s1227">relates to: S1227</h3>

</div>
</div>
</div>