<div class="paragraph">
<p>Using <code>size()</code> or <code>count()</code> method to test for emptiness works, but using <code>empty()</code> or <code>is_empty()</code> methods makes the code more readable and can be more performant. The time complexity of any <code>empty()</code>/<code>is_empty()</code> method implementation should be <code>O(1)</code> whereas some implementations of <code>size()</code> or <code>count()</code> can be <code>O(n)</code>.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>void fun(const std::vector&lt;int&gt; &amp;myVector) {
  if (myVector.size() == 0) { // Noncompliant
    // do something
  }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>void fun(const std::vector&lt;int&gt; &amp;myVector) {
  if (myVector.empty()) {
    // do something
  }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://github.com/isocpp/CppCoreGuidelines/blob/c553535fb8dda2839d13ab5f807ffbc66b63d67b/CppCoreGuidelines.md#enforcement-324">C&#43;&#43; Core Guidelines - T.143</a> - Don&#8217;t write unintentionally nongeneric code</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use empty() or is_empty() to check whether the container is empty or not.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s3981">is related to: S3981</h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s2250">is related to: S2250</h3>

</div>
<div class="sect2">
<h3 id="_on_30_jul_2013_170625_dinesh_bolkensteyn_wrote">on 30 Jul 2013, 17:06:25 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>Implemented by https://jira.codehaus.org/browse/SONARJAVA-243</p>
</div>
</div>
</div>
</div>