<div class="paragraph">
<p>Using <code>count()</code> to test for emptiness works, but using <code>empty()</code> makes the code more readable and can be more performant</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>if (count($a) &gt; 0) { // Noncompliant
 echo $a[0];
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>if (!empty($a)) {
  echo $a[0];
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use isEmpty() to check whether the collection is empty or not.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s3981">is related to: S3981</h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s2250">is related to: S2250</h3>

</div>
<div class="sect2">
<h3 id="_on_30_jul_2013_170625_dinesh_bolkensteyn_wrote">on 30 Jul 2013, 17:06:25 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>Implemented by https://jira.codehaus.org/browse/SONARJAVA-243</p>
</div>
</div>
</div>
</div>