<div class="paragraph">
<p>Classes can register themselves to receive notifications using <code>NotificationCenter.add</code>. Having done so, it seems suspicious that a class would opt to stop receiving notifications before de-initialization. For that reason, this rule raises an issue when <code>NotificationCenter.default.removeObserver(self)</code> is called anywhere but in <code>deinit</code></p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class MyClass {
  func doTheThing() {
    //...
    NotificationCenter.default.removeObserver(self)  // Noncompliant
  }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class MyClass {
  func doTheThing() {
    //...
  }

  func deinit() {
    NotificationCenter.default.removeObserver(self)
  }
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Move this call to (a|the) "deinit" method.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p><code>removeObserver(self)</code></p>
</div>
</div>
</div>
</div>