<div class="paragraph">
<p>Aligning common keywords in a series of statements makes the code easier to read. therefore, it is better to align the <code>TO</code> keywords in a series of successive MOVE statements.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>MOVE "Hi There" TO field
MOVE temp TO b
MOVE 123 TO item</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>MOVE "Hi There" TO field
MOVE temp       TO b
MOVE 123        TO item</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Align the "TO" keywords in this sequence of "MOVE" statements.</p>
</div>
</div>
</div>
</div>