<div class="paragraph">
<p>It is possible to use <code>await</code> on values which are not <code>Promise</code>s, but it&#8217;s useless and misleading. The point of <code>await</code> is to pause execution until the <code>Promise</code>'s asynchronous code has run to completion. With anything other than a <code>Promise</code>, there&#8217;s nothing to wait for.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when an <code>await</code>ed value is guaranteed not to be a <code>Promise</code>.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>let x = 42;
await x; // Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>let x = new Promise(resolve =&gt; resolve(42));
await x;

let y = p ? 42 : new Promise(resolve =&gt; resolve(42));
await y;</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Refactor this redundant 'await' on a non-promise.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_26_jul_2017_180255_ann_campbell_wrote">on 26 Jul 2017, 18:02:55 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~carlo.bottiglieri] this was a bit bare-bones. I&#8217;ve fluffed it up some. Please double-check me.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_26_jul_2017_182610_carlo_bottiglieri_wrote">on 26 Jul 2017, 18:26:10 Carlo Bottiglieri wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I made a small change, for the rest it&#8217;s fine for me, but the rendering of the code highlight is mostly broken in my browser (I checked and the syntax looks correct)</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_26_jul_2017_192911_ann_campbell_wrote">on 26 Jul 2017, 19:29:11 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Yes [~carlo.bottiglieri], Jira&#8217;s handling of its own markdown is broken. It will come out fine when RuleAPI generates the description tho.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_27_jul_2017_134242_freddy_mallet_wrote">on 27 Jul 2017, 13:42:42 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>\[~carlo.bottiglieri] and [~ann.campbell.2], let&#8217;s go for the creation of the implementation tickets for JS and TS ?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_9_aug_2017_180957_elena_vilchik_wrote">on 9 Aug 2017, 18:09:57 Elena Vilchik wrote:</h3>
<div class="paragraph">
<p>I removed from default profile. While we can add it back if make rule configurable (list of promise-like classes)</p>
</div>
</div>
</div>
</div>