<div class="paragraph">
<p>Accessing a non-existing member on an object will raise in most case an <code>AttributeError</code> exception.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when a non-existing member is accessed on a class instance and nothing indicates that this was expected.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>def access_attribute():
    x = 42
    return x.isnumeric()  # Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>def access_attribute():
    x = "42"
    return x.isnumeric()</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_exceptions">Exceptions</h2>
<div class="sectionbody">
<div class="paragraph">
<p>No issue will be raised:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>in <code>try ... except</code> blocks when <code>AttributeError</code> is caught.</p>
</li>
<li>
<p>in <code>if hasattr(...)</code> blocks.</p>
</li>
<li>
<p>when the accessed object has methods <code>__getattribute__</code> or <code>__getattr__</code>. These methods will override the default attribute access behavior.</p>
</li>
<li>
<p>when a class or its parent classes use <code>setattr</code>. This can be used to add dynamically members to the class, even if it is not recommended.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>You can also disable any issue for a given class by adding it to this rule&#8217;s ignoredClassesForMemberValidation property.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://youtu.be/NiSqG6s8skA">The Magic of Attribute Access - by Petr Viktorin - EuroPython Conference 2014</a></p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Fix this access to a non-existing member "XXX"</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="ulist">
<ul>
<li>
<p>key = ignoredClassesForMemberValidation</p>
</li>
<li>
<p>description: Comma-separated list of classes whose members will not be validated.</p>
</li>
<li>
<p>no default value</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Primary: the member access</p>
</div>
<div class="paragraph">
<p>Secondary: the code giving us a clue on the type of the object whose member we try to access. Example: the assignment, the "if x is None" check&#8230;&#8203;</p>
</div>
</div>
</div>
</div>