<div class="paragraph">
<p>Because the <code>is</code> operator performs a cast if the object is not null, using <code>is</code> to check type and then casting the same argument to that type, necessarily performs two casts. The same result can be achieved more efficiently with a single cast using <code>as</code>, followed by a null-check.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>if (x is Fruit)  // Noncompliant
{
  var f = (Fruit)x; // or x as Fruit
  // ...
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>// C# 6
var f = x as Fruit;
if (f != null)
{
  // ...
}
// C# 7
if (x is Fruit fruit)
{
  // ...
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Primary: Replace this type-check-and-cast sequence with an "as" and a null check.</p>
</div>
<div class="paragraph">
<p>Secondary: Replace this cast</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>Primary: on the "is" check</p>
</li>
<li>
<p>Secondary: on the following casts</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s1905">is related to: S1905</h3>

</div>
<div class="sect2">
<h3 id="_on_8_jul_2015_152840_ann_campbell_wrote">on 8 Jul 2015, 15:28:40 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~tamas.vajk] note that I&#8217;m a little concerned the title might be too broad. What do you think?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_9_jul_2015_121838_tamas_vajk_wrote">on 9 Jul 2015, 12:18:38 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I think the title is okay. Maybe we could add "useless duplicate casts&#8230;&#8203;"</p>
</div>
<div class="paragraph">
<p>I found this: https://msdn.microsoft.com/en-us/library/ms182271.aspx, which also uses the phrase "duplicate casts", so there is no need to modify it.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_9_jul_2015_123534_ann_campbell_wrote">on 9 Jul 2015, 12:35:34 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~tamas.vajk] I vote against adding "useless" to the title since that, in itself, would be redundant. :-)</p>
</div>
</div>
</div>
</div>