<div class="paragraph">
<p>While some <code>TestRule</code> classes have the desired effect without ever being directly referenced by a test, several others do not, and there&#8217;s no reason to leave them cluttering up the file if they&#8217;re not in use.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when <code>Test</code> class fields of the following types aren&#8217;t used by any of the test methods: <code>TemporaryFolder</code>, and <code>TestName</code>.</p>
</div>
<div class="paragraph">
<p>This rule also applies to the JUnit 5 equivalent classes: <code>TempDir</code>, and <code>TestInfo</code>.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public class ProjectDefinitionTest {

  @Rule
  public TemporaryFolder temp = new TemporaryFolder();  // Noncompliant

  @Test
  public void shouldSetKey() {
    ProjectDefinition def = ProjectDefinition.create();
    def.setKey("mykey");
    assertThat(def.getKey(), is("mykey"));
  }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public class ProjectDefinitionTest {

  @Test
  public void shouldSetKey() {
    ProjectDefinition def = ProjectDefinition.create();
    def.setKey("mykey");
    assertThat(def.getKey(), is("mykey"));
  }
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this unused "TestRule".</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_12_may_2015_142537_ann_campbell_wrote">on 12 May 2015, 14:25:37 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~david.gageot] this may not be as broad as you wanted&#8230;&#8203;?</p>
</div>
<div class="paragraph">
<p>Also, I&#8217;m not quite sure about the impact of the @Rule annotation, so I didn&#8217;t mention it in the description but left it in the code sample you provided. What happens if the annotation is missing?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_12_may_2015_145412_david_gageot_wrote">on 12 May 2015, 14:54:12 David Gageot wrote:</h3>
<div class="paragraph">
<p>If the annotation is missing, it will not do anything either. So this is really dead code</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_12_may_2015_145416_david_gageot_wrote">on 12 May 2015, 14:54:16 David Gageot wrote:</h3>
<div class="paragraph">
<p>lgtm</p>
</div>
</div>
</div>
</div>