<div class="paragraph">
<p>The Java language authors have been quite frank that <code>Optional</code> was intended for use only as a return type, as a way to convey that a method may or may not return a value.</p>
</div>
<div class="paragraph">
<p>And for that, it&#8217;s valuable but using <code>Optional</code> on the input side increases the work you have to do in the method without really increasing the value. With an <code>Optional</code> parameter, you go from having 2 possible inputs: null and not-null, to three: null, non-null-without-value, and non-null-with-value. Add to that the fact that overloading has long been available to convey that some parameters are optional, and there&#8217;s really no reason to have <code>Optional</code> parameters.</p>
</div>
<div class="paragraph">
<p>The rule also checks for Guava&#8217;s <code>Optional</code>, as it was the inspiration for the JDK <code>Optional</code>. Although it is different in some aspects (serialization, being recommended for use as collection elements), using it as a parameter type causes exactly the same problems as for JDK <code>Optional</code>.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public String sayHello(Optional&lt;String&gt; name) {  // Noncompliant
  if (name == null || !name.isPresent()) {
    return "Hello World";
  } else {
    return "Hello " + name;
  }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public String sayHello(String name) {
  if (name == null) {
    return "Hello World";
  } else {
    return "Hello " + name;
  }
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Specify a "xxx" parameter instead.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p><code>Optional&lt;T&gt;</code></p>
</div>
</div>
</div>
</div>