<div class="paragraph">
<p>Overriding the default parameter value inherited from a parent class will lead to unexpected results when the child class is referenced from a pointer to the parent class.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>enum E_ShapeColor {E_RED, E_GREEN, E_BLUE};

class Shape
{
public:
  virtual void draw(E_ShapeColor color = E_RED) const
  {
    ...
  }
};
class Rectangle : public Shape
{
public:
  virtual void draw(E_ShapeColor color = E_BLUE) const override // Non-compliant
  {
   ...
  }
};

int main() {
  Shape *shape = new Rectangle{};
  shape-&gt;draw();  // unexpectedly calls Rectangle::draw(RED)
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>enum E_ShapeColor {E_RED, E_GREEN, E_BLUE};

class Shape
{
public:
  virtual void draw(E_ShapeColor color = E_RED) const
  {
    ...
  }
};

class Rectangle : public Shape
{
public:
  virtual void draw(E_ShapeColor color) const override
  // OR: virtual void draw(E_ShapeColor color = E_RED) const override
  {
    ...
  }
};

int main() {
  Shape *shape = new Rectangle{};
  shape-&gt;draw(); // expectedly calls Rectangle::draw(RED)
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C&#43;&#43; 2008, 8-3-1 - Parameters in a overriding virtual function shall either use the same default arguments as the function they override, or else shall not specify any default arguments.</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove the default value for parameter "xxx" or set it to the same value as in the base class.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s3719">relates to: S3719</h3>

</div>
</div>
</div>