<div class="paragraph">
<p>Comparisons of objects of different types are likely programming errors since the comparisons will always return false.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when objects with no common ancestor closer than <code>Object</code> are compared.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public class MyClass {
  private Mountain mountain = new Mountain("Saint Helens");

  public doSomething(Apple apple) {
    if (mountain.equals(apple)) {  // Noncompliant. Huh?
      // ...
    }
  }
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>Remove this comparison of "XXX" and "YYY", which are of different types.</p>
</li>
<li>
<p>Remove this comparison of "xxx" and "yyy", which have no common ancestor closer than "Object".</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_25_aug_2014_201654_ann_campbell_wrote">on 25 Aug 2014, 20:16:54 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~pierre-yves.nicolas] what happens if they&#8217;re not?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_20_jan_2015_075202_ann_campbell_wrote">on 20 Jan 2015, 07:52:02 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Ping [~pierre-yves.nicolas] &#8230;&#8203;</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_12_jun_2015_135316_ann_campbell_wrote">on 12 Jun 2015, 13:53:16 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Original description from [~pierre-yves.nicolas]:</p>
</div>
<div class="paragraph">
<p><a href="http://www.csis.ul.ie/cobol/course/Selection.htm#part1" class="bare">http://www.csis.ul.ie/cobol/course/Selection.htm#part1</a></p>
</div>
<div class="paragraph">
<p>Note that the values of the compared items must be type compatible. For instance, it is not valid to have a statement that says IF "mike" IS EQUAL TO 123 THEN etc</p>
</div>
<div class="paragraph">
<p>Maybe that can also be applied to EVALUATE.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_12_jun_2015_135336_ann_campbell_wrote">on 12 Jun 2015, 13:53:36 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>CodePro: Comparison Of Incompatible Types</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_12_jun_2015_135501_ann_campbell_wrote">on 12 Jun 2015, 13:55:01 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>FYI [~pierre-yves.nicolas] I&#8217;ve used this unfinished rule-stub to write up a CodePro Java rule (it already had the title I needed), but left Cobol targeted and added RPG.</p>
</div>
</div>
</div>
</div>