<div class="paragraph">
<p>Explicitly defining a cursor as read-only can improve performance by avoiding table locking. This allows other SQL requests to execute in parallel. Therefore when a cursor will only be used to read data, without modifying anything, the <code>FOR READ ONLY</code> clause or its synonyn, <code>FOR FETCH ONLY</code>, should be used.</p>
</div>
<div class="paragraph">
<p>Conversely when a cursor will modify data, that too should be specified using the <code>FOR UPDATE</code> clause.</p>
</div>
<div class="paragraph">
<p>In short, it&#8217;s better to always explicitly define the purpose of the cursor with help of the <code>FOR READ ONLY</code>, <code>FOR FETCH ONLY</code> or <code>FOR UPDATE</code> clauses.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>EXEC SQL DECLARE CMAJ_0A CURSOR
  FOR SELECT C_BQ
    FROM       S1ORDCOU
    WHERE C_BQ = :TORD-C-BQ
END-EXEC</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>EXEC SQL DECLARE CMAJ_0A CURSOR
  FOR SELECT C_BQ
    FROM       S1ORDCOU
    WHERE C_BQ = :TORD-C-BQ
  FOR READ ONLY
END-EXEC</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Explicitly specify whether the "XXXX" cursor is read-only or not.</p>
</div>
</div>
</div>
</div>