<div class="paragraph">
<p>While it is legal to place <code>#define</code> and <code>#undef</code> directives anywhere in a source file, placing them outside of the global namespace is misleading since their scope is not actually restricted. This may be inconsistent with developer expectations.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>namespace NS
{
  #ifndef MY_HDR
  #define MY_HDR    /* Noncompliant */
  #undef FOO        /* Noncompliant */
  #endif
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>#ifndef MY_HDR
#define MY_HDR
#undef FOO
#endif</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C:2004, 19.5 - Macros shall not be #define&#8217;d or #undef&#8217;d within a block.</p>
</li>
<li>
<p>MISRA C&#43;&#43;:2008, 16-0-2 - Macros shall only be #define&#8217;d or #undef&#8217;d in the global namespace.</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_31_mar_2015_190543_evgeny_mandrikov_wrote">on 31 Mar 2015, 19:05:43 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] implementation seems more complete (SQALE, description) than this spec.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_13_apr_2015_195104_evgeny_mandrikov_wrote">on 13 Apr 2015, 19:51:04 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I&#8217;m wondering why blocker, but not active by default? Note that in implementation currently major and active.</p>
</div>
</div>
</div>
</div>