<div class="paragraph">
<p>A lambda consisting only of a method call with exactly the same parameters passed to the lambda itself is unneeded. Simply call the enclosed method directly, instead.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>def add(x, y):
    return x + y

add_lambda = lambda x, y: add(x, y)  # Noncompliant

print(add_lambda(1, 2))</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>def add(x, y):
    return x + y

print(add(1, 2))</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this lambda definition and call "xxx" directly.</p>
</div>
</div>
</div>
</div>