<div class="paragraph">
<p>The <code>EXEC SQL ... END-EXEC</code> statement can be used to embed Native SQL statically in ABAP programs.</p>
</div>
<div class="paragraph">
<p>According to the SAP documentation:</p>
</div>
<div class="quoteblock">
<blockquote>
<div class="paragraph">
<p>Alongside ADBC, it is also possible to embed Native SQL statically between <code>EXEC SQL</code> and <code>ENDEXEC</code> in ABAP programs. The recommendation, however, is to use ADBC. While the static embedding of Native SQL offers exclusively static access to the Native SQL interface, ADBC makes modern object-orientated and dynamic access possible. New developments and improvements, such as optimized performance using bulk access across internal tables, are now made only for ADBC.</p>
</div>
<div class="paragraph">
<p>The existing static embedding of Native SQL statements is still supported but should no longer be used in new programs.</p>
</div>
</blockquote>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>EXEC SQL.
      CREATE TABLE abap_docu_demo_mytab (
               val1 char(10) NOT NULL,
               val2 char(10) NOT NULL,
               PRIMARY KEY (val1)      )
ENDEXEC.</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>NEW cl_sql_statement( )-&gt;execute_ddl(
      `CREATE TABLE ` &amp;&amp; dbname   &amp;&amp;
      `( val1 char(10) NOT NULL,` &amp;&amp;
      `  val2 char(10) NOT NULL,` &amp;&amp;
      `  PRIMARY KEY (val1) )` ).</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>If there is no other option than using native SQL, at least the ADBC API should be used.</p>
</div>
</div>
</div>
</div>