<div class="paragraph">
<p>Some AWS services create <code>Log Groups</code> implicitly and don’t let the developer choose the <code>Log Group</code> name. This means that CloudFormation does not require the developer to declare the <code>Log Group</code> that the resource will write to.</p>
</div>
<div class="paragraph">
<p>If a <code>Log Group</code> is not declared within CloudFormation, then this <code>Log Group</code> will be automatically created at run time and it won’t be managed by CloudFormation. The consequences are that the:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>Log Group</code> will not be removed when the <code>Stack</code> is removed.</p>
</li>
<li>
<p><code>Log Group</code> attributes (such as retention) won’t be managed "as code" within CloudFormation.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>These automatically created <code>Log Groups</code> will remain forever which can have huge impact on costs and security (the data stay forever while this is unexpected from company or regulatory rules).</p>
</div>
<div class="paragraph">
<p>A good practice is to declare the target <code>Log Group</code> with CloudFormation, matching the name that the AWS will use so that the <code>Log Group</code> resource is managed as code.</p>
</div>
<div class="paragraph">
<p>This rule applies to the following resources:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>AWS::Lambda::Function</p>
</li>
<li>
<p>AWS::Serverless::Function</p>
</li>
<li>
<p>AWS::ApiGatewayV2::Api</p>
</li>
<li>
<p>AWS::CodeBuild::Project</p>
</li>
</ul>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>  # There is no "Log Group" declared corresponding to "MyLambdaFunction": logs will not be managed by CloudFormation
  MyLambdaFunction:
    Type: AWS::Lambda::Function
    Properties:
      Runtime: nodejs12.x
      Description: Example of Lambda Function</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Example with Ref</p>
</div>
<div class="listingblock">
<div class="content">
<pre>  MyLambdaFunction:
    Type: AWS::Lambda::Function
    Properties:
      Runtime: nodejs12.x
      Description: Example of Lambda Function

  MyFunctionLogGroup:
    Type: AWS::Logs::LogGroup
    Properties:
      LogGroupName: !Join ['/', ['/aws/lambda', !Ref MyLambdaFunction]]
      RetentionInDays: 30</pre>
</div>
</div>
<div class="paragraph">
<p>Example with Sub</p>
</div>
<div class="listingblock">
<div class="content">
<pre>  MyLambdaFunction:
    Type: AWS::Lambda::Function
    Properties:
      Runtime: nodejs12.x
      Description: Example of Lambda Function

  MyFunctionLogGroupSub:
    Type: AWS::Logs::LogGroup
    Properties:
      LogGroupName: !Sub '/aws/lambda/${MyLambdaFunction}’
      RetentionInDays: 30</pre>
</div>
</div>
</div>
</div>