<div class="paragraph">
<p>Using the same name for multiple purposes reduces the understandability of the code and might eventually lead to bugs.</p>
</div>
<div class="paragraph">
<p>This rule verifies that no name is reused in an inner scope.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>&lt;&lt;foo&gt;&gt;                                    -- Compliant
DECLARE
  a CONSTANT PLS_INTEGER := 0;
BEGIN
  &lt;&lt;foo&gt;&gt;                                  -- Non-Compliant
  DECLARE
    b CONSTANT PLS_INTEGER := 42;
  BEGIN
    DBMS_OUTPUT.PUT_LINE('x = ' || foo.b); -- Confusing
  END;
END;
/</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>&lt;&lt;foo&gt;&gt;                                    -- Compliant
DECLARE
  a CONSTANT PLS_INTEGER := 0;
BEGIN
  &lt;&lt;bar&gt;&gt;                                  -- Compliant
  DECLARE
    b CONSTANT PLS_INTEGER := 42;
  BEGIN
    DBMS_OUTPUT.PUT_LINE('x = ' || bar.b); -- Clear
  END;
END;
/</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_duplicates_s1524">duplicates: S1524</h3>

</div>
<div class="sect2">
<h3 id="_on_21_may_2013_141155_fabrice_bellingard_wrote">on 21 May 2013, 14:11:55 Fabrice Bellingard wrote:</h3>
<div class="paragraph">
<p>Implementation: http://jira.sonarsource.com/browse/PLSQL-348</p>
</div>
</div>
</div>
</div>