<div class="paragraph">
<p>Using the output of a non-boolean assignment as a condition is highly suspicious. Either it was done accidentally, and the assignment operator <code>=</code> was used in place of the equality operator <code>==</code>, or it was done with the intent to rely on the automatic conversion of the non-boolean value to a boolean, which could be confusing for maintainers.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>void process(int a) {
   if(a = BAD_OPS) {...}
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_exceptions">Exceptions</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In C&#43;&#43;, it is common to null-check a pointer by using it as a condition. In such cases, if the pointer is <code>NULL</code>, the condition evaluates to <code>false</code>, and otherwise it is <code>true</code>. Such pointer assignments are ignored:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>int *ip;
if (ip = get_status() ) {...}</pre>
</div>
</div>
<div class="paragraph">
<p>CLang is able to detect the same kind of quality issues and allows to switch-off those issues by surrounding the non-boolean assignment with some parentheses. To prevent raising some issues which were intentionally switched off, this rule supports the same coding convention:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>if((my_status = get_status())) {...}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C:2004, 13.1 - Assignment operators shall not be used in expressions that yield a Boolean value</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Refactor this suspicious code to not use the result of this non-boolean assignment as a condition</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The overall non-boolean assignment</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_21_mar_2017_112105_ann_campbell_wrote">on 21 Mar 2017, 11:21:05 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~freddy.mallet] does this double-parens exception need to be added in the double-parens rule as well?</p>
</div>
</div>
<div class="sect2">
<h3 id="_relates_to_s1121">relates to: S1121</h3>

</div>
<div class="sect2">
<h3 id="_on_21_mar_2017_095054_freddy_mallet_wrote">on 21 Mar 2017, 09:50:54 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Before C99, the C language didn&#8217;t have any boolean type so applying this rule on C might raise a lot of false-positives.</p>
</div>
</div>
</div>
</div>