<div class="paragraph">
<p>Non <code>final</code> classes shouldn&#8217;t use a hardcoded class name in the <code>equals</code> method. Doing so breaks the method for subclasses. Instead, make the comparison dynamic.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public class Fruit {
  private Season ripe;

  public boolean equals(Object obj) {
    if (obj == this) {
      return true;
    }
    if (Fruit.class == obj.class) { // Noncompliant
      return false;
    }
    // ...</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public class Fruit {
  private Season ripe;

  public boolean equals(Object obj) {
    if (obj == this) {
      return true;
    }
    if (this.class == obj.class) { // will work for subclasses too
      return false;
    }
    // ...</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_replaced_by_s2162">is replaced by: S2162</h3>

</div>
<div class="sect2">
<h3 id="_on_15_oct_2014_215316_freddy_mallet_wrote">on 15 Oct 2014, 21:53:16 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>@Ann, I would merge this rule with RSPEC-2162 as they both have exactly the same goal.</p>
</div>
</div>
</div>
</div>