<div class="paragraph">
<p>The use of positional notation to describe the overlapping of one field on another can quickly become confusing, particularly in large, or complicated data structures. Instead, the <code>OVERLAY</code> keyword should be used to make such overlaps more readily identifiable.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>     D Message         DS                  QUALIFIED
     D   Id                           7
      * Noncompliant
     D   Type                  1      3</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>     D Message         DS                  QUALIFIED
     D   Id                           7
     D   Type                         3    OVERLAY(Id)</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace this positional notation with the "OVERLAY" keyword.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_2_apr_2015_185452_ann_campbell_wrote">on 2 Apr 2015, 18:54:52 Ann Campbell wrote:</h3>
<div class="paragraph">
<p><a href="http://www.bmeyers.net/faqs/14-tips/32-rpg-iv-style?start=3" class="bare">http://www.bmeyers.net/faqs/14-tips/32-rpg-iv-style?start=3</a></p>
</div>
</div>
</div>
</div>