<div class="paragraph">
<p>Template literals (previously named "template strings") are an elegant way to build a string without using the <code>+</code> operator to make strings concatenation more readable.</p>
</div>
<div class="paragraph">
<p>However, it&#8217;s possible to build complex string literals by nesting together multiple template literals, and therefore lose readability and maintainability.</p>
</div>
<div class="paragraph">
<p>In such situations, it&#8217;s preferable to move the nested template into a separate statement.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>let color = "red";
let count = 3;
let message = `I have ${color ? `${count} ${color}` : count} apples`; // Noncompliant; nested template strings not easy to read</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>let color = "red";
let count = 3;
let apples = color ? `${count} ${color}` : count;
let message = `I have ${apples} apples`;</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Refactor this code to not use nested template literals.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_3_may_2018_165351_alexandre_gigleux_wrote">on 3 May 2018, 16:53:51 Alexandre Gigleux wrote:</h3>
<div class="paragraph">
<p><a href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals" class="bare">https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals</a></p>
</div>
</div>
</div>
</div>