<div class="paragraph">
<p><code>RAISE_APPLICATION_ERROR</code> may only be called with an error code from <code>-20,000</code> to <code>-20,999</code>, which is the range reserved for application errors. When called with another value, Oracle raises the exception: <code>ORA-21000: error number argument to raise_application_error of 0 is out of range.</code></p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>BEGIN
  RAISE_APPLICATION_ERROR(0, 'This is an application error'); -- Non-Compliant - raises ORA-21000
END;
/</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>BEGIN
  RAISE_APPLICATION_ERROR(-20000, 'This is an application error'); -- Compliant
END;
/</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>"nnn" is an invalid error code.</p>
</div>
</div>
</div>
</div>