<div class="paragraph">
<p>Numbers are infinite, but the types that hold them are not. Each numeric type has hard upper and lower bounds. Try to calculate or assign numbers beyond those bounds, and the result will be a value that has silently wrapped around from the expected positive value to a negative one, or vice versa.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public int getTheNumber(int positiveInt) {
  int num = Integer.MAX_VALUE;
  return num + positiveInt;  // Noncompliant
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>This calculation is guaranteed to overflow the maximum value of [type].</p>
</li>
<li>
<p>This calculation is likely to overflow the maximum value of [type].</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>primary: the entire calculation</p>
</li>
<li>
<p>secondary: the type into which the overflowing value is expected to fit: variable, return type (in the case of <code>return x*y</code>, &#8230;&#8203;</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_supercedes_s879">supercedes: S879</h3>

</div>
<div class="sect2">
<h3 id="_on_28_mar_2017_163925_ann_campbell_wrote">on 28 Mar 2017, 16:39:25 Ann Campbell wrote:</h3>
<div class="paragraph">
<p><a href="https://groups.google.com/forum/#!topic/sonarqube/HGIpDMj7bpA" class="bare">https://groups.google.com/forum/#!topic/sonarqube/HGIpDMj7bpA</a></p>
</div>
</div>
</div>
</div>