<div class="paragraph">
<p>Defining variables in the minimum block scope possible reduces the visibility of those variables and therefore reduces the possibility that these identifiers will be used accidentally. A corollary of this is that global objects (including singleton function objects) shall be used in more than one function.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>void f ( int32_t k )
{
  int32_t i = k * k; // Noncompliant, visibility could be reduced
  {
    std::cout &lt;&lt; i &lt;&lt; std::endl;
  }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>void f ( int32_t k )
{
  {
    int32_t i = k * k; // Compliant
    std::cout &lt;&lt; i &lt;&lt; std::endl;
  }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C&#43;&#43;:2008, 3-4-1</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Scope of this declaration should be reduced.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s1899">relates to: S1899</h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s5955">relates to: S5955</h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s3798">is related to: S3798</h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s831">relates to: S831</h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s825">is related to: S825</h3>

</div>
<div class="sect2">
<h3 id="_on_19_feb_2016_224544_evgeny_mandrikov_wrote">on 19 Feb 2016, 22:45:44 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] looks very similar to RSPEC-2392</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_22_feb_2016_165223_ann_campbell_wrote">on 22 Feb 2016, 16:52:23 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Actually [~evgeny.mandrikov] RSPEC-2392 used to be a subtask of RSPEC-1899 but was promoted because RSPEC-1899, like this rule, is only about moving vars down from global to local scope. In JavaScript they can move in both directions.</p>
</div>
<div class="paragraph">
<p>And because this is a MISRA rule, there are of course related rules (RSPEC-825). Since neither this rule nor RSPEC-825 is implemented yet, would you like to roll them all together?</p>
</div>
</div>
</div>
</div>