<div class="paragraph">
<p>Assistive technologies, such as screen readers, will not be able to render <code>&lt;object&gt;</code> elements, in such cases it is the content of the <code>&lt;object&gt;</code> which is provided to the user. This alternative content needs to be accessible or the screen readers won&#8217;t be able to use it. For example, if an <code>&lt;img&gt;</code> is used it must contain an <code>alt</code> attribute (see corresponding rule Web:ImgWithoutAltCheck).</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when an <code>&lt;object&gt;</code> tag does not have any alternative content.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>&lt;object&gt;&lt;/object&gt; &lt;!-- Noncompliant --&gt;

&lt;object&gt;
  &lt;object&gt;&lt;/object&gt; &lt;!-- Noncompliant --&gt;
&lt;/object&gt;</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>&lt;object&gt;This application shows the simulation of two particles colliding&lt;/object&gt;

&lt;object&gt;
  &lt;img src="flower.png" alt="Flower growing in a pot" /&gt;
&lt;/object&gt;

&lt;object&gt;
  &lt;object&gt;
    This application shows the simulation of two particles colliding
  &lt;/object&gt;
&lt;/object&gt;</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://www.w3.org/TR/WCAG20-TECHS/H53.html">WCAG2, H53</a> - Using the body of the object element</p>
</li>
<li>
<p><a href="https://www.w3.org/WAI/WCAG21/quickref/?versions=2.0#qr-text-equiv-all">WCAG2, 1.1.1</a> - Non-text Content</p>
</li>
<li>
<p><a href="https://www.w3.org/WAI/WCAG21/quickref/?versions=2.0#qr-media-equiv-audio-desc">WCAG2, 1.2.3</a> - Audio Description or Media Alternative (Prerecorded)</p>
</li>
<li>
<p><a href="https://www.w3.org/WAI/WCAG21/quickref/?versions=2.0#qr-media-equiv-text-doc">WCAG2, 1.2.8</a> - Media Alternative (Prerecorded)</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add an accessible content to this "&lt;object&gt;" tag.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The opening &lt;object&gt; tag without its content</p>
</div>
</div>
</div>
</div>