<div class="paragraph">
<p>Catching an exception class by value rather than by reference has several bad effects:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Slicing occurs, yielding an instance of the exception&#8217;s base class, rather than the potentially more specific exception class that was actually thrown. This means that only the base class' functions will be available; any additional data or functionality that is offered by the extended class will not be accessible.</p>
</li>
<li>
<p>Memory is allocated unnecessarily.</p>
</li>
<li>
<p>Copying the exception class might potentially throw an exception.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>You might also be tempted to catch an exception by pointer, but this causes issues related to the exception lifetime, and should also be avoided. This situation is detected by rule S1035.</p>
</div>
<div class="paragraph">
<p>Therefore exception classes should always be caught by reference.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>try
{
  // ...
}
catch(ExceptionClass ex)
{
  //...
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>try
{
  // ...
}
catch(ExceptionClass &amp;ex)
{
  //...
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C&#43;&#43;:2008, 15-3-5 - A class type exception shall always be caught by reference</p>
</li>
<li>
<p><a href="https://wiki.sei.cmu.edu/confluence/x/SXs-BQ">CERT, ERR61-CPP.</a> - Catch exceptions by lvalue reference</p>
</li>
<li>
<p><a href="https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#e15-catch-exceptions-from-a-hierarchy-by-reference">C&#43;&#43; Core Guidelines E.15</a> - Catch exceptions from a hierarchy by reference</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Catch the exception by reference.</p>
</div>
</div>
</div>
</div>