<div class="paragraph">
<p>Shadowing a builtin makes your code more difficult to read and maintain. It may also be a source of bugs as you can reference the builtin by mistake.</p>
</div>
<div class="paragraph">
<p>It is sometimes ok to shadow a builtin to improve the readability of a public API or to support multiple versions of a library. In these cases the value is higher than the maintainability cost. Just be careful when you do it.</p>
</div>
<div class="paragraph">
<p>It is not ok to shadow builtins with variables which are local to a function or method. These variables are not public and can be easily renamed, thus reducing the confusion and making the code less error-prone.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when the name of a local variable matches the name of a builtin.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>def a_function():
    int = 42  # Noncompliant; int is a builtin</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>def a_function():
    value = 42</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.python.org/3.8/library/stdtypes.html">Python documentation - Built-in Types</a></p>
</li>
<li>
<p><a href="https://docs.python.org/3/library/functions.html">Python documentation - Built-in Functions</a></p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Rename this variable; it shadows a builtin.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Primary:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>The first definition of a variable shadowing a builtin in a local scope (function)</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Secondary:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Any other assignment to the variable.</p>
</li>
<li>
<p>message: "Variable also assigned here."</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s2137">relates to: S2137</h3>

</div>
<div class="sect2">
<h3 id="_on_30_apr_2020_191203_nicolas_harraudeau_wrote">on 30 Apr 2020, 19:12:03 Nicolas Harraudeau wrote:</h3>
<div class="paragraph">
<p>This rule is similar to RSPEC-2137 but it is a code smell because shadowing builtins will not make your code crash most of the time.</p>
</div>
</div>
</div>
</div>