<div class="paragraph">
<p>PHP 5.3 introduces <a href="http://www.php.net/namespaces">namespaces</a> to the language. Use of this mechanism should be preferred to <code>include</code> or <code>include_once</code> or <code>require</code> or <code>require_once</code> because it solves two common problems:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>it avoids name collisions</p>
</li>
<li>
<p>it provides the ability to create alias which improve readability of the code</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Starting from its version 8, Drupal is relying on namespaces to be compliant with <a href="https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-4-autoloader.md">PSR-4 standard</a>. Drupal&#8217;s modules should be compliant with PSR-4 standard and therefore should no longer rely on <code>include</code> or <code>include_once</code> or <code>require</code> or <code>require_once</code> functions.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>require_once('./modules/vegetable/src/Entity/Tomato.php');</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>use Drupal\vegetable\Entity\Tomato</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_exceptions">Exceptions</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule doesn&#8217;t raise issues on <code>autoload.php</code> or <code>ScriptHandler.php</code> files.</p>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace [include|include_once|require|require_once] with namespace import mechanism through the "use" keyword.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_14_oct_2018_111858_pierre_loup_tristant_wrote">on 14 Oct 2018, 11:18:58 Pierre-Loup Tristant wrote:</h3>
<div class="paragraph">
<p>There is no "Sonar way recommended" quality profile in SonarPHP.</p>
</div>
<div class="paragraph">
<p>Should I put the rule in "Sonar way" ?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_14_oct_2018_111904_pierre_loup_tristant_wrote">on 14 Oct 2018, 11:19:04 Pierre-Loup Tristant wrote:</h3>
<div class="paragraph">
<p>Pull request :</p>
</div>
<div class="paragraph">
<p><a href="https://github.com/SonarSource/sonar-php/pull/362" class="bare">https://github.com/SonarSource/sonar-php/pull/362</a></p>
</div>
</div>
</div>
</div>