<div class="paragraph">
<p>When two locks are held simultaneously, a <code>wait</code> call only releases one of them. The other will be held until some other thread requests a lock on the awaited object. If no unrelated code tries to lock on that object, then all other threads will be locked out, resulting in a deadlock.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>synchronized (this.mon1) {  // threadB can't enter this block to request this.mon2 lock &amp; release threadA
	synchronized (this.mon2) {
		this.mon2.wait();  // Noncompliant; threadA is stuck here holding lock on this.mon1
	}
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Don&#8217;t use "wait()" here; multiple locks are held.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s2446">is related to: S2446</h3>

</div>
<div class="sect2">
<h3 id="_on_11_jun_2015_185101_ann_campbell_wrote">on 11 Jun 2015, 18:51:01 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>CodePro: Notify method invoked while two locks are held</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_11_jun_2015_185146_ann_campbell_wrote">on 11 Jun 2015, 18:51:46 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>FYI [~tamas.vajk]. I wrote this for Java but I&#8217;m guessing it applies to C# as well, so I&#8217;ve targeted it to the language. Simply un-target it if it&#8217;s not, please.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_23_jan_2017_144408_tibor_blenessy_wrote">on 23 Jan 2017, 14:44:08 Tibor Blenessy wrote:</h3>
<div class="paragraph">
<p><code>notify</code> call is already covered in RSPEC-2446 and there will be two issues raised on notify call in some cases. Therefore I will remove it from the implementation of this rule and change this RSPEC accordingly.</p>
</div>
</div>
</div>
</div>