<div class="paragraph">
<p>Shared naming conventions allow teams to collaborate efficiently. This rule raises an issue when a test method name does not match the provided regular expression.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="paragraph">
<p>With the default value: <code>^test[A-Z][a-zA-Z0-9]*$</code></p>
</div>
<div class="listingblock">
<div class="content">
<pre>@Test
public void foo() {  // Noncompliant
  //...
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>@Test
public void testFoo() {
  // ...
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Rename this method to match the regular expression: yyyy</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="ulist">
<ul>
<li>
<p>key = format</p>
</li>
<li>
<p>Description : Regular expression the test method names are checked against.</p>
</li>
<li>
<p>Default value : ^test[A-Z][a-zA-Z0-9]*$</p>
</li>
</ul>
</div>
</div>
</div>
</div>