<div class="paragraph">
<p>Tables without primary keys are largely unusable in a relational database because they cannot be joined to. A primary key should be specified at table creation to guarantee that all its records have primary key values.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>CREATE TABLE employee
(
  employee_id INTEGER NOT NULL,
  first_name VARCHAR2(42) NOT NULL,
  last_name VARCHAR2(42) NOT NULL
);</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>CREATE TABLE employee
(
  employee_id INTEGER PRIMARY KEY,
  first_name VARCHAR2(42) NOT NULL,
  last_name VARCHAR2(42) NOT NULL
);</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add a primary key constraint to this table.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_6_may_2014_160649_ann_campbell_wrote">on 6 May 2014, 16:06:49 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Not relevant for abap : no table creation.</p>
</div>
</div>
</div>
</div>