<div class="paragraph">
<p>It is not very common for a struct to have base classes. When they do, by default, they will have public inheritance. Since this is not a fact known by everybody, it&#8217;s usually better to be explicit about the visibility of base classes in a struct.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class B {
};

struct C : B {
};</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class B {
};

struct C :  public B { // Or private, if it was public by mistake
};</pre>
</div>
</div>
<div class="sect2">
<h3 id="_see_also">See Also</h3>
<div class="ulist">
<ul>
<li>
<p>S5965 is a similar rule that deals with base visibility in classes</p>
</li>
<li>
<p>S3540 is another rule that favors a different coding style for base classes. It should not be activated at the same time as this rule.</p>
</li>
</ul>
</div>
</div>
</div>
</div>