<div class="paragraph">
<p>A <code>LocalConnection</code> object is used to invoke a method in another <code>LocalConnection</code> object, either within a single SWF file or between multiple SWF files. This kind of local connection should be authorized only when the origin (domain) of the other Flex applications is perfectly defined.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>localConnection.allowDomain("*");</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>localConnection.allowDomain("www.myDomain.com");</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace this wildcard character '*' with a well defined domain</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_22_nov_2013_101321_freddy_mallet_wrote">on 22 Nov 2013, 10:13:21 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3276</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_24_nov_2013_191011_ann_campbell_wrote">on 24 Nov 2013, 19:10:11 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>I&#8217;m wondering how broadly this rule will apply. Clearly it will flag "*" but what about other wildcard combinations? If so, we might want to take another look at the wording.</p>
</div>
</div>
</div>
</div>