<div class="paragraph">
<p>Shared coding conventions allow teams to collaborate efficiently. This rule checks that all namespace names match a provided regular expression.</p>
</div>
<div class="paragraph">
<p>The default configuration is the one recommended by Microsoft:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Pascal casing, starting with an upper case character, e.g. Microsoft, System</p>
</li>
<li>
<p>Short abbreviations of 2 letters can be capitalized, e.g. System.IO</p>
</li>
<li>
<p>Longer abbreviations need to be lower cased</p>
</li>
</ul>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="paragraph">
<p>With the default regular expression: <code>^([A-Z]{1,3}[a-z0-9]+)*([A-Z]{2})?(\.([A-Z]{1,3}[a-z0-9]+)*([A-Z]{2}))*$</code></p>
</div>
<div class="listingblock">
<div class="content">
<pre>Namespace foo  ' Noncompliant
End Namespace</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>Namespace Foo  ' Compliant
End Namespace</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Rename this namespace to match the regular expression "rrr".</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="ulist">
<ul>
<li>
<p>key = format
description: Regular expression used to check the namespace names against.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>default: ^([A-Z]{1,3}<span class="a-z0-9">)*([A-Z]{2})?(\.([A-Z]{1,3}[a-z0-9]</span>)<strong>([A-Z]{2}))</strong>$</p>
</div>
</div>
</div>
</div>