<div class="paragraph">
<p>The use of virtual base classes can introduce a number of undefined and potentially confusing behaviours. The use of virtual bases is not recommended.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class B {};
class D: public virtual B {}; // Noncompliant, B is a virtual base</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C&#43;&#43;:2008, 10-1-1 - Classes should not be derived from virtual bases.</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Refactor this "virtual" inheritance.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_21_oct_2014_185514_ann_campbell_wrote">on 21 Oct 2014, 18:55:14 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~samuel.mercier]</p>
</div>
<div class="ulist">
<ul>
<li>
<p>for SQALE, we go with the worst-possible outcome. Since this is an "undefined behaviors" rule, Maintainability is not the best fit</p>
</li>
<li>
<p>this is likely a "bug"</p>
</li>
</ul>
</div>
</div>
</div>
</div>