<div class="paragraph">
<p>The methods <code>wait(...)</code>, <code>notify()</code> and <code>notifyAll()</code> are available on a <code>Thread</code> instance, but only because all classes in Java extend <code>Object</code> and therefore automatically inherit those methods. But there are two very good reasons for not calling them on a <code>Thread</code>:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Internally, the JVM relies on these methods to change the state of the Thread (<code>BLOCKED</code>, <code>WAITING</code>, &#8230;&#8203;), so calling them will corrupt the behavior of the JVM.</p>
</li>
<li>
<p>It is not clear (perhaps even to the original coder) what is really expected. For instance, it is waiting for the execution of the Thread to suspended, or is it the acquisition of the object monitor that is waited for?</p>
</li>
</ul>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>Thread myThread = new Thread(new RunnableJob());
...
myThread.wait(2000);</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Refactor the synchronisation mechanism to not use a Thread instance as a monitor.</p>
</div>
</div>
</div>
</div>