<div class="paragraph">
<p><code>ASC</code> or <code>DESC</code> should be specified for every column of an <code>ORDER BY</code> clause to improve readability.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>BEGIN
  SELECT col1, col2, col3
  BULK COLLECT INTO result
  FROM my_table
  ORDER BY
    col1 ASC,
    col2,            -- Noncompliant - ASC or DESC should be specified
    col3 DESC;
END;
/</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>BEGIN
  SELECT col1, col2, col3
  BULK COLLECT INTO result
  FROM my_table
  ORDER BY
    col1 ASC,
    col2 ASC,
    col3 DESC;
END;
/</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add "ASC" or "DESC" to this "ORDER BY".</p>
</div>
</div>
</div>
</div>