<div class="paragraph">
<p>Availability of rare resources, such as mutexes, often has a direct impact on the performance of an application. Holding onto rare resources for too long might lead to a significant slowdown or a deadlock.</p>
</div>
<div class="paragraph">
<p><code>co_yield</code> and <code>co_await</code> suspend the execution of a coroutine for an indefinite time, and <code>co_return</code> terminates it. At this point, all the rare resources acquired by the coroutine must be released to enable other potential consumers to use them while the coroutine is inactive.</p>
</div>
<div class="paragraph">
<p>This rule reports execution scenarios that reach <code>co_yield</code>, <code>co_await</code>, or <code>co_return</code> without releasing a rare resource, such as a mutex.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>std::mutex m;
int shared = 0;
cppcoro::generator&lt;int&gt; coroutine()
{
  while(true) {
    m.lock();
    co_yield ++shared; // Noncompliant: m is locked
    m.unlock();
  }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>std::mutex m;
int shared = 0;
cppcoro::generator&lt;int&gt; coroutine()
{
  while(true) {
    m.lock();
    ++shared;
    m.unlock();
    co_yield shared; // Compliant: m is available
  }
}</pre>
</div>
</div>
</div>
</div>