<div class="paragraph">
<p><code>ToCharArray</code> can be omitted when the operation on the array could have been done directly on the string, such as when iterating over the characters in a string, and when accessing a character in a string via an array index. In those cases, explicit <code>ToCharArray</code> calls should be omitted.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>string str = "some string";
foreach (var c in str.ToCharArray()) // Noncompliant
{
  // ...
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>string str = "some string";
foreach (var c in str)
{
  // ...
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this redundant "ToCharArray" call.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p><code>ToCharArray()</code></p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_10_dec_2015_090547_tamas_vajk_wrote">on 10 Dec 2015, 09:05:47 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Applied minor changes.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_10_dec_2015_144316_ann_campbell_wrote">on 10 Dec 2015, 14:43:16 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>looks good [~tamas.vajk]</p>
</div>
</div>
</div>
</div>