<div class="paragraph">
<p>Most people will reasonably assume that a class with "Test" in the name will contain test methods. Writing "Test" classes without tests will inevitably lead to confusion, and perhaps to gaps in test coverage. Such classes should either be renamed or corrected.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public class MathTest {
  private List questions;
  private int averageScore;
  private int highScore;
  // ...
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public class MathExam {
  private List questions;
  private int averageScore;
  private int highScore;
  // ...
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_replaced_by_s2187">is replaced by: S2187</h3>

</div>
</div>
</div>