<div class="paragraph">
<p>Just because you <em>can</em> do something, doesn&#8217;t mean you should, and that&#8217;s the case with nested conditional expressions. Nesting conditional expressions results in the kind of code that may seem clear as day when you write it, but six months later will leave maintainers (or worse - future you) scratching their heads and cursing.</p>
</div>
<div class="paragraph">
<p>Instead, err on the side of clarity, and use another line to express the nested operation as a separate statement.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class Job:
    @property
    def readable_status(self):
        return "Running" if job.is_running else "Failed" if job.errors else "Succeeded"  # Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class Job:
    @property
    def readable_status(self):
        if job.is_running:
            return "Running"
        return "Failed" if job.errors else "Succeeded"</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_exceptions">Exceptions</h2>
<div class="sectionbody">
<div class="paragraph">
<p>No issue is raised on conditional expressions in comprehensions.</p>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Extract this nested conditional expression into an independent statement.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>Primary: highlight the entire nested conditional expression</p>
</li>
<li>
<p>Secondary: highlight the "if" and "else" of the parent conditional expression
message: 'Parent conditional expression.'</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s134">relates to: S134</h3>

</div>
<div class="sect2">
<h3 id="_on_7_mar_2018_133132_janos_gyerik_wrote">on 7 Mar 2018, 13:31:32 Janos Gyerik wrote:</h3>
<div class="paragraph">
<p>This rule does not apply to Go, because there is no ternary operator (as of version 1.10).</p>
</div>
</div>
</div>
</div>