<div class="paragraph">
<p>The size of a collection and the length of an array are always greater than or equal to zero. So testing that a size or length is greater than or equal to zero doesn&#8217;t make sense, since the result is always <code>true</code>. Similarly testing that it is less than zero will always return <code>false</code>. Perhaps the intent was to check the non-emptiness of the collection or array instead.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>if (myList.size() &gt;= 0) { ... }

if (myList.size() &lt; 0) { ... }

boolean result = myArray.length &gt;= 0;

if (0 &gt; myArray.length) { ... }</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>if (!myList.isEmpty()) { ... }

if (myArray.length &gt;= 42) { ... }</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>The [size|length] of ["XXX"|an array|a collection] is always "&gt;=0", so update this test to [either "==0" or "&gt;0"|use "isEmpty()"].</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The boolean expression with the "&gt;=" or "&lt;" operator and the two operands</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s1155">relates to: S1155</h3>

</div>
<div class="sect2">
<h3 id="_on_27_apr_2018_162106_elena_vilchik_wrote">on 27 Apr 2018, 16:21:06 Elena Vilchik wrote:</h3>
<div class="paragraph">
<p>\[~alexandre.gigleux] could you create go subtask?</p>
</div>
</div>
</div>
</div>