<div class="paragraph">
<p>Method references are specialized lambda expressions for methods that already have a name. They take the form of <code>objectReference::methodName</code> or <code>ClassName::methodName</code>. When a lambda does nothing but call an existing method, it can be much clearer to use a method reference instead.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>Set roster = transferElements(rosterSource,
      () -&gt; { return new HashSet&lt;&gt;(); }  // Noncompliant
);</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>Set roster = transferElements(rosterSource, HashSet::new);</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_duplicates_s1612">duplicates: S1612</h3>

</div>
</div>
</div>