<div class="paragraph">
<p>Shared coding conventions allow teams to collaborate effectively. This rule raises an issue when the use of spaces within a template does not conform to the configured style.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="paragraph">
<p>With the configured default forbidding spaces:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>`Hello, ${world }!`; // Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>`Hello, ${world}!`;</pre>
</div>
</div>
</div>
</div>