<div class="paragraph">
<p>Using a scalar function or an arithmetic expression in a <code>WHERE</code> condition can prevent the database from using indexes on the relevant column(s), and could therefore lead to performance issues.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>SELECT * FROM MY_TABE WHERE C2 = C3 + :HostVar1  -- Noncompliant

SELECT * FROM MY_TABLE WHERE YEAR(BIRTHDATE) &gt; 2000  -- Noncompliant</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Refactor this SQL request to eliminate [the arithmetic expression|"xxx" scalar function] in the "WHERE" clause</p>
</div>
</div>
</div>
</div>