<div class="paragraph">
<p><code>SELECT</code> with <code>JOIN</code> always performs better than nested selects.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>SELECT * FROM SPFL INTO SPFLI_WA.
  SELECT * FROM SFLOGHT INTO SFLIGHT_WA
    WHERE CARRID = SPFLI_WA-CARRID
    AND CONNID = SPFLIGHT_WA_CONNID.
  ENDSELECT.
ENDSELECT.</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>SELECT * INTO WA
  FROM SPFLI AS P INNER JOIN SFLIGHT AS F
    ON P~CARRID = F~CARRID AND
        P~CONNID = F~CONNID.
END-SELECT.</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove the nested "SELECT" statements by using the "JOIN" clause</p>
</div>
</div>
</div>
</div>