<div class="paragraph">
<p>When a function is called, it accepts only one value per parameter. Python interpreters will raise a SyntaxError when they see something like <code>myfunction(a=1, a=2)</code>, but there are other cases which will only fail at runtime:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>An argument is provided by value and position at the same time.</p>
</li>
<li>
<p>Some arguments are provided via unpacking and the same argument is provided twice.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>This rule raises an issue when a function is called with multiple values for the same parameter.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>def func(a, b, c):
    return a * b * c

func(6, 93, 31, c=62) # Noncompliant: argument "c" is duplicated

params = {'c':31}
func(6, 93, 31, **params) # Noncompliant: argument "c" is duplicated
func(6, 93, c=62, **params) # Noncompliant: argument "c" is duplicated</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>def func(a, b, c):
    return a * b * c

print(func(c=31, b=93, a=6)) # Compliant</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>{xxx} argument is duplicated in {function} call</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Primary: the first appearance of duplicated argument</p>
</div>
<div class="paragraph">
<p>Secondary:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>location: the others argument duplicated</p>
<div class="ulist">
<ul>
<li>
<p>message: argument also passed here</p>
</li>
</ul>
</div>
</li>
<li>
<p>location: the signature of the called function</p>
</li>
<li>
<p>message: function definition</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_23_apr_2020_172822_nicolas_harraudeau_wrote">on 23 Apr 2020, 17:28:22 Nicolas Harraudeau wrote:</h3>
<div class="paragraph">
<p>The following use case was removed because it is a syntax error and we don&#8217;t want to target those:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>func(c=31, b=93, c=62)</pre>
</div>
</div>
</div>
</div>
</div>