<div class="paragraph">
<p>If a type is declared but not used, then it is unclear to a reviewer if the type is redundant or it has been left unused by mistake.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>void unusedtype()
{
  typedef int local_Type; // Noncompliant, unused
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C&#43;&#43;:2008, 0-1-5 - A project shall not contain unused type declarations.</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove the unused declaration of  type "xxx".</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s1763">is related to: S1763</h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s901">relates to: S901</h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s1029">is related to: S1029</h3>

</div>
<div class="sect2">
<h3 id="_on_22_oct_2014_164743_samuel_mercier_wrote">on 22 Oct 2014, 16:47:43 Samuel Mercier wrote:</h3>
<div class="paragraph">
<p>Should not be enabled while analysing a library.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_23_oct_2014_144658_ann_campbell_wrote">on 23 Oct 2014, 14:46:58 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~samuel.mercier] please consider other relevant tags. E.g. 'unused' for this rule</p>
</div>
</div>
</div>
</div>