<div class="paragraph">
<p>Most of the time a block of code is empty when a piece of code is really missing. So such empty block must be either filled or removed.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>For Index = 1 To 42 ' Empty on purpose or missing piece of code?
Next</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>For Index = 1 To 42
    Process Index
Next</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_exceptions">Exceptions</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When a block contains a comment, this block is not considered to be empty.</p>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Either remove or fill this block of code.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s3532">is related to: S3532</h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s1186">relates to: S1186</h3>

</div>
<div class="sect2">
<h3 id="_on_20_may_2013_145434_fabrice_bellingard_wrote">on 20 May 2013, 14:54:34 Fabrice Bellingard wrote:</h3>
<div class="paragraph">
<p>Implementation: http://jira.codehaus.org/browse/SONARJAVA-128</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_31_oct_2013_154823_freddy_mallet_wrote">on 31 Oct 2013, 15:48:23 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3226 for Flex</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_12_feb_2014_214850_freddy_mallet_wrote">on 12 Feb 2014, 21:48:50 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3469 for PHP</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_13_aug_2014_093324_linda_martin_wrote">on 13 Aug 2014, 09:33:24 Linda Martin wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARJS-134 for JavaScript</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_20_oct_2014_184731_ann_campbell_wrote">on 20 Oct 2014, 18:47:31 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~nicolas.peru] note the updated exception for Java.</p>
</div>
</div>
</div>
</div>