<div class="paragraph">
<p>Assigning default values to variables in an initialization section is a good practice as it increases the readability of the code: when starting reading a block, one will know exactly which variables have default values.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>DECLARE
   var1 integer;
   var2 integer;
   var3 integer;
BEGIN
   var1 := 1; -- Noncompliant
   var2 := 2; -- Noncompliant
   var3 := var1 + var2;
END;
/</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>DECLARE
   var1 integer := 1;
   var2 integer := 2;
   var3 integer;
BEGIN
   var3 := var1 + var2;
END;
/</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Move this devault value assignment in the initialization section</p>
</div>
</div>
</div>
</div>