<div class="paragraph">
<p>By default, the parameter mode is <code>IN</code>. However, specifying it explicitly makes the code easier to read.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>SET SERVEROUTPUT ON

DECLARE
  PROCEDURE printName(name VARCHAR2) AS -- Noncompliant; relies on default mode
  BEGIN
    DBMS_OUTPUT.PUT_LINE('name: ' || name);
  END;

BEGIN
  printName('Foo');
END;
/</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>SET SERVEROUTPUT ON

DECLARE
  PROCEDURE printName(name IN VARCHAR2) AS
  BEGIN
    DBMS_OUTPUT.PUT_LINE('name: ' || name);
  END;
BEGIN
  printName('Foo');
END;
/</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Specify this parmeter mode as "IN".</p>
</div>
</div>
</div>
</div>