<div class="paragraph">
<p>The use of <code>break</code> with a label is only needed to break out of multiple, nested control blocks, and this kind of usage adds further complication to already complex code, making it difficult to maintain. Instead the code should be restructured so that a label <code>break</code> isn&#8217;t required.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>outer: for (int i = 0; i &lt; outerLimit; i++) {
  for (int j = 0; j &lt; innerLimit) {
    if (i * j == 400) {
      break outer;  // Noncompliant
    }
    // ...
  }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>int j = 0;
for (int i = 0; i &lt; outerLimit &amp;&amp; i * j != 400; i++) {
  for (; j &lt; innerLimit) {
    // ...
  }
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_duplicates_s1119">duplicates: S1119</h3>

</div>
<div class="sect2">
<h3 id="_on_11_jun_2015_184031_ann_campbell_wrote">on 11 Jun 2015, 18:40:31 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>CodePro: Break with Label</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_16_jun_2015_131310_nicolas_peru_wrote">on 16 Jun 2015, 13:13:10 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I think this one is covered by LabelsShouldNotBeUsedCheck</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_16_jun_2015_134355_ann_campbell_wrote">on 16 Jun 2015, 13:43:55 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>thanks [~nicolas.peru]. Closing.</p>
</div>
</div>
</div>
</div>