<div class="paragraph">
<p><code>mktemp</code>, <code>mkstemp</code>, <code>mkstemps</code> or <code>mkdtemp</code> are useful functions to create temporary files and directories. They accept a <code>template</code> parameter which defines the path where the file should be created and which should contain at least six trailing "X"s to ensure the uniqueness of the filename.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>mkstemp("/tmp/file_XXXX"); // Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>mkstemp("/tmp/file_XXXXXX"); // Compliant</pre>
</div>
</div>
</div>
</div>