<div class="paragraph">
<p>For readability, the indentation of source code comments should match the indentation of the commented code.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>    " GAC -  13 June 13 - output user data
WRITE: / 'Firstname'.
" GAC -  13 June 13 - output user data
    WRITE: / 'Lasttname'.</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>" GAC -  13 June 13 - output user data
WRITE: / 'Firstname'.
    " GAC -  13 June 13 - output user data
    WRITE: / 'Lasttname'.</pre>
</div>
</div>
</div>
</div>