<div class="paragraph">
<p>Portability issues may restrict which characters should be used in an identifier.</p>
</div>
<div class="paragraph">
<p>This rule checks identifier names against a regular expression of disallowed characters. Due to a technical limitation, the COBOL analyzer is not able for the time-being to differentiate lowercase from uppercase characters.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="paragraph">
<p>With the default regular expression <code>[^a-zA-Z0-9-]</code>:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>MOVE DATA-1 TO DATA_2 *&gt; Noncompliant; '_' not allowed</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>MOVE DATA-1 TO DATA-2</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove the prohibited characters from the identifier "{&#8230;&#8203;}".</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="ulist">
<ul>
<li>
<p>key = regex</p>
</li>
<li>
<p>Description: Regular expression to be used to detect prohibited characters</p>
</li>
<li>
<p>Default value: [^a-zA-Z0-9-]</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_12_apr_2016_203711_ann_campbell_wrote">on 12 Apr 2016, 20:37:11 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~pierre-yves.nicolas] I just came across this rule and I&#8217;m wondering if this part is still true:</p>
</div>
<div class="quoteblock">
<blockquote>
<div class="paragraph">
<p>Due to a technical limitation, the COBOL analyzer is not able for the time-being to differentiate lowercase from uppercase characters.</p>
</div>
</blockquote>
</div>
</div>
<div class="sect2">
<h3 id="_on_14_apr_2016_095956_pierre_yves_nicolas_wrote">on 14 Apr 2016, 09:59:56 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] This limitation still exists.</p>
</div>
</div>
</div>
</div>