<div class="paragraph">
<p>In Swift, the semicolon (<code>;</code>) is optional as a statement separator, but omitting semicolons can be confusing.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>var x = 1</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>var x = 1;</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add a semicolon at the end of this statement.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s3964">is related to: S3964</h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s1472">is related to: S1472</h3>

</div>
<div class="sect2">
<h3 id="_on_30_oct_2013_155334_freddy_mallet_wrote">on 30 Oct 2013, 15:53:34 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3217 for Flex</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_3_nov_2013_090707_ann_campbell_wrote">on 3 Nov 2013, 09:07:07 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>I like the "what was the goal" part after the Non-compliant example, but should it be followed up with a Compliant solution that shows the two options, separated with "or"? Or perhaps just show the single choice with the added semicolon?</p>
</div>
</div>
</div>
</div>