<div class="paragraph">
<p>A non-static inner class has a reference to its outer class, and access to the outer class' fields and methods. That class reference makes the inner class larger and could cause the outer class instance to live in memory longer than necessary.</p>
</div>
<div class="paragraph">
<p>If the reference to the outer class isn&#8217;t used, it is more efficient to make the inner class <code>static</code> (also called nested). If the reference is used only in the class constructor, then explicitly pass a class reference to the constructor. If the inner class is anonymous, it will also be necessary to name it.</p>
</div>
<div class="paragraph">
<p>However, while a nested/<code>static</code> class would be more efficient, it&#8217;s worth noting that there are semantic differences between an inner class and a nested one:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>an inner class can only be instantiated within the context of an instance of the outer class.</p>
</li>
<li>
<p>a nested (<code>static</code>) class can be instantiated independently of the outer class.</p>
</li>
</ul>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public class Fruit {
  // ...

  public class Seed {  // Noncompliant; there's no use of the outer class reference so make it static
    int germinationDays = 0;
    public Seed(int germinationDays) {
      this.germinationDays = germinationDays;
    }
    public int getGerminationDays() {
      return germinationDays;
    }
  }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public class Fruit {
  // ...

  public static class Seed {
    int germinationDays = 0;
    public Seed(int germinationDays) {
      this.germinationDays = germinationDays;
    }
    public int getGerminationDays() {
      return germinationDays;
    }
  }
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Make this a [named] "static" inner class.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_6_oct_2015_193330_ann_campbell_wrote">on 6 Oct 2015, 19:33:30 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~nicolas.peru], <em><a href="http://www.securingjava.com/chapter-seven/chapter-seven-1.html">Securing Java</a></em> (see Rule 5) says that inner classes (presumably only non-<code>static</code>) are security holes because the compiler translates them to ordinary classes with <code>package</code> accessibility, and "upgrades" the owning&#8217;s class&#8217;s <code>private</code> member visibility to <code>package</code>.</p>
</div>
<div class="paragraph">
<p>The upshot is a recommendation against using inner classes. Since those problems go away if the inner class is <code>static</code>, I&#8217;m wondering whether to combine "Don&#8217;t use non-static inner classes" with this rule or handle it in a separate RSpec. WDYT?</p>
</div>
</div>
</div>
</div>