<div class="paragraph">
<p>Shared coding conventions allow teams to collaborate efficiently. This rule checks that all record field names match the provided regular rexpression.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="paragraph">
<p>With the default regular expression <code>[a-zA-Z](_?+[a-zA-Z0-9])*+</code>:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>DECLARE
  TYPE my_type IS RECORD(
    foo__bar PLS_INTEGER   -- Non-Compliant
  );
BEGIN
  NULL;
END;
/</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>DECLARE
  TYPE my_type IS RECORD(
    foo_bar PLS_INTEGER    -- Compliant
  );
BEGIN
  NULL;
END;
/</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Rename this record field to match the provided regular expression: xxx.</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="ulist">
<ul>
<li>
<p>key = regexp</p>
</li>
<li>
<p>description: The regular expression the field should match.</p>
</li>
<li>
<p>default: [a-zA-Z](_?[a-zA-Z0-9])*</p>
</li>
</ul>
</div>
</div>
</div>
</div>