<div class="paragraph">
<p>Developers should use the <code>FROM ... OUTER JOIN</code> syntax rather than the Oracle join operator (<code>+</code>). The reason is that outer join queries that use <code>+</code> are subject to several restrictions which do not apply to the <code>FROM ... OUTER JOIN</code> syntax. For instance, a <code>WHERE</code> condition containing the <code>+</code> operator cannot be combined with another condition using the <code>OR</code> logical operator.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>BEGIN
  -- Noncompliant
  SELECT *
    INTO employeesArray
    FROM employee, department
    WHERE employee.DepartmentID = department.ID(+);
END;
/</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>BEGIN
  -- Compliant
  SELECT *
    INTO employeesArray
    FROM employee LEFT OUTER JOIN department
    ON employee.DepartmentID = department.ID;
END;
/</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Refactor this SQL to use standard join syntax.</p>
</div>
</div>
</div>
</div>