<div class="paragraph">
<p>Python does not check returned types by default, however some methods are expected to return a specific type otherwise builtin functions will fail.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when a builtin function or method:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>does not return a value of the expected type.</p>
</li>
<li>
<p>returns a value when no value should be returned.</p>
</li>
<li>
<p>returns no value when a return value is expected.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>In such way that the resulting code would result in a runtime error.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class MyInt:
    def __init__(self):
        return self  # Noncompliant. __init__ should return nothing. This will raise a TypeError.

    def __int__(self):
        return 3.0  # Noncompliant. __int__ should always return an integer

int(MyInt())  # This will fail with "TypeError: __int__ returned non-int (type float)"</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class MyInt:
    def __int__(self):
        return 3

int(MyInt())</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.python.org/3/reference/datamodel.html#special-method-names">Python documentation - Special method names</a></p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>Return a value of type "XXX" instead of "YYY", as should any "ZZZ" method.</p>
</li>
<li>
<p>Return "None" instead of a value of type "XXX", as should any "ZZZ" method.</p>
</li>
<li>
<p>Return a value of type "XXX"; "ZZZ" may reach its end and return None.</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>if the returned type is wrong or nothing should be returned.</p>
<div class="ulist">
<ul>
<li>
<p>Primary: the <code>return</code> statement.</p>
</li>
<li>
<p>Secondary: The function/method name.</p>
</li>
</ul>
</div>
</li>
<li>
<p>if no value is returned and one is expected.</p>
<div class="ulist">
<ul>
<li>
<p>Primary: the function/method name</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s3801">is related to: S3801</h3>

</div>
<div class="sect2">
<h3 id="_supercedes_s2734">supercedes: S2734</h3>

</div>
<div class="sect2">
<h3 id="_on_10_dec_2014_233745_evgeny_mandrikov_wrote">on 10 Dec 2014, 23:37:45 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I think that we can safely add link to <a href="https://www.securecoding.cert.org/confluence/x/ZoCKC">CERT MSC36-CPP</a> ? WDYT?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_11_dec_2014_124347_ann_campbell_wrote">on 11 Dec 2014, 12:43:47 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Agreed, [~evgeny.mandrikov]. Done</p>
</div>
</div>
</div>
</div>