<div class="paragraph">
<p><code>C&#43;&#43;17</code> introduced structured binding, a syntax that initializes multiple entities by elements or members of an object. It is especially useful to emulate multiple return values from a function.</p>
</div>
<div class="paragraph">
<p>Suppose you have a function that returns a pair:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>std::pair&lt;std::string, std::string&gt; getPair();</pre>
</div>
</div>
<div class="paragraph">
<p>Structured binding allows you to bind the members of the <code>std::pair</code> class directly to new names:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>auto [firstName, lastName] = getPair();</pre>
</div>
</div>
<div class="paragraph">
<p>The names <code>firstName</code> and <code>lastName</code> are called structured bindings. As you can see, structured binding makes the code more readable as they allow binding values to names that carry information about their purpose.</p>
</div>
<div class="paragraph">
<p>Structured binding works with:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Raw arrays, by binding a name to each element</p>
</li>
<li>
<p>Any type that has a tuple-like API</p>
</li>
<li>
<p>Classes and structures where all non-static data member are publicly accessible</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>This rule will detect places where <code>std::pair</code> and <code>std::tuple</code> can be effortlessly replaced by a structured binding.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>void printingMap(const std::map&lt;int, std::string&gt;&amp; map) {
  for (const auto&amp; elem : map) { // Noncompliant
    std::cout &lt;&lt; elem.first &lt;&lt; ": " &lt;&lt; elem.second &lt;&lt; "\n";
  }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>void printingMap(const std::map&lt;int, std::string&gt;&amp; map) {
  for (const auto&amp; [key, value] : map) { // Compliant
    std::cout &lt;&lt; key &lt;&lt; ": " &lt;&lt; value &lt;&lt; "\n";
  }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#f21-to-return-multiple-out-values-prefer-returning-a-struct-or-tuple">C&#43;&#43; Core Guidelines F.21</a> - To return multiple “out” values, prefer returning a struct or tuple</p>
</li>
</ul>
</div>
</div>
</div>