<div class="paragraph">
<p>Non-static variables initialized with <code>INZ()</code> are only reliably initialized on the first run through the code in a  "terminate and stay resident" (TSR) program. Subsequent calls to the program within the same job do not re-initialize the variable with the value from <code>INZ()</code> unless the last record indicator is set at the end of the program.</p>
</div>
<div class="paragraph">
<p>Without setting the last record indicator, the assumptions the code makes about the variable&#8217;s initial value will be wrong every time but one, potentially leading to bad program behavior.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="paragraph">
<p>{code}     D TESTINZ         PR                  extpgm('TESTINZ')</p>
</div>
<div class="literalblock">
<div class="content">
<pre>D  Parm1                        15A</pre>
</div>
</div>
<div class="literalblock">
<div class="content">
<pre>D  Parm2                        15A   CONST</pre>
</div>
</div>
<div class="literalblock">
<div class="content">
<pre>DMUTVAR           S             15A   INZ('ABC')  // Noncompliant; *inlr not set</pre>
</div>
</div>
<div class="literalblock">
<div class="content">
<pre>D TESTINZ         PI</pre>
</div>
</div>
<div class="literalblock">
<div class="content">
<pre>D  Parm1                        15A</pre>
</div>
</div>
<div class="literalblock">
<div class="content">
<pre>D  Parm2                        15A   CONST</pre>
</div>
</div>
<div class="literalblock">
<div class="content">
<pre>/Free</pre>
</div>
</div>
<div class="literalblock">
<div class="content">
<pre>IF %PARMS &gt; 1 and Parm2 &lt;&gt; '';</pre>
</div>
</div>
<div class="literalblock">
<div class="content">
<pre>MUTVAR=Parm2;</pre>
</div>
</div>
<div class="literalblock">
<div class="content">
<pre>ENDIF;</pre>
</div>
</div>
<div class="literalblock">
<div class="content">
<pre>DSPLY(E) ('MUTVAR:' + MUTVAR);</pre>
</div>
</div>
<div class="literalblock">
<div class="content">
<pre>Parm1=MUTVAR;</pre>
</div>
</div>
<div class="literalblock">
<div class="content">
<pre>return;</pre>
</div>
</div>
<div class="literalblock">
<div class="content">
<pre>/End-free</pre>
</div>
</div>
<div class="listingblock">
<div class="content">
<pre></pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>     D TESTINZ         PR                  extpgm('TESTINZ')
     D  Parm1                        15A
     D  Parm2                        15A   CONST

     DMUTVAR           S             15A   INZ('ABC')

     D TESTINZ         PI
     D  Parm1                        15A
     D  Parm2                        15A   CONST
      /Free
       IF %PARMS &gt; 1 and Parm2 &lt;&gt; '';
         MUTVAR=Parm2;
       ENDIF;
       DSPLY(E) ('MUTVAR:' + MUTVAR);
       Parm1=MUTVAR;
       *inlr = *on;
       return;
       /End-free</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this use of "Inz()", and set "xxx"'s default value at the beginning of the program.</p>
</div>
</div>
</div>
</div>