<div class="paragraph">
<p>Swing interfaces should be constructed and shown from the Swing event dispatch thread. Doing so from any other thread, such as from <code>main</code> risks deadlocks since you run the risk of multiple threads accessing things which are inherently not thread-safe.</p>
</div>
<div class="paragraph">
<p>Instead, use <code>SwingUtilities.invokeLater</code> or <code>SwingUtilities.invokeAndWait</code> to kick off a <code>new Runnable</code> that handles your GUI creation.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public static void main(String args[]) {
  makeGui(); // Noncompliant
}

public void makeGui() {
  JFrame frame = new JFrame();
  // ...
  frame.show();
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public static void main(String args[]) {
  SwingUtilities.invokeAndWait(new Runnable() {
    public void run() {
      makeGui();
    }
  }
}

public void makeGui() {
  JFrame frame = new JFrame();
  // ...
  frame.show();
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Move this "xxx" call to the event dispatch thread.</p>
</div>
</div>
</div>
</div>