<div class="paragraph">
<p>Any class that doesn&#8217;t directly extend <code>Object</code> should invoke <code>super</code> in the constructor that performs initialization to ensure that all parts of the class are fully initialized.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public class Human extends Animal {
  private Date dateOfBirth;

  public Human(Gender gender) {  // Compliant; doesn't perform initialization directly
    this(gender, new Date());
  }

  public Human (Gender gender, Date dob) {  // Noncompliant
    this.dateOfBirth = dob;
  }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public class Human extends Animal {
  private Date dateOfBirth;

  public Human(Gender gender) {
    this(gender, new Date());
  }

  public Human (Gender gender, Date dob) {
    super(gender);
    this.dateOfBirth = dob;
  }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add a "super" call to this constructor.</p>
</div>
</div>
</div>
</div>