<div class="paragraph">
<p>The conventional expectation of operators that end with <code>=</code>, such as <code>+=</code>, <code>-=</code>, <code>*=</code>, and so on, is that the result of the operation will be assigned to the operand on the left-hand side of the operator.</p>
</div>
<div class="paragraph">
<p>Define any other behavior and you almost guarantee that the users of your code will misunderstand and therefore misuse your operator.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>func **= (p1:Int, p2:Int) -&gt; Int {   // Noncompliant. Change operator name or update value of first parameter
    return p1 ** p2
}

func =&gt; (p1:Int, p2:Int) -&gt; Int {  // Compliant; doesn't end with '='
    return p1 ** p1 ** p2
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>func **= (inout p1:Int, p2:Int) {
    p1 = p1 ** p2
}

func =&gt; (p1:Int, p2:Int) -&gt; Int {
    return p1 ** p1 ** p2
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Assign the value of the operation to "xxx".</p>
</div>
</div>
</div>
</div>