<div class="paragraph">
<p>An unordered pair is easy to handle, but the longer an unordered set gets, the harder it is to deal with.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when a tuple of more than the allowed number of elements is returned.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="paragraph">
<p>With the default threshold of 2</p>
</div>
<div class="listingblock">
<div class="content">
<pre>func doTheThing() -&gt; (Int, Int, Int) {}  // Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>func doTheThing() -&gt; MyIntStruct {}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Reduce the size of this tuple or convert it to a named structure (class or struct).</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="ulist">
<ul>
<li>
<p>key = threshold</p>
</li>
<li>
<p>description: The maximum number of items allowed in a threshold</p>
</li>
<li>
<p>default value: 2</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p><code>(...)</code></p>
</div>
</div>
</div>
</div>