<div class="paragraph">
<p>The <code>nodiscard</code> attribute can be used with or without explanations, but explaining why a result should not be discarded can only improve oneâ€™s understanding of the code, and would prevent developers from wasting time figuring those things out by themselves.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when <code>nodiscard</code> is used on function without any explanation.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>[[nodiscard]] std::vector&lt;int&gt; generateRandomValues(int count); // Noncompliant

generateRandomValues(100);</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>[[nodiscard("Computation of values is expensive")]] std::vector&lt;int&gt; generateRandomValues(int count);

generateRandomValues(100);</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s6222">is related to: S6222</h3>

</div>
</div>
</div>