<div class="paragraph">
<p>Labeled loops are useful, especially when the code is badly indented, to match the begin and end of each loop. Within a labeled loop, the code&#8217;s maintainability is increased by explicitly providing the loop&#8217;s label in every <code>EXIT</code> statement. Indeed, if a nested loop is added afterwards, it is clear which loop has to be exited.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>BEGIN
  &lt;&lt;myLoopLabel1&gt;&gt;
  LOOP
    EXIT; -- Noncompliant, the loop label is missing
  END LOOP myLoopLabel1;

  LOOP
    EXIT; -- Compliant, this EXIT is not in a labeled loop
  END LOOP;
END;
/</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>BEGIN
 &lt;&lt;myLoopLabel1&gt;&gt;
  LOOP
    EXIT myLoopLabel1;
  END LOOP myLoopLabel1;

  LOOP
    EXIT;
  END LOOP;
END;
/</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Label this "EXIT" statement.</p>
</div>
</div>
</div>
</div>