<div class="paragraph">
<p>When a function parameter has a nullable type, e.g., parameter <code>param</code> in <code>f(?int param)</code>, it must be explicitly provided in every function call. A nullable-type parameter has no default value.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>function f(?int param) {}
f();</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>function f(?int param) {}
f(0);</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_2_nov_2016_133150_ann_campbell_wrote">on 2 Nov 2016, 13:31:50 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~yves.duboispelerin] what "bad thing" will happen if you violate this rule? We need to show the justification for making this a Bug, and also need to know that when evaluating severity.</p>
</div>
</div>
</div>
</div>