<div class="paragraph">
<p>This includes file and I/O functions <code>fgetpos</code>, <code>fopen</code>, <code>ftell</code>, <code>gets</code>, <code>perror</code>, <code>remove</code>, <code>rename</code>, etc.</p>
</div>
<div class="paragraph">
<p>Streams and file I/O have a large number of unspecified, undefined and implementation-defined behaviors associated with them.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>#include &lt;cstdio&gt;  /* Noncompliant */

void fn()
{
  char_t array[10];
  gets(array);           /* Can lead to buffer over-run */
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C&#43;&#43;:2008, 27-0-1 - The stream input/output library <em>&lt;cstdio&gt;</em> shall not be used.</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s988">relates to: S988</h3>

</div>
<div class="sect2">
<h3 id="_on_31_mar_2015_190332_evgeny_mandrikov_wrote">on 31 Mar 2015, 19:03:32 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] implementation seems more complete (SQALE, description) than this spec.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_13_apr_2015_194505_evgeny_mandrikov_wrote">on 13 Apr 2015, 19:45:05 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I&#8217;m wondering why blocker, but not active by default? Note that in implementation currently major and not active.</p>
</div>
</div>
</div>
</div>