<div class="paragraph">
<p>An incorrectly defined platform invoke method (marked with <code>System.Runtime.InteropServices.DllImportAttribute</code>) can lead to runtime exceptions because of issues such as a misnamed function, faulty mapping of parameter and return value data types, and incorrect field specifications. If available, it is a lot less error prone, not to mention more secure and quite simpler, to call the equivalent managed method.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when a platform invoke method is called and a method with the equivalent functionality exists in the .Net framework library.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>using System;
using System.Runtime.InteropServices;
using System.Text;

namespace MyLibrary
{
   public class Foo
   {
      public Foo() {}

      [DllImport("kernel32.dll", CharSet = CharSet.Unicode, SetLastError = true)]
      internal static extern int ExpandEnvironmentStrings(
         string lpSrc, StringBuilder lpDst, int nSize);

      public void Bar()
      {
         string environmentVariable = "%TEMP%";
         StringBuilder expandedVariable = new StringBuilder(100);

         ExpandEnvironmentStrings( // Noncompliant
            environmentVariable,
            expandedVariable,
            expandedVariable.Capacity);
      }
   }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>using System;
using System.Text;

namespace MyLibrary
{
   public class Foo
   {
      public Foo() {}

      public void Bar()
      {
         string environmentVariable = "%TEMP%";
         StringBuilder expandedVariable = new StringBuilder(100);

         Environment.ExpandEnvironmentVariables(environmentVariable);
      }
   }
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace the call to 'xxx' to its managed equivalent.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The unmanaged Win32 function call.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_9_jan_2018_122011_amaury_levé_wrote">on 9 Jan 2018, 12:20:11 Amaury Levé wrote:</h3>
<div class="paragraph">
<p>We need to define the list of methods with their equivalents.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_9_jan_2018_153023_amaury_levé_wrote">on 9 Jan 2018, 15:30:23 Amaury Levé wrote:</h3>
<div class="paragraph">
<p>The following list was suggested by [~valeri.hristov] : https://msdn.microsoft.com/en-us/library/aa302340.aspx?f=255&amp;MSPPError=-2147217396#win32map_memorymanagementfunctions</p>
</div>
</div>
</div>
</div>