<div class="paragraph">
<p>An invalid color definition will by default be interpreted as black, which is likely to have unintended impacts on the expected look and feel of the website.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when a color definition (<code>color</code>, <code>background-color</code>) is not valid. The color definition is considered valid when it is made of hexadecimal characters:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>longhand: 6 or 8 characters (when alpha is defined)</p>
</li>
<li>
<p>shorthand variant: 3 or 4 characters (when alpha is defined)</p>
</li>
</ul>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>a {
 color: #3c; /* Noncompliant; shorthand should be made of 3 characters */
}
div {
  background-color: #3cb371a; /* Noncompliant; alpha should have 2 characters */
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>a {
 color: #3cc;
}
div {
  background-color: #3cb371ac;
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://developer.mozilla.org/en-US/docs/Web/CSS/color_value">Mozilla Web Technology for Developers</a> - CSS Color</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_4_jun_2018_092819_ann_campbell_wrote">on 4 Jun 2018, 09:28:19 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~alexandre.gigleux], there&#8217;s no "why" here.</p>
</div>
</div>
</div>
</div>