<div class="paragraph">
<p>There no point in using the <code>DISTINCT</code> qualifier on columns with unique values; there can be no duplicates, so there&#8217;s nothing to eliminate. Further, using this keyword redundantly can slow performance because many optimizers don&#8217;t remove it, so an unnecessary de-duplication step is run anyway.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when <code>DISTINCT</code> is applied to non-compound key columns, and to columns with <code>UNIQUE</code> constraints.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>SELECT DISTINCT id, -- Noncompliant
  name
FROM employee
WHERE ...</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>SELECT id,
  name
FROM employee
WHERE ...</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this redundant "DISTINCT" qualifier.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p><code>DISTINCT</code></p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_16_jun_2016_145137_ann_campbell_wrote">on 16 Jun 2016, 14:51:37 Ann Campbell wrote:</h3>
<div class="paragraph">
<p><a href="https://www.researchgate.net/publication/222432151_Semantic_errors_in_SQL_queries_A_quite_complete_list" class="bare">https://www.researchgate.net/publication/222432151_Semantic_errors_in_SQL_queries_A_quite_complete_list</a></p>
</div>
</div>
</div>
</div>