<div class="paragraph">
<p>Initializing a variable to zero or <code>nil</code> in an <code>init</code> method is completely redundant; the compiler takes care of that for you. Therefore initializing class instance variables to 0 or <code>nil</code> is simply wasted keystrokes.</p>
</div>
<div class="paragraph">
<p>This rule applies to methods that return an <code>id</code> and have names that begin with "init".</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>- (id)initWithBlah:(NSString *)blah {
    self = [super init];
    if (self) {
      _count = 0;  // Noncompliant;
      _blah = [blah copy];
    }
    return self;
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>- (id)initWithBlah:(NSString *)blah {
    self = [super init];
    if (self) {
      _blah = [blah copy];
    }
    return self;
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this redundant initialization.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_19_sep_2014_113742_freddy_mallet_wrote">on 19 Sep 2014, 11:37:42 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>@Ann, I would reduce the default severity to minor.</p>
</div>
</div>
</div>
</div>