<div class="paragraph">
<p>In general, <code>async</code> <code>void</code> test methods are not executed by test frameworks, therefore it&#8217;s better to avoid them altogether.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>    [TestMethod]
    public async void MyIgnoredTestMethod()  // Noncompliant
    { /* ... */ }</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_1_dec_2015_143237_tamas_vajk_wrote">on 1 Dec 2015, 14:32:37 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I changed the description to not mention <code>[TestMethod]</code> attribute, because that is framework specific. In NUnit, xUnit, and MS test frameworks there are different attributes for marking test methods.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_1_dec_2015_150204_ann_campbell_wrote">on 1 Dec 2015, 15:02:04 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Thanks [~tamas.vajk]</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_31_jul_2017_160705_jean_christophe_collet_wrote">on 31 Jul 2017, 16:07:05 Jean-Christophe Collet wrote:</h3>
<div class="paragraph">
<p>Merged with RSPEC-3433</p>
</div>
</div>
</div>
</div>