<div class="paragraph">
<p>Throwing an exception from within a finally block will mask any exception which was previously thrown in the <code>try</code> or <code>catch</code> block, and the masked&#8217;s exception message and stack trace will be lost.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>try
{
  /* some work which end up throwing an exception */
  throw new ArgumentException();
}
finally
{
  /* clean up */
  throw new InvalidOperationException();       // Noncompliant; will mask the ArgumentException
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>try
{
  /* some work which end up throwing an exception */
  throw new ArgumentException();
}
finally
{
  /* clean up */                       // Compliant
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Refactor this code to not throw exceptions in finally blocks.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_29_jul_2013_235740_ann_campbell_wrote">on 29 Jul 2013, 23:57:40 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>There&#8217;s no compliant example. Is that okay?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_30_jul_2013_064149_freddy_mallet_wrote">on 30 Jul 2013, 06:41:49 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Indeed Ann, that&#8217;s fine to not have a compliant example.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_2_aug_2013_144930_dinesh_bolkensteyn_wrote">on 2 Aug 2013, 14:49:30 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>Implemented by https://jira.codehaus.org/browse/SONARJAVA-251</p>
</div>
</div>
</div>
</div>