<div class="paragraph">
<p>In programming languages keywords have a special meaning and are reserved to the language. It is hence a bad idea to define macros with keywords as macro identifier as it can easily lead to undefined behavior:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>The same object might be defined differently in different places, which violates the One Definition Rule</p>
</li>
<li>
<p>If you include any header from the standard library, it is undefined behavior to define such macros</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Additionally, it is very awkward for anyone reading the code to have a keyword that means something different.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>#define int some_other_type // Noncompliant
#include &lt;stdlib.h&gt;;</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C:2012, 20.4 -Â A macro shall not be defined with the same name as a keyword</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this keyword redefinition.</p>
</div>
</div>
</div>
</div>