<div class="paragraph">
<p>Shared naming conventions allow teams to collaborate efficiently. This rule checks that all coroutine names match a provided regular expression.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="paragraph">
<p>With default provided regular expression: <code>^[a-z][a-zA-Z0-9]*$</code>:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>generator&lt;int&gt; Iota(int n = 0) {
  while(true)
    co_yield n++;
}{code}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>generator&lt;int&gt; iota(int n = 0) {
  while(true)
    co_yield n++;
}{code}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Rename coroutine "XXXX" to match the regular expression ${format}.</p>
</div>
</div>
</div>
</div>