<div class="paragraph">
<p>A class with only <code>private</code> constructors can&#8217;t be instantiated, thus, it seems to be pointless code.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public class MyClass // Noncompliant
{
  private MyClass() { ... }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public class MyClass
{
  public MyClass() { ... }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_exceptions">Exceptions</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Classes that themselves access their private constructors (singletons or smart enums) are ignored. Classes with only <code>static</code> members are also ignored because they are covered by Rule S1118.</p>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>This class can&#8217;t be instantiated; make [at least one of its constructors|its constructor] "public".</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>class name</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_8_dec_2015_172353_tamas_vajk_wrote">on 8 Dec 2015, 17:23:53 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Shouldn&#8217;t we add a reference to RSPEC-1118 as the other rule?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_8_dec_2015_175656_ann_campbell_wrote">on 8 Dec 2015, 17:56:56 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>done [~tamas.vajk]</p>
</div>
</div>
</div>
</div>