<div class="paragraph">
<p>Since Java 9, <code>@Deprecated</code> has two additional arguments to the annotation:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>since</code> allows you to describe when the deprecation took place</p>
</li>
<li>
<p><code>forRemoval</code>, indicates whether the deprecated element will be removed at some future date</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>In order to ease the maintainers work, it is recommended to always add one or both of these arguments.</p>
</div>
<div class="paragraph">
<p>This rule reports an issue when <code>@Deprecated</code> is used without any argument.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>@Deprecated</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>@Deprecated(since="4.2", forRemoval=true)</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_exceptions">Exceptions</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The members and methods of a deprecated class or interface are ignored by this rule. The classes and interfaces themselves are still subject to it.</p>
</div>
<div class="sect2">
<h3 id="_see_also">See Also</h3>
<div class="ulist">
<ul>
<li>
<p>S1123</p>
</li>
</ul>
</div>
</div>
</div>
</div>