<div class="paragraph">
<p>Comparing dissimilar types using the strict equality operators <code>===</code> and <code>!==</code> will always return the same value, respectively <code>false</code> and <code>true</code>, because no type conversion is done before the comparison. Thus, such comparisons can only be bugs.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>var a = 8;
var b = "8";

if (a === b) {  // Noncompliant; always false
  // ...
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>var a = 8;
var b = "8";

if (a == b) {
  // ...
}</pre>
</div>
</div>
<div class="paragraph">
<p>or</p>
</div>
<div class="listingblock">
<div class="content">
<pre>var a = 8;
var b = "8";

if (a === Number(b)) {
  // ...
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this ["==="|"!=="] check; it will always be [false|true]. Did you mean to use ["=="|"!="]?</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>Primary: identity operator</p>
</li>
<li>
<p>Secondary: both operands</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_20_nov_2015_172431_elena_vilchik_wrote">on 20 Nov 2015, 17:24:31 Elena Vilchik wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Thanks!</p>
</div>
</div>
</div>
</div>