<div class="paragraph">
<p>Operators <a href="https://docs.python.org/3/reference/expressions.html#is-not"><code>is</code></a> and <a href="https://docs.python.org/3/reference/expressions.html#is-not"><code>is not</code></a> check if their operands point to the same instance, thus they will always return respectively True and False when they are used to compare objects of different type. Such comparisons can only be bugs.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>myint = 1
mystring = "1"
value = myint is mystring  # Noncompliant. Always False
value = myint is not mystring  # Noncompliant. Always True</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this ["is"|"is not"] check between types X and Y; it will always be [False|True].</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>Primary: identity operator</p>
</li>
<li>
<p>Secondary: both operands</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_20_nov_2015_172431_elena_vilchik_wrote">on 20 Nov 2015, 17:24:31 Elena Vilchik wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Thanks!</p>
</div>
</div>
</div>
</div>