<div class="paragraph">
<p>A test case without assertions ensures only that no exceptions are thrown. Beyond basic runnability, it ensures nothing about the behavior of the code under test.</p>
</div>
<div class="paragraph">
<p>This rule raises an exception when no assertions from any of the following frameworks are found in a test:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>MSTest</code></p>
</li>
<li>
<p><code>NUnit</code></p>
</li>
<li>
<p><code>xUnit</code></p>
</li>
<li>
<p><code>FluentAssertions</code> (4.x and 5.x)</p>
</li>
<li>
<p><code>NFluent</code></p>
</li>
<li>
<p><code>NSubstitute</code></p>
</li>
<li>
<p><code>Shoudly</code></p>
</li>
</ul>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>[TestMethod]
public void MyMethod_WhenSomething_ExpectsSomething()
{
    var myClass = new Class();
    var result = myClass.GetFoo();
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>[TestMethod]
public void MyMethod_WhenSomething_ExpectsSomething()
{
    var myClass = new Class();
    var result = myClass.GetFoo();
    Assert.IsTrue(result);
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_exceptions">Exceptions</h2>
<div class="sectionbody">
<div class="paragraph">
<p>To create a custom assertion method declare an attribute with name <code>AssertionMethodAttribute</code> and mark the method with it:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>using System;
using Microsoft.VisualStudio.TestTools.UnitTesting;

[TestClass]
public class CustomTest
{
    [TestMethod]
    public void TestMethod1() =&gt; Validator.CustomMethod(42);    // Compliant
}

public static class Validator
{
    [AssertionMethod]
    public static void CustomMethod(int value) { }
}

public class AssertionMethodAttribute : Attribute { }</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add at least one assertion to this test case.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_2_may_2019_144407_andrei_epure_wrote">on 2 May 2019, 14:44:07 Andrei Epure wrote:</h3>
<div class="paragraph">
<p>This is implemented in <a href="https://github.com/SonarSource/sonar-dotnet/blob/7.13.0.8313/sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/TestMethodShouldContainAssertion.cs">TestMethodShouldContainAssertion </a></p>
</div>
</div>
<div class="sect2">
<h3 id="_on_9_apr_2015_074738_nicolas_peru_wrote">on 9 Apr 2015, 07:47:38 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I would maybe include a junit3 example in this rule. Otherwise looks fine.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_9_apr_2015_124716_ann_campbell_wrote">on 9 Apr 2015, 12:47:16 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Per our discussion [~nicolas.peru] I&#8217;m going to skip an explicit JUnit 3 example</p>
</div>
</div>
</div>
</div>