<div class="paragraph">
<p>A test case without assertions ensures only that no exceptions are thrown. Beyond basic runnability, it ensures nothing about the behavior of the code under test.</p>
</div>
<div class="paragraph">
<p>This rule raises an exception when the assertion library <code>chai</code> is imported but no assertion is used in a test.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>const expect = require('chai').expect;

describe("No assertion", function() {
    it("doesn't test anything", function() { // Noncompliant
        const str = "";
    });
});</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>const expect = require('chai').expect;

describe("Has assertions", function() {
    it("tests a string", function() {
        const str = "";
        expect(str).to.be.a('string');
    });
});</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add at least one assertion to this test case.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_9_apr_2015_074738_nicolas_peru_wrote">on 9 Apr 2015, 07:47:38 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I would maybe include a junit3 example in this rule. Otherwise looks fine.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_9_apr_2015_124716_ann_campbell_wrote">on 9 Apr 2015, 12:47:16 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Per our discussion [~nicolas.peru] I&#8217;m going to skip an explicit JUnit 3 example</p>
</div>
</div>
</div>
</div>