<div class="paragraph">
<p>According to the Java Language Specification:</p>
</div>
<div class="quoteblock">
<blockquote>
<div class="paragraph">
<p>Unnamed packages are provided by the Java platform principally for convenience when developing small or temporary applications or when just beginning development.</p>
</div>
</blockquote>
</div>
<div class="paragraph">
<p>To enforce this best practice, classes located in default package can no longer be accessed from named ones since Java 1.4.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public class MyClass { /* ... */ }</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>package org.example;

public class MyClass{ /* ... */ }</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Move this file to a named package.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_20_aug_2013_200941_freddy_mallet_wrote">on 20 Aug 2013, 20:09:41 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARJAVA-310</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_16_nov_2015_114010_michael_gumowski_wrote">on 16 Nov 2015, 11:40:10 Michael Gumowski wrote:</h3>
<div class="paragraph">
<p>I removed reference to PMD rule "DefaultPackage", which is not describing the same case (forbid usage of package visibility for members, see RSPEC-1784 and RSPEC-2072)</p>
</div>
</div>
</div>
</div>