<div class="paragraph">
<p>A results indicator should be used on a <code>CHAIN</code> statement because it leads to cleaner, more readable, and more correct code. Using an indicator results in the conditions <code>record not found</code>, and <code>end of file</code> being automatically handled correctly, without the need for further error handling.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>C     K0GMCP        CHAIN     SSGMCPR</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>C     K0GMCP        CHAIN     SSGMCPR                            81</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_exceptions">Exceptions</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Free-form <code>CHAIN</code> statements cannot have indicators, and are ignored.</p>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add an indicator to this <code>CHAIN</code> statement</p>
</div>
</div>
</div>
</div>