<div class="paragraph">
<p>Exceptions are meant to represent the application&#8217;s state at the point at which an error occurred.</p>
</div>
<div class="paragraph">
<p>Making all fields in an <code>Exception</code> class <code>final</code> ensures that this state:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Will be fully defined at the same time the <code>Exception</code> is instantiated.</p>
</li>
<li>
<p>Won&#8217;t be updated or corrupted by a questionable error handler.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>This will enable developers to quickly understand what went wrong.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public class MyException extends Exception {

  private int status;                               // Noncompliant

  public MyException(String message) {
    super(message);
  }

  public int getStatus() {
    return status;
  }

  public void setStatus(int status) {
    this.status = status;
  }

}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public class MyException extends Exception {

  private final int status;

  public MyException(String message, int status) {
    super(message);
    this.status = status;
  }

  public int getStatus() {
    return status;
  }

}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Make this field "xxxx" final.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_30_jul_2013_165545_freddy_mallet_wrote">on 30 Jul 2013, 16:55:45 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARJAVA-254</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_14_mar_2017_114035_amaury_levé_wrote">on 14 Mar 2017, 11:40:35 Amaury Levé wrote:</h3>
<div class="paragraph">
<p>\[~valeri.hristov] This seems to be a nice rule to implement for C#. WDYT?</p>
</div>
</div>
</div>
</div>