<div class="paragraph">
<p>Declaring a variable only to immediately return or throw it is a bad practice.</p>
</div>
<div class="paragraph">
<p>Some developers argue that the practice improves code readability, because it enables them to explicitly name what is being returned. However, this variable is an internal implementation detail that is not exposed to the callers of the method. The method name should be sufficient for callers to know exactly what will be returned.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public long computeDurationInMilliseconds() {
  long duration = (((hours * 60) + minutes) * 60 + seconds ) * 1000 ;
  return duration;
}

public void doSomething() {
  RuntimeException myException = new RuntimeException();
  throw myException;
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public long computeDurationInMilliseconds() {
  return (((hours * 60) + minutes) * 60 + seconds ) * 1000 ;
}

public void doSomething() {
  throw new RuntimeException();
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Immediately {return|throw} this expression instead of assigning it to the temporary variable "{&#8230;&#8203;}".</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_20_dec_2013_135702_dinesh_bolkensteyn_wrote">on 20 Dec 2013, 13:57:02 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARJAVA-417</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_4_jan_2014_172221_ann_campbell_wrote">on 4 Jan 2014, 17:22:21 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Sorry for the delay on this one, [~dinesh.bolkensteyn]. Somehow I missed the assignment notification.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_7_jan_2014_102602_freddy_mallet_wrote">on 7 Jan 2014, 10:26:02 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3359 in JavaScript</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_12_feb_2014_205441_freddy_mallet_wrote">on 12 Feb 2014, 20:54:41 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3459 for PHP</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_3_mar_2014_115044_dinesh_bolkensteyn_wrote">on 3 Mar 2014, 11:50:44 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] IMO we should follow the commonly accepted standards of each language in our examples. So in .NET languages, curly braces start on a new line, and method names start with an upper case letter.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_3_mar_2014_125624_ann_campbell_wrote">on 3 Mar 2014, 12:56:24 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>I agree [~dinesh.bolkensteyn] &amp; transgressed out of ignorance. But I think you&#8217;ll find people to argue vehemently on either side of the curly brace question for any given language&#8230;&#8203;</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_25_apr_2014_132559_ann_campbell_wrote">on 25 Apr 2014, 13:25:59 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>This rule is not appropriate for VB6.</p>
</div>
</div>
</div>
</div>