<div class="paragraph">
<p>These are the only forms for the <code>#include</code> directive permitted by the standard. The behavior is undefined when other forms are used.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>#include filename.h        // Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>#include "filename.h"        // Compliant
#include &lt;filename.h&gt;

#define HEADER "filename.h"
#include HEADER</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C:2004, 19.3 - The #include directive shall be followed by either a &lt;filename&gt; or "filename" sequence.</p>
</li>
<li>
<p>MISRA C&#43;&#43;:2008, 16-2-6 - The #include directive shall be followed by either a &lt;filename&gt; or "filename" sequence.</p>
</li>
<li>
<p>MISRA C:2012, 20.3 - The #include directive shall be followed by either a &lt;filename&gt; or "filename" sequence</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_6_apr_2015_120715_evgeny_mandrikov_wrote">on 6 Apr 2015, 12:07:15 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] implementation seems more complete (SQALE, description) than this spec.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_13_apr_2015_194546_evgeny_mandrikov_wrote">on 13 Apr 2015, 19:45:46 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I&#8217;m wondering why blocker, but not active by default? Note that in implementation currently major and active.</p>
</div>
</div>
</div>
</div>