<div class="paragraph">
<p>Iterators are useful to deal with data inside a container: they point to one of its element and can be incremented or decremented to access other elements of this container. However, it will be undefined behavior if they  try to access data out of bounds:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>You should not try to get an iterator which would be before <code>begin()</code></p>
</li>
<li>
<p>You should not try to get an iterator which would be after <code>end()</code></p>
</li>
<li>
<p>You should not try to dereference <code>end()</code>: ranges are semi open, which means that <code>begin()</code> is the location of the first element, but <code>end()</code> is a location past-the-end of the container, and does not correspond to any data.</p>
</li>
</ul>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>void test(std::vector&lt;int&gt; const &amp;v) {
  auto iter = v.end();
  auto x = *iter; // Noncompliant; access out of range
  // ...
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>iterator accessed outside of its range</p>
</div>
</div>
</div>
</div>