<div class="paragraph">
<p>Using trailing closure syntax for the last parameter in a call is often the most elegant way to handle it. But if the call requires multiple function-type arguments, the use of a trailing closure can be messy and confusing. In such cases, it&#8217;s better to pass closure expressions as normal arguments.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>var x = complexOperation(
  arg: 2,
  op1: {$0 + 10}
) {$0 * $0}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>var x = complexOperation(
  arg: 2,
  op1: {$0 + 10},
  op2: {$0 * $0}
)</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Move this trailing closure inside/outside of the function call.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_21_may_2015_075739_elena_vilchik_wrote">on 21 May 2015, 07:57:39 Elena Vilchik wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] you understand the rule in a wrong way. The thing is that you can use trailing syntax only for the last parameter.</p>
</div>
<div class="paragraph">
<p>I changed description, could you verify it?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_21_may_2015_135757_ann_campbell_wrote">on 21 May 2015, 13:57:57 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Back to you [~elena.vilchik] to check my changes and to update code examples as discussed.</p>
</div>
</div>
</div>
</div>