<div class="paragraph">
<p>Labeled loops are useful, especially when the code is badly indented, to match the begin and end of each loop. This rule raises an issue when the end of a labeled loop is unlabeled.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>BEGIN
  &lt;&lt;myLoopLabel1&gt;&gt;
  LOOP
    EXIT;
  END LOOP; -- Noncompliant; this labeled loop has no ending label

  LOOP
    EXIT;
  END LOOP; -- Compliant; not a labeled loop
END;
/</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>BEGIN
  &lt;&lt;myLoopLabel1&gt;&gt;
  LOOP
    EXIT;
  END LOOP myLoopLabel1;

  LOOP
    EXIT;
  END LOOP;
END;
/</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Label this "END".</p>
</div>
</div>
</div>
</div>