<div class="paragraph">
<p>With the advent of .NET framework version 2, certain practices have become obsolete.</p>
</div>
<div class="paragraph">
<p>In particular, exceptions should now extend <code>System.Exception</code> instead of <code>System.ApplicationException</code>. Similarly, generic collections should be used instead of the older, non-generic, ones. Finally when creating an XML view, you should not extend <code>System.Xml.XmlDocument</code>.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when an externally visible type extends one of these types:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>System.ApplicationException</code></p>
</li>
<li>
<p><code>System.Xml.XmlDocument</code></p>
</li>
<li>
<p><code>System.Collections.CollectionBase</code></p>
</li>
<li>
<p><code>System.Collections.DictionaryBase</code></p>
</li>
<li>
<p><code>System.Collections.Queue</code></p>
</li>
<li>
<p><code>System.Collections.ReadOnlyCollectionBase</code></p>
</li>
<li>
<p><code>System.Collections.SortedList</code></p>
</li>
<li>
<p><code>System.Collections.Stack</code></p>
</li>
</ul>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>using System;
using System.Collections;

namespace MyLibrary
{
  public class MyCollection : CollectionBase  // Noncompliant
  {
  }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>using System;
using System.Collections;

namespace MyLibrary
{
  public class MyCollection : Collection&lt;T&gt;
  {
  }
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Refactor this type not to derive from an outdated type '{0}'.</p>
</div>
</div>
</div>
</div>