<div class="paragraph">
<p>The <code>&lt;html&gt;</code> element should provide the <code>lang</code> and/or <code>xml:lang</code> attribute in order to identify the default language of a document.</p>
</div>
<div class="paragraph">
<p>It enables assistive technologies, such as screen readers, to provide a comfortable reading experience by adapting the pronunciation and accent to the language. It also helps braille translation software, telling it to switch the control codes for accented characters for instance.</p>
</div>
<div class="paragraph">
<p>Other benefits of marking the language include:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>assisting user agents in providing dictionary definitions or helping users benefit from translation tools.</p>
</li>
<li>
<p>improving <a href="https://blogs.bing.com/webmaster/2011/03/01/how-to-tell-bing-your-websites-country-and-language/">search engine ranking</a>.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Both the <code>lang</code> and the <code>xml:lang</code> attributes can take only one value.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>&lt;!DOCTYPE html&gt;
&lt;html&gt; &lt;!-- Noncompliant --&gt;
    &lt;head&gt;
          &lt;title&gt;A page written in english&lt;/title&gt;
          &lt;meta content="text/html; charset=utf-8" /&gt;
    &lt;/head&gt;


    &lt;body&gt;
    ...
    &lt;/body&gt;
&lt;/html&gt;</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>&lt;!DOCTYPE html&gt;
&lt;html lang="en"&gt;
    &lt;head&gt;
          &lt;title&gt;A page written in english&lt;/title&gt;
          &lt;meta content="text/html; charset=utf-8" /&gt;
    &lt;/head&gt;


    &lt;body&gt;
    ...
    &lt;/body&gt;
&lt;/html&gt;</pre>
</div>
</div>
<div class="listingblock">
<div class="content">
<pre>&lt;!DOCTYPE html&gt;
&lt;html lang="en" xml:lang="en"&gt;
    &lt;head&gt;
          &lt;title&gt;A page written in english&lt;/title&gt;
          &lt;meta content="text/html; charset=utf-8" /&gt;
    &lt;/head&gt;


    &lt;body&gt;
    ...
    &lt;/body&gt;
&lt;/html&gt;</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://www.w3.org/TR/WCAG20-TECHS/html.html#H57">WCAG2, H57</a> - Using language attributes on the html element</p>
</li>
<li>
<p><a href="https://www.w3.org/WAI/WCAG21/quickref/?versions=2.0#qr-meaning-doc-lang-id">WCAG2, 3.1.1</a> - Language of Page</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add "lang" and/or "xml:lang" attributes to this "&lt;html&gt;" element</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Highlight the &lt;html&gt; element</p>
</div>
</div>
</div>
</div>