<div class="paragraph">
<p>A magic number is a number that comes out of nowhere, and is directly used in a statement. Magic numbers are often used, for instance to limit the number of iterations of a loops, to test the value of a property, etc.</p>
</div>
<div class="paragraph">
<p>Using magic numbers may seem obvious and straightforward when you&#8217;re writing a piece of code, but they are much less obvious and straightforward at debugging time.</p>
</div>
<div class="paragraph">
<p>That is why magic numbers must be demystified by first being assigned to clearly named constants before being used.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>Class Foo
  Sub DoSomething(Param As Integer)
    If Param &gt; 100 Then ' Magic Number
      ' Do something
    End If
  End Sub
End Class</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>Class Foo
  Private Const MaxOfSomething As Integer = 100

  Sub DoSomething(Param As Integer)
    If Param &gt; MaxOfSomething Then
      ' Do something
    End If
  End Sub
End Class</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Assign this magic number X to a well-named (variable|constant), and use the (variable|constant) instead.</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="ulist">
<ul>
<li>
<p>key = exceptions</p>
</li>
<li>
<p>default = -1,0,1,2</p>
</li>
<li>
<p>type = STRING</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_duplicated_by_s1977">is duplicated by: S1977</h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s6164">relates to: S6164</h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s2480">is related to: S2480</h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s1583">is related to: S1583</h3>

</div>
<div class="sect2">
<h3 id="_on_4_nov_2013_091654_freddy_mallet_wrote">on 4 Nov 2013, 09:16:54 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by https://jira.codehaus.org/browse/SONARJAVA-130</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_28_apr_2014_085008_dinesh_bolkensteyn_wrote">on 28 Apr 2014, 08:50:08 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>In COBOL, this rule has been dropped in favor of "String literals should not be duplicated", see http://jira.sonarsource.com/browse/COBOL-997</p>
</div>
</div>
</div>
</div>