<div class="paragraph">
<p><em>Forwarding references</em> are a special kind of references that both ignore and preserve the <em>value category</em> of a function argument, making it possible to forward it by means of <code>std::forward</code>.</p>
</div>
<div class="paragraph">
<p>Any code using such a reference for any other purpose than forwarding is actually ignoring rvalue-ness and const-ness of the associated parameter.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>#include &lt;utility&gt;
#include &lt;string&gt;
#include &lt;iostream&gt;

template&lt;typename TP&gt; void f( TP&amp;&amp; arg ) {
    std::string s(arg);
}

int main() {
    std::string s("test");
    f(std::move(s));
    std::cout&lt;&lt;"f:"&lt;&lt;s&lt;&lt;std::endl; // output is "f:test"

    return 0;
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>#include &lt;utility&gt;
#include &lt;string&gt;
#include &lt;iostream&gt;

template&lt;typename TP&gt; void f( TP&amp;&amp; arg ) {
    std::string s(std::forward&lt;TP&gt;(arg));
}

int main() {
    std::string s("test");
    f(std::move(s));
    std::cout&lt;&lt;"f:"&lt;&lt;s&lt;&lt;std::endl; // output is "f:"

    return 0;
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://github.com/isocpp/CppCoreGuidelines/blob/036324/CppCoreGuidelines.md#f19-for-forward-parameters-pass-by-tp-and-only-stdforward-the-parameter">C&#43;&#43; Core Guidelines F.19</a> - For “forward” parameters, pass by TP&amp;&amp; and only std::forward the parameter</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use only "std::forward" on forwarding references.</p>
</div>
</div>
</div>
</div>