<div class="paragraph">
<p>Function pointer syntax can be hard on the eyes, particularly when one function is used as a parameter to another. Providing and using a <code>typedef</code> instead (or a <code>using</code> in C&#43;&#43;) can make code easier to read, and should be preferred.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>extern void (*signal(int, void(*)(int)))(int);</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>typedef void (*SignalHandler)(int signum);
extern SignalHandler signal(int signum, SignalHandler handler);</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Define and use a "typedef" for XXX</p>
</div>
</div>
</div>
</div>