<div class="paragraph">
<p>Using a unique <code>KLIST</code> to access files ensures that the correct key is used at all times and creates a standard method of accessing the data throughout the code, simplifying maintenance and improving readability.</p>
</div>
<div class="paragraph">
<p>This rule flags non-compliant instances of <code>CHAIN</code>, <code>DELETE</code>, <code>READE</code>, <code>READPE</code>, <code>SETGT</code>, and <code>SETLL</code> operations.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>C     'D001 B'      CHAIN     KYCUSTN                            20
...
C     DEPT          CHAIN     KYCUSTN                            20</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>C     DEPTKY        KLIST
C                   KFLD                    DEPT
C                   KFLD                    SHIFT

C     DEPTKY        CHAIN     KYCUSTN                            20</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Rewrite this statement to use a "KLIST"</p>
</div>
</div>
</div>
</div>