<div class="paragraph">
<p><code>break</code> and <code>continue</code> are unstructured control flow statements which make code harder to read. Additionally, more recent versions of Python raise a SyntaxError when modules containing <code>break</code> or <code>continue</code> outside of a loop are imported.</p>
</div>
<div class="paragraph">
<p>Therefore, these statements should not be used outside of loops.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>narg=len(sys.argv)
if narg == 1:
        print('@Usage: input_filename nelements nintervals')
        break</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>if narg == 1:
        print('@Usage: input_filename nelements nintervals')
        sys.exit()</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this "xxx" statement</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s910">is related to: S910</h3>

</div>
<div class="sect2">
<h3 id="_on_4_nov_2014_155914_ann_campbell_wrote">on 4 Nov 2014, 15:59:14 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>pylint:E0103</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_9_may_2016_154147_evgeny_mandrikov_wrote">on 9 May 2016, 15:41:47 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>I don&#8217;t know any C/C&#43;&#43; compiler, which will allow to use "continue" outside of loop and "break" outside of switch and loop, hence not applicable.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_9_may_2016_154426_evgeny_mandrikov_wrote">on 9 May 2016, 15:44:26 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>Note that PC-Lint error codes from 1 to 199 for C and from 1001 to 1199 for C&#43;&#43; are syntax errors.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_16_may_2016_170600_ann_campbell_wrote">on 16 May 2016, 17:06:00 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~evgeny.mandrikov] I guess this means that PC-Lint doesn&#8217;t expect compilable code?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_16_may_2016_171419_evgeny_mandrikov_wrote">on 16 May 2016, 17:14:19 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I suppose that it expects for proper analysis in general, but when this is not the case it generates more precise "parse error".</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_16_may_2016_175032_ann_campbell_wrote">on 16 May 2016, 17:50:32 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>For the record PC-Lint rules in these ranges are Syntax errors: 1-199, 1001-1199.</p>
</div>
</div>
</div>
</div>