<div class="paragraph">
<p>If a private variable is declared but not used in the program, it can be considered dead code and should therefore be removed. This will improve maintainability because developers will not wonder what the variable is used for.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>Private Foo as Integer

Function Compute(A As Integer)
   Compute = A * 42
End Function</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>Function Compute(A As Integer)
   Compute = A * 42
End Function</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this unused "xxx" private variable. It is set n times but never read.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s1144">relates to: S1144</h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s3985">is related to: S3985</h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s895">relates to: S895</h3>

</div>
<div class="sect2">
<h3 id="_on_15_oct_2013_084525_freddy_mallet_wrote">on 15 Oct 2013, 08:45:25 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARJAVA-358</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_16_oct_2013_235216_ann_campbell_wrote">on 16 Oct 2013, 23:52:16 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>I&#8217;ve added Compliant Solutions, but someone should double-check my guess at PL/SQL</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_22_oct_2013_144538_freddy_mallet_wrote">on 22 Oct 2013, 14:45:38 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3195 in Flex</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_9_dec_2013_083703_dinesh_bolkensteyn_wrote">on 9 Dec 2013, 08:37:03 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>The PL/SQL case is not a field, but just a local variable (for which we have another rule, I&#8217;ll remove it for now.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_9_dec_2013_084512_dinesh_bolkensteyn_wrote">on 9 Dec 2013, 08:45:12 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>I guess that we need to allow any annotated private field? - in fact perhaps not, I can imagine that some private fields are injected, but read from outside?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_23_feb_2014_225134_freddy_mallet_wrote">on 23 Feb 2014, 22:51:34 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3505 for PHP</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_10_mar_2017_143245_ann_campbell_wrote">on 10 Mar 2017, 14:32:45 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Irrelevant for C# because fields are covered in the language&#8217;s implementation of RSPEC-1144</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_5_nov_2019_105434_nicolas_harraudeau_wrote">on 5 Nov 2019, 10:54:34 Nicolas Harraudeau wrote:</h3>
<div class="paragraph">
<p>Python fields only exist once they are set. Thus RSPEC-4487 is more appropriate.</p>
</div>
</div>
</div>
</div>