<div class="paragraph">
<p>According to the PSR2 coding standard:</p>
</div>
<div class="quoteblock">
<blockquote>
<div class="paragraph">
<p>The keyword <code>elseif</code> SHOULD be used instead of <code>else if</code> so that all control keywords look like single words.</p>
</div>
</blockquote>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>if ($expr1) {
  ...
} else if ($expr2) {
  ...
} else {...}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>if ($expr1) {
  ...
} elseif ($expr2) {
  ...
} else {...}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace this "else if" keyword sequence by "elseif" keyword.</p>
</div>
</div>
</div>
</div>