<div class="paragraph">
<p>Python has no pre/post increment/decrement operator. For instance, <code>x&#43;&#43;</code> and <code>x--</code> will fail to parse. More importantly, <code>&#43;&#43;x</code> and <code>--x</code> will do nothing. To increment a number, simply write <code>x += 1</code>.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>++x # Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>x += 1</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Python does not include the [pre|post][increment|decrement] operator.</p>
</div>
</div>
</div>
</div>