<div class="paragraph">
<p>The CPython interpreter does not check arguments type when functions are called. However a function can express the type it expects for each argument in its documentation or by using <a href="https://www.python.org/dev/peps/pep-0484/">Type Hints</a>. Calling such a function with an argument of a different type can easily create a bug. Even if it works right now it can fail later when APIs evolve or when type checks are added (ex: with <code>isinstance</code>).</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when a function or method is called with an argument of a different type than the one described in its type annotations. It also checks argument types for builtin functions.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>def func(var: str):
    pass

func(42)  # Noncompliant

len(1)  # Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>def func(var: str):
    pass

func("42")

len("1")</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.python.org/3/library/functions.html#built-in-funcs">Python documentation - builtins</a></p>
</li>
<li>
<p><a href="https://www.python.org/dev/peps/pep-0484/">PEP 484 — Type Hints</a></p>
</li>
<li>
<p><a href="https://docs.python.org/3/library/typing.html">Python documentation - typing — Support for type hints</a></p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>Change this argument of type XXX; Function FFF expects type YYY</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Primary: the expression provided as argument</p>
</div>
<div class="paragraph">
<p>Secondary:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>location: definition of the function called</p>
</li>
<li>
<p>message: "Function definition"</p>
</li>
</ul>
</div>
</div>
</div>
</div>