<div class="paragraph">
<p>The goal of this coding rule is to make it obvious at first look that a class <code>extends</code> some other classes and/or <code>implements</code> some interfaces. The names of extended classes or implemented interfaces can be located on next lines.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>&lt;?php
class ClassName
extends ParentClass {...}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>&lt;?php
class ClassName extends ParentClass {...}</pre>
</div>
</div>
<div class="paragraph">
<p>or</p>
</div>
<div class="listingblock">
<div class="content">
<pre>&lt;?php
class ClassName extends
ParentClass {...}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Move ["extends"] [and] ["implements"] keyword[s] on line {0} where declaration of class name "xxxx" begin.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_6_oct_2014_130139_ann_campbell_wrote">on 6 Oct 2014, 13:01:39 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>For PHP this was rolled into RSPEC-1808</p>
</div>
</div>
</div>
</div>