<div class="paragraph">
<p>A common good practice is to write test methods targeting only one logical concept, that can only fail for one reason.</p>
</div>
<div class="paragraph">
<p>While it might make sense to have more than one assertion to test one concept, having too many is a sign that a test became too complex and should be refactored to multiples ones.</p>
</div>
<div class="paragraph">
<p>This rule will report any test method containing more than a given number of assertion.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="paragraph">
<p>With a parameter of 2.</p>
</div>
<div class="listingblock">
<div class="content">
<pre>@Test
void test() { // Refactor this method.
  assertEquals(1, f(1));
  assertEquals(2, f(2));
  assertEquals(3, g(1));
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>void test_f() {
  assertEquals(1, f(1));
  assertEquals(2, f(2));
}
void test_g() {
  assertEquals(3, g(1));
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Refactor this method to reduce the number of assertions from {number of assertions} to less than {default value}.</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="ulist">
<ul>
<li>
<p>key = maximumAssertionNumber</p>
</li>
<li>
<p>description = The maximum authorized number of assertions in a test method.</p>
</li>
<li>
<p>default = 25</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Primary locations:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Method name</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Secondary locations:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Assertions</p>
</li>
</ul>
</div>
</div>
</div>
</div>