<div class="paragraph">
<p>When using a floating-point <code>for</code> loop counter, an accumulation of rounding errors may result in a mismatch between the expected and actual number of iterations.</p>
</div>
<div class="paragraph">
<p>Even if floating-point loop counters appears to behave correctly on one implementation, it may give a different number of iterations on another implementation.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>for (float counter = 0.0f; counter &lt; 1.0f; counter += 0.001f) {
  ...
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>for (int counter = 0; counter &lt; 1000; ++counter) {
  ...
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C:2004, 13.4 - The controlling expression of a for statement shall not contain any objects of floating type.</p>
</li>
<li>
<p>MISRA C&#43;&#43;:2008, 6-5-1 - A <em>for</em> loop shall contain a single <em>loop-counter</em> which shall not have floating type.</p>
</li>
<li>
<p>MISRA C:2012, 14.1 - A <em>loop counter</em> shall not have essentially <em>floating type</em>.</p>
</li>
<li>
<p><a href="https://wiki.sei.cmu.edu/confluence/display/c/FLP30-C.+Do+not+use+floating-point+variables+as+loop+counters">CERT, FLP30-C.</a> - Do not use floating-point variables as loop counters</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Do not use a counter of type float.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_duplicates_s887">duplicates: S887</h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s5311">is related to: S5311</h3>

</div>
</div>
</div>