<div class="paragraph">
<p>Modulo by 1 operation always returns 0, it is likely a typo.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>void f(int value1, int value2) {
  if (fun() % 1 == 1) { // Noncompliant
    // ...
  }
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Review this modulo by 1 operation.</p>
</div>
</div>
</div>
</div>