<div class="paragraph">
<p>When constraints enforce a limited set of values for a column, conditioning a SQL statement on a value outside that set is simply an error.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>...
CHECK(SIZE = ’S’ OR SIZE = ’M’ OR SIZE = 'L')
...

SELECT id, name, price
FROM product
WHERE size != 'XXL'  -- Noncompliant; XXL is not a valid value, so clause has no effect

SELECT id, name, price
FROM product
WHERE size = 'XXL'  -- Noncompliant; XXL is not a valid value, so no rows selected</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>"xxx" is not a possible value of "ccc".</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p><code>colname opr value</code></p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_16_jun_2016_144131_ann_campbell_wrote">on 16 Jun 2016, 14:41:31 Ann Campbell wrote:</h3>
<div class="paragraph">
<p><a href="https://www.researchgate.net/publication/222432151_Semantic_errors_in_SQL_queries_A_quite_complete_list" class="bare">https://www.researchgate.net/publication/222432151_Semantic_errors_in_SQL_queries_A_quite_complete_list</a></p>
</div>
</div>
</div>
</div>