<div class="paragraph">
<p>Shared naming conventions allow teams to collaborate efficiently. This rule checks that all generic type parameter names match a provided regular expression.</p>
</div>
<div class="paragraph">
<p>The default configuration is the one recommended by Microsoft:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Must start with an upper case 'T' character, e.g. T</p>
</li>
<li>
<p>Followed by Pascal casing, starting with an upper case character, e.g. TKey</p>
</li>
<li>
<p>Short abbreviations of 2 letters can be capitalized, e.g. TFooID</p>
</li>
<li>
<p>Longer abbreviations need to be lower cased, e.g. TFooHtml</p>
</li>
</ul>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="paragraph">
<p>With the default parameter value <code>^T(([A-Z]{1,3}[a-z0-9]+)*([A-Z]{2})?)?$</code>:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>Public Class Foo(Of t) ' Noncompliant
End Class</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>Public Class Foo(Of T) ' Compliant
End Class</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Rename "xxx" to match the regular expression: "yyy".</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="ulist">
<ul>
<li>
<p>key = format
description: Regular expression used to check the generic type parameter names against.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>default: ^T(([A-Z]{1,3}[a-z0-9]+)*([A-Z]{2})?)?$</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s119">is related to: S119</h3>

</div>
</div>
</div>