<div class="paragraph">
<p>JavaScript regular expressions provide  Unicode character classses and Unicode property escapes for matching characters based on their Unicode values and Unicode properties respectively. When using Unicode property escapes like <code>\p{Alpha}</code> without the <code>u</code> flag, the regular expression will not match alphabetic characters but rather the <code>+'\p{Alpha}+'</code> string literal, which is likely a mistake.</p>
</div>
<div class="paragraph">
<p>This rules raises an issue when Unicode character classses and Unicode property escapes are used without the <code>u</code> flag.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>/\u{1234}/
/\p{Alpha}/</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>/\u{1234}/u
/\p{Alpha}/u</pre>
</div>
</div>
</div>
</div>