<div class="paragraph">
<p>There&#8217;s no point in checking a variable against the value you&#8217;re about to assign it. Save the cycles and lines of code, and simply perform the assignment.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>if (x != a)  // Noncompliant; why bother?
{
    x = a;
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>x = a;</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_exceptions">Exceptions</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Properties and checks inside setters are excluded from this rule because they could have side effects and removing the check could lead to undesired side effects.</p>
</div>
<div class="listingblock">
<div class="content">
<pre>if (MyProperty != a)
{
    MyProperty = a; // Compliant because the setter could be expensive call
}</pre>
</div>
</div>
<div class="listingblock">
<div class="content">
<pre>private int myField;
public int SomeProperty
{
    get
    {
        return myField;
    }
    set
    {
        if (myField != value)
        {
            myField = value;
        }
    }
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this useless conditional.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>if statement</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_8_dec_2015_083842_tamas_vajk_wrote">on 8 Dec 2015, 08:38:42 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Added it to SonarQube Way profile.</p>
</div>
</div>
</div>
</div>