<div class="paragraph">
<p>This rule allows you to check for certain types of comments before each method. For example, you may want to check for a Doxygen @author comment before each method.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="paragraph">
<p>With regular expression: <code>//! @brief[ \t]+[a-zA-Z ]+</code>:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>bool TEST_CLASS_OK::Method(int nArg)  // Noncompliant
{
  return true;
}{code}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>//-------------------------------------------------------------------------
//! @brief    Method
//-------------------------------------------------------------------------
bool TEST_CLASS_OK::Method(int nArg)
{
  return true;
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>The missing documentation should be added</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="ulist">
<ul>
<li>
<p>key = required_comment</p>
</li>
<li>
<p>description: Regex describing a line that should be found in a comment block before each method.</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_16_jul_2014_134119_ann_campbell_wrote">on 16 Jul 2014, 13:41:19 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>please review</p>
</div>
</div>
</div>
</div>