<div class="paragraph">
<p>Property getter, setter and deleter methods are called by the python interpreter with a specific number or arguments:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>only "self" for property getter and deleter methods.</p>
</li>
<li>
<p>"self" and a value for setter methods.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Adding any other parameter, or removing these mandatory parameters will make method calls fail.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>too many parameters are defined in a property getter, setter or deleter method.</p>
</li>
<li>
<p>the value parameter is missing in a property setter method.</p>
</li>
</ul>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class A:
    @property
    def foo(self, unexpected, unexpected2):  # Noncompliant. Too many parameters.
        return self._foo

    @foo.setter
    def foo(self, value, unexpected):  # Noncompliant. Too many parameters.
        self._foo = value

    @foo.deleter
    def foo(self, unexpected):  # Noncompliant. Too many parameters.
        del self._foo

class B:
    def get_foo(self, unexpected):  # Noncompliant. Too many parameters.
        return self._foo

    def set_foo(self, value, unexpected):  # Noncompliant. Too many parameters.
        self._foo = value

    def del_foo(self, unexpected):  # Noncompliant. Too many parameters.
        del self._foo

    foo = property(get_foo, set_foo, del_foo, "'foo' property.")</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class A:
    @property
    def foo(self):
        return self._foo

    @foo.setter
    def foo(self, value):
        self._foo = value

    @foo.deleter
    def foo(self):
        del self._foo

class B:
    def get_foo(self):
        return self._foo

    def set_foo(self, value):
        self._foo = value

    def del_foo(self):
        del self._foo

    foo = property(get_foo, set_foo, del_foo, "'foo' property.")</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.python.org/3/library/functions.html#property">Python Documentation - Built-in Functions - property</a></p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>Remove XXX parameters; property getter methods receive only "self".</p>
</li>
<li>
<p>Remove XXX parameters; property deleter methods receive only "self".</p>
</li>
<li>
<p>Remove XXX parameters; property setter methods receive "self" and a value.</p>
</li>
<li>
<p>Add the value parameter; property setter methods receive "self" and a value.</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The method signature.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_11_feb_2020_182423_nicolas_harraudeau_wrote">on 11 Feb 2020, 18:24:23 Nicolas Harraudeau wrote:</h3>
<div class="paragraph">
<p>Note that we don&#8217;t raise any issue for missing "self" parameter because this is already covered by RSPEC-5720.</p>
</div>
</div>
</div>
</div>