<div class="paragraph">
<p>Shared coding conventions allow teams to collaborate effectively. This rule checks that classes extending <code>React.PureComponent</code> or <code>React.Component</code> are named with UpperCamelCase (aka PascalCase).</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>export default class my-super-component extends React.PureComponent { // Noncompliant
 ...
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>export default class MySuperComponent extends React.PureComponent {
 ...
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Rename this class to be compliant with the UpperCamelCase naming convention.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Primary: Class name</p>
</div>
</div>
</div>
</div>