<div class="paragraph">
<p>Shadowing fields or <code>enum</code> <code>case</code>s with a local variable is a bad practice that reduces code readability: It makes it confusing to know whether the field or the variable is being used.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public class Foo {
  public var myField:Int = 0

  public func doSomething() {
    var myField = 0
    ...
  }
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Rename "${elementName}" which has the same name as the (field|case) declared at line {}.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>Primary: shadowing variable</p>
</li>
<li>
<p>Secondary: shadowed field/case</p>
<div class="ulist">
<ul>
<li>
<p>message: 'Shadowed field/case.'</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_deprecates_s1524">deprecates: S1524</h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s801">relates to: S801</h3>

</div>
<div class="sect2">
<h3 id="_on_11_jul_2013_100430_dinesh_bolkensteyn_wrote">on 11 Jul 2013, 10:04:30 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>Ignore constructors</p>
</div>
<div class="paragraph">
<p>Ignore setters, method which start with 'set'</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_15_jul_2013_135906_dinesh_bolkensteyn_wrote">on 15 Jul 2013, 13:59:06 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>Implemented by http://jira.codehaus.org/browse/SONARJAVA-211</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_31_oct_2013_163605_freddy_mallet_wrote">on 31 Oct 2013, 16:36:05 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3232 for Flex</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_23_feb_2014_224904_freddy_mallet_wrote">on 23 Feb 2014, 22:49:04 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3504 for PHP</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_12_mar_2019_164940_alexandre_gigleux_wrote">on 12 Mar 2019, 16:49:40 Alexandre Gigleux wrote:</h3>
<div class="paragraph">
<p>Decision was taken to deprecate in favour of RSPEC-1117. As a consequence the following maintenance tickets should be implemented:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>https://jira.sonarsource.com/browse/SONARPLSQL-722</p>
</li>
<li>
<p>https://github.com/SonarSource/SonarTS/issues/838</p>
</li>
<li>
<p>https://github.com/SonarSource/SonarJS/issues/1214</p>
</li>
</ul>
</div>
</div>
</div>
</div>