<div class="paragraph">
<p>While the ternary operator is pleasingly compact, its use can make code more difficult to read. It should therefore be avoided in favor of the more verbose <code>if</code>/<code>else</code> structure.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>function foo(a) {
  var b = (a === 'A') ? 'is A' : 'is not A'; // Noncompliant
  // ...
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>function foo(a) {
  var b;
  if (a === 'A') {
    b = 'is A';
  }
  else {
    b = 'is not A';
  }
  // ...
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Convert this usage of the ternary operator to an "if"/"else" structure.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_deprecates_s1521">deprecates: S1521</h3>

</div>
<div class="sect2">
<h3 id="_on_7_apr_2016_093902_elena_vilchik_wrote">on 7 Apr 2016, 09:39:02 Elena Vilchik wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Shouldn&#8217;t it be <code>its</code> instead of <code>it's</code> in description?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_7_apr_2016_134735_ann_campbell_wrote">on 7 Apr 2016, 13:47:35 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Fixed. Thanks [~elena.vilchik]. That&#8217;s one I chronically screw up</p>
</div>
</div>
</div>
</div>