<div class="paragraph">
<p>Error handling should be defined on file operations to ensure correct processing of program failures rather than defaulting to an abnormal program end. This rule checks two things:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>INFSR</code> error handling is defined on file operations.</p>
</li>
<li>
<p>error handling is delegated to an appropriately named routine.</p>
</li>
</ul>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Given the default parameter value, <code>^*PSSR$</code>:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>     F* Noncompliant
     FIPBDCCP   IF   E           K DISK

     FEP210     CF   E             WORKSTN
     F                                     SFILE(EP21003S:RELN03)

     FEP471R1   O    E             PRINTER
     F                                     USROPN
     F                                     INFDS(W1SF01)</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>     FIPBDCCP   IF   E           K DISK    INFSR(*PSSR)

     FEP210     CF   E             WORKSTN INFSR(*PSSR)
     F                                     SFILE(EP21003S:RELN03)

     FEP471R1   O    E             PRINTER INFSR(*PSSR)
     F                                     USROPN
     F                                     INFDS(W1SF01)</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add the error handling "INFSR(xxxx)" keyword to this F spec</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="ulist">
<ul>
<li>
<p>key = format_error_handling
description: Regular expression describing the names of valid error handling routines</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>default value: ^*PSSR$</p>
</div>
</div>
</div>
</div>