<div class="paragraph">
<p>Type assertion can be done in two ways: with <code>as MyType</code> or with <code>&lt;MyType&gt;</code>. But since there is an ambiguity in the latter when using JSX and there is no ambiguity in the former, <code>as</code> is preferred.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>var foo = &lt;any&gt;"foo";  // Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>var foo = "foo" as any;</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use "as" type assertion here instead.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p><code>&lt;foo&gt;</code></p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_31_jul_2017_181737_elena_vilchik_wrote">on 31 Jul 2017, 18:17:37 Elena Vilchik wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Ann, could you explain in more details why <code>&lt;T&gt;</code> notation is ambiguous?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_31_jul_2017_194012_ann_campbell_wrote">on 31 Jul 2017, 19:40:12 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~elena.vilchik] from https://basarat.gitbooks.io/typescript/content/docs/types/type-assertion.html:</p>
</div>
<div class="paragraph">
<p><em><em>_there is an ambiguity in the language grammar when using &lt;foo&gt; style assertions in JSX</em>_</em></p>
</div>
<div class="paragraph">
<p>In trying to re-find my original reference (I didn&#8217;t) I also came across <a href="https://stackoverflow.com/a/33503842/2662707">this SO answer</a>, which states it slightly differently</p>
</div>
<div class="quoteblock">
<blockquote>
<div class="paragraph">
<p>The difference is that as Circle works in TSX files, but &lt;Circle&gt; conflicts with JSX syntax.</p>
</div>
</blockquote>
</div>
</div>
<div class="sect2">
<h3 id="_on_2_aug_2017_092109_elena_vilchik_wrote">on 2 Aug 2017, 09:21:09 Elena Vilchik wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I think we can say something like:</p>
</div>
<div class="paragraph">
<p><code>But since the latter is ambiguous in language grammar when using JSX and the former ...</code></p>
</div>
</div>
<div class="sect2">
<h3 id="_on_2_aug_2017_154636_ann_campbell_wrote">on 2 Aug 2017, 15:46:36 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Updated [~elena.vilchik]. See what you think.</p>
</div>
</div>
</div>
</div>