<div class="paragraph">
<p>In PHP 4, any function with the same name as the nesting class was considered a class constructor. In PHP 5, this mechanism has been deprecated and the <code>__construct</code> method name should be used instead. If both styles are present in the same class, PHP 5 will treat the function named <code>__construct</code> as the class constructor.</p>
</div>
<div class="paragraph">
<p>This rule rule raises an issue for each method with the same name as the enclosing class.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class Foo {
  function Foo(){...}
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class Foo {
  function __construct(){...}
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>Rename this "XXXXX" function to "__construct".</p>
</li>
<li>
<p>Replace this function name "XXXXX", since a "__construct" method has already been defined in this class.</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_23_feb_2014_231958_freddy_mallet_wrote">on 23 Feb 2014, 23:19:58 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3513 for PHP</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_12_oct_2015_123559_ann_campbell_wrote">on 12 Oct 2015, 12:35:59 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~linda.martin] I&#8217;ve updated your edits. Double-check me.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_13_oct_2015_101857_linda_martin_wrote">on 13 Oct 2015, 10:18:57 Linda Martin wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Perfect thanks!</p>
</div>
</div>
</div>
</div>