<div class="paragraph">
<p>By convention, the first argument to class methods, i.e. methods decorated with <code>@classmethod</code>, is named <code>cls</code> as a representation and a reminder that the argument is the class itself. Name the argument something else, and you stand a good chance of confusing both users and maintainers of the code. It might also indicate that the <code>cls</code> parameter was forgotten, in which case calling the method will most probably fail. This rule also applies to methods <code>__init_subclass__</code>, <code>__class_getitem__</code> and <code>__new__</code> as their first argument is always the class instead of "self".</p>
</div>
<div class="paragraph">
<p>By default this rule accepts <code>cls</code> and <code>mcs</code>, which is sometime used in metaclasses, as valid names for class parameters. You can set your own list of accepted names via the parameter <code>classParameterNames</code>.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when the first parameter of a class method is not an accepted name.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class Rectangle(object):

  @classmethod
  def area(bob, height, width):  #Noncompliant
    return height * width</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class Rectangle(object):

  @classmethod
  def area(cls, height, width):
    return height * width</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>PEP8 - <a href="https://www.python.org/dev/peps/pep-0008/#function-and-method-arguments">Function and Method Arguments</a></p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Rename XXX to a valid class parameter name or add the missing class parameter.</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="ulist">
<ul>
<li>
<p>key = classParameterNames
Description: Comma separated list of accepted values for a class parameter.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Default value : cls,mcs</p>
</div>
<div class="paragraph">
<p>Type: string</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The first parameter of the class method.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_17_mar_2015_084957_elena_vilchik_wrote">on 17 Mar 2015, 08:49:57 Elena Vilchik wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I would suggest label "convention" here</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_18_mar_2015_162414_ann_campbell_wrote">on 18 Mar 2015, 16:24:14 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>There is some question about class vs metaclass method vs instance method. Postponing this rule until that&#8217;s settled. Updates may be needed at that time.</p>
</div>
</div>
</div>
</div>