<div class="paragraph">
<p>In order to be callable, a python class should implement the <a href="https://docs.python.org/3/reference/datamodel.html#object.__call__"><code>__call__</code></a> method.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when a non-callable object is called.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class MyClass:
    pass

myvar = MyClass()
myvar()  # Noncompliant

none_var = None
none_var()  # Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class MyClass:
    def __call__(self):
        print("called")

myvar = MyClass()
myvar()</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.python.org/3/reference/datamodel.html#object.__call__">Python documentation - __call__ method</a></p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Fix this call; XX is not callable.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Primary: The name before the opening parenthesis</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Secondary 1 (if the call is made on a variable)</p>
<div class="ulist">
<ul>
<li>
<p>location: The last value assigned.</p>
</li>
<li>
<p>message: "Assigned value."</p>
</li>
</ul>
</div>
</li>
<li>
<p>Secondary 2</p>
<div class="ulist">
<ul>
<li>
<p>location: The class definition</p>
</li>
<li>
<p>message: "Definition."</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s2873">relates to: S2873</h3>

</div>
</div>
</div>