<div class="paragraph">
<p><code>#pragma once</code> is a preprocessor directive meant to ensure that a file is only included once in a compilation unit. However, there are several reasons to avoid it:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>While it is widely implemented, it&#8217;s not universally supported.</p>
</li>
<li>
<p>It&#8217;s not part of the standard, so its behavior may vary from compiler to compiler.</p>
</li>
<li>
<p>If the same file exists in two different locations, it will <em>not</em> prevent the second file from being included in the compilation.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>For these reasons, include guards, or include guards along with <code>#pragma once</code> should be used instead.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>// my_header.h
#pragma once
...</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>// my_header.h
#ifndef MY_HEADER
#define MY_HEADER
...
#endif</pre>
</div>
</div>
<div class="paragraph">
<p>or</p>
</div>
<div class="listingblock">
<div class="content">
<pre>// my_header.h
#pragma once
#ifndef MY_HEADER
#define MY_HEADER
...
#endif</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this use of "#pragma once" and use include guards instead</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_30_jul_2014_202537_freddy_mallet_wrote">on 30 Jul 2014, 20:25:37 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>If my feeling is correct :</p>
</div>
<div class="ulist">
<ul>
<li>
<p>The purpose of this rule is mainly to prevent a portability issue from one compiler to another or from one version of the language to another. So I would use the "Portability" category.</p>
</li>
</ul>
</div>
</div>
</div>
</div>