<div class="paragraph">
<p>Many assertions compare two objects or properties of these objects. Passing twice the same argument is likely to be a bug due to developer&#8217;s carelessness.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when a Chai assertion is given twice the same argument.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>const assert = require('chai').assert;

describe("test the same object", function() {
    it("uses chai 'assert'", function() {
        const expected = '1'
        const actual = (1).toString()
        assert.equal(actual, actual);  // Noncompliant
    });
});</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>const assert = require('chai').assert;

describe("test the same object", function() {
    it("uses chai 'assert'", function() {
        const expected = '1'
        const actual = (1).toString()
        assert.equal(actual, expected);
    });
});</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace this argument or its duplicate at position [X].</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>If the same argument is passed twice to the same function</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Primary: the first expression occurence
message: 'Refactor this assertion to not pass the same argument twice.'</p>
</li>
<li>
<p>Secondary: the second expression occurence
message: 'Argument [POSITION] has the same value.'</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>If the assertion is of the form <code>expect(obj).a(obj, other1)</code></p>
</div>
<div class="ulist">
<ul>
<li>
<p>Primary: the duplicate argument.
message: 'Refactor this assertion to not pass the same value as in "expect()".'</p>
</li>
<li>
<p>Secondary: the argument of "expect".</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>If the assertion is of the form <code>obj.should.be.closeTo(obj, other1, other2)</code></p>
</div>
<div class="ulist">
<ul>
<li>
<p>Primary: the duplicate argument.
message: Refactor this assertion to not use the same value as the one asserted by "should".</p>
</li>
<li>
<p>Secondary: the variable on which "should" is called.
no message</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_17_nov_2020_161530_nicolas_harraudeau_wrote">on 17 Nov 2020, 16:15:30 Nicolas Harraudeau wrote:</h3>
<div class="paragraph">
<p>Note that this rule is a bit more generic for dynamically typed languages. In Java calling <code>assertThat(x).isInstanceOf(x)</code> is likely to fail because <code>isInstanceOf</code> wants a <code>Class</code> as parameter. For languages like PHP or Javascript it is possible to make such a typo. If the test is skipped the issue will go undetected.</p>
</div>
</div>
</div>
</div>