<div class="paragraph">
<p>Shared naming conventions allow teams to collaborate effectively. This rule raises an issue when a local variable or function parameter name does not match the provided regular expression.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="paragraph">
<p>With the default regular expression <code>@[a-zA-Z0-9_]*</code>:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>CREATE PROCEDURE proc1
    @@var1 INT -- Noncompliant
AS
BEGIN
    DECLARE @@var2 INT; -- Noncompliant
END</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Rename this [local variable|parameter] "XXXX" to match the regular expression ${format}.</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="ulist">
<ul>
<li>
<p>key = format
Description: Regular expression used to check the names against.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Default Value for Java : ^[a-z][a-zA-Z0-9]*$</p>
</div>
<div class="paragraph">
<p>Default Value for Flex : ^[_a-z][a-zA-Z0-9]*$</p>
</div>
<div class="paragraph">
<p>Default Value for Python : ^[<em>a-z][a-z0-9</em>]*$</p>
</div>
<div class="paragraph">
<p>Default Value for Swift : ^[a-z][a-zA-Z0-9]*$</p>
</div>
<div class="paragraph">
<p>Default Value for PHP : ^[a-z][a-zA-Z0-9]*$</p>
</div>
<div class="paragraph">
<p>Default Value for VB.NET: ^[a-z]<strong class="a-z0-9">([A-Z]{1,3}[a-z0-9]+)</strong>([A-Z]{2})?$</p>
</div>
<div class="paragraph">
<p>Default Value for TSQL: @[a-zA-Z0-9_]*</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s116">relates to: S116</h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s1545">is related to: S1545</h3>

</div>
<div class="sect2">
<h3 id="_on_20_may_2013_142618_fabrice_bellingard_wrote">on 20 May 2013, 14:26:18 Fabrice Bellingard wrote:</h3>
<div class="paragraph">
<p>Implementation: http://jira.codehaus.org/browse/SONARJAVA-146</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_30_oct_2013_122023_freddy_mallet_wrote">on 30 Oct 2013, 12:20:23 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3215 for Flex</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_19_feb_2014_155318_freddy_mallet_wrote">on 19 Feb 2014, 15:53:18 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3497 for PHP</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_5_dec_2016_144326_tamas_vajk_wrote">on 5 Dec 2016, 14:43:26 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~pierre-yves.nicolas], [~evgeny.mandrikov] I removed the exception part from the root rule and added it to Java. If any of your languages also implement that exception, could you add it to the appropriate subtask?</p>
</div>
</div>
</div>
</div>