<div class="paragraph">
<p>Sharing some naming conventions is a key point to make it possible for a team to efficiently collaborate.This rule checks that all local variables follow a naming convention.</p>
</div>
<div class="paragraph">
<p>The default configuration is:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Camel casing, starting with a lower case character, e.g. backColor</p>
</li>
<li>
<p>Short abbreviations of 2 letters can be capitalized only when not at the beginning, e.g. id, productID</p>
</li>
<li>
<p>Longer abbreviations need to be lower cased, e.g. html</p>
</li>
</ul>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="paragraph">
<p>With the default regular expression <code>^[a-z][a-z0-9]*([A-Z]{1,3}[a-z0-9]+)*([A-Z]{2})?$</code>:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>Module Module1
    Sub Main()
        Dim Foo = 0 ' Noncompliant
    End Sub
End Module</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>Module Module1
    Sub Main()
        Dim foo = 0 ' Compliant
    End Sub
End Module</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Rename this local variable to match the regular expression: "xxx".</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="ulist">
<ul>
<li>
<p>key = format
Description: Regular expression used to check the names against.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Default Value for Java : ^[a-z][a-zA-Z0-9]*$</p>
</div>
<div class="paragraph">
<p>Default Value for Flex : ^[_a-z][a-zA-Z0-9]*$</p>
</div>
<div class="paragraph">
<p>Default Value for Python : ^[<em>a-z][a-z0-9</em>]*$</p>
</div>
<div class="paragraph">
<p>Default Value for Swift : ^[a-z][a-zA-Z0-9]*$</p>
</div>
<div class="paragraph">
<p>Default Value for PHP : ^[a-z][a-zA-Z0-9]*$</p>
</div>
<div class="paragraph">
<p>Default Value for VB.NET: ^[a-z]<strong class="a-z0-9">([A-Z]{1,3}[a-z0-9]+)</strong>([A-Z]{2})?$</p>
</div>
<div class="paragraph">
<p>Default Value for TSQL: @[a-zA-Z0-9_]*</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s116">relates to: S116</h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s1545">is related to: S1545</h3>

</div>
<div class="sect2">
<h3 id="_on_20_may_2013_142618_fabrice_bellingard_wrote">on 20 May 2013, 14:26:18 Fabrice Bellingard wrote:</h3>
<div class="paragraph">
<p>Implementation: http://jira.codehaus.org/browse/SONARJAVA-146</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_30_oct_2013_122023_freddy_mallet_wrote">on 30 Oct 2013, 12:20:23 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3215 for Flex</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_19_feb_2014_155318_freddy_mallet_wrote">on 19 Feb 2014, 15:53:18 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3497 for PHP</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_5_dec_2016_144326_tamas_vajk_wrote">on 5 Dec 2016, 14:43:26 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~pierre-yves.nicolas], [~evgeny.mandrikov] I removed the exception part from the root rule and added it to Java. If any of your languages also implement that exception, could you add it to the appropriate subtask?</p>
</div>
</div>
</div>
</div>