<div class="paragraph">
<p>Rethrowing an unmodified copy of the caught exception is a waste of resources. Additionally, doing so may lead to a loss of precision in the object type and its data, since the copy will be an instance of the base class, rather than of the potentially more specific exception class originally caught.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>try {
  throw std::invalid_argument("x");
} catch (const std::exception&amp; ex) {
  /* ... */
  throw ex; // Noncompliant; the received "std::invalid_argument" is copied into a less specialized class "std::exception"
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>try {
  throw std::invalid_argument("x");
} catch (const std::exception&amp; ex) {
  /* ... */
  throw; // rethrows the initial "std::invalid_argument"
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove "XXX" from this "throw" statement to rethrow the original exception.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>variable after the "throw" keyword</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_4_aug_2015_195727_ann_campbell_wrote">on 4 Aug 2015, 19:57:27 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~tamas.vajk] I&#8217;ve mapped this to FxCop RethrowToPreserveStackDetails even though the reasoning seems to be different, the title and code samples are remarkably similar.</p>
</div>
<div class="paragraph">
<p>\[~evgeny.mandrikov] I have found only nebulous hints that the stacktrace is reset in C&#43;&#43; in this Noncompliant scenario (which is the logic behind the C# FxCop rule). Can you confirm/deny? If true, it would be a valuable addition to the description IMO.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_21_feb_2017_154102_ann_campbell_wrote">on 21 Feb 2017, 15:41:02 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Nice find [~alexandre.gigleux]. Mapped.</p>
</div>
</div>
</div>
</div>