<div class="paragraph">
<p>Marking a non-public method <code>@Transactional</code> is both useless and misleading because Spring doesn&#8217;t "see" non-<code>public</code> methods, and so makes no provision for their proper invocation. Nor does Spring make provision for the methods invoked by the method it called.</p>
</div>
<div class="paragraph">
<p>Therefore marking a <code>private</code>  method, for instance, <code>@Transactional</code> can only result in a runtime error or exception if the method is actually written to be <code>@Transactional</code>.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>@Transactional  // Noncompliant
private void doTheThing(ArgClass arg) {
  // ...
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Make this method "public" or remove the "@Transactional" annotation</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_21_nov_2014_122829_freddy_mallet_wrote">on 21 Nov 2014, 12:28:29 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Two questions/remarks:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Are we talking about private methods or about non-public methods ? If my feeling is correct this rule should only target private methods</p>
</li>
<li>
<p>I would tag the rule with the label "spring"</p>
</li>
<li>
<p>As this rule is associated to the Reliability characteristic, I think the default severity should be "Critical"</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_on_21_nov_2014_132808_ann_campbell_wrote">on 21 Nov 2014, 13:28:08 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>The Spring docs are pretty clear that only <code>public</code> method can actually be <code>@Transactional</code></p>
</div>
</div>
<div class="sect2">
<h3 id="_on_21_nov_2014_141444_freddy_mallet_wrote">on 21 Nov 2014, 14:14:44 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Ok Ann, so I would replace :</p>
</div>
<div class="paragraph">
<p>"Therefore marking a private method"</p>
</div>
<div class="paragraph">
<p>by</p>
</div>
<div class="paragraph">
<p>"Therefore marking for instance a private method"</p>
</div>
<div class="paragraph">
<p>to prevent any misunderstanding</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_27_nov_2018_130643_semyon_danilov_wrote">on 27 Nov 2018, 13:06:43 Semyon Danilov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Actually, any method can be Transactional if you&#8217;re using AspectJ compiler, it&#8217;s stated in the docs <a href="https://docs.spring.io/spring/docs/4.2.x/spring-framework-reference/html/transaction.html">here</a>. The excerpt:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>Method visibility and @Transactional

When using proxies, you should apply the @Transactional annotation only to methods with public visibility. If you do annotate protected, private or package-visible methods with the @Transactional annotation, no error is raised, but the annotated method does not exhibit the configured transactional settings. Consider the use of AspectJ (see below) if you need to annotate non-public methods.</pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_on_27_nov_2018_133330_ann_campbell_wrote">on 27 Nov 2018, 13:33:30 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>FYI [~alexandre.gigleux] ^</p>
</div>
</div>
</div>
</div>