<div class="paragraph">
<p>When multiple tables are involved in a query, using table aliases helps to make it more understandable and keeps it short.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>BEGIN
  SELECT
    name,
    firstname,
    location
  INTO employeesArray
  FROM employee -- Noncompliant - should be aliased
  INNER JOIN department -- Noncompliant - should be aliased
  ON employee.DepartmentID = department.ID;
END;
/</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>BEGIN
  SELECT
    empl.name,
    empl.firstname,
    dpt.location
  INTO employeesArray
  FROM employee empl
  INNER JOIN department dpt
  ON empl.DepartmentID = dpt.ID;
END;
/</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add aliases for the tables in this query.</p>
</div>
</div>
</div>
</div>