<div class="paragraph">
<p>Shared coding conventions allow teams to collaborate efficiently. This rule checks that all constant names match a provided regular expression.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="paragraph">
<p>With the default regular expression <code>^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$</code>:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>define("const1", true);

class Foo {
    const const2 = "bar";
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>define("CONST1", true);

class Foo {
    const CONST2 = "bar";
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Rename this constant "XXX" to match the regular expression ${format}.</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="ulist">
<ul>
<li>
<p>key = format
Description: Regular expression used to check the constant names against.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Default Value: ^[A-Z]<strong class="A-Z0-9">(_[A-Z0-9]+)</strong>$</p>
</div>
<div class="paragraph">
<p>Python default: [A-Z_][A-Z0-9_]*)|(<em>.*</em>$</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_20_may_2013_122824_fabrice_bellingard_wrote">on 20 May 2013, 12:28:24 Fabrice Bellingard wrote:</h3>
<div class="paragraph">
<p>Implementation: http://jira.codehaus.org/browse/SONARJAVA-144</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_21_oct_2013_155648_freddy_mallet_wrote">on 21 Oct 2013, 15:56:48 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3188 for Flex</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_23_feb_2014_230247_freddy_mallet_wrote">on 23 Feb 2014, 23:02:47 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3510 for PHP</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_4_nov_2014_160255_ann_campbell_wrote">on 4 Nov 2014, 16:02:55 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>pylint:C0103</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_11_jun_2015_202126_ann_campbell_wrote">on 11 Jun 2015, 20:21:26 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>CodePro: Enumerated Type Naming Convention</p>
</div>
</div>
</div>
</div>