<div class="paragraph">
<p>The <code>open</code> builtin can open files in different modes, which are provided as a combination of characters. Using an invalid sequence of characters will at best make <code>open</code> fail, or worse, it will have an undefined behavior (ex: it might ignore some characters).</p>
</div>
<div class="paragraph">
<p>A valid mode:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>should contain one and only one of the following characters: "r" (read), "w" (write), "a" (append), "x" (create).</p>
</li>
<li>
<p>should contain zero or one of the following characters: "t" (text), "b" (binary).</p>
</li>
<li>
<p>should contain zero or one "+" character (open for updating)</p>
</li>
<li>
<p>cannot contain "a", "w", "+", or "x" if "U" (universal newlines) is used. Note that "U" has no effect in python 3, it is deprecated and shouldn&#8217;t be used anymore.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>This rule raises an issue when an invalid "mode" is provided to the <code>open</code> builtin.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre># In python 3 the following fails
# In python 2.7.16 on MacOs, "open" will just ignore the "w" flag
with open("test.txt", "aw") as f:  # Noncompliant
    pass</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>with open("test.txt", "a") as f:
    pass</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.python.org/3/library/functions.html#open">Python documentation - the <code>open</code> builtin</a></p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Fix this invalid mode string.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The mode parameter</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s5488">is related to: S5488</h3>

</div>
</div>
</div>