<div class="paragraph">
<p><code>switch</code> can contain a <code>default</code> clause for various reasons: to handle unexpected values, to show that all the cases were properly considered.</p>
</div>
<div class="paragraph">
<p>For readability purpose, to help a developer to quickly find the default behavior of a <code>switch</code> statement, it is recommended to put the <code>default</code> clause at the beginning or the end of the <code>switch</code> statement. This rule raises an issue if the <code>default</code> clause is not the first or the last one of the <code>switch</code>'s cases.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>switch tag {
case 0, 1, 2, 3:
	foo()
default:  // Noncompliant; default case should be the first or last one
	qix()
case 4, 5, 6, 7:
	bar()
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>switch tag {
default:
	qix() // Compliant; default is the first one
case 0, 1, 2, 3:
	foo()
case 4, 5, 6, 7:
	bar()
}</pre>
</div>
</div>
<div class="listingblock">
<div class="content">
<pre>switch tag {
case 0, 1, 2, 3:
	foo()
case 4, 5, 6, 7:
	bar()
default:
	qix() // Compliant; default is the last one
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Move this "[case default|WHEN OTHERS|Case Else]" clause to the beginning or end of this "[switch|CASE|Select]" statement.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_24_apr_2018_165220_alban_auzeill_wrote">on 24 Apr 2018, 16:52:20 Alban Auzeill wrote:</h3>
<div class="paragraph">
<p>Irrelevant for Swift because the compiler already enforce "default" to be the latest.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_26_jul_2018_140209_pierre_yves_nicolas_wrote">on 26 Jul 2018, 14:02:09 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>\[~alexandre.gigleux] The change of the title from "last" to "first or last" require:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>some changes in descriptions, code examples, issue message</p>
</li>
<li>
<p>changes in the implementation of the rule for the languages which already implemented the rule and for which it makes sense</p>
</li>
</ul>
</div>
</div>
</div>
</div>