<div class="paragraph">
<p>A <code>for</code> loop with a stop condition that can never be reached, such as one with a counter that moves in the wrong direction, will run infinitely. While there are occasions when an infinite loop is intended, the convention is to construct such loops as <code>while</code> loops. More typically, an infinite <code>for</code> loop is a bug.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>for (var i = 0; i &lt; strings.length; i--) { // Noncompliant;
  //...
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>for (var i = 0; i &lt; strings.length; i++) {
  //...
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>"xxx" is (inc|dec)remented and will never reach "stop condition".</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_13_mar_2015_140309_pierre_yves_nicolas_wrote">on 13 Mar 2015, 14:03:09 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>In JavaScript, moving the counter in the wrong direction causes an infinite loop because there is no integer type. All numbers are 64 bit floating point numbers.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_2_dec_2014_092956_pierre_yves_nicolas_wrote">on 2 Dec 2014, 09:29:56 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>In fact, it&#8217;s not an infinite loop because of integer overflow. On my computer, the noncompliant example above finishes in less than a second.</p>
</div>
</div>
</div>
</div>