<div class="paragraph">
<p>As stated per effective java :</p>
</div>
<div class="quoteblock">
<blockquote>
<div class="paragraph">
<p>Varargs methods are a convenient way to define methods that require a variable number of arguments, but they should not be overused. They can produce confusing results if used inappropriately.</p>
</div>
</blockquote>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>void fun ( String... strings )	// Noncompliant
{
  // ...
}</pre>
</div>
</div>
</div>
</div>