<div class="paragraph">
<p>A selector argument is a <code>boolean</code> argument that&#8217;s used to determine which of two paths to take through a method. Specifying such a parameter may seem innocuous, particularly if it&#8217;s well named.</p>
</div>
<div class="paragraph">
<p>Unfortunately, the maintainers of the code calling the method won&#8217;t see the parameter name, only its value. They&#8217;ll be forced either to guess at the meaning or to take extra time to look the method up.</p>
</div>
<div class="paragraph">
<p>Instead, separate methods should be written.</p>
</div>
<div class="paragraph">
<p>This rule finds methods with a <code>boolean</code> that&#8217;s used to determine which path to take through the method.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public String tempt(String name, boolean ofAge) {
  if (ofAge) {
    offerLiquor(name);
  } else {
    offerCandy(name);
  }
}

// ...
public void corrupt() {
  tempt("Joe", false); // does this mean not to temp Joe?
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public void temptAdult(String name) {
  offerLiquor(name);
}

public void temptChild(String name) {
    offerCandy(name);
}

// ...
public void corrupt() {
  age &lt; legalAge ? temptChild("Joe") : temptAdult("Joe");
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Provide multiple methods instead of using "xxx" to determine which action to take.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_duplicated_by_s2495">is duplicated by: S2495</h3>

</div>
</div>
</div>