<div class="paragraph">
<p>Using upper case literal suffixes removes the potential ambiguity between "1" (digit 1) and "l" (letter el) for declaring literals.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>const long b = 0l;      // Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>const long b = 0L;</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Upper case this literal suffix.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_deprecates_s1129">deprecates: S1129</h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s1129">is related to: S1129</h3>

</div>
<div class="sect2">
<h3 id="_on_31_mar_2015_190623_evgeny_mandrikov_wrote">on 31 Mar 2015, 19:06:23 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] implementation seems more complete (SQALE, description) than this spec.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_31_mar_2015_190646_evgeny_mandrikov_wrote">on 31 Mar 2015, 19:06:46 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>Also I believed that this rule is implemented in Java or at least applicable.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_2_oct_2015_101753_tamas_vajk_wrote">on 2 Oct 2015, 10:17:53 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] The message contains upper-case with a dash, while none of the other occurrences use the dash. Is this on purpose? Also, the description doesn&#8217;t explain why we should upper-case <code>f</code>. Shouldn&#8217;t we add something like "For consistency, all literal suffixes should be upper-case."</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_28_jun_2016_081456_michael_piefel_wrote">on 28 Jun 2016, 08:14:56 Michael Piefel wrote:</h3>
<div class="paragraph">
<p>Whether <code>d</code> and <code>f</code> must be upper-case is quite debatable; itâ€™s a matter of style and personal preference. I have yet to see a font where <code>D</code> and <code>0</code> really get mixed up; but this would even be in favour of letting it be lower-case.</p>
</div>
<div class="paragraph">
<p>This should be made configurable. I do not fancy changing a big codebase to comply to this; and neither do I want to give up the check on <code>l</code> versus <code>L</code>.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_28_jun_2016_150824_michael_gumowski_wrote">on 28 Jun 2016, 15:08:24 Michael Gumowski wrote:</h3>
<div class="paragraph">
<p>Thanks for your feedback [~piefel]. We updated the JAVA part of the rule (RSPEC-3339) in order to add a parameter allowing the rule to only focus on long.</p>
</div>
<div class="paragraph">
<p>See corresponding JIRA ticket SONARJAVA-1755.</p>
</div>
<div class="paragraph">
<p>For next time, please use directly our mailing list (sonarqube@googlegroups.com) rather than directly posting into JIRA tickets. It will allow other members of the community to participate to discussions.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_9_mar_2018_160605_janos_gyerik_wrote">on 9 Mar 2018, 16:06:05 Janos Gyerik wrote:</h3>
<div class="paragraph">
<p>Irrelevant for Go. There are no such nonsensical suffixes for numbers.</p>
</div>
</div>
</div>
</div>