<div class="paragraph">
<p>Shared naming conventions allow teams to collaborate efficiently. This rule checks that all element names match a provided regular expression.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="paragraph">
<p>With the default regular expression:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>&lt;xs:element name="Fruit"&gt;     &lt;!-- Noncompliant --&gt;
  ...
&lt;/xs:element&gt;</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>&lt;xs:element name="fruit"&gt;
  ...
&lt;/xs:element&gt;</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Rename this element/attribute to match the regular expression: ${format}</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="ulist">
<ul>
<li>
<p>key = format
Description: Regular expression used to check element names against.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Default: ^<span class="a-z">([A-Z][a-z0-9]</span>)*$</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>name value</p>
</div>
</div>
</div>
</div>