<div class="paragraph">
<p>To perform calculations when specifying a CSS property <code>calc()</code> function can be used. This function takes single expression as parameter. When writing this expression some rules must be respected:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>no empty <code>calc()</code></p>
</li>
<li>
<p>there should be an operator between the arguments, spacing should be respected</p>
</li>
<li>
<p>there should not be any division by zero</p>
</li>
<li>
<p>the resolved type should be valid for where the expression is placed</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Otherwise <code>calc()</code> function will be invalid and the entire rule using it will be ignored.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>.btn {
  border: solid black 1px;
  width: calc(100% 80px);  /* Noncompliant */
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>.btn {
  border: solid black 1px;
  width: calc(100% - 80px);
}</pre>
</div>
</div>
</div>
</div>