<div class="paragraph">
<p>An array declared with missing ("sparse") elements is likely to be an error: an extra comma was inserted or perhaps the developer meant to insert the missing value and forgot.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>let a = [1, , 3, 6, 9];  // Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>let a = [1, 3, 6, 9];</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Either supply the missing element or remove the extra comma</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The sparse comma</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_8_nov_2017_095607_michael_gumowski_wrote">on 8 Nov 2017, 09:56:07 Michael Gumowski wrote:</h3>
<div class="paragraph">
<p>Removing PHP from the list of targeted language. You can not declare sparse arrays in PHP. The following code does not compile:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>$array = [1, 2, 3, , 4];</pre>
</div>
</div>
<div class="paragraph">
<p>However, sparse arrays can also be seen as a feature of the language. In the following code, the array is defined with 3 values, but the internal representation of the array is a map, which will use 3 keys: <code>14</code>, <code>15</code> and <code>42</code>.</p>
</div>
<div class="listingblock">
<div class="content">
<pre>$array[14] = "hello";
$array[15] = "world";
$array[42] = "yolo";</pre>
</div>
</div>
</div>
</div>
</div>