<div class="paragraph">
<p>Creating an extension method that extends <code>object</code> is not recommended because it makes the method available on <em>every</em> type. Extensions should be applied at the most specialized level possible, and that is very unlikely to be <code>object</code>.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>using System;

namespace MyLibrary
{
    public static class MyExtensions
    {
        public static void Foo(this object o)  //Noncompliant
        {
            // ...
        }
    }
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Refactor this extension to extend a more concrete type.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The extension method declaration</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_9_jan_2018_101007_amaury_levé_wrote">on 9 Jan 2018, 10:10:07 Amaury Levé wrote:</h3>
<div class="paragraph">
<p>The rule wasn&#8217;t picked up by the rule-api automatic change.</p>
</div>
</div>
</div>
</div>