<div class="paragraph">
<p>According to the Java Language Specification:</p>
</div>
<div class="quoteblock">
<blockquote>
<div class="paragraph">
<p>It is permitted, but discouraged as a matter of style, to redundantly specify the <code>public</code> and/or <code>abstract</code> modifier for a method declared in an interface.</p>
</div>
</blockquote>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public interface Task{
  public abstract void execute();
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public interface Task{
  void execute();
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove useless 'xxx' modifier</p>
</div>
</div>
</div>
</div>