<div class="paragraph">
<p>Classes without <code>public static</code> members cannot be used without being instantiated, but classes with only <code>private</code> constructors cannot be instantiated. When a class has only <code>private</code> constructors and no <code>static</code> members, it is useless and should be removed or refactored.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public class MyClass {  // Noncompliant

  private MyClass () {
  }

  public double getPi(){
    return 3.14;
  }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public class MyClass {

  public MyClass () {
  }

  public double getPi(){
    return 3.14;
  }
}</pre>
</div>
</div>
<div class="paragraph">
<p>or</p>
</div>
<div class="listingblock">
<div class="content">
<pre>public class MyClass {

  private MyClass () {
  }

  public static double getPi(){
    return 3.14;
  }
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Either add a public constructor or make the methods of this class "static".</p>
</div>
</div>
</div>
</div>