<div class="paragraph">
<p>Catching an exception only to immediately rethrow it without doing anything else is useless and misleading.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>try {
  /* ... */
} catch (Exception e) {                   // Non-Compliant
  throw e;
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_exceptions">Exceptions</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When all instances of a general exception must be handled, but some specific ones not, propagation must be used and so is allowed by this rule.</p>
</div>
<div class="listingblock">
<div class="content">
<pre>try {
  /* ... */
} catch (RuntimeException e) {            // Compliant - propagation of the unchecked exception
  throw e;
} catch (Exception e) {                   // Compliant - catching of the checked exception
  LOGGER.error("...", e);
}</pre>
</div>
</div>
<div class="paragraph">
<p>Throwing the same exception can also makes sense when an action is done before throwing it again.</p>
</div>
<div class="listingblock">
<div class="content">
<pre>try {
  /* ... */
} catch (MyException e) {            // Compliant - something is done before throwing again the exception
  fixSomething();
  throw e;
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_duplicates_s2737">duplicates: S2737</h3>

</div>
</div>
</div>