<div class="paragraph">
<p>This rule raised an issue when a component is used in the <code>render()</code> method and not <code>import</code>ed or defined.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>&lt;Rating value={this.props.measure.reliability_rating} /&gt;;</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>import Rating from './Rating'
[...]
&lt;Rating value={this.props.measure.reliability_rating} /&gt;</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add the missing import of "XXX".</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Primary: first occurrence of the component usage</p>
</div>
<div class="paragraph">
<p>Secondary: all other places in the file where this component is used</p>
</div>
</div>
</div>
</div>