<div class="paragraph">
<p>In property and indexer <code>set</code> methods, and in event <code>add</code> and <code>remove</code> methods, the implicit <code>value</code> parameter holds the value the accessor was called with. Not using the <code>value</code> means that the accessor ignores the caller&#8217;s intent which could cause unexpected results at runtime.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>private int count;
public int Count
{
  get { return count; }
  set { count = 42; } // Noncompliant
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>private int count;
public int Count
{
  get { return count; }
  set { count = value; }
}</pre>
</div>
</div>
<div class="paragraph">
<p>or</p>
</div>
<div class="listingblock">
<div class="content">
<pre>public int Count
{
  get { return count; }
  set { throw new InvalidOperationException(); }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_exceptions">Exceptions</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule doesn&#8217;t raise an issue when the setter is empty and part of the implementation of an <code>interface</code> . The assumption is that this part of the interface is not meaningful to that particular implementation. A good example of that would be a "sink" logger that discards any logs.</p>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use the "value" parameter in this [property set|indexer set|event] accessor.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_8_jul_2015_135134_ann_campbell_wrote">on 8 Jul 2015, 13:51:34 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~tamas.vajk] 10min seems like a high remediation cost. Does that mean that the Compliant Solution I added is off-base?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_20_jul_2015_115905_tamas_vajk_wrote">on 20 Jul 2015, 11:59:05 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] We can reduce the required time.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_20_jul_2015_135316_ann_campbell_wrote">on 20 Jul 2015, 13:53:16 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>I halved it to 5min, [~tamas.vajk]</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_27_jul_2015_153903_ann_campbell_wrote">on 27 Jul 2015, 15:39:03 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~dinesh.bolkensteyn] note that I&#8217;ve edited the first line. Your version, "a property and indexer <code>set</code> method" speaks of one, collective method for both a property and an indexer at one time.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_27_jul_2015_154458_dinesh_bolkensteyn_wrote">on 27 Jul 2015, 15:44:58 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>thanks [~ann.campbell.2]</p>
</div>
</div>
</div>
</div>