<div class="paragraph">
<p>Any statement after a <code>STOP RUN</code> or <code>GOBACK</code> is unreachable and therefore dead code which should be removed.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>PARAGRAPH1.
  MOVE A TO B.
  STOP RUN.
  MOVE B TO C.</pre>
</div>
</div>
<div class="paragraph">
<p>or</p>
</div>
<div class="listingblock">
<div class="content">
<pre>PARAGRAPH1.
  MOVE A TO B.
  GOBACK.
  MOVE B TO C.</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>PARAGRAPH1.
  MOVE A TO B.
  MOVE B TO C.
  STOP RUN.</pre>
</div>
</div>
<div class="paragraph">
<p>or</p>
</div>
<div class="listingblock">
<div class="content">
<pre>PARAGRAPH1.
  MOVE A TO B.
  MOVE B TO C.
  GOBACK.</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>This [STOP RUN|GOBACK] statement should be moved to the end the statement sequence.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s1577">relates to: S1577</h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s1763">is related to: S1763</h3>

</div>
</div>
</div>