<div class="paragraph">
<p>Naming the parameters in a function prototype helps identify how they&#8217;ll be used by the function, thereby acting as a thin layer of documentation for the function.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>void divide (int, int);</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>void divide (int numerator, int denominator);</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C:2004, 16.3 - Identifiers shall be given for all of the parameters in a function prototype declaration</p>
</li>
<li>
<p>MISRA C:2012, 8.2 - Function types shall be in prototype form with named parameters</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Give names to the parameters of this function prototype.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s819">relates to: S819</h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s927">relates to: S927</h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s829">is related to: S829</h3>

</div>
</div>
</div>