<div class="paragraph">
<p>Multi-line comments are more readable when each line is aligned using the "*" character. At most one violation is created for each comment</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>/*
this line is not aligned and ugly Non-Compliant
no violation is created on this line, even though is it also bad
 */

/* this is Compliant */</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>/*
 * this is much better Compliant
 */

/* this is Compliant */</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Start the lines in this comment block with "*".</p>
</div>
</div>
</div>
</div>