<div class="paragraph">
<p>There&#8217;s no reason to import modules you don&#8217;t use; and every reason not to: doing so needlessly increases the load.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>import A from 'a';      // Noncompliant, A isn't used
import { B1 } from 'b';

console.log(B1);</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>import { B1 } from 'b';

console.log(B1);</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>Remove this unused import of {}.</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Primary: Import to remove/merge</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_28_jan_2019_115108_nicolas_harraudeau_wrote">on 28 Jan 2019, 11:51:08 Nicolas Harraudeau wrote:</h3>
<div class="paragraph">
<p>Message updated to match the implementation. This is part of the title cleaning for this rule.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_12_jul_2013_124132_dinesh_bolkensteyn_wrote">on 12 Jul 2013, 12:41:32 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>Implemented by http://jira.codehaus.org/browse/SONARJAVA-207</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_12_jul_2013_132112_dinesh_bolkensteyn_wrote">on 12 Jul 2013, 13:21:12 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>Why not enabled by default?</p>
</div>
<div class="paragraph">
<p>I guess this should be set to MINOR and enabled by default, it&#8217;s simply a matter of proper IDE configuration, and should never happen.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_12_jul_2013_153331_dinesh_bolkensteyn_wrote">on 12 Jul 2013, 15:33:31 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>IMO this is not much different from an unused variable, it does not harm, but it should be avoided.</p>
</div>
<div class="paragraph">
<p>I&#8217;ve updated the spec.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_9_nov_2016_171826_ann_campbell_wrote">on 9 Nov 2016, 17:18:26 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>XML: https://groups.google.com/forum/?pli=1#!topic/sonarqube/wcQMDPPC0OM</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_24_oct_2019_183200_li_huan_wrote">on 24 Oct 2019, 18:32:00 Li Huan wrote:</h3>
<div class="paragraph">
<p>In C#, this rule incorrectly removes imports used in XML comments.</p>
</div>
</div>
</div>
</div>