<div class="paragraph">
<p>Although unnecessary won&#8217;t change anything to the produced application, removing them:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Will help readability and maintenance.</p>
</li>
<li>
<p>Will help reduce the number of items in the IDE auto-completion list when coding.</p>
</li>
<li>
<p>May avoid some name collisions.</p>
</li>
<li>
<p>May improve compilation time because the compiler has fewer namespaces to look-up when it resolves types.</p>
</li>
</ul>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>Imports System.Collections.Generic // Noncompliant - unnecessary using

Module Module1
    Sub Main(path As String)
        File.ReadAllLines(path);
    End Sub
End Module</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>Imports System.IO

Module Module1
    Sub Main(path As String)
        File.ReadAllLines(path);
    End Sub
End Module</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>Remove this unused import.</p>
</li>
<li>
<p>Remove this duplicate import.</p>
</li>
<li>
<p>Remove this unnecessary import.</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_23_sep_2019_163348_nicolas_harraudeau_wrote">on 23 Sep 2019, 16:33:48 Nicolas Harraudeau wrote:</h3>
<div class="paragraph">
<p><strong>OUT OF SCOPE</strong></p>
</div>
<div class="paragraph">
<p>Duplicate imports are out of scope as dotnet won&#8217;t even compile code in this case.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_12_jul_2013_124132_dinesh_bolkensteyn_wrote">on 12 Jul 2013, 12:41:32 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>Implemented by http://jira.codehaus.org/browse/SONARJAVA-207</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_12_jul_2013_132112_dinesh_bolkensteyn_wrote">on 12 Jul 2013, 13:21:12 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>Why not enabled by default?</p>
</div>
<div class="paragraph">
<p>I guess this should be set to MINOR and enabled by default, it&#8217;s simply a matter of proper IDE configuration, and should never happen.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_12_jul_2013_153331_dinesh_bolkensteyn_wrote">on 12 Jul 2013, 15:33:31 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>IMO this is not much different from an unused variable, it does not harm, but it should be avoided.</p>
</div>
<div class="paragraph">
<p>I&#8217;ve updated the spec.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_9_nov_2016_171826_ann_campbell_wrote">on 9 Nov 2016, 17:18:26 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>XML: https://groups.google.com/forum/?pli=1#!topic/sonarqube/wcQMDPPC0OM</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_24_oct_2019_183200_li_huan_wrote">on 24 Oct 2019, 18:32:00 Li Huan wrote:</h3>
<div class="paragraph">
<p>In C#, this rule incorrectly removes imports used in XML comments.</p>
</div>
</div>
</div>
</div>