<div class="paragraph">
<p><code>System.Collections.Generic.List&lt;T&gt;</code> is a generic collection that is designed for performance and not inheritance. For example, it does not contain virtual members that make it easier to change the behavior of an inherited class. That means that future attempts to expand the behavior will be spoiled because the extension points simply aren&#8217;t there. Instead, one of the following generic collections should be used:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>System.Collections.Generic.IEnumerable&lt;T&gt;</code></p>
</li>
<li>
<p><code>System.Collections.Generic.IReadOnlyCollection&lt;T&gt;</code></p>
</li>
<li>
<p><code>System.Collections.Generic.ICollection&lt;TKey&gt;</code></p>
</li>
<li>
<p><code>System.Collections.Generic.IReadOnlyList&lt;T&gt;</code></p>
</li>
<li>
<p><code>System.Collections.Generic.IList&lt;TKey&gt;</code></p>
</li>
<li>
<p><code>System.Collections.ObjectModel.Collection&lt;T&gt;</code></p>
</li>
<li>
<p><code>System.Collections.ObjectModel.ReadOnlyCollection&lt;T&gt;</code></p>
</li>
<li>
<p><code>System.Collections.ObjectModel.KeyedCollection&lt;TKey, Titem&gt;</code></p>
</li>
</ul>
</div>
<div class="paragraph">
<p>This rule raises an issue every time a <code>System.Collections.Generic.List&lt;T&gt;</code> is exposed:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>As an externally visible member.</p>
</li>
<li>
<p>As the return type of an externally visible method.</p>
</li>
<li>
<p>As a parameter type of an an externally visible method.</p>
</li>
</ul>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>namespace Foo
{
   public class Bar
   {
      public List&lt;T&gt; Method1(T arg) // Noncompliant
      {
           //...
      }
   }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>namespace Foo
{
   public class Bar
   {
      public Collection&lt;T&gt; Method1(T arg)
      {
           //...
      }
   }
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Refactor this [method|field|property] to use a generic collection designed for inheritance.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The generic list declaration</p>
</div>
</div>
</div>
</div>