<div class="paragraph">
<p>Parameters that are declared as "INONLY" or "NONASSIGNABLE" are flagged as non-assignable by the compiler.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when a non-assignable parameter is written to.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>sum: PROC(a, b);
   dcl a fixed dec(15,2) NONASSIGNABLE;
   dcl b fixed dec(15,2) NONASSIGNABLE;
   a = a + b; /* Noncompliant */
   return (a);
end;</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Refactor this code in order not to assign a value to this parameter.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Primary: Assignation</p>
</div>
<div class="paragraph">
<p>Secondary: Parameter declaration</p>
</div>
</div>
</div>
</div>