<div class="paragraph">
<p>The <code>defined</code> preprocessing directive is used in the context of <code>#if</code> and <code>#elif</code> expressions to see whether a given identifier has been defined as a macro. It returns a value of 0 (false) or 1 (true), and has two valid forms, <code>defined IDENTIFIER</code> and <code>defined ( IDENTIFIER )</code>. Since it is essentially a macro existence check, it cannot take expressions as arguments.</p>
</div>
<div class="paragraph">
<p>Note that since</p>
</div>
<div class="paragraph">
<p><code>#if defined AN_IDENTIFIER</code></p>
</div>
<div class="paragraph">
<p>is equivalent to</p>
</div>
<div class="paragraph">
<p><code>#ifdef AN_IDENTIFIER</code></p>
</div>
<div class="paragraph">
<p><code>defined</code> is most useful when there are multiple arguments to check, E.G.</p>
</div>
<div class="paragraph">
<p><code>#if defined AAA || defined BBB</code></p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>#if defined ( X &gt; Y ) // Noncompliant; expressions not allowed</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>#if defined X &amp;&amp; defined Y &amp;&amp; X &gt; Y</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C:2004, 19.14 - The defined preprocessor operator shall only be used in one of the two standard forms.</p>
</li>
<li>
<p>MISRA C&#43;&#43;:2008, 16-1-1 - The defined preprocessor operator shall only be used in one of the two standard forms.</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Correct or remove this use of "defined".</p>
</div>
</div>
</div>
</div>