<div class="paragraph">
<p>When the equality operator '==' is unexpectedly used in place of the assignment operator '=', this leads to execute a useless condition expression and to not do the expected assignment.</p>
</div>
<div class="paragraph">
<p>By default, clang compiler for instance generates a warning in such case but doesn&#8217;t fail the build.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>a == 4;    //noncompliant
...
for( a == 0; a &lt; 10; a++){  //noncompliant
 ...
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>a = 4;
...
for( a = 0; a &lt; 10; a++){
 ...
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_duplicates_s1875">duplicates: S1875</h3>

</div>
</div>
</div>