<div class="paragraph">
<p>It is well documented that while <code>HttpClient</code> implements <code>IDisposable</code>, it is intended to be reused, which means not wrapping it in a <code>using</code> statement.</p>
</div>
<div class="paragraph">
<p>As noted in the documentation:</p>
</div>
<div class="quoteblock">
<blockquote>
<div class="paragraph">
<p>HttpClient is intended to be instantiated once and re-used throughout the life of an application. Instantiating an HttpClient class for every request will exhaust the number of sockets available under heavy loads. This will result in SocketException errors.</p>
</div>
</blockquote>
</div>
<div class="paragraph">
<p>This rule raises an issue when an instance of a <code>HttpClient</code> is created within a <code>using</code> statement.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>namespace MyLibrary
{
    public class Foo
    {
        public async Task Bar()
        {
            using (var client = new HttpClient()) // Noncompliant
            {
                await client.GetAsync("https://www.sonarqube.org/");
            }
        }
    }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>namespace MyLibrary
{
    public class Foo
    {
        private static HttpClient client = new HttpClient();

        public async Task Bar()
        {
            await client.GetAsync("https://www.sonarqube.org/");
        }
    }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://msdn.microsoft.com/en-us/library/system.net.http.httpclient(v=vs.110).aspx?f=255&amp;mspperror=-2147217396#Anchor_5">HttpClient Class</a></p>
</li>
<li>
<p><a href="https://docs.microsoft.com/en-us/aspnet/web-api/overview/advanced/calling-a-web-api-from-a-net-client">Call a Web API From a .NET Client (C#)</a></p>
</li>
<li>
<p><a href="https://github.com/mspnp/performance-optimization/blob/master/ImproperInstantiation/docs/LoadTesting.md">Microsoft Patterns &amp; Practices</a></p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Refactor this use of the "HttpClient".</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Object creation.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_15_feb_2018_125215_amaury_levé_wrote">on 15 Feb 2018, 12:52:15 Amaury Levé wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Could you please review?</p>
</div>
</div>
</div>
</div>