<div class="paragraph">
<p>The <code>ACCEPT</code> keyword does no editing or error checking of the data it stores, therefore its use can be dangerous. For this reason, <code>ACCEPT</code> should be avoided.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre> 01 USER-INPUT PIC X(4).

  GET-USER-INPUT.
       MOVE 'N' TO WS-NUMERIC.
       PERFORM UNTIL WS-NUMERIC = 'Y'
           DISPLAY 'ENTER YOUR 4 DIGIT RECORD NUMBER: ' NO ADVANCING
           ACCEPT USER-RECORD *&gt; Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_exceptions">Exceptions</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule ignores uses of <code>ACCEPT FROM</code> with date/time-related inputs.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://owasp.org/Top10/A03_2021-Injection/">OWASP Top 10 2021 Category A3</a> - Injection</p>
</li>
<li>
<p><a href="https://www.owasp.org/index.php/Top_10-2017_A1-Injection">OWASP Top 10 2017 Category A1</a> - Injection</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this use of "ACCEPT".</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>the <code>ACCEPT</code> statement - so both <code>ACCEPT</code> and what the input is being accepted into.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_3_nov_2015_202313_ann_campbell_wrote">on 3 Nov 2015, 20:23:13 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~pierre-yves.nicolas], I wrote the exception based on <a href="http://supportline.microfocus.com/Documentation/AcucorpProducts/docs/v6_online_doc/gtman3/gt3678.htm">this</a> but suspect it should be expanded to include more cases.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_5_nov_2015_120530_pierre_yves_nicolas_wrote">on 5 Nov 2015, 12:05:30 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] The documentation you found is specific to AcuCobol which adds many extensions on top of "standard" COBOL. I suggest to keep only the exception about date/time and to not activate that rule by default.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_5_nov_2015_142813_ann_campbell_wrote">on 5 Nov 2015, 14:28:13 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Okay [~pierre-yves.nicolas]. Done</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_21_mar_2017_175400_pierre_yves_nicolas_wrote">on 21 Mar 2017, 17:54:00 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] The scope of this rule changed completely: it used to be "ACCEPT should not be used" and is now "Track uses of forbidden statements". I don&#8217;t see how we can keep the exception for <code>ACCEPT FROM</code> with the new scope.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_22_mar_2017_090425_ann_campbell_wrote">on 22 Mar 2017, 09:04:25 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~pierre-yves.nicolas] I&#8217;ve reverted this rule &amp; created RSPEC-3938</p>
</div>
</div>
</div>
</div>