<div class="paragraph">
<p>Re-exporting is a feature that allows a file to import a component (object, function or primitive) from a module and add it to the exported components of the file&#8217;s module. However, as the language allows a file to export a component without explicitly importing it, a reader may mistake a re-export with a simple export.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>export { foo } from 'someModule';</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>import { foo } from 'someModule';
export { foo };</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Explicitly import this/these component/s before re-exporting it/them.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The exported component</p>
</div>
</div>
</div>
</div>