<div class="paragraph">
<p>A cast operation makes it possible to convert an object from one type to another one. The compiler raises an error when it can determine that the target type is incompatible with the declared type of the object. Other cases are accepted by the compiler. However, depending on the actual runtime type of the object, a cast operation may fail at runtime with a <code>ClassCastException</code> which can crash the program. This kind of failure happens when a piece of code makes an invalid assumption about the runtime type of an object.</p>
</div>
<div class="paragraph">
<p>The appropriate fix for those cast operations highly depends on the context: adding a condition before the cast or changing the runtime type of the object are some of the possible solutions.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>List&lt;String&gt; list = new LinkedList&lt;&gt;();
if (someCondition) {
  list = new ArrayList&lt;&gt;();
}
((LinkedList&lt;String&gt;) list).addLast("abc"); // Noncompliant, throws a ClassCastException when someCondition is true</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>List&lt;String&gt; list = new LinkedList&lt;&gt;();
if (someCondition) {
  list = new ArrayList&lt;&gt;();
}
if (list instanceof LinkedList) {
  ((LinkedList&lt;String&gt;) list).addLast("abc");
}</pre>
</div>
</div>
<div class="paragraph">
<p>or</p>
</div>
<div class="listingblock">
<div class="content">
<pre>List&lt;String&gt; list = new LinkedList&lt;&gt;();
if (someCondition) {
  list = new ArrayList&lt;&gt;();
}
list.add("abc");</pre>
</div>
</div>
<div class="paragraph">
<p>or</p>
</div>
<div class="listingblock">
<div class="content">
<pre>LinkedList&lt;String&gt; list = new LinkedList&lt;&gt;();
list.addLast("abc");</pre>
</div>
</div>
</div>
</div>