<div class="paragraph">
<p>The <a href="https://www.python.org/dev/peps/pep-0572">walrus operator</a> <code>:=</code> (also known as "assignment expression") should be used with caution as it can easily make code more difficult to understand and thus maintain. In such case it is advised to refactor the code and use an assignment statement (i.e. <code>=</code>) instead.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when the walrus operator is used in a way which makes the code confusing, as described in <a href="https://www.python.org/dev/peps/pep-0572/#exceptional-cases">PEP 572</a>.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre># using an assignment expression (:=) as an assignment statement (=) is more explicit
(v := f(p))  # Noncompliant
v0 = (v1 := f(p))  # Noncompliant

# using an assignment expression in a function call when keyword arguments are also used.
func(a=(b := f(p)))  # Noncompliant
func(a := f(p), b=2)  # Noncompliant
def func(param=(p := 21)):  # Noncompliant
    pass

# using an assignment expression in an annotation
def func(param: (p := 21) = 3):  # Noncompliant
    pass

# using assignment expression in an f-string. Character ":" is also used as a formatting marker in f-strings.
f'{(x:=10)}'  # Noncompliant
f'{x:=10}' # No issue raised but still not recommended. This is not an assignment expression. '=10' is passed to the f-string formatter.</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>v = f(p)
v0 = v1 = f(p)

value = f(p)
func(a=value)
func(value, b=2)
def func(param=21):
    p = 21

p = 21
def func(param: p = 3):
    pass

x = 10
f'{x}'</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://www.python.org/dev/peps/pep-0572/#exceptional-cases">PEP 572 - Assignment Expressions</a></p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>Use an assignment statement ("=") instead; ":=" operator is confusing in this context</p>
</li>
<li>
<p>Move this assignment out of the f-string; ":=" operator is confusing in this context</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The := operator</p>
</div>
</div>
</div>
</div>