<div class="paragraph">
<p>Having duplicated font names doesn&#8217;t help to read the font declaration and may be an indicator the author of the line was not sure how to configure it. This rule raises an issue when <code>font</code> or <code>font-family</code> properties contain a duplicated font name. This rule ignores <code>$sass</code>, <code>@less</code>, and <code>var(--custom-property)</code> variable syntaxes.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>a {
  font-family: 'Georgia', Georgia, serif; /* Noncompliant; 'Georgia' is duplicated */
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>a {
  font-family: Georgia, serif;
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_4_jun_2018_092845_ann_campbell_wrote">on 4 Jun 2018, 09:28:45 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~alexandre.gigleux], there&#8217;s no "why" here.</p>
</div>
</div>
</div>
</div>