<div class="paragraph">
<p><code>fixed bin(31)</code> type has been optimized to run faster than <code>int</code> especially when used in loops. For this reason, <code>fixed bin(31)</code> should be preferred to <code>int</code>.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>dcl i int(2) init (0);
do i = 1 to 20 by 1;
...
end;</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>dcl i fixed bin(31) init (0);
do i = 1 to 20 by 1;
...
end;</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace INT with FIXED BIN(31)</p>
</div>
</div>
</div>
</div>