<div class="paragraph">
<p>Starting from C&#43;&#43;20, mathematical constants are defined in the header <code>&lt;numbers&gt;</code>.</p>
</div>
<div class="paragraph">
<p>You should prefer using them instead of hardcoding your own constants.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>const double pi = 3.14159265358979323846; // Noncompliant

auto computeCirclePerimeter(double radius) {
    return 2 * pi * radius;
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>#include &lt;numbers&gt;

auto computeCirclePerimeter(double radius) {
    return 2 * std::numbers::pi * radius;
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s2904">is related to: S2904</h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s109">is related to: S109</h3>

</div>
</div>
</div>