<div class="paragraph">
<p>Developers sometimes "reserve" an <code>enum</code> member for future use, i.e. these members won&#8217;t be used right but they might change in the future. However changing an <code>enum</code> member is a breaking change and can create significant problems. There is no need to reserve an <code>enum</code> member since a new member can be added in the future, and such an addition will usually not be a breaking change.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when an <code>enum</code> contains multiple members whose name match the pattern <code>.*reserved.*\d+</code>Â (case insensitive) and they all have exactly the same name except for the number at the end.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>using System;

namespace MyLibrary
{
  public enum Color // Noncompliant
  {
        None,
        Red,
        Orange,
        Yellow,
        Reserved_1,  // reserved
        Reserved_2 // reserved
    }
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove reserved members from this "enum".</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>Primary: the enum type name.</p>
</li>
<li>
<p>Secondaries: each reserved member name.</p>
<div class="ulist">
<ul>
<li>
<p>message: 'This member appears to be reserved.'</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_deprecates_s4016">deprecates: S4016</h3>

</div>
</div>
</div>