<div class="paragraph">
<p>String data types (<code>char</code>, <code>varchar</code>, <code>nchar</code>, <code>nvarchar</code>) default to a size of 1 if no size is specified in the declaration. For <code>char</code> and <code>nchar</code> this is confusing at best, but it is most probably a mistake for <code>varchar</code> and <code>nvarchar</code>.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when no size is specified for <code>varchar</code> or <code>nvarchar</code>.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>DECLARE @myStr varchar;  -- Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>DECLARE @myStr varchar(255);</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add a size to this [variable|parameter] declaration</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>variable type</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_6_jul_2017_102801_pierre_yves_nicolas_wrote">on 6 Jul 2017, 10:28:01 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I don&#8217;t think the rule should cover <code>char</code> and <code>nchar</code>. I think it would be quite noisy.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_6_jul_2017_110055_pierre_yves_nicolas_wrote">on 6 Jul 2017, 11:00:55 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Should we update the rule title?</p>
</div>
</div>
</div>
</div>