<div class="paragraph">
<p><code>isMounted()</code> is generally used before calling <code>setState</code> to avoid changing the state on an already unmounted component. A user should never be able to do an action on an unmount component unless the component is having a bug in its unmount phase.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue on each use of <code>isMounted()</code>.</p>
</div>
<div class="paragraph">
<p><code>isMounted()</code> is not longer available in ES6. Use of <code>isMounted()</code> must be avoided to ease the upgrade to ES6.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>if (this.isMounted()) { // Noncompliant
  this.setState({...});
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class MyComponent extends React.Component {

  _isMounted = false;

  componentDidMount() {
    _isMounted = true;
    ...
    if (_isMounted) {
       this.setState({...});
     }
  }
  render() {
    ...
  }
  componentWillUnmount() {
    _isMounted = false;
  }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://reactjs.org/blog/2015/12/16/ismounted-antipattern.html">React.js blog</a> - isMounted is an Antipattern</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_23_feb_2018_114616_alexandre_gigleux_wrote">on 23 Feb 2018, 11:46:16 Alexandre Gigleux wrote:</h3>
<div class="paragraph">
<p>Covered by ESLint for React: https://github.com/yannickcr/eslint-plugin-react/blob/HEAD/docs/rules/no-is-mounted.md (react/no-is-mounted)</p>
</div>
</div>
</div>
</div>