<div class="paragraph">
<p>The use of <code>org.fest.assertions.Assertions.assertThat</code> by itself does nothing. You must combine it with another method that actually tests the value in use.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>assertThat(name);  // Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>assertThat(name).isNotNull();</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_duplicates_s2970">duplicates: S2970</h3>

</div>
<div class="sect2">
<h3 id="_on_22_apr_2015_160223_nicolas_peru_wrote">on 22 Apr 2015, 16:02:23 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] This rule should precise that it checks for <code>org.fest.assertions.Assertions.assertThat</code> methods. As this is not applicable for junit <code>Assert.assertThat </code></p>
</div>
</div>
</div>
</div>