<div class="paragraph">
<p>This rule allows banning certain statements.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Given a parameter of <code>CANCEL</code> for this rule:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>DISPLAY "Cancelling action".
CANCEL PROGRAM1. *&gt; Noncompliant</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this use of "xxx".</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="ulist">
<ul>
<li>
<p>key = statements</p>
<div class="ulist">
<ul>
<li>
<p>description: Comma-separated list of disallowed statements, such as "EXIT PROGRAM, CANCEL"</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>the forbidden statement</p>
</div>
</div>
</div>
</div>