<div class="paragraph">
<p>The <code>delete</code> operator can be used to remove a property from any object. Arrays are objects, so the <code>delete</code> operator can be used here too, but if it is, a hole will be left in the array because the indexes/keys won&#8217;t be shifted to reflect the deletion.</p>
</div>
<div class="paragraph">
<p>The proper method for removing an element at a certain index would be:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>Array.prototype.splice</code> - add/remove elements from the array</p>
</li>
<li>
<p><code>Array.prototype.pop</code> - add/remove elements from the end of the array</p>
</li>
<li>
<p><code>Array.prototype.shift</code> - add/remove elements from the beginning of the array</p>
</li>
</ul>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>var myArray = ['a', 'b', 'c', 'd'];

delete myArray[2];  // Noncompliant. myArray =&gt; ['a', 'b', undefined, 'd']
console.log(myArray[2]); // expected value was 'd' but output is undefined</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>var myArray = ['a', 'b', 'c', 'd'];

// removes 1 element from index 2
removed = myArray.splice(2, 1);  // myArray =&gt; ['a', 'b', 'd']
console.log(myArray[2]); // outputs 'd'</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this use of "delete".</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_28_apr_2015_153329_ann_campbell_wrote">on 28 Apr 2015, 15:33:29 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>back to you [~linda.martin]</p>
</div>
<div class="paragraph">
<p>You didn&#8217;t have any languages targeted. I assumed JavaScript&#8230;&#8203;</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_29_apr_2015_090241_linda_martin_wrote">on 29 Apr 2015, 09:02:41 Linda Martin wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Indeed thanks!</p>
</div>
<div class="paragraph">
<p>I updated the comment in the code snippet to show how the array look like after the deleting the element. I let you remove it or update if you think it is to heavy and not necessary.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_29_apr_2015_090947_ann_campbell_wrote">on 29 Apr 2015, 09:09:47 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Your updated looks good to me [~linda.martin]. I just modified it&#8217;s place in the code slightly.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_29_apr_2015_114126_linda_martin_wrote">on 29 Apr 2015, 11:41:26 Linda Martin wrote:</h3>
<div class="paragraph">
<p>Perfect this way! Thanks.</p>
</div>
</div>
</div>
</div>