<div class="paragraph">
<p>Sharing naming conventions allows teams to collaborate efficiently. This rule checks that type names match a provided regular expression.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="paragraph">
<p>With the default regular expression:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>&lt;xs:complexType name="fruit"&gt;    &lt;!-- Noncompliant --&gt;
  ...
&lt;/xs:complexType&gt;</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>&lt;xs:complexType name="fruitType"&gt;
  ...
&lt;/xs:complexType&gt;</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Rename this type to match the regular expression: ${format}.</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="ulist">
<ul>
<li>
<p>key = format</p>
</li>
<li>
<p>Description: Regular expression used to check types against</p>
</li>
<li>
<p>Default value: ^<span class="a-z">([A-Z][a-z0-9]</span>)*Type$</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>name value</p>
</div>
</div>
</div>
</div>