<div class="paragraph">
<p>Invoking a method designed to return a string representation of an object which is already a string is a waste of keystrokes. This redundant construction may be optimized by the compiler, but will be confusing in the meantime.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>String message = "hello world";
System.out.println(message.toString()); // Noncompliant;</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>String message = "hello world";
System.out.println(message);</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>there&#8217;s no need to call "toString()" on a string literal.</p>
</div>
<div class="paragraph">
<p>"XXX" is already a string, there&#8217;s no need to call "toString()" on it.</p>
</div>
<div class="paragraph">
<p>"XXX" returns a string, there&#8217;s no need to call "toString()".</p>
</div>
<div class="paragraph">
<p>"XXX" is an array of string, there&#8217;s no need to call "toString()".</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_duplicated_by_s2135">is duplicated by: S2135</h3>

</div>
</div>
</div>