<div class="paragraph">
<p>Recursion is acceptable in methods, where you can break out of it. But with class types, you end up with code that will compile but not run if you try to instantiate the class.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class C1&lt;T&gt;
{
}
class C2&lt;T&gt; : C1&lt;C2&lt;C2&lt;T&gt;&gt;&gt; // Noncompliant
{
}

...
var c2 = new C2&lt;int&gt;();</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Refactor this class so that the generic inheritance chain is not recursive.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>class name &amp; type &amp; inherited class &amp; type</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_10_dec_2015_173209_ann_campbell_wrote">on 10 Dec 2015, 17:32:09 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Targeting to C&#43;&#43; because according to [~evgeny.mandrikov],</p>
</div>
<div class="quoteblock">
<blockquote>
<div class="paragraph">
<p>from language specification - there is no limit</p>
</div>
<div class="paragraph">
<p>templates are turing complete</p>
</div>
<div class="paragraph">
<p>so in theory - this is valid</p>
</div>
</blockquote>
</div>
</div>
<div class="sect2">
<h3 id="_on_11_dec_2015_091458_tamas_vajk_wrote">on 11 Dec 2015, 09:14:58 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I changed the description a bit</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_11_dec_2015_135831_ann_campbell_wrote">on 11 Dec 2015, 13:58:31 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>looks good [~tamas.vajk]</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_27_sep_2018_201118_loïc_joly_wrote">on 27 Sep 2018, 20:11:18 Loïc Joly wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] -I would remove C&#43;&#43;, because such inheritance is indeed totally valid, and is even a commonly used idiom see for instance the https://en.wikipedia.org/wiki/Curiously_recurring_template_pattern link-</p>
</div>
<div class="paragraph">
<p>After better understanding of this rule, I would still remove it for C&#43;&#43;, but for other reasons:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>There could be places where that works, because template specialization could break the recursion</p>
</li>
<li>
<p>In places where it does not work, since templates are fully resolved during compilation, the code would not compile, so no need to report anything during analysis.</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_on_27_sep_2018_204807_ann_campbell_wrote">on 27 Sep 2018, 20:48:07 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Feel free, [~loic.joly] :-)</p>
</div>
</div>
</div>
</div>