<div class="paragraph">
<p>Appending <code>String.valueOf()</code> to a <code>String</code> decreases the code readability.</p>
</div>
<div class="paragraph">
<p>The argument passed to <code>String.valueOf()</code> should be directly appended instead.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public void display(int i){
  System.out.println("Output is " + String.valueOf(i));    // Noncompliant
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public void display(int i){
  System.out.println("Output is " + i);                    // Compliant
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Directly append the argument of String.valueOf().</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_16_aug_2013_125137_dinesh_bolkensteyn_wrote">on 16 Aug 2013, 12:51:37 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I though you didn&#8217;t like 'source code' &#8230;&#8203; but here it is "and decreases the readability of the source code" ;-)</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_16_aug_2013_143004_dinesh_bolkensteyn_wrote">on 16 Aug 2013, 14:30:04 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>The deprecatied rule covered also this case AFAIK:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>String a = String.valueOf("foo");</pre>
</div>
</div>
<div class="paragraph">
<p>which will not be covered by this rule.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_16_aug_2013_152024_dinesh_bolkensteyn_wrote">on 16 Aug 2013, 15:20:24 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>Implemented by http://jira.codehaus.org/browse/SONARJAVA-241</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_17_aug_2013_155450_ann_campbell_wrote">on 17 Aug 2013, 15:54:50 Ann Campbell wrote:</h3>
<div class="literalblock">
<div class="content">
<pre>I don't [~dinesh.bolkensteyn]], but without a list of rules, I try to conform to what appears to be the established standard. :-)</pre>
</div>
</div>
</div>
</div>
</div>