<div class="paragraph">
<p>PHPUnit provides helper functions and annotations to verify that a given block of code throws an exception and to assert different properties of that exception. The provided helper functions are:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>expectException()</code></p>
</li>
<li>
<p><code>expectExceptionCode()</code></p>
</li>
<li>
<p><code>expectExceptionMessage()</code></p>
</li>
<li>
<p><code>expectExceptionMessageRegExp()</code></p>
</li>
</ul>
</div>
<div class="paragraph">
<p>This check raises an issue when the throw of an exception is verified using a <em>try-catch</em> approach instead of relying on the provided helper functions.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public function testA()
{
    try {
        doSomething();
        $this-&gt;fail("Assertion should have been thrown");
    } catch (MyException $e) {
        assertEquals("Exception message", $e-&gt;getMessage());
    }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public function testB()
{
    $this-&gt;expectException(MyException::class);
    $this-&gt;expectExceptionMessage("Exception message");

    doSomething();
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="paragraph">
<p>thePHP.cc: <a href="https://thephp.cc/news/2016/02/questioning-phpunit-best-practices">Questioning PHPUnit Best Practice</a></p>
</div>
</div>
</div>