<div class="paragraph">
<p>Creating temporary primitive wrapper objects only for <code>String</code> conversion or the use of the <code>compareTo</code> method is inefficient.</p>
</div>
<div class="paragraph">
<p>Instead, the static <code>toString()</code> or <code>compare</code> method of the primitive wrapper class should be used.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>new Integer(myInteger).toString();  // Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>Integer.toString(myInteger);        // Compliant</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Call the static method {wrapperClass}.[toString(&#8230;&#8203;)|compare(&#8230;&#8203;)} instead of instantiating a temporary object to perform this to string conversion.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_29_jul_2013_152120_freddy_mallet_wrote">on 29 Jul 2013, 15:21:20 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARJAVA-246</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_16_aug_2013_142929_dinesh_bolkensteyn_wrote">on 16 Aug 2013, 14:29:29 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>\[~freddy.mallet] You can&#8217;t use {primitiveWrapperClass} in the rule description ;-)</p>
</div>
</div>
</div>
</div>