<div class="paragraph">
<p>OS/VS COBOL accepted the <code>ON</code> statement, but IBM Enterprise COBOL does not accept it anymore. The <code>ON</code> statement allows the selective execution of statements it contains. Similar functions are provided in Enterprise COBOL by <code>EVALUATE</code> and <code>IF</code></p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>       IDENTIFICATION DIVISION.
       PROGRAM-ID. foo.

       PROCEDURE DIVISION.
      * Non-Compliant
         ON 1
           DISPLAY 'First time'
         ELSE
           DISPLAY 'Other times'.
       END PROGRAM foo.</pre>
</div>
</div>
</div>
</div>