<div class="paragraph">
<p>For better readability, do not put multiple property declarations in the same statement.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>&lt;?php
class Foo
{
   private $bar = 1, $bar2 = 2;
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>&lt;?php
class Foo
{
   private $bar1 = 1;
   private $bar2 = 2;
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>{0} property declarations were found in this statement. Reformat the code to declare only one property per statement.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s122">relates to: S122</h3>

</div>
<div class="sect2">
<h3 id="_on_27_may_2014_113254_ann_campbell_wrote">on 27 May 2014, 11:32:54 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~linda.martin] how is this different from the standard rule against having more than one statement per line?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_27_may_2014_151818_linda_martin_wrote">on 27 May 2014, 15:18:18 Linda Martin wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] here we are talking about 2 property declarations in the same statement, so it won&#8217;t be detected by the rule "one statement per line".</p>
</div>
</div>
</div>
</div>