<div class="paragraph">
<p>There are two ways to define asynchronous functions in Kotlin:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>using the modifier <code>suspend</code> in the function declaration</p>
</li>
<li>
<p>creating an extension function on <code>CoroutineScope</code></p>
</li>
</ul>
</div>
<div class="paragraph">
<p>The <code>suspend</code> modifier is generally used for functions that might take some time to complete. The caller coroutine might potentially be suspended.</p>
</div>
<div class="paragraph">
<p>Functions that start a coroutine in the background and return before said coroutine has completed running should be extension functions on <code>CoroutineScope</code>. This helps to clarify the intention of such a function. Further, such functions should not be suspending, as suspending functions should only return once all the work they are designed to perform is complete.</p>
</div>
<div class="paragraph">
<p>Functions returning <code>Flow</code> or <code>Channel</code> should return the result immediately and may start a new coroutine in the background. As a consequence, such functions should not be suspending and if they launch a coroutine in the background, they should be declared as extension functions on <code>CoroutineScope</code>.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>suspend fun f(): Flow&lt;Int&gt; {
    val flow = flow {
        emit(1)
    }
    delay(500L)
    return flow
}</pre>
</div>
</div>
<div class="listingblock">
<div class="content">
<pre>suspend fun f(): Channel&lt;Int&gt; {
    val ch = Channel&lt;Int&gt;()
    ch.send(1)
    return ch
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>fun f(): Flow&lt;Int&gt; = flow {
    emit(1)
}</pre>
</div>
</div>
<div class="listingblock">
<div class="content">
<pre>fun CoroutineScope.f(): Channel&lt;Int&gt; {
    val ch = Channel&lt;Int&gt;()
    launch {
        ch.send(1)
    }
    return ch
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://elizarov.medium.com/coroutine-context-and-scope-c8b255d59055">Coroutine Context and Scope</a></p>
</li>
</ul>
</div>
</div>
</div>