<div class="paragraph">
<p>TODO</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>PERFORM UNTIL FIN-RGL0
     PERFORM DEBUT-RGL0
     PERFORM UNTIL FIN-RGL1
     PERFORM DEBUT-RGL1
          PERFORM UNTIL FIN-RGL2
           PERFORM DEBUT-RGL2
           PERFORM ALIM-RGL2
           PERFORM FIN-RGL2
     END-PERFORM
     PERFORM FIN-RGL1
END-PERFORM
     PERFORM FIN-RGL0
END-PERFORM</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>PERFORM TRAITEMENT-RGL0 UNTIL FIN-RGL0.
TRAITEMENT-RGL0.
PERFORM DEBUT-RGL0
PERFORM TRAITEMEN-RGL1 UNTIL FIN-RGL1
     PERFORM FIN-RGL0
	.
TRAITEMENT-RGL1.
     PERFORM DEBUT-RGL1
     PERFORM UNTIL FIN-RGL2
           PERFORM DEBUT-RGL2
           PERFORM ALIM-RGL2
           PERFORM FIN-RGL2
     END-PERFORM
     PERFORM FIN-RGL1</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s134">relates to: S134</h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s1306">relates to: S1306</h3>

</div>
<div class="sect2">
<h3 id="_on_13_may_2014_083941_freddy_mallet_wrote">on 13 May 2014, 08:39:41 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Might in fact duplicates RSPEC-134</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_13_may_2014_084901_ann_campbell_wrote">on 13 May 2014, 08:49:01 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Given the way RSPEC-134 was implemented, this is indeed a duplicate.</p>
</div>
</div>
</div>
</div>