<div class="paragraph">
<p>Union and intersection types are convenient but can make code harder to read and maintain. So if a particular union or intersection is used in multiple places, the use of a type alias is recommended.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>function foo(x:string|null|number) { // Noncompliant
  // ...
}
function bar(x:string|null|number) {
  // ...
}
function zoo(): string|null|number {
  return null;
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>type MyType = string | null | number;

function foo(x: MyType) {
  // ...
}
function bar(x: MyType) {
  // ...
}
function zoo(): MyType {
  return null;
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace this [union|intersection] type with a type alias.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>Primary: first occurrence of the union/intersection</p>
</li>
<li>
<p>Secondary: All other occurrences</p>
<div class="ulist">
<ul>
<li>
<p>message: 'Following occurrence.'</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_14_nov_2017_220210_ann_campbell_wrote">on 14 Nov 2017, 22:02:10 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~jeanchristophe.collet] there are no <code>//Noncompliant</code> comments.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_15_nov_2017_091900_jean_christophe_collet_wrote">on 15 Nov 2017, 09:19:00 Jean-Christophe Collet wrote:</h3>
<div class="paragraph">
<p>That&#8217;s because it&#8217;s not a particular line that is non compliant, but the repetition of the specific union.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_27_nov_2017_190440_ann_campbell_wrote">on 27 Nov 2017, 19:04:40 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~jeanchristophe.collet] I&#8217;ve updated the highlighting to match the 'Noncompliant' comment.</p>
</div>
</div>
</div>
</div>