<div class="paragraph">
<p>The attribute <code>noreturn</code> indicates that a function does not return.</p>
</div>
<div class="paragraph">
<p>Using this attribute allows the compilerÂ to do some assumptions that can lead to optimizations. However, if a function with this attribute ever returns, the behavior becomes undefined.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>[[noreturn]] void f () {
  while (1) {
    // ...
    if (/* something*/) {
      return; // Noncompliant, this function should not return
    }
  }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>[[noreturn]] void f() {  // Compliant
  while (true) {
    // ...
  }
}</pre>
</div>
</div>
<div class="paragraph">
<p>Or</p>
</div>
<div class="listingblock">
<div class="content">
<pre>void f() {
  while (true) {
    // ...
    if (/* something*/) {
      return; // Compliant
    }
  }
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>function "XXX" declared 'noreturn' should not return</p>
</div>
</div>
</div>
</div>