<div class="paragraph">
<p><code>Cloneable</code> is the marker <code>Interface</code> that indicates that <code>clone()</code> may be called on an object. Overriding <code>clone()</code> without implementing <code>Cloneable</code> can be useful if you want to control how subclasses clone themselves, but otherwise, it&#8217;s probably a mistake.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class Team  {  // Noncompliant
  private Person coach;
  private List&lt;Person&gt; players;
  public void addPlayer(Person p) { ... }
  public Person getCoach() { ... }

  @Override
  public Team clone() { ... }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class Team implements Cloneable {
  private Person coach;
  private List&lt;Person&gt; players;
  public void addPlayer(Person p) { ... }
  public Person getCoach() { ... }

  @Override
  public Team clone() { ... }
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_25_nov_2014_163547_freddy_mallet_wrote">on 25 Nov 2014, 16:35:47 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>My 2 cents @Ann: I would merge this rule with RSPEC-1182 and so will also change the title of RSPEC-1182 like this :</p>
</div>
<div class="paragraph">
<p>"When overriding Object.clone(), Cloneable should be implemented and super.clone() should be called"</p>
</div>
<div class="paragraph">
<p>Indeed, that&#8217;s an issue when overriding the Object.clone() method to not implement the Cloneable interface if and only if super.clone() method is called.</p>
</div>
</div>
</div>
</div>