<div class="paragraph">
<p>Log streams created on AWS will stay forever unless the <code>AWS::Logs::LogGroup</code> to which they belong to was configured with a retention policy.
<code>Log Groups</code> should have their “RetentionInDays” property set with a valid value to be sure the log events are kept only for the expected duration.</p>
</div>
<div class="paragraph">
<p>When the property is not set, the log events will be kept for ever or will be deleted only when the Log Group is removed.</p>
</div>
<div class="paragraph">
<p>Keeping the logs for ever doesn&#8217;t come for free: AWS will charge for keeping these logs.
Also from a security point of view, keeping the data for ever may be not compliant with company policy or regulatory rules.</p>
</div>
<div class="paragraph">
<p>Note: this rule doesn&#8217;t check if the value provided to "RetentionInDays" is valid because AWS CloudFormation Linter (cfn-lint) do it already</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>  MyLambdaFunction:
    Type: AWS::Lambda::Function
    Properties:
      Runtime: nodejs12.x
      Description: Example of Lambda Function

  MyFunctionLogGroup:                                                                                                                                                 Type: AWS::Logs::LogGroup
    Properties:
      LogGroupName: !Join ['/', ['/aws/lambda', !Ref MyLambdaFunction]]
      # Noncompliant: "RetentionInDays" property is not set: logs are kept for ever</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>  MyLambdaFunction:
    Type: AWS::Lambda::Function
    Properties:
      Runtime: nodejs12.x
      Description: Example of Lambda Function

  MyFunctionLogGroup:                                                                                                                                                 Type: AWS::Logs::LogGroup
    Properties:
      LogGroupName: !Join ['/', ['/aws/lambda', !Ref MyLambdaFunction]]
      RetentionInDays: 30</pre>
</div>
</div>
</div>
</div>