<div class="paragraph">
<p>Calling an operator in python is equivalent to calling a special method (except for the identity operator <code>is</code>). Python provides a set of built-in operations. It is for example possible to add two integers: <code>1 + 2</code>. It is however not possible to add a string and an integer: <code>1 + "2"</code> and such an operation will raise a <code>TypeError</code>.</p>
</div>
<div class="paragraph">
<p>It is possible to define how an operator will behave with a custom class by defining the corresponding special method. See python documentation for a complete list of operators and their methods: <a href="https://docs.python.org/3/reference/datamodel.html?highlight=__exit__%20special%20methods#emulating-numeric-types">arithmetic and bitwise operators</a>, <a href="https://docs.python.org/3/reference/datamodel.html?highlight=__exit__%20special%20methods#object.__lt__">comparison operators</a>.</p>
</div>
<div class="paragraph">
<p>For symmetrical binary operators you need to define two methods so that the order of operands doesn&#8217;t matter, ex: <code>__add__</code> and <code>__radd__</code>.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when an operator is used on incompatible types. Types are considered incompatible if no built-in operations between those types exist and none of the operands has implemented the corresponding special methods.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class Empty:
    pass

class Add:
    def __add__(self, other):
        return 42

1 + 2
1 + "2"  # Noncompliant
Empty() + 1  # Noncompliant
Add() + 1
1 + Add()  # Noncompliant
Add() + Empty()
Empty() + Add()  # Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class Empty:
    pass

class Add:
    def __add__(self, other):
        return 42

    def __radd__(self, other):
        return 42

Add() + 1
1 + Add()
Add() + Empty()
Empty() + Add()</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>Python documentation - <a href="https://docs.python.org/3/reference/datamodel.html?highlight=__exit__%20special%20methods#object.__lt__">Rich comparison methods</a></p>
</li>
<li>
<p>Python documentation - <a href="https://docs.python.org/3/reference/datamodel.html?highlight=__exit__%20special%20methods#emulating-numeric-types">Emulating numeric types</a></p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>Fix this invalid XXX operation between incompatible types.</p>
</li>
<li>
<p>Fix this invalid XXX operation on a type which doesn&#8217;t support it.</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Primary location: the operator</p>
</div>
<div class="paragraph">
<p>Secondary locations: the operand(s)</p>
</div>
</div>
</div>
</div>