<div class="paragraph">
<p>While not technically incorrect, the omission of <code>BEGIN...END</code> can be misleading, and may lead to the introduction of errors during maintenance.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>IF @flag = 1 -- Noncompliant
  EXEC something;</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>IF @flag = 1
    BEGIN
        EXEC something;
    END;</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add curly braces around the nested statement(s) in this "[if|else|for|while|do]" block.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_deprecates_s911">deprecates: S911</h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s5305">relates to: S5305</h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s5306">relates to: S5306</h3>

</div>
<div class="sect2">
<h3 id="_replaces_s913">replaces: S913</h3>

</div>
<div class="sect2">
<h3 id="_on_20_may_2013_125106_fabrice_bellingard_wrote">on 20 May 2013, 12:51:06 Fabrice Bellingard wrote:</h3>
<div class="paragraph">
<p>Implementation: http://jira.codehaus.org/browse/SONARJAVA-150</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_12_feb_2014_215811_freddy_mallet_wrote">on 12 Feb 2014, 21:58:11 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3470 for PHP</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_24_feb_2014_180358_freddy_mallet_wrote">on 24 Feb 2014, 18:03:58 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>This rule description is very light @Ann. Feel free to improve it.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_25_feb_2014_002931_ann_campbell_wrote">on 25 Feb 2014, 00:29:31 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Didn&#8217;t beef it up much. :-/</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_28_apr_2014_222332_evgeny_mandrikov_wrote">on 28 Apr 2014, 22:23:32 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] how this ticket connected with RSPEC-911 and RSPEC-913, which are already implemented for C/C&#43;&#43; ?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_29_apr_2014_193746_ann_campbell_wrote">on 29 Apr 2014, 19:37:46 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>How indeed [~evgeny.mandrikov]. It appears to supersede them.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_9_mar_2018_162242_janos_gyerik_wrote">on 9 Mar 2018, 16:22:42 Janos Gyerik wrote:</h3>
<div class="paragraph">
<p>Irrelevant for Go. The compiler requires the braces, always.</p>
</div>
</div>
</div>
</div>