<div class="paragraph">
<p>Some implementations of <code>java.sql.ResultSet#getMetaData()</code> suffer from performance issue and should not be called in a loop. Instead, multiple calls in a row should be replaced by a single cached call.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>ResultSetMetaData rsmd = rs.getMetaData();
for (int i=1; i&lt;rsmd.getColumnCount(); i++) {
  System.out.println(rs.getMetaData().getAutoIncrement()); // Noncompliant; "rs.getMetaData()" will be called for each columns
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>ResultSetMetaData rsmd = rs.getMetaData(); // Compliant; result of "rs.getMetaData()" is cached
for (int i=1; i&lt;rsmd.getColumnCount(); i++) {
  System.out.println(rsmd.getAutoIncrement());
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Cache the result of "java.sql.ResultSet#getMetaData()" out of this loop.</p>
</div>
</div>
</div>
</div>