<div class="paragraph">
<p>There are several reasons for a method not to have a method body:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>It is an unintentional omission, and should be fixed.</p>
</li>
<li>
<p>It is not yet, or never will be, supported. In this case an error should be thrown.</p>
</li>
<li>
<p>The method is an intentionally-blank. In this case a nested comment should explain the reason for the blank method.</p>
</li>
</ul>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>Sub DoSomething()
End Sub

Function DoSomething() As String
End Function</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>Sub DoSomething()
    ' Not implemented because of reason
End Sub

Function DoSomething() As String
    DoSomething = "Value"
End Function</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add a nested comment explaining why this method is empty, throw an error or complete the implementation.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s3575">relates to: S3575</h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s3050">relates to: S3050</h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s108">is related to: S108</h3>

</div>
<div class="sect2">
<h3 id="_on_7_aug_2013_171231_freddy_mallet_wrote">on 7 Aug 2013, 17:12:31 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARJAVA-273</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_26_aug_2013_143316_dinesh_bolkensteyn_wrote">on 26 Aug 2013, 14:33:16 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>As discussed, this rule should no longer apply to abstract classes to allow default visitor implementations for example</p>
</div>
</div>
</div>
</div>