<div class="paragraph">
<p>When a reluctant (or lazy) quantifier is followed by a pattern that can match the empty string or directly by the end of the regex, it will always match zero times for <code>*?</code> or one time for <code>+?</code>. If a reluctant quantifier is followed directly by the end anchor (<code>$</code>), it  behaves indistinguishably from a greedy quantifier while being less efficient.</p>
</div>
<div class="paragraph">
<p>This is likely a sign that the regex does not work as intended.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>re.replace(r"start\w*?(end)?", "x", "start123endstart456") # Noncompliant. In contrast to what one would expect, the result is not "xx"

re.match(r"^\d*?$", "123456789") # Noncompliant. Matches the same as "/^\d*$/", but will backtrack in every position.</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>re.replace(r"start\w*?(end|$)", "x", "start123endstart456") # Result is "xx"

re.match(r"^\d*$", "123456789")</pre>
</div>
</div>
</div>
</div>