<div class="paragraph">
<p><code>break;</code> is an unstructured control flow statement which makes code harder to read.</p>
</div>
<div class="paragraph">
<p>Ideally, every loop should have a single termination condition.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>for (element = list.first; element != null; element = element-&gt;next) { // First termination condition
  if (!matches(element-&gt;value)) {                                      // Second termination condition
    break; // Noncompliant
  }

  /* ... */
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>// Compliant
for (element = list.first; element != null &amp;&amp; matches(element-&gt;value); element = element-&gt;next) {
  /* ... */
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Refactor the code to remove this break statement.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s3385">is related to: S3385</h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s924">is related to: S924</h3>

</div>
</div>
</div>