<div class="paragraph">
<p>By contract, every Python function returns something, even if it&#8217;s the <code>None</code> value, which can be returned implicitly by omitting the <code>return</code> statement, or explicitly.</p>
</div>
<div class="paragraph">
<p>The <code>__init__</code> method is required to return <code>None</code>. A <code>TypeError</code> will be raised if the <code>__init__</code> method either <code>yield</code>s or <code>return</code>s any expression other than <code>None</code>. Returning some expression that evaluates to <code>None</code> will not raise an error, but is considered bad practice.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class MyClass(object):
    def __init__(self):
        self.message = 'Hello'
        return self  # Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class MyClass(object):
    def __init__(self):
        self.message = 'Hello'</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>Remove this yield statement.</p>
</li>
<li>
<p>Remove this return value.</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s5654">relates to: S5654</h3>

</div>
<div class="sect2">
<h3 id="_on_18_mar_2015_175542_ann_campbell_wrote">on 18 Mar 2015, 17:55:42 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>-E0100: <em>init</em> method is a generator</p>
</div>
<div class="paragraph">
<p>-E0101: Explicit return in <em>init</em></p>
</div>
</div>
</div>
</div>