<div class="paragraph">
<p><code>ObjectOutputStream</code>s are used with serialization, and the first thing an <code>ObjectOutputStream</code> writes is the serialization stream header. This header should appear once per file, at the beginning. Pass a file opened in append mode into an <code>ObjectOutputStream</code> constructor and the serialization stream header will be added to the end of the file before your object is then also appended.</p>
</div>
<div class="paragraph">
<p>When you&#8217;re trying to read your object(s) back from the file, only the first one will be read successfully, and a <code>StreamCorruptedException</code> will be thrown after that.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>val fos = FileOutputStream(fileName, true) // fos opened in append mode
val out = ObjectOutputStream(fos) // Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>val fos = FileOutputStream(fileName)
val out = ObjectOutputStream(fos)</pre>
</div>
</div>
</div>
</div>