<div class="paragraph">
<p>Arrow functions <code>=&gt;</code> use a syntax similar to certain comparison operators (<code>&lt;=</code>, <code>&gt;=</code>). This can create confusion when used in certain contexts.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>let foo = (x) =&gt; x ? "Africa" : "Asia";   // Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>let foo = (x) =&gt; { return x ? "Africa" : "Asia"; };</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Refactor this code to disambiguate the use of arrow function.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The arrow function "&#8658;"</p>
</div>
</div>
</div>
</div>