<div class="paragraph">
<p>The following words may be used as keywords in future evolutions of the language, so using them as identifiers should be avoided to allow an easier adoption of those potential future versions:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>await</code></p>
</li>
<li>
<p><code>class</code></p>
</li>
<li>
<p><code>const</code></p>
</li>
<li>
<p><code>enum</code></p>
</li>
<li>
<p><code>export</code></p>
</li>
<li>
<p><code>extends</code></p>
</li>
<li>
<p><code>implements</code></p>
</li>
<li>
<p><code>import</code></p>
</li>
<li>
<p><code>interface</code></p>
</li>
<li>
<p><code>let</code></p>
</li>
<li>
<p><code>package</code></p>
</li>
<li>
<p><code>private</code></p>
</li>
<li>
<p><code>protected</code></p>
</li>
<li>
<p><code>public</code></p>
</li>
<li>
<p><code>static</code></p>
</li>
<li>
<p><code>super</code></p>
</li>
<li>
<p><code>yield</code></p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Use of these words as identifiers would produce an error in JavaScript <code>strict</code> mode code.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>var package = document.getElementsByName("foo"); // Noncompliant
var someData = { package: true };                 // Compliant, as it is not used as an identifier here</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>var elements = document.getElementsByName("foo"); // Compliant</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Rename "{0}" identifier to prevent potential conflicts with future evolutions of the JavaScript language.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s1190">relates to: S1190</h3>

</div>
<div class="sect2">
<h3 id="_on_10_apr_2015_125318_elena_vilchik_wrote">on 10 Apr 2015, 12:53:18 Elena Vilchik wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I added two full lists of future keywords. Could you verify it?</p>
</div>
<div class="paragraph">
<p><a href="http://www.ecma-international.org/ecma-262/5.1/#sec-7.6.1.2" class="bare">http://www.ecma-international.org/ecma-262/5.1/#sec-7.6.1.2</a> (ES5)</p>
</div>
<div class="paragraph">
<p><a href="http://people.mozilla.org/~jorendorff/es6-draft.html#sec-future-reserved-words" class="bare">http://people.mozilla.org/~jorendorff/es6-draft.html#sec-future-reserved-words</a> (ES6)</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_10_apr_2015_134203_ann_campbell_wrote">on 10 Apr 2015, 13:42:03 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>updated per discussion [~elena.vilchik]</p>
</div>
</div>
</div>
</div>