<div class="paragraph">
<p>Attempting to make a comparison between pointers using &gt;, &gt;=, &lt; or &lt;= will produce undefined behavior if the two pointers point to different arrays.</p>
</div>
<div class="paragraph">
<p>Additionally, directly comparing two arrays for equality or inequality has been deprecated in C&#43;&#43;.</p>
</div>
<div class="paragraph">
<p>However, equality or inequality between an array and a pointer is still valid</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>void f1 ( )
{
  int a1[ 10 ];
  int a2[ 10 ];
  int * p1 = a1;
  if ( p1 &lt; a2 ) // Non-compliant, p1 and a2 point to different arrays.
  {
  }
  if ( p1 - a2 &gt; 0 ) // Non-compliant, p1 and a2 point to different arrays.
  {
  }
  if ( a1 == a2) // Non-compliant (in C++). Comparing different array for equality is deprecated
  {
  }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>void f1 ( )
{
  int a1[ 10 ];
  int * p1 = a1;
  if ( p1 &lt; a1 ) // Compliant, p1 and a1 point to the same array.
  {
  }
  if ( p1 - a1 &gt; 0 )  // Compliant, p1 and a1 point to the same array.
  {
  }
  if ( p1 == a2 ) // Compliant, comparing a pointer and an array for equality is valid
  {
  }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C:2004, 17.3 - &gt;, &gt;=, &lt;, &lt;= shall not be applied to pointer types except where they point to the same array.</p>
</li>
<li>
<p>MISRA C&#43;&#43;:2008, 5-0-18 - &gt;, &gt;=, &lt;, &lt;= shall not be applied to objects of pointer type, except where they point to the same array.</p>
</li>
<li>
<p><a href="https://github.com/isocpp/CppCoreGuidelines/blob/036324/CppCoreGuidelines.md#es62-dont-compare-pointers-into-different-arrays">C&#43;&#43; Core Guidelines ES.62</a> - Don&#8217;t compare pointers into different arrays</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s939">is related to: S939</h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s940">is related to: S940</h3>

</div>
</div>
</div>