<div class="paragraph">
<p>There&#8217;s no need to repeat the program name in the free-format definition of an external program.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>DCL-PR PGM001 EXTPGM('PGM001');
END-PR;</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>DCL-PR PGM001 EXTPGM ;
END-PR;</pre>
</div>
</div>
<div class="paragraph">
<p>or if the program is renamed</p>
</div>
<div class="listingblock">
<div class="content">
<pre>DCL-PR TheProgram EXTPGM('PGM001');
END-PR;</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove "xxx" at position n.</p>
</div>
</div>
</div>
</div>