<div class="paragraph">
<p><code>SELECT</code> queries that return too many columns may be complex or difficult to maintain.</p>
</div>
<div class="paragraph">
<p>This rule identifies queries that <code>SELECT</code> more than the specified number of columns.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="paragraph">
<p>With a threshold of 6:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>BEGIN
  SELECT id, name, firstname, gender, height, weight, age -- Noncompliant
    INTO peopleArray
    FROM people
    WHERE age &gt; 60;
END;
/</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Refactor this query to select no more than n columns.</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="ulist">
<ul>
<li>
<p>key = maxColumnsCount</p>
</li>
<li>
<p>description: Maximum number of columns allowed</p>
</li>
<li>
<p>default: 10</p>
</li>
</ul>
</div>
</div>
</div>
</div>