<div class="paragraph">
<p>One thing that makes good code good is the clarity with which it conveys the intent of the original programmer to maintainers, and the proper choice of <code>indexOf</code> methods can help move code from confusing to clear.</p>
</div>
<div class="paragraph">
<p>If you need to see whether a substring is located beyond a certain point in a string, you can test the <code>indexOf</code> the substring versus the target point, or you can use the version of <code>indexOf</code> which takes a starting point argument. The latter is arguably clearer because the result is tested against -1, which is an easily recognizable "not found" indicator.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>String name = "ismael";

if (name.indexOf("ae") &gt; 2) { // Noncompliant
  // ...
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>String name = "ismael";

if (name.indexOf("ae", 2) &gt; -1) {
  // ...
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use ".indexOf(xxx,n) &gt; -1" instead.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s2692">is related to: S2692</h3>

</div>
<div class="sect2">
<h3 id="_on_20_may_2015_131910_massimo_paladin_wrote">on 20 May 2015, 13:19:10 Massimo PALADIN wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] LGTM.</p>
</div>
</div>
</div>
</div>