<div class="paragraph">
<p>During dictionary use, it is normal that you might want to replace the value associated with one of the dictionary&#8217;s keys. But do so during the initial definition of a dictionary, and you&#8217;ve probably made an error.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when the same key is used multiple times in the initial definition of a dictionary.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>fruit = {"apple":4, "pear":6, "quince":29, "apple": 8}  # Noncompliant; "apple" is used in positions 0 and 3</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>fruit = {"apple":4, "pear":6, "quince":29, "potato": 8}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>"xxx" is used in positions n, o, p&#8230;&#8203;</p>
</div>
</div>
</div>
</div>