<div class="paragraph">
<p>The multi-line comment syntax <code>/* ... */</code> should not be used for single line comments; the <code>--</code> syntax is more appropriate.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>/*
 These comment lines are Compliant
 comment 1
 comment 2
*/

/*
 This comment is also Compliant
 */

/* This comment is Noncompliant */</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>/*
 These comment lines are Compliant
 comment 1
 comment 2
*/

/*
 This comment is also Compliant
 */

-- This comment is compliant</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use the single-line format for this comment.</p>
</div>
</div>
</div>
</div>