<div class="paragraph">
<p>The following predefined variables are deprecated and should be replaced by the new versions:</p>
</div>
<table class="tableblock frame-all grid-all stretch">
<colgroup>
<col style="width: 50%;"/>
<col style="width: 50%;"/>
</colgroup>
<thead>
<tr>
<th class="tableblock halign-center valign-top">Replace</th>
<th class="tableblock halign-center valign-top">With</th>
</tr>
</thead>
<tbody>
<tr>
<td class="tableblock halign-center valign-top"><p class="tableblock">$HTTP_SERVER_VARS</p></td>
<td class="tableblock halign-center valign-top"><p class="tableblock">$_SERVER</p></td>
</tr>
<tr>
<td class="tableblock halign-center valign-top"><p class="tableblock">$HTTP_GET_VARS</p></td>
<td class="tableblock halign-center valign-top"><p class="tableblock">$_GET</p></td>
</tr>
<tr>
<td class="tableblock halign-center valign-top"><p class="tableblock">$HTTP_POST_VARS</p></td>
<td class="tableblock halign-center valign-top"><p class="tableblock">$_POST</p></td>
</tr>
<tr>
<td class="tableblock halign-center valign-top"><p class="tableblock">$HTTP_POST_FILES</p></td>
<td class="tableblock halign-center valign-top"><p class="tableblock">$_FILES</p></td>
</tr>
<tr>
<td class="tableblock halign-center valign-top"><p class="tableblock">$HTTP_SESSION_VARS</p></td>
<td class="tableblock halign-center valign-top"><p class="tableblock">$_SESSION</p></td>
</tr>
<tr>
<td class="tableblock halign-center valign-top"><p class="tableblock">$HTTP_ENV_VARS</p></td>
<td class="tableblock halign-center valign-top"><p class="tableblock">$_ENV</p></td>
</tr>
<tr>
<td class="tableblock halign-center valign-top"><p class="tableblock">$HTTP_COOKIE_VARS</p></td>
<td class="tableblock halign-center valign-top"><p class="tableblock">$_COOKIE</p></td>
</tr>
<tr>
<td class="tableblock halign-center valign-top"><p class="tableblock">$php_errormsg</p></td>
<td class="tableblock halign-center valign-top"><p class="tableblock">error_get_last()</p></td>
</tr>
</tbody>
</table>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>echo 'Name parameter value: ' . $HTTP_GET_VARS["name"];</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>echo 'Name parameter value: ' . $_GET["name"];</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace this use of the deprecated "XXXXXX" variable with "YYYYY".</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_12_feb_2014_203817_freddy_mallet_wrote">on 12 Feb 2014, 20:38:17 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-2338</p>
</div>
</div>
</div>
</div>