<div class="paragraph">
<p>This rule allows banning usage of certain constructors.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Given parameters:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>className: java.util.Date</p>
</li>
<li>
<p>argumentTypes: java.lang.String</p>
</li>
</ul>
</div>
<div class="listingblock">
<div class="content">
<pre>Date birthday;
birthday = new Date("Sat Sep 27 05:42:21 EDT 1986");  // Noncompliant
birthday = new Date(528176541000L); // Compliant</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this forbidden initialization</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="ulist">
<ul>
<li>
<p>key = className</p>
<div class="ulist">
<ul>
<li>
<p>description: Name of the class whose constructor is forbidden. This parameter is mandatory, if absent the rule is disabled.</p>
</li>
</ul>
</div>
</li>
<li>
<p>key: argumentTypes</p>
<div class="ulist">
<ul>
<li>
<p>description: Comma-delimited list of argument types, E.G. java.lang.String, int[], int</p>
</li>
</ul>
</div>
</li>
<li>
<p>key: allOverloads</p>
<div class="ulist">
<ul>
<li>
<p>description: Boolean. Set to true to flag all overloads regardless of parameter type (default: false)</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_deprecates_s1212">deprecates: S1212</h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s2253">relates to: S2253</h3>

</div>
</div>
</div>