<div class="paragraph">
<p>Developers may define a list named <code>__all__</code> in a module to limit the names imported from it by wildcard imports (<code>from mymodule import *</code>). This list can only reference defined names, otherwise an <code>AttributeError</code> will be raised when the module is imported.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>from mymodule import my_func

__all__ = ["unknown_func"]  # Noncompliant. "unknown_func" is undefined</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>from mymodule import my_func

__all__ = ["my_func"]</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.python.org/3/tutorial/modules.html#importing-from-a-package">Python documentation  - Importing * From a Package</a></p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Change or remove this string; "XX" is not defined.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>Primary: The string with an undefined name.</p>
</li>
<li>
<p>Secondary: the variable assignment if a variable is used.
message: 'Assigned here.'</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s2823">relates to: S2823</h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s3827">relates to: S3827</h3>

</div>
</div>
</div>