<div class="paragraph">
<p>Every paragraph should be commented to explain its goal and how it works. This comment can be placed either just before or just after the paragraph label. Moreover paragraphs used to close a module can be left uncommented.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre> PROCEDURE DIVISION.

    PARAGRAPH1.           *&gt; Noncompliant
    ...

*-------
    PARAGRAPH2.           *&gt; Noncompliant; the comment is empty
    ...

      PERFORM P1 THRU P2.
    ...

*Some comments                                  *&gt; Compliant
    P1.
      ....

    P2.                                         *&gt; No violation as the this P2 paragraph close a module
       MOVE A TO B.
       ...
       EXIT.</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre> PROCEDURE DIVISION.

* Some comments
    PARAGRAPH1.     *&gt;  Compliant; the comment is just before
    ...

    PARAGRAPH2.     *&gt;  Compliant; the comment is just after
* Some comments
    ...
      PERFORM P1 THRU P2.
    ...

*Some comments
    P1.
      ....

    P2.
       MOVE A TO B.
       ...
       EXIT.</pre>
</div>
</div>
</div>
</div>