<div class="paragraph">
<p>The rules for method resolution are complex and perhaps not properly understood by all coders. Having overloads with optional parameter values makes the matter even harder to understand.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when an overload with default parameter values is hidden by one without the optional parameters.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public class MyClass
{
  void Print(string[] messages) {...}
  void Print(string[] messages, string delimiter = "\n") {...} // Noncompliant; default parameter value is hidden by overload
}

// ...
MyClass myClass = new MyClass();

myClass.Print(new string[3] {"yes", "no", "maybe"});  // which version of Print will be called?</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>This method signature overlaps the one defined on line n [in "filename"], the default parameter value can only be used with named arguments.</p>
</li>
<li>
<p>This method signature overlaps the one defined on line n [in "filename"], the default parameter value can&#8217;t be used.</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>First default parameter value</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_30_nov_2015_105412_tamas_vajk_wrote">on 30 Nov 2015, 10:54:12 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Could you check this RSPEC? I splitted RSPEC-3220 as it covered two similar, but different cases. (One reports on declarations, the other on invocations.)</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_1_dec_2015_145949_ann_campbell_wrote">on 1 Dec 2015, 14:59:49 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>I&#8217;ve updated the end of the description with a "this rule raises an issue" sentence. Please check my accuracy.</p>
</div>
</div>
</div>
</div>