<div class="paragraph">
<p>Since ANSI SQL-92, explicit joins using the <code>JOIN</code> keyword have been possible, and are preferred. Therefore table joins should be done with help of the one of the following clauses: <code>JOIN</code>, <code>INNER  JOIN</code>, <code>LEFT OUTER  JOIN</code>, <code>RIGHT OUTER JOIN</code>, and <code>FULL OUTER  JOIN</code>. The old way to join tables is deprecated and should not be used anymore.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>SELECT *
FROM PARTS, PRODUCTS
WHERE PARTS.PROD = PRODUCTS.PROD</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>SELECT *
FROM PARTS
INNER JOIN PRODUCTS ON PARTS.PROD = PRODUCTS.PROD</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Refactor this SQL to join tables with help of the JOIN keyword.</p>
</div>
</div>
</div>
</div>