<div class="paragraph">
<p>Call stacks containing lot of <code>PERFORM</code> statements is a key ingredient for making what&#8217;s known as "Spaghetti code".</p>
</div>
<div class="paragraph">
<p>Such code is hard to read, refactor and therefore maintain.</p>
</div>
<div class="paragraph">
<p>This rule supports both sections and paragraphs.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="paragraph">
<p>With a threshold of 3:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>PERFORM FIRST.

FIRST.
  PERFORM SECOND.

SECOND.
  PERFORM THIRD.

THIRD.
  PERFORM FOURTH.  *&gt; Noncompliant

FOURTH.
  DISPLAY something.</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>This module can trigger long chain of up to 6 PERFORMs, for example using the ones at line: 123 &#8594; 23 &#8594; 54 &#8594; &#8230;&#8203;</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="ulist">
<ul>
<li>
<p>key = max</p>
</li>
<li>
<p>Description : Maximum allowed call stack</p>
</li>
<li>
<p>Default value : 5</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s1684">is related to: S1684</h3>

</div>
</div>
</div>