<div class="paragraph">
<p>Shared coding conventions allow teams to collaborate efficiently. This rule checks that cursor parameters match the provided regular expression.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="paragraph">
<p>With the default regular expression <code>[a-zA-Z]([a-zA-Z0-9_]*[a-zA-Z0-9])?</code>:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>CREATE TABLE employee(
  name VARCHAR2(42)
);

DECLARE
  CURSOR mycursor2(Employee-name-parameter_ VARCHAR2) RETURN employee%ROWTYPE; -- Noncompliant

  CURSOR mycursor2(Employee-name-parameter_ VARCHAR2) RETURN employee%ROWTYPE IS SELECT * FROM employee WHERE name = Employee-name-parameter_; -- Noncompliant
BEGIN
  NULL;
END;
/

DROP TABLE employee;</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>CREATE TABLE employee(
  name VARCHAR2(42)
);

DECLARE
  CURSOR mycursor2(employeeNameParameter VARCHAR2) RETURN employee%ROWTYPE;

  CURSOR mycursor2(employeeNameParameter VARCHAR2) RETURN employee%ROWTYPE IS SELECT * FROM employee WHERE name = employeeNameParameter;
BEGIN
  NULL;
END;
/

DROP TABLE employee;</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Rename this cursor parameter to match the regular expression: xxx.</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="ulist">
<ul>
<li>
<p>key = regexp
description: The regular expression the name should match</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>default: [a-zA-Z]([a-zA-Z0-9_]*[a-zA-Z0-9])?</p>
</div>
</div>
</div>
</div>