<div class="paragraph">
<p>When a method is designed to return an invariant value, it may be poor design, but it shouldn&#8217;t adversely affect the outcome of your program. However, when it happens on all paths through the logic, it is surely a bug.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when a method contains several <code>return</code> statements that all return the same value.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>int foo(int a) {
  int b = 12;
  if (a == 1) {
    return b;
  }
  return b;  // Noncompliant
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Refactor this method to not always return the same value.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>Primary: method name</p>
</li>
<li>
<p>Secondary: each return statement value</p>
<div class="ulist">
<ul>
<li>
<p>message: "Returned value."</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s3400">relates to: S3400</h3>

</div>
</div>
</div>