<div class="paragraph">
<p><code>DATA BEGIN OF ... OCCURS</code> has been deprecated and will eventually be removed. All usages should be replaced.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>DATA BEGIN OF itab OCCURS n. "Noncompliant
  ...
DATA END OF itab [VALID BETWEEN intlim1 AND intlim2].</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>DATA BEGIN OF wa.
       ...
DATA END OF wa.
DATA itab LIKE TABLE OF wa.</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Declare this data structure differently to remove the use of "DATA BEGIN OF OCCURS".</p>
</div>
</div>
</div>
</div>