<div class="paragraph">
<p>In Python 3, attempting to raise an object which does not derive from BaseException will raise a <code>TypeError</code>. In Python 2 it is possible to raise old-style classes but this shouldn&#8217;t be done anymore in order to be compatible with Python 3.</p>
</div>
<div class="paragraph">
<p>If you are about to create a custom Exception class, note that custom exceptions should inherit from <code>Exception</code>, not <code>BaseException</code>. <code>Exception</code> allows people to catch all exceptions except the ones explicitly asking the interpreter to stop, such as <code>KeyboardInterrupt</code> and <a href="https://docs.python.org/3/library/exceptions.html#GeneratorExit"><code>GeneratorExit</code></a> which is not an error. See <a href="https://www.python.org/dev/peps/pep-0352/#exception-hierarchy-changes">PEP 352</a> for more information.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when an object which doesn&#8217;t derive from BaseException is raised.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>raise "Something went wrong"  # Noncompliant

class A:
    pass

raise A  # Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class MyError(Exception):
    pass

raise MyError("Something went wrong")
raise MyError</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.python.org/3/tutorial/errors.html">Python documentation - Errors and Exceptions</a></p>
</li>
<li>
<p><a href="https://www.python.org/dev/peps/pep-0352/#exception-hierarchy-changes">PEP 352 - Required Superclass for Exceptions</a></p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Change this code so that it raises an object deriving from BaseException.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s3696">relates to: S3696</h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s5708">is related to: S5708</h3>

</div>
<div class="sect2">
<h3 id="_on_13_nov_2019_162349_nicolas_harraudeau_wrote">on 13 Nov 2019, 16:23:49 Nicolas Harraudeau wrote:</h3>
<div class="paragraph">
<p>Note: This rule is similar to RSPEC-3696 "Non-exception types should not be thrown" but this one is a Bug instead of a Code Smell because raising a non-exception in Python will raise a TypeError.</p>
</div>
</div>
</div>
</div>