<div class="paragraph">
<p>By definition, primitive types are not Objects and so they can&#8217;t be <code>null</code>. Adding <code>@CheckForNull</code> or <code>@Nullable</code> on primitive types adds confusion and is useless.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when <code>@CheckForNull</code> or <code>@Nullable</code> is set on a method returning a primitive type: byte, short, int, long, float, double, boolean, char.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>@CheckForNull
boolean isFoo() {
 ...
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>boolean isFoo() {
 ...
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>"XXX" annotation should not be used on primitive types</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Primitive type</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_13_jun_2018_173029_alexandre_gigleux_wrote">on 13 Jun 2018, 17:30:29 Alexandre Gigleux wrote:</h3>
<div class="paragraph">
<p>@CheckForNull = javax.annotation.Nullable</p>
</div>
<div class="paragraph">
<p>@Nullable = javax.annotation.CheckForNull</p>
</div>
</div>
</div>
</div>