<div class="paragraph">
<p>Operators <code>in</code> and <code>not in</code>, also called <a href="https://docs.python.org/3/reference/expressions.html#membership-test-operations">"membership test operators"</a>, require that the right operand supports membership protocol.</p>
</div>
<div class="paragraph">
<p>In order to support this membership protocol a user-defined class should implement at least one of the following methods: <code>__contains__</code>, <code>__iter__</code>, <code>__getitem__</code>.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when operators <code>in</code> and <code>not in</code> are called with a right operand not supporting membership protocol.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>myint = 42

if 42 in myint:  # Noncompliant. Integers don't support membership protocol
    pass

class A:
    def __init__(self, values):
        self._values = values

if "mystring" in A(["mystring"]):  # Noncompliant. Class A doesn't support membership protocol
    pass</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>mylist = [42]

if 42 in mylist:
    pass

class MyContains:
    def __init__(self, values):
        self._values = values

    def __contains__(self, value):
        return value in self._values

if "mystring" in MyContains(["mystring"]):  # Noncompliant. Class A don't support membership protocol
    pass

# OR

class MyIterable:
    def __init__(self, values):
        self._values = values

    def __iter__(self):
        return iter(self._values)

if "mystring" in MyIterable(["mystring"]):  # Noncompliant. Class A don't support membership protocol
    pass

# OR

class MyGetItem:
    def __init__(self, values):
        self._values = values

    def __getitem__(self, key):
        return self._values[key]

if "mystring" in MyGetItem(["mystring"]):  # Noncompliant. Class A don't support membership protocol
    pass</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.python.org/3/reference/expressions.html#membership-test-operations">Python documentation - Membership test operations</a></p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Change the type of X; type Y does not support membership protocol.</p>
</div>
<div class="paragraph">
<p>For an expression like "a in X" where X has type Y</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Primary: The "in" or "not in" operator</p>
</div>
<div class="paragraph">
<p>Secondary: The operator&#8217;s right operand</p>
</div>
</div>
</div>
</div>