<div class="paragraph">
<p>Duplicated string literals make the process of refactoring error-prone, since you must be sure to update all occurrences.</p>
</div>
<div class="paragraph">
<p>On the other hand, constants can be referenced from many places, but only need to be updated in a single place.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="paragraph">
<p>With the default threshold of 3:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>public def run() {
  prepare("action random1")    // Noncompliant - "action random1" is duplicated 3 times
  execute("action random1")
  release("action random1")
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public def run() {
  val action = "action random1"
  prepare(action)
  execute(action)
  release(action)
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_exceptions">Exceptions</h2>
<div class="sectionbody">
<div class="paragraph">
<p>To prevent generating some false-positives, literals having 5 or less characters are excluded as well as literals containing only letters, digits and '_'.</p>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Define a constant instead of duplicating this literal "{string}" {number} times.</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="ulist">
<ul>
<li>
<p>key = threshold
Description: Number of times a literal must be duplicated to trigger an issue</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Default: 3</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>Primary: The first occurrence</p>
</li>
<li>
<p>Secondary: the following occurrences</p>
<div class="ulist">
<ul>
<li>
<p>message: 'Duplication.'</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_16_aug_2013_081335_freddy_mallet_wrote">on 16 Aug 2013, 08:13:35 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARJAVA-291</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_13_jan_2014_142946_dinesh_bolkensteyn_wrote">on 13 Jan 2014, 14:29:46 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>For COBOL, this rule should apply only to literals contained with tine PROCEDURE DIVISION.</p>
</div>
<div class="paragraph">
<p>Moreover, occurences coming from copybooks should be excluded.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_12_feb_2014_210634_freddy_mallet_wrote">on 12 Feb 2014, 21:06:34 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3460 for PHP</p>
</div>
</div>
</div>
</div>