<div class="paragraph">
<p>When constructing an <code>std::scoped_lock</code>, the constructor arguments are used to list the mutexes that the <code>scoped_lock</code> will lock on the creation and unlock on destruction. It is possible to construct a <code>scoped_lock</code> without any parameter, but in that case, it does absolutely nothing and is just dead code, which was probably not the intent of the user.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>void f1(std::mutex &amp;m) {
  std::scoped_lock lock; // Noncompliant
  // Do some work
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>void f1(std::mutex &amp;m) {
  std::scoped_lock lock {m}; // Compliant
  // Do some work
}

template&lt;class... D&gt;
void processAll(D &amp;...data) {
  scoped_lock lock {data.getMutex()...}; // Compliant, even if the list might be empty in some cases
  // Do some work
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s5997">relates to: S5997</h3>

</div>
</div>
</div>