<div class="paragraph">
<p>It is highly recommended <strong>not</strong> to use wildcard types as return types. Because the type inference rules are fairly complex it is unlikely the user of that API will know how to use it correctly.</p>
</div>
<div class="paragraph">
<p>Let&#8217;s take the example of method returning a "List&lt;? extends Animal&gt;". Is it possible on this list to add a Dog, a Cat, &#8230;&#8203; we simply don&#8217;t know. And neither does the compiler, which is why it will not allow such a direct use. The use of wildcard types should be limited to method parameters.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when a method returns a wildcard type.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>List&lt;? extends Animal&gt; getAnimals(){...}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>List&lt;Animal&gt; getAnimals(){...}</pre>
</div>
</div>
<div class="paragraph">
<p>or</p>
</div>
<div class="listingblock">
<div class="content">
<pre>List&lt;Dog&gt; getAnimals(){...}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove usage of generic wildcard type</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_1_nov_2013_192211_freddy_mallet_wrote">on 1 Nov 2013, 19:22:11 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by https://jira.sonarsource.com/browse/SONARJAVA-374</p>
</div>
</div>
</div>
</div>