<div class="paragraph">
<p>The output of <code>DBMS_OUTPUT.PUT_LINE</code> is not always visible, for example when <code>SERVEROUTPUT</code> is set to <code>OFF</code>. Moreover, there is no standardized way to specify the importance of the message. It is better to use a logging mechanism instead.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>SET SERVEROUTPUT ON

BEGIN
  DBMS_OUTPUT.PUT_LINE('An error occured'); -- Noncompliant
END;
/</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this use of "DBMS_OUTPUT.PUT_LINE".</p>
</div>
</div>
</div>
</div>