<div class="paragraph">
<p>When an object is created with a setter for a property but without a getter for that property, the property is inaccessible and is thus useless.</p>
</div>
<div class="paragraph">
<p>This rule also enforces the reverse situation (getter but no setter).</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>var obj = {
    set foo(value) {
        this.fooval = value;
    }
};</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>var obj = {
    set foo(value) {
        this.fooval = value;
    },
    get foo() {
        return this.fooval;
    }
};</pre>
</div>
</div>
<div class="paragraph">
<p>or</p>
</div>
<div class="listingblock">
<div class="content">
<pre>var obj = {
    setFoo(value) {    // a standard method, not a setter
        this.fooval = value;
    }
};</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Provide a [setter|getter] matching this [getter|setter] or replace this accessor with a simple method.</p>
</div>
</div>
</div>
</div>