<div class="paragraph">
<p>Java 8 adds <code>Comparator.comparing</code> to allow the creation of a single-value comparator to be shorthanded into a single call. This cleaner syntax should be preferred.</p>
</div>
<div class="paragraph">
<p><strong>Note</strong> that this rule is automatically disabled when the project&#8217;s <code>sonar.java.source</code> is lower than <code>8</code>.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>Comparator&lt;Foo&gt; compartor = (foo1, foo2) -&gt; foo.getName().compareTo(foo2.getName());  // Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>Comparator&lt;Foo&gt; compartor = Comparator.comparing(Foo::getName);</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Simplify this code with the use of "Comparator.comparing" instead.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>the comparator definition</p>
</div>
</div>
</div>
</div>