<div class="paragraph">
<p>The <code>else</code> clause of a loop is skipped when a <code>break</code> is executed in this loop. In other words, a loop with an <code>else</code> but no <code>break</code> statement will always execute the <code>else</code> part (unless of course an exception is raised or <code>return</code> is used). If this is what the developer intended, it would be much simpler to have the <code>else</code> statement removed and its body unindented. Thus having a loop with an <code>else</code> and no <code>break</code> is most likely an error.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>from typing import List

def search_first_number_without_break(elements: List[str]):
    for elt in elements:
        if elt.isnumeric():
            return elt
    else:  # Noncompliant. This will be executed every time
        raise ValueError("List does not contain any number")</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>from typing import List

def search_first_number_with_break(elements: List[str]):
    for elt in elements:
        if elt.isnumeric():
            break
    else:
        raise ValueError("List does not contain any number")
    return elt</pre>
</div>
</div>
<div class="paragraph">
<p>or</p>
</div>
<div class="listingblock">
<div class="content">
<pre>from typing import List

def search_first_number_without_else(elements: List[str]):
    for elt in elements:
        if elt.isnumeric():
            return elt
    raise ValueError("List does not contain any number")</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.python.org/3/tutorial/controlflow.html#break-and-continue-statements-and-else-clauses-on-loops">Python documentation - break and continue Statements, and else Clauses on Loops</a></p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add a "break" statement or remove this "else" clause.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_14_apr_2020_094958_pierre_yves_nicolas_wrote">on 14 Apr 2020, 09:49:58 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>I would expect the compliant solution to suggest something like:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>for i in range(50):
  if i == 42:
    print('Magic number in range')
print('Magic number not found')</pre>
</div>
</div>
<div class="paragraph">
<p>That behaves the same way as the noncompliant example but it doesn&#8217;t contain the suspicious <code>else</code>.</p>
</div>
</div>
</div>
</div>