<div class="paragraph">
<p>When a generic class is declaring wildcard with an upper bound that is <code>final</code>, this generic class is not generic at all because it only accepts one and only one type at runtime: the one that is <code>final</code>. Instead of using <code>Generics</code>, it&#8217;s simpler to directly use the concrete <code>final</code> class.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public static &lt;T extends String&gt; T getMyString() { // Noncompliant; String is a "final" class and so can't be extended
 [...]
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public static String getMyString() { // Compliant
  [...]
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Refactor this code to use XXX only.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_31_oct_2018_093436_nicolas_peru_wrote">on 31 Oct 2018, 09:34:36 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>Title of the rule is dodgy : there is no wildcard at all in this example.</p>
</div>
<div class="paragraph">
<p>"upper bound of type variables" would be more correct.</p>
</div>
</div>
</div>
</div>