<div class="paragraph">
<p>This rule checks that the specified child elements are present inside the specified parent elements.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Given a parent/child combination of <code>&lt;head&gt;</code>/<code>&lt;title&gt;</code>:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>&lt;html&gt;
  &lt;head&gt;
  &lt;/head&gt;  &lt;!-- Noncompliant; no title element --&gt;
  &lt;body&gt;
    ...</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>&lt;html&gt;
  &lt;head&gt;
    &lt;title&gt;My Page&lt;/title&gt;
  &lt;/head&gt;
  &lt;body&gt;
    ...</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add the missing "xxx" element to this "yyy".</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="ulist">
<ul>
<li>
<p>key = child</p>
<div class="ulist">
<ul>
<li>
<p>description: Name of the child element</p>
</li>
</ul>
</div>
</li>
<li>
<p>key: parent</p>
<div class="ulist">
<ul>
<li>
<p>description: Name of the parent element</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
</div>
</div>
</div>