<div class="paragraph">
<p>This rule enforces the idea to use ES6 style to create React Component instead of the ES5 way.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>var Rating = createReactClass({ // Noncompliant
  [...]
});</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>export default class Rating extends React.Component { // Compliant
[...]
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_23_feb_2018_114851_alexandre_gigleux_wrote">on 23 Feb 2018, 11:48:51 Alexandre Gigleux wrote:</h3>
<div class="paragraph">
<p>Covered by ESLint for React: https://github.com/yannickcr/eslint-plugin-react/blob/HEAD/docs/rules/prefer-es6-class.md (react/prefer-es6-class)</p>
</div>
</div>
</div>
</div>