<div class="paragraph">
<p>C&#43;&#43;20 extends the <code>using</code> declaration to <code>class enum</code>. <code>using enum</code> introduces all the <code>enum</code> constants into the current scope.</p>
</div>
<div class="listingblock">
<div class="content">
<pre>enum class rgbaColorChannel { red, green, blue, alpha };

std::string_view toString(rgbaColorChannel channel) {
  switch (channel) {
    using enum rgbaColorChannel;
    case red:   return "red";
    case green: return "green";
    case blue:  return "blue";
    case alpha: return "alpha";
  }
}</pre>
</div>
</div>
<div class="paragraph">
<p>As with other <code>using</code> declarations, <code>using enum</code> improves readability when used in small scopes yet might generate confusion in large scopes.</p>
</div>
<div class="paragraph">
<p>The <code>switch</code> statement, as in the example above, when applied to a <code>class enum</code> value, is a natural scope for <code>using enum</code>.</p>
</div>
<div class="paragraph">
<p>This rule reports scopes that use a particular <code>class enum</code> extensively and could benefit from <code>using enum</code> declaration. For example, it reports most <code>switch</code> statements applied to an <code>enum</code> value.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>enum class DayOfTheWeek { mon, tue, wed, thu, fri, sat, sun };

bool isWorkDay(DayOfTheWeek day) {
  switch(day) { // Noncompliant: case statements are too verbose
    case DayOfTheWeek::mon: return true;
    case DayOfTheWeek::tue: return true;
    case DayOfTheWeek::wed: return true;
    case DayOfTheWeek::thu: return true;
    case DayOfTheWeek::fri: return true;
    case DayOfTheWeek::sat: return false;
    case DayOfTheWeek::sun: return false;
  }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>enum class DayOfTheWeek { mon, tue, wed, thu, fri, sat, sun };

bool isWorkDay(DayOfTheWeek day) {
  switch(day) {
    using enum DayOfTheWeek;

    case mon: return true;
    case tue: return true;
    case wed: return true;
    case thu: return true;
    case fri: return true;
    case sat: return false;
    case sun: return false;
  }
}</pre>
</div>
</div>
</div>
</div>