<div class="paragraph">
<p>While the assignment of default parameter values is typically a good thing, it can go very wrong very quickly when mutable objects are used. That&#8217;s because a new instance of the object <em>is not</em> created for each function invocation. Instead, all invocations share the same instance, and the changes made for one caller are made for all!</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>def get_attr_array(obj, arr=[]):  # Noncompliant
  props = (name for name in dir(obj) if not name.startswith('_'))
  arr.extend(props)  # after only a few calls, this is a big array!
  return arr</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>def get_attr_array(obj, arr):
  props = (name for name in dir(obj) if not name.startswith('_'))
  arr.extend(props)
  return arr</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove the default value of "xxx".</p>
</div>
</div>
</div>
</div>