<div class="paragraph">
<p>Using a unique symbolic <code>PLIST</code> name provides a single definition point for the parameters used by any external routine being called. This ensures that the same information is passed every time the external routine is called, reducing the potential for mistakes in operation.</p>
</div>
<div class="paragraph">
<p>This rule flags non-compliant instances of <code>CALL</code> and <code>CALLB</code>.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>C                   CALLB(D)  'XR01'
C                   PARM                    WSXR07
C                   PARM                    WSFLD1</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>C     P0XR07        PLIST
C                   PARM                    WSXR07
C                   PARM                    WSFLD1

C                   CALLB     'XR07'        P0XR07</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Create a PLIST for this invocation.</p>
</div>
</div>
</div>
</div>