<div class="paragraph">
<p>Using unvalidated values can expose an application to injection attacks.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
  // ...
  Employee employee = new Employee();
  employee.setFirstName(request.getParameter("firstName")); // Noncompliant
  // ...

  save(employee); // Uh-oh!</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>"xxx" is provided externally to the method an is not sanitized before use.</p>
</div>
</div>
</div>
</div>