<div class="paragraph">
<p>Multiple spaces in a regular expression can make it hard to tell how many spaces should be matched. It&#8217;s more readable to use only one space and then indicate with a quantifier how many spaces are expected.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>const pattern = /Hello,   world!/;</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>const pattern = /Hello, {3}world!/;</pre>
</div>
</div>
</div>
</div>