<div class="paragraph">
<p>When the second and third operands of a ternary operator are the same, the operator will always return the same value regardless of the condition. Either the operator itself is pointless, or a mistake was made in coding it.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>func canVote(person:Person) -&gt; Bool {
  return person.age &gt; 18 ? true : true // Noncompliant; is this what was intended?
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>func canVote(person:Person) -&gt; Bool {
  return person.age &gt; 18 ? true : false
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>This operation returns the same value whether the condition is "true" or "false".</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_25_mar_2015_202333_evgeny_mandrikov_wrote">on 25 Mar 2015, 20:23:33 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I slightly adjusted description and message.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_26_mar_2015_140604_ann_campbell_wrote">on 26 Mar 2015, 14:06:04 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~evgeny.mandrikov] I put the message back the way it was - your version was a repetition of the title without the "The" at the front.</p>
</div>
<div class="paragraph">
<p>Also updated description.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_26_mar_2015_141253_evgeny_mandrikov_wrote">on 26 Mar 2015, 14:12:53 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Thanks! LGTM.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_16_feb_2018_194412_alban_auzeill_wrote">on 16 Feb 2018, 19:44:12 Alban Auzeill wrote:</h3>
<div class="paragraph">
<p>\[~freddy.mallet] I removed this rule from the "Common Rule: Yes" because it is deprecated in favor of [RSPEC-1871] that is also part of the "Common Rule: Yes"</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_19_feb_2018_105330_freddy_mallet_wrote">on 19 Feb 2018, 10:53:30 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>I see your point [~alban.auzeill] and at the same time I&#8217;m pretty sure that few implementations of RSPEC-1871 raise an issue when a ternary operator returns the same value regardless of the condition as initially RSPEC-1871 was only targeting the <code>switch</code> and <code>if</code> control statements. But fine.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_9_mar_2018_160434_freddy_mallet_wrote">on 9 Mar 2018, 16:04:34 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>In fact [~alban.auzeill], this RSPEC is not deprecated by RSPEC-1871 but by RSPEC-3923 and indeed all implementations of RSPEC-3923 already cover the case of ternary operator. So no need to touche anything. Thanks.</p>
</div>
</div>
</div>
</div>