<div class="paragraph">
<p>Most checks against an <code>indexOf</code> call against an array compare it with -1 because 0 is a valid index. Any checks which look for values &gt;0 ignore the first element, which is likely a bug. If you&#8217;re merely checking the presence of the element, consider using <code>includes</code> instead. Before usingÂ <code>includes</code> method make sure that your browser version is supporting it.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>var color = "blue";
var name = "ishmael";

var arr = [color, name];

if (arr.indexOf("blue") &gt; 0) { // Noncompliant
  // ...
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>var color = "blue";
var name = "ishmael";

var arr = [color, name];

if (arr.indexOf("blue") &gt;= 0) {
  // ...
}
if (arr.includes("blue")) {
  // ...
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="paragraph">
<p><a href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes">Array.prototype.includes()</a> documentation at MDN</p>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>This check ignores index 0; consider using 'includes' method to make this check safe and explicit.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_10_apr_2017_152904_elena_vilchik_wrote">on 10 Apr 2017, 15:29:04 Elena Vilchik wrote:</h3>
<div class="paragraph">
<p>Temporary removed from SW before FP are removed: SONARJS-991</p>
</div>
</div>
<div class="sect2">
<h3 id="_relates_to_s2912">relates to: S2912</h3>

</div>
<div class="sect2">
<h3 id="_on_5_may_2015_123922_ann_campbell_wrote">on 5 May 2015, 12:39:22 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>After discussion, rule only checks for tests against <code>&gt;0</code>, so this is ignored: <code>if (str.indexOf('a') &gt;  1)</code></p>
</div>
</div>
<div class="sect2">
<h3 id="_on_26_jun_2015_133635_tamas_vajk_wrote">on 26 Jun 2015, 13:36:35 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] this rule talks about both lists and strings, but the error message explicitly mentions strings. Also, the rule seems to only check for <code>&gt;0</code>, but the message says <code>n</code>.</p>
</div>
<div class="paragraph">
<p>I would probably go with something like this: <em>0 is a valid index, but this check ignores it.</em></p>
</div>
</div>
<div class="sect2">
<h3 id="_on_26_jun_2015_152759_ann_campbell_wrote">on 26 Jun 2015, 15:27:59 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Done [~tamas.vajk]</p>
</div>
</div>
</div>
</div>