<div class="paragraph">
<p>Most checks against an <code>IndexOf</code> value compare it with -1 because 0 is a valid index. Any checks which look for values <code>&gt; 0</code> ignore the first element, which is likely a bug. If the intent is merely to check inclusion of a value in a <code>string</code>, <code>List</code>, or an array, consider using the <code>Contains</code> method instead.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when an <code>IndexOf</code> value retrieved from a <code>string</code>, <code>List</code> or array is tested against <code>&gt; 0</code>.</p>
</div>
<div class="paragraph">
<p>This rule also raises an issue when <code>IndexOfAny</code>, <code>LastIndexOf</code> or <code>LastIndexOfAny</code> from a <code>string</code> is tested against <code>&gt; 0</code></p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>Dim Color As String = "blue"
Dim Name As String = "ishmael"

Dim Strings As New List(Of String)
Strings.Add(Color)
Strings.Add(Name)
Dim StringArray As String() = Strings.ToArray()

If Strings.IndexOf(Color) &gt; 0 Then ' Noncompliant
  ' ...
End If

If Name.IndexOf("ish") &gt; 0 Then ' Noncompliant
  ' ...
End If

If Name.IndexOf("ae") &gt; 0 Then ' Noncompliant
  ' ...
End If

If Array.IndexOf(StringArray, Color) &gt; 0 Then ' Noncompliant
  ' ...
End If</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>Dim Color As String = "blue"
Dim Name As String = "ishmael"

Dim Strings As New List(Of String)
Strings.Add(Color)
Strings.Add(Name)
Dim StringArray As String() = Strings.ToArray()

If Strings.IndexOf(Color) &gt; -1 Then
  ' ...
End If

If Name.IndexOf("ish") &gt;= 0 Then
  ' ...
End If

If Name.Contains("ae") Then
  ' ...
End If

If Array.IndexOf(StringArray, Color) &gt;= 0 Then
  ' ...
End If</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>0 is a valid index, but this check against n ignores it.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s2912">relates to: S2912</h3>

</div>
<div class="sect2">
<h3 id="_on_5_may_2015_123922_ann_campbell_wrote">on 5 May 2015, 12:39:22 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>After discussion, rule only checks for tests against <code>&gt;0</code>, so this is ignored: <code>if (str.indexOf('a') &gt;  1)</code></p>
</div>
</div>
<div class="sect2">
<h3 id="_on_26_jun_2015_133635_tamas_vajk_wrote">on 26 Jun 2015, 13:36:35 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] this rule talks about both lists and strings, but the error message explicitly mentions strings. Also, the rule seems to only check for <code>&gt;0</code>, but the message says <code>n</code>.</p>
</div>
<div class="paragraph">
<p>I would probably go with something like this: <em>0 is a valid index, but this check ignores it.</em></p>
</div>
</div>
<div class="sect2">
<h3 id="_on_26_jun_2015_152759_ann_campbell_wrote">on 26 Jun 2015, 15:27:59 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Done [~tamas.vajk]</p>
</div>
</div>
</div>
</div>