<div class="paragraph">
<p>Using the same name for multiple purposes reduces the understandability of the code and might eventually lead to bugs.</p>
</div>
<div class="paragraph">
<p>This rule verifies that no label is reused in an inner scope.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>&lt;&lt;foo&gt;&gt;
DECLARE
  a CONSTANT PLS_INTEGER := 0;
BEGIN
  &lt;&lt;foo&gt;&gt;                                  -- Noncompliant
  DECLARE
    b CONSTANT PLS_INTEGER := 42;
  BEGIN
    DBMS_OUTPUT.PUT_LINE('x = ' || foo.b); -- Confusing
  END;
END;
/</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>&lt;&lt;foo&gt;&gt;
DECLARE
  a CONSTANT PLS_INTEGER := 0;
BEGIN
  &lt;&lt;bar&gt;&gt;
  DECLARE
    b CONSTANT PLS_INTEGER := 42;
  BEGIN
    DBMS_OUTPUT.PUT_LINE('x = ' || bar.b);
  END;
END;
/</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Give a unique name to this identifier which was first declared at line X</p>
</div>
</div>
</div>
</div>