<div class="paragraph">
<p>Making a <code>base</code> call in an overriding method is generally a good idea, but not in <code>GetHashCode</code> and <code>Equals</code> for classes that directly extend <code>object</code> because those methods are based on the object reference. Meaning that no two <code>objects</code> that use those <code>base</code> methods will ever be equal or have the same hash.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public class Point
{
  private readonly int x;
  public MyClass(int x)
  {
    this.x = x;
  }
  public override int GetHashCode()
  {
    return x.GetHashCode() ^ base.GetHashCode(); //Noncompliant
  }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public class Point
{
  private readonly int x;
  public MyClass(int x)
  {
    this.x = x;
  }
  public override int GetHashCode()
  {
    return x.GetHashCode();
  }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_exceptions">Exceptions</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule doesn&#8217;t report on guard conditions checking for reference equality.</p>
</div>
<div class="listingblock">
<div class="content">
<pre>public class Point
{
  public override bool Equals(object obj)
  {
    if (base.Equals(obj)) // Compliant, although it could be replaced with object.ReferenceEquals(obj, this), which is clearer
    {
      return true;
    }
    ...
  }
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this "base" call to "object.[GetHashCode|Equals]", which is directly based on the object reference.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_8_jul_2015_173308_ann_campbell_wrote">on 8 Jul 2015, 17:33:08 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>FYI [~nicolas.peru]. I&#8217;ve targeted this to Java because I believe the underlying mechanisms are the same.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_20_jul_2015_114355_tamas_vajk_wrote">on 20 Jul 2015, 11:43:55 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Looks good.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_22_jul_2015_071305_nicolas_peru_wrote">on 22 Jul 2015, 07:13:05 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>Looks good for java</p>
</div>
</div>
</div>
</div>