<div class="paragraph">
<p>Having two paragraphs with the same name in the same section or in no section at all is bad practice. At best, each copy contains the same code, and the redefinition is simply useless, duplicated code. At worst, the paragraphs contain different logic, potentially leading to confusion and unexpected results as a programmer who was aware of the first paragraph definition inadvertently invokes the second. For these reasons, paragraphs with duplicated names should be either removed or renamed.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre> LOAD-DATA.
     EXEC SQL
         INSERT INTO EMP (EMPNO, ENAME, DEPTNO)
             VALUES (:EMP-NUMBER, :EMP-NAME, :DEPT-NUMBER)
     END-EXEC.

 LOAD-DATA.
     IF EMP-NUMBER = ZERO
         MOVE FALSE TO VALID-DATA
         PERFORM GET-EMP-NUM UNTIL VALID-DATA = TRUE
     ELSE
         EXEC SQL DELETE FROM EMP
             WHERE EMPNO = :EMP-NUMBER
         END-EXEC
         ADD 1 TO DELETE-TOTAL.
     END-IF.

 LOAD-DATA.
     EXEC SQL
         INSERT INTO EMP (EMPNO, ENAME, DEPTNO)
             VALUES (:EMP-NUMBER, :EMP-NAME, :DEPT-NUMBER)
     END-EXEC.</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre> LOAD-DATA.
     EXEC SQL
         INSERT INTO EMP (EMPNO, ENAME, DEPTNO)
             VALUES (:EMP-NUMBER, :EMP-NAME, :DEPT-NUMBER)
     END-EXEC.

 CLEAR-EMP.
     IF EMP-NUMBER = ZERO
         MOVE FALSE TO VALID-DATA
         PERFORM GET-EMP-NUM UNTIL VALID-DATA = TRUE
     ELSE
         EXEC SQL DELETE FROM EMP
             WHERE EMPNO = :EMP-NUMBER
         END-EXEC
         ADD 1 TO DELETE-TOTAL.
     END-IF.</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Rename or remove this copy of XXX</p>
</div>
</div>
</div>
</div>