<div class="paragraph">
<p>Typically you want to use <code>using</code> to create a local <code>IDisposable</code> variable; it will trigger disposal of the object when control passes out of the block&#8217;s scope. The exception to this rule is when your method returns that <code>IDisposable</code>. In that case <code>using</code> disposes of the object before the caller can make use of it, likely causing exceptions at runtime. So you should either remove <code>using</code> or avoid returning the <code>IDisposable</code>.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public FileStream WriteToFile(string path, string text)
{
  using (var fs = File.Create(path)) // Noncompliant
  {
    var bytes = Encoding.UTF8.GetBytes(text);
    fs.Write(bytes, 0, bytes.Length);
    return fs;
  }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public FileStream WriteToFile(string path, string text)
{
  var fs = File.Create(path);
  var bytes = Encoding.UTF8.GetBytes(text);
  fs.Write(bytes, 0, bytes.Length);
  return fs;
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove the "using" statement; it will cause automatic disposal of "xxx".</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s837">is related to: S837</h3>

</div>
<div class="sect2">
<h3 id="_on_8_jun_2015_093621_tamas_vajk_wrote">on 8 Jun 2015, 09:36:21 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] LGTM, maybe we could emphasize another compliant solution, that is not to return an <code>IDisposable</code> at all.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_8_jun_2015_121329_ann_campbell_wrote">on 8 Jun 2015, 12:13:29 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>See what you think [~tamas.vajk]</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_9_jun_2015_113923_tamas_vajk_wrote">on 9 Jun 2015, 11:39:23 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Looks good.</p>
</div>
</div>
</div>
</div>