<div class="paragraph">
<p><code>DISTINCT</code> operator causes the <code>SELECT</code> statement to avoid the SAP buffering and to read directly from the database and not from the buffer on the application server.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>SELECT DISTINCT carrid
       FROM spfli
       INTO count
       WHERE cityto = 'NEW YORK'.</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Double check that usage of "DISTINCT" operator is required as it leads to bypass SAP buffering.</p>
</div>
</div>
</div>
</div>