<div class="paragraph">
<p>Declaring a parameter without specifying its data type leaves the compiler to assign the type that seems the most appropriate - whether it&#8217;s what you need or not. Therefore you should always specify the data type.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>Sub SubComputeArea(Length, TheWidth)
  [. . .]
End Sub</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>Sub SubComputeArea(Length As Integer, TheWidth As Integer)
  [. . .]
End Sub</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Specify the data type for XXX</p>
</div>
</div>
</div>
</div>