<div class="paragraph">
<p>While working with bitwise operators <code>&amp;</code> or <code>|</code> it is easy to make a typo andÂ write the equivalent logical operators <code>&amp;&amp;</code> or <code>||</code>. This rule is raising issues when the right operand of a logical expression  <code>&amp;&amp;</code> or <code>||</code> is a constant of integral type, as the developer probably meant to use the corresponding bitwise operator <code>&amp;</code> or <code>|</code>.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>int fun(int a) {
  return a || 4; // Noncompliant, did you mean to use bitwise operator '|'?
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>int fun(int a) {
 return a | 4;
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Review this logical (&amp;&amp; | ||) expression with constant operand.</p>
</div>
</div>
</div>
</div>