<div class="paragraph">
<p>The point of declaring an optional property or parameter is to make explicit the fact that it might contain no valid value, i.e. <code>null</code> or <code>undefined</code>. Using a non-null assertion (the <code>!.</code> operator) will lead to a runtime error if the optional does contain <code>null</code> or <code>undefined</code>. Even if the value is tested first, it&#8217;s still considered a bad practice to use a non-null assertion.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>function doTheThing(foo?: Foo) {
  let s = foo!.bar;  // Noncompliant
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>function doTheThing(foo?: Foo) {
  if (foo) {
    let s = foo.bar;
  }
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use optional binding or chaining instead to access the value of "x".</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s3655">is related to: S3655</h3>

</div>
<div class="sect2">
<h3 id="_on_9_apr_2018_133528_stas_vilchik_wrote">on 9 Apr 2018, 13:35:28 Stas Vilchik wrote:</h3>
<div class="paragraph">
<p>Because the case described in the description is a bad practice, I changed the type to Code Smell.</p>
</div>
</div>
</div>
</div>