<div class="paragraph">
<p>The <code>void</code> operator evaluates its argument and unconditionally returns <code>undefined</code>. It can be useful in pre-ECMAScript 5 environments, where <code>undefined</code> could be reassigned, but generally, its use makes code harder to understand.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>void doSomething();</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>doSomething();</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_exceptions">Exceptions</h2>
<div class="sectionbody">
<div class="paragraph">
<p>No issue is raised when <code>void 0</code> is used in place of <code>undefined</code>.</p>
</div>
<div class="listingblock">
<div class="content">
<pre>if (parameter === void 0) {...}</pre>
</div>
</div>
<div class="paragraph">
<p>No issue is also raised when <code>void</code> is used before immediately invoked function expressions.</p>
</div>
<div class="listingblock">
<div class="content">
<pre>void (function() {
   ...
}());</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this use of the 'void' operator.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p><code>void arg</code></p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_30_sep_2016_101428_elena_vilchik_wrote">on 30 Sep 2016, 10:14:28 Elena Vilchik wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Hi! Could you finish this ticket? Thanks!</p>
</div>
<div class="paragraph">
<p>(I&#8217;m struggling to define severity)</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_30_sep_2016_164730_ann_campbell_wrote">on 30 Sep 2016, 16:47:30 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~elena.vilchik] could you supply some code samples?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_30_sep_2016_171313_elena_vilchik_wrote">on 30 Sep 2016, 17:13:13 Elena Vilchik wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] done</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_14_mar_2017_103327_elena_vilchik_wrote">on 14 Mar 2017, 10:33:27 Elena Vilchik wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Hi! we added exclusion to this rule: <code>void 0</code> used as <code>undefined</code>. Could you update main part of description, as it&#8217;s outdated now. Thanks!</p>
</div>
</div>
</div>
</div>