<div class="paragraph">
<p>This rule raises an issue when a configured Java package or class is used in a JSP file.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Given a disallowed list of <code>java.sql,ArrayList</code>:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>&lt;%@ page import="java.sql.*" %&gt;     &lt;!-- Noncompliant --&gt;
&lt;% java.util.ArrayList clients; %&gt;  &lt;!-- Noncompliant --&gt;
&lt;% java.lang.String name; %&gt;        &lt;!-- Compliant --&gt;</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove the use of "xxx".</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="ulist">
<ul>
<li>
<p>key = libraries</p>
<div class="ulist">
<ul>
<li>
<p>description: Comma-separated list of Java packages or classes, such as java.sql or java.util.ArrayList</p>
</li>
</ul>
</div>
</li>
<li>
<p>key: message</p>
<div class="ulist">
<ul>
<li>
<p>description: Issue message which is displayed in case of violation</p>
</li>
<li>
<p>default value: Remove the usage of this library which is not allowed.</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
</div>
</div>
</div>