<div class="paragraph">
<p>A thread acquiring a lock on an object that can be accessed across application domain boundaries runs the risk of being blocked by another thread in a different application domain. Objects that can be accessed across application domain boundaries are said to have weak identity. Types with weak identity are:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>MarshalByRefObject</code></p>
</li>
<li>
<p><code>ExecutionEngineException</code></p>
</li>
<li>
<p><code>OutOfMemoryException</code></p>
</li>
<li>
<p><code>StackOverflowException</code></p>
</li>
<li>
<p><code>String</code></p>
</li>
<li>
<p><code>MemberInfo</code></p>
</li>
<li>
<p><code>ParameterInfo</code></p>
</li>
<li>
<p><code>Thread</code></p>
</li>
</ul>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>Public Class Sample

    Private myString As String = "foo"

    Public Sub Go()
        SyncLock myString   ' Noncompliant
        End SyncLock
    End Sub

End Class</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>Public Class Sample

    Private Shared ReadOnly fLock As New Object

    Public Sub Go()
        SyncLock fLock
        End SyncLock
    End Sub

End Class</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace this with a lock against an object with strong-identity.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>the lock call</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_8_may_2017_130557_valeri_hristov_wrote">on 8 May 2017, 13:05:57 Valeri Hristov wrote:</h3>
<div class="paragraph">
<p>Changed the text a bit because we automatically generate description for sonarlint.org from the first paragraph of the RSPEC and the old one was too short. Please, review.</p>
</div>
</div>
</div>
</div>