<div class="paragraph">
<p>The <code>SELECT</code> statement should be used only to clearly define some new branches in the control flow. As soon as a <code>WHEN</code> clause contains too many statements this highly decreases the readability of the overall control flow statement. In such case, the content of <code>WHEN</code> clause should be extracted in a dedicated subroutine.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>C                   SELECT
C                   WHEN      X=1
C                   MOVE      A             B
C                   MOVE      C             D
C                   MOVE      E             F
C                   MOVE      J             K
C                   MOVE      L             M
C                   WHEN      X=2
C                   ....
C                   ENDSL
 /free
   select;
     when X=1;
       B = A;
       D = C;
       F = E;
       K = J;
       M = L;
     when X=2;
       ...
   endsl;
 /end-free</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>C                   SELECT
C                   WHEN      X=1
C                   EXSR      SR011
C                   WHEN      X=2
C                   ....
C                   ENDSL
C                   ...
C     SR011         BEGSR
C                   MOVE      A             B
C                   MOVE      C             D
C                   MOVE      E             F
C                   MOVE      J             K
C                   MOVE      L             M
C                   ENDSR
 /free
   select;
     when X=1;
       exsr SR011;
     when X=2;
       ...
   endsl;
 /end-free</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Reduce this "[switch/CASE|CASE/WHEN]" number of lines from {} to at most {}, for example by extracting code into a [method|subroutine|function].</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="ulist">
<ul>
<li>
<p>key = max</p>
</li>
<li>
<p>Description: Maximum number of lines of code</p>
</li>
<li>
<p>Default value : 5</p>
</li>
<li>
<p>Default for PHP: 10</p>
</li>
<li>
<p>Default for T-SQL: 10</p>
</li>
<li>
<p>Type: Integer</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_duplicated_by_s137">is duplicated by: S137</h3>

</div>
<div class="sect2">
<h3 id="_on_8_aug_2013_093718_dinesh_bolkensteyn_wrote">on 8 Aug 2013, 09:37:18 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>Implemented by https://jira.codehaus.org/browse/SONARJAVA-238</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_31_oct_2013_164100_freddy_mallet_wrote">on 31 Oct 2013, 16:41:00 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3235 for Flex</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_12_feb_2014_175822_freddy_mallet_wrote">on 12 Feb 2014, 17:58:22 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3449 for PHP</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_2_jul_2015_081108_evgeny_mandrikov_wrote">on 2 Jul 2015, 08:11:08 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I find it weird to see colon before the header in description.</p>
</div>
</div>
</div>
</div>