<div class="paragraph">
<p>Even if your document is arranged in many layers, you don&#8217;t have to step through those layers explicitly to select an element. In fact, it&#8217;s more efficiently to skim over some of those layers rather than making the JQuery selector engine explicitly step through each one.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>$( "div.data table.attendees td.campbell" );  // Noncompliant; too many selectors. There's no need to trace through each layer of the structure</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>$( ".data td.campbell");</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_6_may_2015_095744_elena_vilchik_wrote">on 6 May 2015, 09:57:44 Elena Vilchik wrote:</h3>
<div class="paragraph">
<p>Performance tests show that long selector might be significantly faster than the shorter one (as in code examples).</p>
</div>
<div class="paragraph">
<p><a href="http://jsperf.com/complex-selectors-sq" class="bare">http://jsperf.com/complex-selectors-sq</a></p>
</div>
<div class="paragraph">
<p>Screenshot of pert test results is attached.</p>
</div>
<div class="paragraph">
<p>So recommendations of this rule could affect code performance in a bad way.</p>
</div>
</div>
</div>
</div>