<div class="paragraph">
<p>AssertJ assertions methods targeting the same object can be chained instead of using multiple <code>assertThat</code>. It avoids duplication and increases the clarity of the code.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when multiples <code>assertThat</code> target the same tested value.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>assertThat(someList).hasSize(3);
assertThat(someList).contains("something");</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>assertThat(someList)
  .hasSize(3)
  .contains("something");</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Join these multiple assertions subject to one assertion chain</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>Primary location: first assertThat</p>
</li>
<li>
<p>Secondaries: all subsequent assertThat</p>
</li>
</ul>
</div>
</div>
</div>
</div>