<div class="paragraph">
<p>Initializing an alphanumeric field with the literal character <code>' '</code>, <code>'1'</code>, or <code>'0'</code> has the same effect as using the equivalent figurative constant <code>*BLANK</code>, <code>*ON</code>, or <code>*OFF</code>, but the constant version removes ambiguity and makes the code more readable.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>C                   MOVE      ' '           FLD1</pre>
</div>
</div>
<div class="listingblock">
<div class="content">
<pre>  /free
    fld1 = ' ';
  /end-free</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>C                   MOVE      *BLANK        FLD1</pre>
</div>
</div>
<div class="listingblock">
<div class="content">
<pre>  /free
    fld1 = *BLANK;
  /end-free</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace this string literal 'x' literal with YYY</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_4_mar_2014_163623_ann_campbell_wrote">on 4 Mar 2014, 16:36:23 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>updated spec per email from customer, which does not use empty string:</p>
</div>
<div class="literalblock">
<div class="content">
<pre>the ' ' literal is not to be used to represent *BLANK</pre>
</div>
</div>
</div>
</div>
</div>