<div class="paragraph">
<p>It&#8217;s possible to name the members of an inner class the same as the <code>static</code> members of its enclosing class - possible, but a bad idea. That&#8217;s because maintainers may be confused about which members are being used where. Instead the inner class' members should be renamed and all the references updated.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class Outer
{
  public static int A;

  public class Inner
  {
    public int A; //Noncompliant
    public int MyProp
    {
      get { return A; }  // Returns inner A. Was that intended?
    }
  }
}</pre>
</div>
</div>
<div class="paragraph">
<p>After a rename</p>
</div>
<div class="listingblock">
<div class="content">
<pre>class Outer
{
  public static int A;

  public class Inner
  {
    public int B;
    public int MyProp
    {
      get { return A; }  // Still compiles and runs but functionality has changed
    }
  }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class Outer
{
  public static int A;

  public class Inner
  {
    public int InnerA;
    public int MyProp
    {
      get { return InnerA; }
    }
  }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://wiki.sei.cmu.edu/confluence/x/5DVGBQ">CERT, DCL51-J.</a> - Do not shadow or obscure identifiers in subscopes</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Rename this [field|method|delegate|type|event|property] to not shadow the outer class' member with the same name.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_1_jul_2015_163356_ann_campbell_wrote">on 1 Jul 2015, 16:33:56 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~tamas.vajk] please make sure I annotated the code samples correctly</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_2_jul_2015_064938_tamas_vajk_wrote">on 2 Jul 2015, 06:49:38 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I&#8217;ve fixed the code sample. LGTM now.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_2_jul_2015_114340_ann_campbell_wrote">on 2 Jul 2015, 11:43:40 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Thanks [~tamas.vajk]</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_7_aug_2015_092915_tamas_vajk_wrote">on 7 Aug 2015, 09:29:15 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] could you check the error message. Thanks.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_7_aug_2015_121212_ann_campbell_wrote">on 7 Aug 2015, 12:12:12 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Looks fine [~tamas.vajk]</p>
</div>
</div>
</div>
</div>