<div class="paragraph">
<p>Apart from a few operators (notably <code>&amp;&amp;</code>, <code>||</code>, <code>?:</code> and <code>,</code>) the order in which sub-expressions are evaluated is unspecified and can vary. This means that no reliance can be placed on the order of evaluation of sub-expressions and, in particular, no reliance can be placed on the order in which side effects occur. Those points in the evaluation of an expression at which all previous side effects can be guaranteed to have taken place are called “sequence points”. Sequence points and side effects are described in Section 1.9(7) of ISO/IEC 14882:2003 [1].</p>
</div>
<div class="paragraph">
<p>Note that the “order of evaluation” problem is not solved by the use of parentheses, as this is not a precedence issue.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre></pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C:2004, 12.2 - The value of an expression shall be the same under any order of evaluation that the standard permits.</p>
</li>
<li>
<p>MISRA C&#43;&#43;:2008, 5-0-1 - The value of an expression shall be the same under any order of evaluation that the standard permits.</p>
</li>
<li>
<p><a href="https://www.securecoding.cert.org/confluence/x/WQD3">CERT, EXP10-C.</a> - Do not depend on the order of evaluation of subexpressions or the order in which side effects take place</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_duplicates_s864">duplicates: S864</h3>

</div>
<div class="sect2">
<h3 id="_duplicates_s1987">duplicates: S1987</h3>

</div>
</div>
</div>