<div class="paragraph">
<p>Use of the bang (<code>!</code>) operator leads to late binding and results in inefficient code. Use the slightly more verbose dot (<code>.</code>) notation instead.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>MsgBox ObjectWithDefaultProperty!Param</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>MsgBox ObjectWithDefaultProperty.DefaultProperty("Param")</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace this use of bang syntax with dot syntax</p>
</div>
</div>
</div>
</div>