<div class="paragraph">
<p>Rather than creating a boxed primitive from a <code>String</code> to extract the primitive value, use the relevant <code>parse</code> method instead. It will be clearer and more efficient.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>String myNum = "12.2";

float f = (new Float(myNum)).floatValue();  // Noncompliant; creates &amp; discards a Float</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>String myNum = "12.2";

float f = Float.parseFloat(myNum);</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use "XXX.parseYyy" for this string-to-yyy conversion.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_10_oct_2014_135853_freddy_mallet_wrote">on 10 Oct 2014, 13:58:53 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>@Ann, for me this RSPEC fully duplicates RSPEC-2131, what&#8217;s the difference ? Thanks</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_10_oct_2014_175539_ann_campbell_wrote">on 10 Oct 2014, 17:55:39 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~freddy.mallet] they go in opposite directions. This one is string-to-primitive. The other is primitive-to-string. S&#8217;okay?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_11_oct_2014_121320_freddy_mallet_wrote">on 11 Oct 2014, 12:13:20 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Ok [~ann.campbell.2] !</p>
</div>
</div>
</div>
</div>