<div class="paragraph">
<p><code>yield</code> and <code>return</code> only make sense in the context of functions. Using them outside a function raises a <code>SyntaxError</code>. To break out of a loop, use <code>break</code> instead.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class MyClass:
    while True:
        return False #Noncompliant</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this use of "xxx".</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_17_mar_2015_075940_elena_vilchik_wrote">on 17 Mar 2015, 07:59:40 Elena Vilchik wrote:</h3>
<div class="paragraph">
<p>Pylint rules</p>
</div>
<div class="paragraph">
<p>E0104: Return outside function</p>
</div>
<div class="paragraph">
<p>E0105: Yield outside function</p>
</div>
</div>
</div>
</div>