<div class="paragraph">
<p>Sometimes there is the need to cancel the execution of a coroutine after a given period of time. You can do this manually by combining the <code>delay()</code> and <code>cancel()</code> functions. However, this technique is verbose and error-prone. An easier way to manage timeouts is using the function <code>withTimeout()</code> or <code>withTimeoutOrNull()</code>.</p>
</div>
<div class="paragraph">
<p>The <code>withTimeout</code> function will throw a <code>TimeoutCancellationException</code> when the timeout is reached, while <code>withTimeoutOrNull</code> will simply return <code>null</code> instead.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue if timeout mechanisms are implemented manually instead of using appropriate built-in functions.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>suspend fun main() {
    coroutineScope {
        val job = launch {
            delay(2000L)
            println("Finished")
        }
        delay(500L)
        job.cancel()
    }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>suspend fun main() {
    coroutineScope {
        withTimeoutOrNull(1000L){
            delay(2000L)
            println("Finished")
        }
    }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://kotlinlang.org/docs/cancellation-and-timeouts.html">Cancellation and timeouts</a></p>
</li>
</ul>
</div>
</div>
</div>