<div class="paragraph">
<p>Every class definition should be commented to explain its goal and how it works. This comment must be located before [or after] the class definition.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>CLASS my_class DEFINITION.
  ...
ENDCLASS</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>* here is
* my comment
CLASS my_class DEFINITION.
  ...
ENDCLASS</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Document this class "XXX".</p>
</div>
</div>
</div>
</div>