<div class="paragraph">
<p>Using an empty driver table in a <code>SELECT</code>/<code>FOR ALL ENTRIES</code> table has a very important side effect: the complete <code>WHERE</code> clause is not taken into account because a <code>NO WHERE</code> condition is generated. Thus a full table scan is unexpectedly executed.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>SELECT carrid , connid , seatsocc FROM flights
INTO TABLE seatsocc_tab
FOR ALL ENTRIES IN conn_tab  " Noncompliant; conn_tab may be empty.
WHERE carrid = conn_tab-carrid
AND connid = conn_tab-connid.</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>IF conn_tab is not initial.
  " ...
  SELECT carrid , connid , seatsocc FROM flights
  INTO TABLE seatsocc_tab
  FOR ALL ENTRIES IN conn_tab
  WHERE carrid = conn_tab-carrid
  AND connid = conn_tab-connid.
  " ...
ENDIF.</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Driver table "xxxxx" should be checked for emptiness before being used.</p>
</div>
</div>
</div>
</div>