<div class="paragraph">
<p>Labeled blocks are useful, especially when the code is badly indented, to match the begin and end of each block. This check detects labeled blocks which are missing an ending label.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>&lt;&lt;myBlockLabel1&gt;&gt;
BEGIN
  NULL;
END; -- Noncompliant; this labeled loop has no ending label
/

BEGIN
  NULL; -- Compliant; not a labeled block
END;
/</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>&lt;&lt;myBlockLabel2&gt;&gt;
BEGIN
  NULL;
END myBlockLabel2;
/

BEGIN
  NULL;
END;
/</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add the missing "xxx" label to this statement.</p>
</div>
</div>
</div>
</div>