<div class="paragraph">
<p>"using-directives" add additional scopes to the set of scopes searched during name lookup. All identifiers in these scopes become visible. If two different namespaces contain some identifiers with same names then it can quickly become error-prone as it increases the possibility that the identifier found by the compiler does not meet developer expectations. This rule raises an issue as soon a "using" directive leads to introduce an ambiguity.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>namespace NS1 {
  int foo;
  int bar;
}

using namespace NS1; // OK

namespace NS2 {
  int foo;
  int somethingElse;
}

using namespace NS2; // Noncompliant as there might be an ambiguity between NS1::foo and NS2::foo</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C&#43;&#43;:2008, 7-3-4 - using-directives shall not be used.</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s1003">relates to: S1003</h3>

</div>
<div class="sect2">
<h3 id="_on_31_mar_2015_190703_evgeny_mandrikov_wrote">on 31 Mar 2015, 19:07:03 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] implementation seems more complete (SQALE, description) than this spec.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_13_apr_2015_191638_evgeny_mandrikov_wrote">on 13 Apr 2015, 19:16:38 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Changed spelling of "using-directives" and "using-declarations" to match C&#43;&#43; language specification. Also wondering why blocker, but not active by default? Note that in implementation currently major and active.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_20_apr_2015_140033_ann_campbell_wrote">on 20 Apr 2015, 14:00:33 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>in fact [~evgeny.mandrikov] I don&#8217;t think I had previously edited either of those two attributes</p>
</div>
</div>
</div>
</div>