<div class="paragraph">
<p>A set cannot have two identical values. When a value is repeated in a set literal, only the last occurence will remain. Thus duplicate values should be either modified or removed.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when the same value is used multiple times as a value in a set literal.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>{"one", "two", "one"}  # Noncompliant

def func(a1, a2, a3):
    {a1, a2, a1}  # Noncompliant.</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>{"one", "two", "three"}

def func(a1, a2, a3):
    {a1, a2, a3}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.python.org/3/reference/expressions.html#set-displays">Python documentation - Set displays</a></p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Change or remove duplicates of value "XXX"</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Primary: the first occurence of a duplicated value.</p>
</div>
<div class="paragraph">
<p>Secondary:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>location: other occurences of the value.</p>
</li>
<li>
<p>message: "Duplicate value"</p>
</li>
</ul>
</div>
</div>
</div>
</div>