<div class="paragraph">
<p>Most browsers parse and discard a meaningless, trailing comma. Unfortunately, that&#8217;s not the case for Internet Explorer below version 9, which throws a meaningless error. Therefore trailing commas should be eliminated.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>var settings = {
    'foo'  : oof,
    'bar' : rab,    // Noncompliant - trailing comma
};</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>var settings = {
    'foo'  : oof,
    'bar' : rab
};</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this trailing comma.</p>
</div>
</div>
</div>
</div>