<div class="paragraph">
<p>C&#43;&#43; language is highly elastic regarding the placement of type specifiers on type declarations. This makes it possible to place a <code>const</code> qualifier before the type ("west const") or after it ("east const").</p>
</div>
<div class="paragraph">
<p>Mixing both <code>const</code> placement in a single project or, even worse, in a single file, makes the code harder to understand. It will require more effort to notice that the object cannot be modified.</p>
</div>
<div class="paragraph">
<p>This rule checks if the <code>const</code> qualifier placement matches the configuration.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="paragraph">
<p>With default value of "eastConst" parameter.</p>
</div>
<div class="listingblock">
<div class="content">
<pre>long double const tau = 6.28L; // Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>const long double tau = 6.28L;</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rl-const">C&#43;&#43; Core Guidelines NL.26</a> - Use conventional const notation</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="ulist">
<ul>
<li>
<p>key = eastConst</p>
</li>
<li>
<p>description = Determines if the "const" qualifier should be placed after type</p>
</li>
<li>
<p>default = false</p>
</li>
<li>
<p>type = BOOLEAN</p>
</li>
</ul>
</div>
</div>
</div>
</div>