<div class="paragraph">
<p>JUnit assertions should not be made from the <code>run</code> method of a <code>Runnable</code>, because failed assertions result in <code>AssertionError</code>s being thrown. If the error is thrown from a thread other than the one that ran the test, the thread will exit but the test won&#8217;t fail.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public void run() {
  // ...
  Assert.assertEquals(expected, actual);  // Noncompliant
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this assertion.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_22_oct_2014_192032_ann_campbell_wrote">on 22 Oct 2014, 19:20:32 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Note that Applicability is marked only for Tests</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_24_nov_2014_193309_nicolas_peru_wrote">on 24 Nov 2014, 19:33:09 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>I am guessing here and so the answer should probably part of the RSPEC : The run method is a run method of a Thread class ?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_22_apr_2015_110115_ann_campbell_wrote">on 22 Apr 2015, 11:01:15 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Since you&#8217;re already implementing this [~nicolas.peru], I&#8217;m guessing it&#8217;s okay.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_22_apr_2015_112020_nicolas_peru_wrote">on 22 Apr 2015, 11:20:20 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2]yes, more especially with the the mention of <code>Runnable</code></p>
</div>
</div>
</div>
</div>