<div class="paragraph">
<p>A <code>CROSS JOIN</code> query will return all records where each row from the first table is combined with each row from the second table. This means that such a query returns the Cartesian product of the sets of rows from the joined tables, which is why it is also know as "Cartesian product query".</p>
</div>
<div class="paragraph">
<p>Such a query can return a huge amount of data, and therefore should be used only with great caution and only when really needed.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>BEGIN
  -- Standard ANSI syntax
  SELECT *
    INTO employeeArray
    FROM employee CROSS JOIN department; -- Noncompliant; explicit cross join
END;
/

BEGIN
  -- Old syntax
  SELECT *
    INTO employeeArray
    FROM employee, department; -- Noncompliant; also a cross join
END;
/</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this "CROSS JOIN" query</p>
</div>
</div>
</div>
</div>