<div class="paragraph">
<p>In the interests of clean code, getters should be implicit, rather than explicit for properties that have only getters.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>var luckyNumber: Int {
  get {  // Noncompliant
    return 7
  }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>var luckyNumber: Int {
  return 7
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_duplicates_s2962">duplicates: S2962</h3>

</div>
<div class="sect2">
<h3 id="_on_8_jul_2016_192950_ann_campbell_wrote">on 8 Jul 2016, 19:29:50 Ann Campbell wrote:</h3>
<div class="paragraph">
<p><a href="https://github.com/github/swift-style-guide#prefer-implicit-getters-on-read-only-properties-and-subscripts" class="bare">https://github.com/github/swift-style-guide#prefer-implicit-getters-on-read-only-properties-and-subscripts</a></p>
</div>
</div>
</div>
</div>