<div class="paragraph">
<p>Since <code>abstract</code> classes can&#8217;t be instantiated, there&#8217;s no point in their having <code>public</code> or <code>internal</code> constructors. If there is basic initialization logic that should run when an extending class instance is created, you can by all means put it in a constructor, but make that constructor <code>private</code> or <code>protected</code>.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>abstract class Base
{
    public Base() // Noncompliant, should be private or protected
    {
      //...
    }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>abstract class Base
{
    protected Base()
    {
      //...
    }
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Reduce the visibility of this constructor.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p><code>public</code> keyword</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_8_dec_2015_084844_tamas_vajk_wrote">on 8 Dec 2015, 08:48:44 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] LGTM, changed the label from api-design to clumsy. The constructor can&#8217;t be called directly, because it&#8217;s in an <code>abstract</code> class. So we can only call it from a deriving type&#8217;s constructor, but this is also the case if the constructor is <code>protected</code>, so we are not proposing any real change in api-design. WDYT?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_12_apr_2016_175627_ann_campbell_wrote">on 12 Apr 2016, 17:56:27 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Rule originally taken from R#, but also exists in MSFT Roslyn rule set &amp; therefore doesn&#8217;t need re-implementation.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_26_jan_2017_164921_ann_campbell_wrote">on 26 Jan 2017, 16:49:21 Ann Campbell wrote:</h3>
<div class="paragraph">
<p><a href="https://msdn.microsoft.com/en-us/library/ms182126.aspx" class="bare">https://msdn.microsoft.com/en-us/library/ms182126.aspx</a></p>
</div>
</div>
</div>
</div>