<div class="paragraph">
<p>The execution order of <code>static</code> initializers is unspecified when they are in different compilation units (files). Relying on a particular initialization order can have nasty repercussions and should therefore be avoided. Even if the code works now, it could change in the future without notice.</p>
</div>
<div class="paragraph">
<p>If you need to use <code>static</code> globals, you should put them inside the function that uses them, or create a getter and declare them inside that getter.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>static const std::string airports[] = {"GVA", "SFO", "CDG"}; // Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>std::string GetAirportCode(int i) {
  static const std::string airports[] = {"GVA", "SFO", "CDG"};
  return airports[i];
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Move this "static" initializer into a getter or the function that uses it.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p><code>static</code></p>
</div>
</div>
</div>
</div>