<div class="paragraph">
<p>Every instance method is expected to have at least one positional parameter. This parameter will reference the object instance on which the method is called. Calling an instance method which doesn&#8217;t have at least one parameter will raise a "TypeError". By convention, this first parameter is usually named "self".</p>
</div>
<div class="paragraph">
<p>Class methods, i.e. methods annotated with <code>@classmethod</code>, also require at least one parameter. The only differences is that it will receive the class itself instead of a class instance. By convention, this first parameter is usually named "cls". Note that <code>__new__</code> and <code>__init_subclass__</code> take a class as first argument even thought they are not decorated with <code>@classmethod</code>.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when an instance of class method does not have at least one positional parameter.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class MyClass:
    def instance_method():  # Noncompliant. "self" parameter is missing.
        print("instance_method")

    @classmethod
    def class_method():  # Noncompliant. "cls" parameter is missing.
        print("class_method")</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class MyClass:
    def instance_method(self):
        print("instance_method")

    @classmethod
    def class_method(cls):
        print("class_method")


    @staticmethod
    def static_method():
        print("static_method")</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>Python documentation - <a href="https://docs.python.org/3.8/tutorial/classes.html#method-objects">Method Objects</a></p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Method has no @classmethod or @staticmethod annotation</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Add a "self" or class parameter</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Method has a @classmethod annotation, or method is <em>new</em> or <em>init_subclass</em></p>
</div>
<div class="ulist">
<ul>
<li>
<p>Add a class parameter</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The method signature <code>def name()</code></p>
</div>
</div>
</div>
</div>