<div class="paragraph">
<p>It is a best practice to have variable declarations outside of the loop. Additionally, declaring variables inside a loop is slightly less efficient because memory management is then performed with each iteration of the loop.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>BEGIN
  FOR i IN 1..10
  LOOP
    DECLARE
      variableX NUMBER:= 10;
    BEGIN
        variableX:= variableX+i;
        dbms_output.put_line(variableX);
    END;
 END LOOP;
END;</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>DECLARE
  variableX NUMBER:= 0;
BEGIN
  FOR i IN 1..10
  LOOP
    variableX:= variableX+1;
    dbms_output.put_line(variableX);
  END LOOP;
END;</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Move this variable declaration outside of the loop.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Variable declaration</p>
</div>
</div>
</div>
</div>