<div class="paragraph">
<p>The new style of class creation, with the declaration of a parent class, created a unified object model in Python, so that the type of an instantiated class is equal to its class. In Python 2.2-2.7, this is not the case for old-style classes. In Python 3+ all classes are new-style classes. However, since the behavior can differ from 2.2+ to 3+, explicitly inheriting from <code>object</code> (if there is no better candidate) is recommended.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class MyClass():
    pass</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class MyClass(object):
    pass</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add inheritance from "object" or some other new-style class.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_15_may_2014_135535_pierre_yves_nicolas_wrote">on 15 May 2014, 13:55:35 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>In Python 2.x, a class is an old-style class if it does not subclass a new-style class. If a class subclasses another class, we should check recursively whether the superclass is a new-style class. If a class does not subclass any class, we know for sure that it is an old-style class.</p>
</div>
<div class="paragraph">
<p>See https://docs.python.org/2/whatsnew/2.2.html#old-and-new-classes</p>
</div>
</div>
</div>
</div>