<div class="paragraph">
<p><code>&lt;stdlib.h&gt;</code>'s <code>atof</code>, <code>atoi</code>, and <code>atol</code> functions, which convert strings to numbers, have undefined behavior when the strings cannot be converted, and should therefore be avoided.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>int converter (const char * numstr) {
  return atoi(numstr); // Noncompliant
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>int converter (const char * numstr) {
  return strtol(numstr, NULL, 10);
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C:2004, 20.10 - The library functions atof, atoi and atol from library &lt;stdlib.h&gt; shall not be used.</p>
</li>
<li>
<p>MISRA C&#43;&#43;:2008, 18-0-2 - The library functions atof, atoi and atol from library &lt;cstdlib&gt; shall not be used.</p>
</li>
<li>
<p>MISRA C:2012, 21.7 - The atof, atoi, atol and atoll functions of &lt;stdlib.h&gt; shall not be used</p>
</li>
<li>
<p><a href="https://wiki.sei.cmu.edu/confluence/x/C9cxBQ">CERT, ERR34-C.</a> - Detect errors when converting a string to a number</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this use of "xxx".</p>
</div>
</div>
</div>
</div>