<div class="paragraph">
<p><code>readonly</code> properties can only be assigned in a class constructor or at the point of declaration. If a class has a property that&#8217;s not marked <code>readonly</code> but is only set in the constructor, it could cause confusion about the property&#8217;s intended use. To avoid confusion, such properties should be marked <code>readonly</code> to make their intended use explicit, and to prevent future maintainers from inadvertently changing their use.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class Person {
  private _birthYear: number;  // Noncompliant
  constructor(birthYear: number) {
    this._birthYear = birthYear;
  }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class Person {
  private readonly _birthYear: number;
  constructor(birthYear: number) {
    this._birthYear = birthYear;
  }
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Make "x" "readonly".</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_13_may_2015_200503_ann_campbell_wrote">on 13 May 2015, 20:05:03 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~tamas.vajk] please provide R# id if relevant.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_18_may_2015_083035_tamas_vajk_wrote">on 18 May 2015, 08:30:35 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Couldn&#8217;t find a R# id, just a link: https://confluence.jetbrains.com/display/ReSharper/Field+can+be+made+read-only</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_19_may_2015_201216_ann_campbell_wrote">on 19 May 2015, 20:12:16 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Thanks [~tamas.vajk], I figured it out. Please review the rule.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_20_may_2015_070643_tamas_vajk_wrote">on 20 May 2015, 07:06:43 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>LGTM</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_16_nov_2020_142048_andrei_epure_wrote">on 16 Nov 2020, 14:20:48 Andrei Epure wrote:</h3>
<div class="paragraph">
<p>\[~nicolas.harraudeau] should we extend this to static fields and static constructors as well?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_16_nov_2020_162002_nicolas_harraudeau_wrote">on 16 Nov 2020, 16:20:02 Nicolas Harraudeau wrote:</h3>
<div class="paragraph">
<p>After discussing with [~andrei.epure] we agreed that there isn&#8217;t much value in raising issues on static fields right now. This rule already raises thousands of issues on multiple projects and we should think about how to improve the rule before adding new noncompliant cases.</p>
</div>
</div>
</div>
</div>