<div class="paragraph">
<p>If an alternative in a regular expression only matches things that are already matched by another alternative, that alternative is redundant and serves no purpose.</p>
</div>
<div class="paragraph">
<p>In the best case this means that the offending subpattern is merely redundant and should be removed. In the worst case it&#8217;s a sign that this regex does not match what it was intended to match and should be reworked.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>r"[ab]|a"   # The "|a" is redundant because "[ab]" already matches "a"
r".*|a"     # .* matches everything, so any other alternative is redundant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>r"[ab]"
r".*"</pre>
</div>
</div>
</div>
</div>