<div class="paragraph">
<p>This rule allows banning some modules.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="paragraph">
<p>With <code>moduleName</code> configured with <code>[a-zA-Z-]*UT123[a-zA-Z]*</code>:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>CALL UT123.
CALL UT123L.
CALL UT123-L.
CALL WS-UT123.</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove the use of this forbidden module. Use {replacementModuleName} instead.</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="ulist">
<ul>
<li>
<p>key = moduleName</p>
<div class="ulist">
<ul>
<li>
<p>description: Name of the forbidden module. Use a regex to forbid more widely multiple modules sharing the same root name.</p>
</li>
</ul>
</div>
</li>
<li>
<p>key: replacementModuleName</p>
<div class="ulist">
<ul>
<li>
<p>description: Name of the module to use as a replacement of the forbidden one.</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>First use of forbidden class</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_deprecates_s1212">deprecates: S1212</h3>

</div>
<div class="sect2">
<h3 id="_is_duplicated_by_s5338">is duplicated by: S5338</h3>

</div>
</div>
</div>