<div class="paragraph">
<p>It is possible to subscribe to events with anonymous delegates, but having done so, it is impossible to unsubscribe from them. That&#8217;s because the process of subscribing adds the delegate to a list. The process of unsubscribing essentially says: remove <em>this item</em> from the subscription list. But because an anonymous delegate was used in both cases, the unsubscribe attempt tries to remove a different item from the list than was added. The result: <code>NOOP</code>.</p>
</div>
<div class="paragraph">
<p>Instead, save the delegate to a variable and use the variable to subscribe and unsubscribe.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>listView.PreviewTextInput += (obj,args) =&gt;
        listView_PreviewTextInput(obj,args,listView);

// ...

listView.PreviewTextInput -= (obj, args) =&gt;
        listView_PreviewTextInput(obj, args, listView); // Noncompliant; this delegate was never subscribed</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>EventHandler func = (obj,args) =&gt; listView_PreviewTextInput(obj,args,listView);

listView.PreviewTextInput += func;

// ...

listView.PreviewTextInput -= func;</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Unsubscribe with the same delegate that was used for the subscription.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_7_jul_2015_150846_ann_campbell_wrote">on 7 Jul 2015, 15:08:46 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~tamas.vajk], I flipped this around from the original proposition; the real problem is that you subscribed with an anonymous delegate. Hopefully that&#8217;s not too much harder in the implementation&#8230;&#8203;</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_8_jul_2015_091124_tamas_vajk_wrote">on 8 Jul 2015, 09:11:24 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] We won&#8217;t be able to implement the rule this way. Finding  subscription and unsubscription pairs won&#8217;t be easy.</p>
</div>
<div class="paragraph">
<p>On the other hand, whenever you try to unsubscribe with an anonymous delegate that is an issue. So recognizing this is easy. I would go this way. (I&#8217;ve changed the location of the issue, but both the message and the title would need modification)</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_8_jul_2015_135601_ann_campbell_wrote">on 8 Jul 2015, 13:56:01 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>updated. See what you think [~tamas.vajk]</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_20_jul_2015_114456_tamas_vajk_wrote">on 20 Jul 2015, 11:44:56 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Looks good.</p>
</div>
</div>
</div>
</div>