<div class="paragraph">
<p>Checking if a variable or parameter is <code>None</code> should only be done when you expect that it can be <code>None</code>. Doing so when the variable is always None or never None is confusing at best. At worse, there is a bug and the variable is not updated properly.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when expressions <code>X is None</code>, <code>X is not None</code>, <code>X == None</code> or <code>X != None</code> are constant, i.e. <code>X</code> is always None or never None.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>mynone = None
result = mynone is None: # Noncompliant. Always True.

if mynone == None: # Noncompliant. Always True.
    pass

if mynone is not None:  # Noncompliant. Always False.
    pass

if mynone == None:  # Noncompliant. Always False.
    pass

myint = 42
result = myint is None: # Noncompliant. Always False.

if myint == None: # Noncompliant. Always False.
    pass

if myint is not None:  # Noncompliant. Always True.
    pass

if myint == None:  # Noncompliant. Always True.
    pass</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>Python documentation - <a href="https://docs.python.org/3/reference/expressions.html#is-not">Identity comparisons</a></p>
</li>
<li>
<p>Python documentation - <a href="https://docs.python.org/3/reference/datamodel.html#object.__eq__"><code>__eq__</code> operator</a></p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>Remove this identity check; it will always be True/False</p>
</li>
<li>
<p>Remove this == comparison; it will always be True/False</p>
</li>
<li>
<p>Remove this != comparison; it will always be True/False</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>the "is", "is not", "==" or "!=" operator</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s5914">is related to: S5914</h3>

</div>
</div>
</div>