<div class="paragraph">
<p>Early classes of the Java API, such as <code>Vector</code>, <code>Hashtable</code> and <code>StringBuffer</code>, were synchronized to make them thread-safe. Unfortunately, synchronization has a big negative impact on performance, even when using these collections from a single thread.</p>
</div>
<div class="paragraph">
<p>It is better to use their new unsynchronized replacements:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>ArrayList</code> or <code>LinkedList</code> instead of <code>Vector</code></p>
</li>
<li>
<p><code>Deque</code> instead of <code>Stack</code></p>
</li>
<li>
<p><code>HashMap</code> instead of <code>Hashtable</code></p>
</li>
<li>
<p><code>StringBuilder</code> instead of <code>StringBuffer</code></p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Even when used in synchronized context, you should think twice before using it, since it&#8217;s usage can be tricky. If you are confident the usage is legitimate, you can safely ignore this warning.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>Vector cats = new Vector();</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>ArrayList cats = new ArrayList();</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_exceptions">Exceptions</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Use of those synchronized classes is ignored in the signatures of overriding methods.</p>
</div>
<div class="listingblock">
<div class="content">
<pre>@Override
public Vector getCats() {...}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace the synchronized class "{}" by an unsynchronized one such as "{}".</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s1076">relates to: S1076</h3>

</div>
<div class="sect2">
<h3 id="_on_31_jul_2013_151351_dinesh_bolkensteyn_wrote">on 31 Jul 2013, 15:13:51 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>Can also replace UseArrayListInsteadOfVector</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_31_jul_2013_151808_dinesh_bolkensteyn_wrote">on 31 Jul 2013, 15:18:08 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>Implemented by http://jira.codehaus.org/browse/SONARJAVA-236</p>
</div>
</div>
</div>
</div>