<div class="paragraph">
<p>There is no reason to concatenate literal strings. Doing so is an exercise in reducing code readability. Instead, the strings should be combined.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>$msg = "Hello " . "${name}" . "!";  // Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>$msg = "Hello ${name}!";</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Combine these strings instead of concatenating them.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_19_sep_2014_155249_freddy_mallet_wrote">on 19 Sep 2014, 15:52:49 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>@Ann, I would not activate this rule by default as this would lead to generate a lot of issues/noise on some projects.</p>
</div>
</div>
</div>
</div>