<div class="paragraph">
<p>It is expected that some methods should be called with caution, but others, such as <code>ToString</code>, are expected to "just work". Throwing an exception from such a method is likely to break callers' code unexpectedly.</p>
</div>
<div class="paragraph">
<p>An issue is raised when an exception is thrown from any of the following:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Event accessors</p>
</li>
<li>
<p><code>Object.Equals</code></p>
</li>
<li>
<p><code>IEquatable.Equals</code></p>
</li>
<li>
<p><code>GetHashCode</code></p>
</li>
<li>
<p><code>ToString</code></p>
</li>
<li>
<p><code>static</code> constructors</p>
</li>
<li>
<p>Module initializers</p>
</li>
<li>
<p><code>IDisposable.Dispose</code></p>
</li>
<li>
<p>operators <code>==, !=, &lt;, &gt;, &lt;=, &gt;=</code></p>
</li>
<li>
<p><code>implicit</code> cast operators</p>
</li>
</ul>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public override string ToString()
{
  if (string.IsNullOrEmpty(Name))
  {
    throw new ArgumentException("...");  // Noncompliant
  }
  //...</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_exceptions">Exceptions</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>System.NotImplementedException</code> and its derivatives are ignored.</p>
</div>
<div class="paragraph">
<p><code>System.InvalidOperationException</code>, <code>System.NotSupportedException</code>, and <code>System.ArgumentException</code> and their derivatives are ignored in event accessors.</p>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this "throw" statement.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p><code>throw xxx</code></p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_8_mar_2017_140345_ann_campbell_wrote">on 8 Mar 2017, 14:03:45 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>FYI [~amaury.leve] our standard is AmE, so double quotes, not single quotes.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_8_mar_2017_193218_ann_campbell_wrote">on 8 Mar 2017, 19:32:18 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~amaury.leve] did you mean <code>Equals(Object)</code> rather than <code>Object.Equals</code>?</p>
</div>
</div>
</div>
</div>