<div class="paragraph">
<p><em>Mutexes</em> are synchronization primitives that allow to manage concurrency. It is a common situation to have to lock multiple <em>mutexes</em> simultaneously to get access to several resources at the same time.</p>
</div>
<div class="paragraph">
<p>If this is not done properly, it can lead to deadlocks or crashes. If one thread acquires A then tries to acquire B, while another thread acquires B then tries to acquire A, both threads will wait for each other forever.</p>
</div>
<div class="paragraph">
<p>In such a case, a commonly accepted good practice is to define an order on the <em>mutexes</em> and to lock them in that order and unlock them in the reverse order. However, such an order is not always clearly defined or easy to ensure across a whole program.</p>
</div>
<div class="paragraph">
<p>C&#43;&#43; provides facilities to lock multiple <em>mutexes</em> in one go, with a dedicated deadlock prevention algorithm. They should be used instead. Before C&#43;&#43;17, you should use <code>std::lock</code>, and since C&#43;&#43;17 you can use a variadic constructor of <code>std::scoped_lock</code>. See the examples for more details.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>void g();

std::mutex m1;
std::mutex m2;

void f() {
  // The example would be the same with std::lock_guard if C++17 std::scoped_lock is not available
  std::scoped_lock&lt;std::mutex&gt; lck1(m1); // Compliant: first mutex acquired
  std::scoped_lock&lt;std::mutex&gt; lck2(m2); // Noncompliant: acquiring several mutexes
  g();
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>void g();

std::mutex m1;
std::mutex m2;

void f() { // Compliant: C++11 solution
  std::lock(m1, m2);
  std::lock_guard&lt;std::mutex&gt; lck1(m1, std::adopt_lock);
  std::lock_guard&lt;std::mutex&gt; lck2(m2, std::adopt_lock);
  g();
}

void f() { // Compliant: C++17 solution
  std::scoped_lock&lt;std::mutex, std::mutex&gt; lck1(m1, m2);
  g();
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://github.com/isocpp/CppCoreGuidelines/blob/036324/CppCoreGuidelines.md#cp21-use-stdlock-or-stdscoped_lock-to-acquire-multiple-mutexes">C&#43;&#43; Core Guidelines CP.21</a> - Use std::lock() or std::scoped_lock to acquire multiple mutexes</p>
</li>
<li>
<p><a href="https://en.cppreference.com/w/cpp/thread/scoped_lock">cppreference.com - std::scoped_lock</a></p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use C&#43;&#43; facilities as "std::scoped_lock" to acquire multiple mutexes.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s5506">relates to: S5506</h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s5489">is related to: S5489</h3>

</div>
<div class="sect2">
<h3 id="_on_6_nov_2019_232407_loïc_joly_wrote">on 6 Nov 2019, 23:24:07 Loïc Joly wrote:</h3>
<div class="paragraph">
<p>\[~geoffray.adde]: Could you please review my changes?</p>
</div>
</div>
</div>
</div>