<div class="paragraph">
<p>With Java 8, there&#8217;s no need to write <code>Comparator</code>s that compare primitive values or other <code>Comparable</code>s; they can be generated for you using the <code>Comparator.comparing*</code> functions: <code>comparing</code>, <code>comparingDouble</code>, <code>comparingInt</code>, <code>comparingLong</code>.</p>
</div>
<div class="paragraph">
<p><strong>Note</strong> that this rule is automatically disabled when the project&#8217;s <code>sonar.java.source</code> is lower than <code>8</code>.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>unparsedFiles.stream()
   .sorted((f1, f2) -&gt; f1.lines - f2.lines)  // Noncompliant
   .limit(30);</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>unparsedFiles.stream()
   .sorted(Comparator.comparingInt(UnparsedFile::getLines()))
   .limit(30);</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use "Comparator.comparingXXX" instead.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Custom comparator</p>
</div>
</div>
</div>
</div>