<div class="paragraph">
<p>There is no point in creating a branch in the code only to execute an empty statement. Such code can only be due to the presence of an unexpected trailing semi-colon.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>if(foo);      //Noncompliant, the semi-colon must be removed
  trigger(action1);  // executes unconditionally
else if (bar);    //Noncompliant, the semi-colon must be removed
  trigger(action2);  // executes unconditionally

while (condition);  // Noncompliant
  doTheThing();  // executes once, unconditionally</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_duplicates_s2681">duplicates: S2681</h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s1116">relates to: S1116</h3>

</div>
<div class="sect2">
<h3 id="_on_14_apr_2017_224356_freddy_mallet_wrote">on 14 Apr 2017, 22:43:56 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>I&#8217;m closing this ticket as this rule would be a full subset of RSPEC-2681.</p>
</div>
</div>
</div>
</div>