<div class="paragraph">
<p><code>std::is_constant_evaluated</code> is used to determine whether or not a context is constant-evaluated. This can be useful when, for example, two different implementations are provided for an algorithm: one, usually slow, for compile-time and the other one, faster, for runtime.</p>
</div>
<div class="paragraph">
<p>However, some contexts are either always constant-evaluated or never constant-evaluated. In these cases, a call to <code>std::is_constant_evaluated</code> is unnecessary as it will always return the same result.</p>
</div>
<div class="paragraph">
<p><code>std::is_constant_evaluated</code> will always return <code>true</code> when called in:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>the condition of <code>if constexpr</code></p>
</li>
<li>
<p>the condition of <code>static_assert</code></p>
</li>
<li>
<p><code>consteval</code> functions</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>And it will always return <code>false</code> in:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>non-<code>constexpr</code>/<code>consteval</code> functions</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>This rule raises an issue when <code>std::is_constant_evaluated()</code> is called in an <code>if constexpr</code> or a <code>static_assert</code> condition, where it is always <code>true</code>.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>constexpr double power(double b, int x) {
  if constexpr (std::is_constant_evaluated()) {  // Noncompliant: will always be true
    // compile-time implementation
  } else {
    // runtime implementation
  }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>constexpr double power(double b, int x) {
  if (std::is_constant_evaluated()) {
    // compile-time implementation
  } else {
    // runtime implementation
  }
}</pre>
</div>
</div>
</div>
</div>