<div class="paragraph">
<p>Numbers can be shifted with the <code>&lt;&lt;</code> and <code>&gt;&gt;</code> operators, but the right operand of the operation needs to be an <code>int</code> or a type that has an implicit conversion to <code>int</code>. However, with <code>dynamic</code>, the compiler&#8217;s type checking is turned off, so you can pass anything to a shift operator and have it compile. And if  the argument can&#8217;t be converted to <code>int</code> at runtime, then a <code>RuntimeBinderException</code> will be raised.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>Dim o As Object = 5
Dim x As Integer = 5

x = o &gt;&gt; 5 ' Noncompliant
x = x &gt;&gt; o ' Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_exceptions">Exceptions</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule does not raise when the left or the right expression is <code>Nothing</code>.</p>
</div>
<div class="listingblock">
<div class="content">
<pre>x = Nothing &gt;&gt; 5
x = 5 &gt;&gt; Nothing</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this erroneous shift; it will fail because "XXX" can&#8217;t be implicitly converted to "int".</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>RHS of shift operator</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_8_dec_2015_100652_tamas_vajk_wrote">on 8 Dec 2015, 10:06:52 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I removed the second part of the message to add an overload for <code>operator &gt;&gt;</code>, because even the operator overload would need to have <code>int</code> as the type of the right side operand.</p>
</div>
<div class="paragraph">
<p>Changed the title too.</p>
</div>
</div>
</div>
</div>