<div class="paragraph">
<p>Lambdas are a concise way of creating anonymous functions - when those functions are themselves concise. However, the use of lambdas for sizable functions can make the code difficult to read. More importantly, following variable capture in lambdas can be difficult, potentially leading to dangling pointers. Therefore lambdas should be avoided.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>int main() {
  auto func = [] () { std::cout &lt;&lt; "Hello world" &lt;&lt; std::endl; };
  func();
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>void func () {
  std::cout &lt;&lt; "Hello world" &lt;&lt; std::endl;
}

int main() {
  func();
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this lambda</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_28_nov_2018_195946_loïc_joly_wrote">on 28 Nov 2018, 19:59:46 Loïc Joly wrote:</h3>
<div class="paragraph">
<p>Removing broken link, replacing it by new one.</p>
</div>
<div class="paragraph">
<p>This rule actually goes against what is said in the link (and does not make much sense). We should replace it by a set of rule about safe usage of lambdas.</p>
</div>
<div class="paragraph">
<p>Rules currently in this set: RSPEC-5019</p>
</div>
<div class="paragraph">
<p>Other rules should follow, when the set is done, we should deprecate this rule.</p>
</div>
</div>
</div>
</div>