<div class="paragraph">
<p>The function <code>memcmp</code> can only be used for objects of trivially copyable types. This includes scalar types, arrays, and trivially copyable classes.</p>
</div>
<div class="paragraph">
<p>A class type is trivially copyable if:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>One or more of the following methods is trivial and the rest are deleted: copy constructor, move constructor, copy assignment operator, and move assignment operator,</p>
</li>
<li>
<p>It has a trivial, non-deleted destructor.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Additionally, if the type contains padding, some of its bits might be non-representative, and a strict comparison of raw memory contents might lead to the mistaken belief that two identical objects are actually different.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class Shape { // Trivially copyable, but will contain padding after the bool on most architectures
public:
  bool visible;
  int x;
  int y;
};

bool isSame(Shape *s1, Shape *s2)
{
    return memcmp(s1, s2, sizeof Shape) == 0; // Noncompliant
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class Shape {
public:
  bool visible;
  int x;
  int y;
};

bool operator==(Shape const &amp;s1, Shape const &amp;s2) {
  return s1.visible == s2.visible &amp;&amp; s1.x == s2.x &amp;&amp; s1.y == s2.y;
}

bool isSame(Shape *s1, Shape *s2)
{
    return (*s1) == (*s2);
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use "operator==" to check object equality, "XXX" is not a trivially copyable type without padding.</p>
</div>
<div class="paragraph">
<p>Compare member by member to check object equality, "XXX" contains padding.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_6_nov_2018_204332_ann_campbell_wrote">on 6 Nov 2018, 20:43:32 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Double-check me [~loic.joly]</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_7_nov_2018_085739_loïc_joly_wrote">on 7 Nov 2018, 08:57:39 Loïc Joly wrote:</h3>
<div class="paragraph">
<p>Looks good to me.</p>
</div>
</div>
</div>
</div>