<div class="paragraph">
<p>This rule checks that the disallowed tag libraries are not used.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Using the default parameter for this rule, "http://java.sun.com/jstl/sql":</p>
</div>
<div class="listingblock">
<div class="content">
<pre>&lt;%@ taglib uri="http://java.sun.com/jstl/sql" prefix="prefixOfTag" &gt; &lt;!-- Noncompliant --&gt;
&lt;jsp:directive.taglib uri="http://java.sun.com/jstl/sql" prefix="prefixOfTag" /&gt; &lt;!-- Noncompliant --&gt;</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove the use of "xxx".</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="ulist">
<ul>
<li>
<p>key = tagLibs</p>
</li>
<li>
<p>description: Comma-separated list of URIs of disallowed taglibs</p>
</li>
<li>
<p>default: http://java.sun.com/jstl/sql</p>
</li>
</ul>
</div>
</div>
</div>
</div>