<div class="paragraph">
<p>Because force casting (<code>as!</code>) does not perform any type safety validations, it is capable of performing dangerous conversions between unrelated types. When the types are truly unrelated, the cast will cause a system crash.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>foo as! MyClass  // Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>foo as? MyClass</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>CppCoreGuidelines, Type safety profile - Type.1: Don&#8217;t use reinterpret_cast.</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace this force cast with a safer cast.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p><code>as!</code></p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_8_jul_2016_203055_ann_campbell_wrote">on 8 Jul 2016, 20:30:55 Ann Campbell wrote:</h3>
<div class="paragraph">
<p><a href="http://stackoverflow.com/questions/25708649/downcasting-optionals-in-swift-as-type-or-as-type" class="bare">http://stackoverflow.com/questions/25708649/downcasting-optionals-in-swift-as-type-or-as-type</a></p>
</div>
</div>
<div class="sect2">
<h3 id="_relates_to_s869">relates to: S869</h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s871">relates to: S871</h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s1767">relates to: S1767</h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s6181">is related to: S6181</h3>

</div>
<div class="sect2">
<h3 id="_on_13_jun_2016_121013_ann_campbell_wrote">on 13 Jun 2016, 12:10:13 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>looks good [~alban.auzeill]</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_12_apr_2017_113917_jean_christophe_collet_wrote">on 12 Apr 2017, 11:39:17 Jean-Christophe Collet wrote:</h3>
<div class="paragraph">
<p>Changed to a Code Smell.</p>
</div>
</div>
</div>
</div>