<div class="paragraph">
<p>Shared naming conventions allow teams to collaborate efficiently. This rule checks that all method parameters follow a naming convention.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Using the default importing  and returning regular expressions <code>I_[a-zA-Z_\-0-9]+</code>, and <code>R_[a-zA-Z_\-0-9]+</code>:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>    METHODS m_fact IMPORTING i1  TYPE i
                       value(i2) TYPE i
                   RETURNING value(factorial) TYPE i.</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>    METHODS m_fact IMPORTING I_1  TYPE i
                       value(i2) TYPE i
                   RETURNING value(R_factorial) TYPE i.</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Rename this parameter to comply with the required naming convention: xxx</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="ulist">
<ul>
<li>
<p>key = importing_key
description: Naming convention for Importing parameter</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>default: I_[a-zA-Z_-0-9]+</p>
</div>
<div class="paragraph">
<p>key: exporting_key</p>
</div>
<div class="paragraph">
<p>description: Naming convention for Exporting parameter</p>
</div>
<div class="paragraph">
<p>default: E_[a-zA-Z_-0-9]+</p>
</div>
<div class="paragraph">
<p>key: changing_key</p>
</div>
<div class="paragraph">
<p>description: Naming convention for Changing parameter</p>
</div>
<div class="paragraph">
<p>default: C_[a-zA-Z_-0-9]+</p>
</div>
<div class="paragraph">
<p>key: returning_key</p>
</div>
<div class="paragraph">
<p>description: Naming convention for Returning parameter</p>
</div>
<div class="paragraph">
<p>default: R_[a-zA-Z_-0-9]+</p>
</div>
</div>
</div>
</div>