<div class="paragraph">
<p>Each function and procedure parameter name must match a given regular expression.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>DECLARE
  FUNCTION myfunction2(parameter_ PLS_INTEGER) RETURN PLS_INTEGER; -- Noncompliant

  PROCEDURE myprocedure2(parameter_ PLS_INTEGER); -- Noncompliant

  FUNCTION myfunction2(parameter_ PLS_INTEGER) RETURN PLS_INTEGER AS -- Noncompliant
  BEGIN
    RETURN 42;
  END;

  PROCEDURE myprocedure2(parameter_ PLS_INTEGER) AS -- Noncompliant
  BEGIN
    NULL;
  END;
BEGIN
  NULL;
END;
/</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>DECLARE
  FUNCTION myfunction1(parameter PLS_INTEGER) RETURN PLS_INTEGER; -- Compliant

  PROCEDURE myprocedure1(parameter PLS_INTEGER); -- Compliant

  FUNCTION myfunction1(parameter PLS_INTEGER) RETURN PLS_INTEGER AS -- Compliant
  BEGIN
    RETURN 42;
  END;

  PROCEDURE myprocedure1(parameter PLS_INTEGER) AS -- Compliant
  BEGIN
    NULL;
  END;

BEGIN
  NULL;
END;
/</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Rename this parameter to comply with the required naming convention: xxx</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="ulist">
<ul>
<li>
<p>key = importing_key
description: Naming convention for Importing parameter</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>default: I_[a-zA-Z_-0-9]+</p>
</div>
<div class="paragraph">
<p>key: exporting_key</p>
</div>
<div class="paragraph">
<p>description: Naming convention for Exporting parameter</p>
</div>
<div class="paragraph">
<p>default: E_[a-zA-Z_-0-9]+</p>
</div>
<div class="paragraph">
<p>key: changing_key</p>
</div>
<div class="paragraph">
<p>description: Naming convention for Changing parameter</p>
</div>
<div class="paragraph">
<p>default: C_[a-zA-Z_-0-9]+</p>
</div>
<div class="paragraph">
<p>key: returning_key</p>
</div>
<div class="paragraph">
<p>description: Naming convention for Returning parameter</p>
</div>
<div class="paragraph">
<p>default: R_[a-zA-Z_-0-9]+</p>
</div>
</div>
</div>
</div>