<div class="paragraph">
<p>Unnecessary casting expressions make the code harder to read and understand.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public void example() {
  for (Foo obj : (List&lt;Foo&gt;) getFoos()) {  // Noncompliant; cast unnecessary because List&lt;Foo&gt; is what's returned
    //...
  }
}

public List&lt;Foo&gt; getFoos() {
  return this.foos;
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public void example() {
  for (Foo obj : getFoos()) {
    //...
  }
}

public List&lt;Foo&gt; getFoos() {
  return this.foos;
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_exceptions">Exceptions</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Casting may be required to distinguish the method to call in the case of overloading:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>class A {}
class B extends A{}
class C {
  void fun(A a){}
  void fun(B b){}

  void foo() {
    B b = new B();
    fun(b);
    fun((A) b); //call the first method so cast is not redundant.
  }

}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this unnecessary cast to "XXX".</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s3247">relates to: S3247</h3>

</div>
<div class="sect2">
<h3 id="_on_30_jul_2014_211830_freddy_mallet_wrote">on 30 Jul 2014, 21:18:30 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>My feedback [~ann.campbell.2]:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>For me this rule only targets Java so the "Targeted Languages" field should be empty and RSPEC-1905 and RSPEC-1906 should be merged</p>
</li>
<li>
<p>I would decrease the default severity to Minor</p>
</li>
</ul>
</div>
</div>
</div>
</div>