<div class="paragraph">
<p>Coding conventions allow teams to collaborate effectively. For maximum standardization and readability, PHP code should use the long <code>&lt;?php ?&gt;</code> tags or the short-echo <code>&lt;?= ?&gt;</code> tags; it should not use the other tag variations.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>&lt;?
$foo = 1;
?&gt;</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>&lt;?php
$foo = 1;
?&gt;</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Change this opening tag to either "&lt;?php" or "&lt;?=".</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>primary: first NC tag</p>
</li>
<li>
<p>secondary: subsequent NC tags</p>
<div class="ulist">
<ul>
<li>
<p>message: none</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_12_oct_2015_120358_ann_campbell_wrote">on 12 Oct 2015, 12:03:58 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Highlighting added [~linda.martin]. See what you think.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_12_nov_2015_182501_linda_martin_wrote">on 12 Nov 2015, 18:25:01 Linda Martin wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Perfect! Thanks.</p>
</div>
</div>
</div>
</div>