<div class="paragraph">
<p>The use of the <code>this</code> keyword is only required when the member is shadowed by a local parameter, or variable. In other cases, the code should be simplified by its removal.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public class MyClass
{
  private int count;

  public void DoSomething() {
    this.count++;  // Noncompliant
  }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public class MyClass
{
  private int count;

  public void DoSomething() {
    count++;
  }
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_replaced_by_s2963">is replaced by: S2963</h3>

</div>
<div class="sect2">
<h3 id="_on_20_jul_2015_114021_tamas_vajk_wrote">on 20 Jul 2015, 11:40:21 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] LGTM. The subtask is not saying explicitly which language it targets. It&#8217;s probably Swift.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_20_jul_2015_141424_ann_campbell_wrote">on 20 Jul 2015, 14:14:24 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Thanks for the nudge [~tamas.vajk]. Fixed.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_21_jul_2015_012414_evgeny_mandrikov_wrote">on 21 Jul 2015, 01:24:14 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] IMO title "this should only be used when required" is better than "this should not be used".</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_21_jul_2015_074048_ann_campbell_wrote">on 21 Jul 2015, 07:40:48 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>See what you think [~evgeny.mandrikov]</p>
</div>
</div>
</div>
</div>