<div class="paragraph">
<p>When an SWT <code>Image</code> accesses a file directly, it holds the file handle for the life of the image. Do this many times, and the OS may run out of available file handles. At minimum, SWT <code>Image</code>s which directly access files should not be <code>static</code>. At best, they should access their files through <code>ImageDescriptors</code>, which do not hold open file handles.</p>
</div>
<div class="paragraph">
<p>This rule looks for <code>org.eclipse.swt.graphics.Image</code>s which both directly access a file on the file path and are <code>static</code>.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>import org.eclipse.swt.graphics.Image;

public class MyView {
  static Image myImage = new Image("path/to/file.png"); // Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>import org.eclipse.swt.graphics.Image;
import org.eclipse.jface.resource.ImageDescriptor;

public class MyView {
  static ImageDescription myDescriptor = ImageDescriptor.createFromFile("path/to/file.png");  // Doesn't hold file handle open
  Image myImage = myDescriptor.getImage();</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Either remove the "static" modifier or use an "ImageDescriptor" to feed this "Image".</p>
</div>
</div>
</div>
</div>