<div class="paragraph">
<p>Using the <code>this</code> keyword inside the scope of an object to refer to the object&#8217;s properties and methods yields cleaner, clearer code, and helps avoid confusion when there are variables or functions outside the object scope with the same or similar names.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>function Person(name, birthdate) {
  this.name = name;
  this.birthdate = birthdate;

  get name() {
    return name; // Noncompliant
  }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>function Person(name, birthdate) {
  this.name = name;
  this.birthdate = birthdate;

  get name() {
    return this.name;
  }
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add "this" to this reference.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_12_nov_2015_182714_linda_martin_wrote">on 12 Nov 2015, 18:27:14 Linda Martin wrote:</h3>
<div class="paragraph">
<p>OK!</p>
</div>
</div>
</div>
</div>