<div class="paragraph">
<p>When a closable statement contains nested statements, it can quickly become difficult to see which statements are nested and which are not. That&#8217;s why ending a list of nested statements with <code>END-${STATEMENT-NAME}</code> is advised.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>READ DF-PARAM-SPILOTE AT END
  GO TO F-LECT-SPILOTE.</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>READ DF-PARAM-SPILOTE AT END
  GO TO F-LECT-SPILOTE
END-READ.</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="ulist">
<ul>
<li>
<p>key = evaluate</p>
<div class="ulist">
<ul>
<li>
<p>description:'true' to apply the rule to "EVALUATE" statements, 'false' otherwise</p>
</li>
<li>
<p>default: true</p>
</li>
</ul>
</div>
</li>
<li>
<p>key: perform</p>
<div class="ulist">
<ul>
<li>
<p>description:'true' to apply the rule to "PERFORM" statements, 'false' otherwise</p>
</li>
<li>
<p>default: true</p>
</li>
</ul>
</div>
</li>
<li>
<p>key: add</p>
<div class="ulist">
<ul>
<li>
<p>description:'true' to apply the rule to "ADD" statements, 'false' otherwise</p>
</li>
<li>
<p>default: true</p>
</li>
</ul>
</div>
</li>
<li>
<p>key: subtract</p>
<div class="ulist">
<ul>
<li>
<p>description:'true' to apply the rule to "SUBTRACT" statements, 'false' otherwise</p>
</li>
<li>
<p>default: true</p>
</li>
</ul>
</div>
</li>
<li>
<p>key: multiply</p>
<div class="ulist">
<ul>
<li>
<p>description:'true' to apply the rule to "MULTIPLY" statements, 'false' otherwise</p>
</li>
<li>
<p>default: true</p>
</li>
</ul>
</div>
</li>
<li>
<p>key: divide</p>
<div class="ulist">
<ul>
<li>
<p>description:'true' to apply the rule to "DIVIDE" statements, 'false' otherwise</p>
</li>
<li>
<p>default: true</p>
</li>
</ul>
</div>
</li>
<li>
<p>key: call</p>
<div class="ulist">
<ul>
<li>
<p>description:'true' to apply the rule to "CALL" statements, 'false' otherwise</p>
</li>
<li>
<p>default: true</p>
</li>
</ul>
</div>
</li>
<li>
<p>key: read</p>
<div class="ulist">
<ul>
<li>
<p>description:'true' to apply the rule to "READ" statements, 'false' otherwise</p>
</li>
<li>
<p>default: true</p>
</li>
</ul>
</div>
</li>
<li>
<p>key: write</p>
<div class="ulist">
<ul>
<li>
<p>description:'true' to apply the rule to "WRITE" statements, 'false' otherwise</p>
</li>
<li>
<p>default: true</p>
</li>
</ul>
</div>
</li>
<li>
<p>key: string</p>
<div class="ulist">
<ul>
<li>
<p>description:'true' to apply the rule to "STRING" statements, 'false' otherwise</p>
</li>
<li>
<p>default: true</p>
</li>
</ul>
</div>
</li>
<li>
<p>key: unstring</p>
<div class="ulist">
<ul>
<li>
<p>description:'true' to apply the rule to "UNSTRING" statements, 'false' otherwise</p>
</li>
<li>
<p>default: true</p>
</li>
</ul>
</div>
</li>
<li>
<p>key: if</p>
<div class="ulist">
<ul>
<li>
<p>description:'true' to apply the rule to "IF" statements, 'false' otherwise</p>
</li>
<li>
<p>default: true</p>
</li>
</ul>
</div>
</li>
<li>
<p>key: search</p>
<div class="ulist">
<ul>
<li>
<p>description:'true' to apply the rule to "SEARCH" statements, 'false' otherwise</p>
</li>
<li>
<p>default: true</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
</div>
</div>
</div>