<div class="paragraph">
<p>The default sort order is alphabetic, rather than numeric, regardless of the types in the array. Specifically, even if an array contains only numbers, all values in it will be converted to strings and sorted lexicographically, for an order like this: 1, 15, 2, 20, 5.</p>
</div>
<div class="paragraph">
<p>Fortunately the <code>sort</code> method allows you to pass an optional compare function to specify the sort order. When a compare function is supplied, the returned order depends on the return value of the compare function.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>var myarray = [80, 3, 9, 34, 23, 5, 1];

myarray.sort();
console.log(myarray); // outputs: [1, 23, 3, 34, 5, 80, 9]</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>var myarray = [80, 3, 9, 34, 23, 5, 1];

myarray.sort((a, b) =&gt; (a - b));
console.log(myarray); // outputs: [1, 3,  5, 9, 23, 34, 80]</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Provide a compare function to avoid sorting elements alphabetically.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_27_apr_2015_125727_linda_martin_wrote">on 27 Apr 2015, 12:57:27 Linda Martin wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Assign for review and completion</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_28_apr_2015_132808_ann_campbell_wrote">on 28 Apr 2015, 13:28:08 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Double-check my changes, please [~linda.martin]</p>
</div>
<div class="paragraph">
<p>Also, do you plan to raise this on all arrays, or limit it to when you can tell the array contains numbers?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_29_apr_2015_091626_linda_martin_wrote">on 29 Apr 2015, 09:16:26 Linda Martin wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] That&#8217;s a good question. I think we&#8217;ll do a first implementation and see what are the results and narrow the scope if too much FP shows up.</p>
</div>
<div class="paragraph">
<p>Reviewed.</p>
</div>
</div>
</div>
</div>