<div class="paragraph">
<p>Python developers can customize how code is interpreted by defining special methods (also called magic methods). For example, it is possible to override how the multiplication operator (<code>a * b</code>) will apply to instances of a class by defining in this class the <code>__mul__</code> and <code>__rmul__</code> methods. Whenever a multiplication operation is performed with this class, the python interpreter will call one of these methods instead of performing the default multiplication.</p>
</div>
<div class="paragraph">
<p>The python interpreter will always call these methods with the same number of parameters. Every call to a special method will fail if it is defined with an unexpected number of parameters.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when a special method is defined with an unexpected number of parameters.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class A:
    def __mul__(self, other, unexpected):  # Noncompliant. Too many parameters
        return 42

    def __add__(self):  # Noncompliant. Missing one parameter
        return 42

A() * 3  # TypeError: __mul__() missing 1 required positional argument: 'unexpected'
A() + 3  # TypeError: __add__() takes 1 positional argument but 2 were given</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class A:
    def __mul__(self, other):
        return 42

    def __add__(self, other):
        return 42

A() * 3
A() + 3</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>Python Documentation - <a href="https://docs.python.org/3/reference/datamodel.html#special-method-names">Special method names</a></p>
</li>
<li>
<p>Python Documentation - <a href="https://docs.python.org/3/library/copy.html">copy module</a></p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>Add XXX parameters. Method YYY should have ZZZ parameters.</p>
</li>
<li>
<p>Remove XXX parameters. Method YYY should have ZZZ parameters.</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Primary: The method signature.</p>
</div>
<div class="paragraph">
<p>Secondary: The unexpected parameters if there are too many parameters.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_deprecates_s2733">deprecates: S2733</h3>

</div>
<div class="sect2">
<h3 id="_on_11_feb_2020_172241_nicolas_harraudeau_wrote">on 11 Feb 2020, 17:22:41 Nicolas Harraudeau wrote:</h3>
<div class="paragraph">
<p>Special methods which are out of scope for this rule: __new__, __init__, __call__</p>
</div>
<div class="paragraph">
<p>These methods have no maximum number of parameters and require at minimum a "self" parameter. Missing a "self" parameter is already covered by RSPEC-5720.</p>
</div>
</div>
</div>
</div>