<div class="paragraph">
<p>Functions that invoke other <code>inline</code> functions often become too complex to actually be <code>inline</code>d, despite the fact that they appear to be small. This problem is especially common with constructors and destructors.</p>
</div>
<div class="paragraph">
<p>A constructor always invokes the constructors of its base classes and member data before executing its own code.</p>
</div>
<div class="paragraph">
<p>Code that does not try to <code>inline</code> constructors and destructors, is more portable because it can be handled by compilers that get confused when generating complex nested <code>inline</code> functions.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class String
{
  private:
    // ...
  public:
    String();
    ~String();
  };

  inline String::String()      //Noncompliant
  {
    // ...
  }
  inline String::~String()     //Noncompliant
  {
    // ...
  }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class String
{
  private:
    // ...
  public:
    String();
    ~String();
  };

  String::String()
  {
    // ...
  }
  String::~String()
  {
    // ...
  }
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove the "inline" keyword</p>
</div>
</div>
</div>
</div>