<div class="paragraph">
<p>Changing an inherited member to <code>private</code> will not prevent access to the base class implementation.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when a <code>private</code> method in an unsealed type has a signature that is identical to a <code>public</code> method declared in a base type.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>using System;

namespace MyLibrary
{
  public class Foo
  {
    public void SomeMethod(int count) { }
  }
  public class Bar:Foo
  {
    private void SomeMethod(int count) { } // Noncompliant
  }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>using System;

namespace MyLibrary
{
  public class Foo
  {
    public void SomeMethod(int count) { }
  }
  public sealed class Bar : Foo
  {
    private void SomeMethod(int count) { }
  }
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>This member hides '{0}'. Make it non-private or seal the class.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>member declaration</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s3434">relates to: S3434</h3>

</div>
<div class="sect2">
<h3 id="_on_22_may_2017_131956_ann_campbell_wrote">on 22 May 2017, 13:19:56 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~jeanchristophe.collet] RSPEC-2387 is already covered by C#. On the face of it, these two rules overlap nearly completely.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_22_may_2017_135950_jean_christophe_collet_wrote">on 22 May 2017, 13:59:50 Jean-Christophe Collet wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] RSPEC-2387 only covers fields, this rule covers only methods without an override.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_22_may_2017_151029_ann_campbell_wrote">on 22 May 2017, 15:10:29 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Okay, then RSPEC-3434 is the related RSpec, altho it&#8217;s scope is different. I&#8217;ve added it as 'related'</p>
</div>
</div>
</div>
</div>