<div class="paragraph">
<p>Using a <code>&lt;li&gt;</code> or <code>&lt;dt&gt;</code> item tag outside of a <code>&lt;ul&gt;</code>, <code>&lt;ol&gt;</code> or <code>&lt;dl&gt;</code> one does not make sense and indicates a bug.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>&lt;li&gt;Apple&lt;/li&gt;          &lt;!-- Noncompliant --&gt;
&lt;li&gt;Strawberry&lt;/li&gt;     &lt;!-- Noncompliant --&gt;

&lt;li&gt;Apple&lt;/li&gt;          &lt;!-- Noncompliant --&gt;
&lt;li&gt;Strawberry&lt;/li&gt;     &lt;!-- Noncompliant --&gt;

&lt;dt&gt;Apple&lt;/dt&gt;          &lt;!-- Noncompliant --&gt;
&lt;dt&gt;Strawberry&lt;/dt&gt;     &lt;!-- Noncompliant --&gt;</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>&lt;ul&gt;
  &lt;li&gt;Apple&lt;/li&gt;
  &lt;li&gt;Strawberry&lt;/li&gt;
&lt;/ul&gt;

&lt;ol&gt;
  &lt;li&gt;Apple&lt;/li&gt;
  &lt;li&gt;Strawberry&lt;/li&gt;
&lt;/ol&gt;

&lt;dl&gt;
  &lt;dt&gt;Apple&lt;/dt&gt;
  &lt;dt&gt;Strawberry&lt;/dt&gt;
&lt;/dl&gt;</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Surround this {0} item tag by a {1} or {2} container one.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_21_jun_2013_085948_dinesh_bolkensteyn_wrote">on 21 Jun 2013, 08:59:48 Dinesh Bolkensteyn wrote:</h3>
<div class="listingblock">
<div class="content">
<pre>&lt;?php
echo "&lt;ul&gt;\n";
foreach ($array as $element) {
  &lt;li&gt;&lt;?php echo $element; ?&gt;&lt;/li&gt;
}
echo "&lt;/ul&gt;\n";
?&gt;</pre>
</div>
</div>
<div class="paragraph">
<p>should be refactored into:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>&lt;ul&gt;
&lt;?php
foreach ($array as $element) {
?&gt;
  &lt;li&gt;&lt;?php echo $element; ?&gt;&lt;/li&gt;
&lt;?php
}
?&gt;
&lt;/ul&gt;</pre>
</div>
</div>
<div class="paragraph">
<p>or:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>&lt;?php
echo "&lt;ul&gt;\n";
foreach ($array as $element) {
  echo "  &lt;li&gt;$element&lt;/li&gt;\n";
}
echo "&lt;/ul&gt;\n";
?&gt;</pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_on_21_jun_2013_092859_dinesh_bolkensteyn_wrote">on 21 Jun 2013, 09:28:59 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>We should be careful with nested lists&#8230;&#8203;.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_8_jul_2013_182130_freddy_mallet_wrote">on 8 Jul 2013, 18:21:30 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-2993</p>
</div>
</div>
</div>
</div>