<div class="paragraph">
<p>For readability, SAP recommends that asterisks (<code>*</code>) only be used to comment out header lines and code. Commentary should be commented using a double quote (<code>"</code>)</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>* GAC -  13 June 13 - output user data
* WRITE: / 'Firstname'.</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>" GAC -  13 June 13 - output user data
* WRITE: / 'Firstname'.</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_exceptions">Exceptions</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule ignores code that is included in a commentary block and commented as commentary.</p>
</div>
<div class="listingblock">
<div class="content">
<pre>" This is a commentary block. It could go on for a number of lines.
" As we see in the code sample below...
" WRITE: / 'Firstname'.
" It would have been better to comment the line above as code (i.e. with an asterisk)
" but not doing so is accepted.</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use a double-quote to mark this comment line instead of an asterisk</p>
</div>
</div>
</div>
</div>