<div class="paragraph">
<p>When using some transaction managers like IBM IMS, each COBOL program is in fact considered a sub-program by the transaction manager. The <code>GOBACK</code> statement returns control to the transaction manager, but using <code>STOP RUN</code> might cause unpredictable results or abnormal termination.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>STOP RUN</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>GOBACK</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace this "STOP RUN" statement with a "GOBACK" statement.</p>
</div>
</div>
</div>
</div>