<div class="paragraph">
<p>Marking a variable that is unchanged after initialization <code>final</code> is an indication to future maintainers that "no this isn&#8217;t updated, and it&#8217;s not supposed to be". <code>final</code> should be used in these situations in the interests of code clarity.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public boolean seek(int [] input) {
  int target = 32;  // Noncompliant
  for (int i : input) {
    if (i == target) {
      return true;
    }
  }
  return false;
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public boolean seek(int [] input) {
  final int target = 32;
  for (int i : input) {
    if (i == target) {
      return true;
    }
  }
  return false;
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add the "final" modifier to "xxx".</p>
</div>
</div>
</div>
</div>