<div class="paragraph">
<p>When a <code>Promise</code> needs to only "resolve" or "reject", it&#8217;s more efficient and readable to use the methods specially created for such use cases: <code>Promise.resolve(value)</code> and <code>Promise.reject(error)</code>.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>let fulfilledPromise = new Promise(resolve =&gt; resolve(42));
let rejectedPromise = new Promise(function(resolve, reject) {
  reject('fail');
});</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>let fulfilledPromise = Promise.resolve(42);
let rejectedPromise = Promise.reject('fail');</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace this trivial promise with "[Promise.resolve(XXX)|Promise.reject(XXX)]".</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>entire "new Promise" expression</p>
</div>
</div>
</div>
</div>