<div class="paragraph">
<p>Unlike strongly typed languages, JavaScript does not enforce a return type on a function. This means that different paths through a function can return different types of values, which can be very confusing to the user and significantly harder to maintain.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>function foo(a) {  // Noncompliant
  if (a === 1) {
    return true;
  }
  return 3;
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>function foo(a) {
  if (a === 1) {
    return true;
  }
  return false;
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_exceptions">Exceptions</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Functions returning <code>this</code> are ignored.</p>
</div>
<div class="listingblock">
<div class="content">
<pre>function foo() {
  // ...
  return this;
}</pre>
</div>
</div>
<div class="paragraph">
<p>Functions returning expressions having type <code>any</code> are ignored.</p>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Refactor this function to always return the same type.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Primary: function name/declaration</p>
</div>
<div class="paragraph">
<p>Secondary: Return statements</p>
</div>
</div>
</div>
</div>