<div class="paragraph">
<p>Returning <code>Nothing</code> from a non-<code>async</code> <code>Task</code>/<code>Task(Of T)</code> method will cause a <code>NullReferenceException</code> at runtime. This problem can be avoided by returning <code>Task.FromResult(Of T)(Nothing)</code> instead.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>Public Function GetFooAsync() As Task(Of Object)
    Return Nothing
End Function</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>Public Function GetFooAsync() As Task(Of Object)
    Return Task.FromResult(Of Object)(Nothing)
End Function</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Do not return null from this method, instead return Task.FromResult&lt;T&gt;(null).</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>return null</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_16_apr_2018_150041_amaury_levé_wrote">on 16 Apr 2018, 15:00:41 Amaury Levé wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Could you please review?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_17_apr_2018_214326_ann_campbell_wrote">on 17 Apr 2018, 21:43:26 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~amaury.leve] I&#8217;m confused. The title says "in a method <strong>not</strong> marked as async" and the description talks about "Returning <code>null</code> from a <strong>non</strong>-<code>async</code>".</p>
</div>
<div class="paragraph">
<p>Also, this title doesn&#8217;t follow the format: X should [not] Y</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_18_apr_2018_093040_amaury_levé_wrote">on 18 Apr 2018, 09:30:40 Amaury Levé wrote:</h3>
<div class="paragraph">
<p>Thank you for the review. Does it looks better now?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_18_apr_2018_134319_ann_campbell_wrote">on 18 Apr 2018, 13:43:19 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Yes, looks good [~amaury.leve]</p>
</div>
</div>
</div>
</div>