<div class="paragraph">
<p>Using a type parameter when you don&#8217;t have to simply obfuscates the code. Qualifying an inner type with a type parameter will compile, but confuse maintainers.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>&lt;T extends Map&gt; T doTheThing(T.Entry type) {  // Noncompliant
  //...
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>&lt;T extends Map&gt; T doTheThing(Map.Entry type) {
  //...
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use "xxx" here instead of "y".</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Unnecessary type param</p>
</div>
</div>
</div>
</div>