<div class="paragraph">
<p>Shared coding conventions allow teams to collaborate effectively. This rule checks that labels are on lines by themselves, which makes them more immediately visible as such.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>LBLVAL:    IF         COND(&amp;ERRFLG = ' ') THEN (DO)
           CHGVAR     VAR(&amp;MSG) VALUE('Validation Completed')
           ENDDO</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>LBLVAL:
           IF         COND(&amp;ERRFLG = ' ') THEN (DO)
           CHGVAR     VAR(&amp;MSG) VALUE('Validation Completed')
           ENDDO</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_29_sep_2014_132943_ann_campbell_wrote">on 29 Sep 2014, 13:29:43 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>As it turns out, this rule is applicable to CLP, not RPG.</p>
</div>
</div>
</div>
</div>