<div class="paragraph">
<p>If you only want one instance that matches certain criteria out of a collection, it&#8217;s far more efficient to grab the first matching item than it is to fully filter the collection for your criteria and then only use a single value.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>let one = arr.filter { $0.containsString("yo") }.first  // Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>let one = arr.first(where: { $0.containsString("yo") })</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Convert this "filter&#8230;&#8203;first" operation to "first(where: &#8230;&#8203;)".</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>primary: <code>first</code></p>
</div>
<div class="paragraph">
<p>secondary: <code>filter</code></p>
</div>
</div>
</div>
</div>