<div class="paragraph">
<p>It is undefined behaviour if a function has different exception-specifications in different translation units.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>// Translation unit A
void foo( ) throw ( const char_t * ) {
  throw "Hello World!";
}

// Translation unit B
// foo declared in this translation unit with a different exception
// specification
extern void foo ( ) throw ( int32_t ); // Noncompliant - different specifier
void b ( ) throw ( int32_t ) {
  foo ( ); // The behaviour here is undefined.
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>// Header for translation unit A
extern void foo ( ) throw ( const char_t * );

// Translation unit A
void foo( ) throw ( const char_t * ) {
  throw "Hello World!";
}

// Translation unit B
#include "A.h"
void b ( ) throw ( const char_t * ) {
  foo ( );
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C&#43;&#43;:2008, 15-4-1</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Correct the thrown exception(s) to match those of the declaration.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s822">relates to: S822</h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s927">relates to: S927</h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s928">relates to: S928</h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s1017">relates to: S1017</h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s1049">is related to: S1049</h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s829">is related to: S829</h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s833">is related to: S833</h3>

</div>
<div class="sect2">
<h3 id="_on_14_oct_2014_182912_ann_campbell_wrote">on 14 Oct 2014, 18:29:12 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~samuel.mercier] please correct the following</p>
</div>
<div class="ulist">
<ul>
<li>
<p>There is no message</p>
</li>
<li>
<p>Add a See section to the description listing the appropriate MISRA number</p>
</li>
<li>
<p>Fill in Applicability.</p>
</li>
<li>
<p>Make sure the appropriate MISRA C and MISRA C&#43;&#43; fields on the references tab are filled in</p>
</li>
<li>
<p>I would not use Maintainability/Understandability for an "undefined behavior" rule. In fact, I doubt Maintainability is applicable to many MISRA rules at all. I would assign this one to reliability, instead</p>
</li>
</ul>
</div>
</div>
</div>
</div>