<div class="paragraph">
<p><code>DELETE FROM dbtab</code> without a <code>WHERE</code> condition deletes all the entries of the table. Check whether dataset to be deleted can be limited by a suitable WHERE condition.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>DELETE FROM COUNTRIES.</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>DELETE FROM COUNTRIES WHERE CODE = country_code.</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add a "WHERE" clause to this "DELETE" statement to prevent deleting all rows in table "XXX".</p>
</div>
</div>
</div>
</div>