<div class="paragraph">
<p>Given a consistent set of data, the results from any given database query will be consistent, but any random subset of that data may not be. Any time you perform a query that retrieves only the first n rows, without having ordered the data in the query, you&#8217;re grabbing that random subset and putting the reliability of your program at risk. Thus queries that use <code>FETCH FIRST ... ROWS</code> must always include an <code>ORDER BY</code>.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>SELECT name   -- Noncompliant
FROM employee
WHERE sick_days &gt; 5
FETCH FIRST 10 ROWS ONLY</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>SELECT name
FROM employee
WHERE sick_days &gt; 5
ORDER BY sick_days desc
FETCH FIRST 10 ROWS ONLY</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add an "ORDER BY" clause to this query to ensure consistency in the returned set.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_14_jul_2015_150059_pierre_yves_nicolas_wrote">on 14 Jul 2015, 15:00:59 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>Looks good.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_14_jul_2015_150123_ann_campbell_wrote">on 14 Jul 2015, 15:01:23 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Mysql uses <code>Limit n,n</code></p>
</div>
<div class="paragraph">
<p>SQL Server uses <code>select top N &lt;field&gt;â€¦.</code></p>
</div>
<div class="paragraph">
<p>Oracle uses <code>TOP</code></p>
</div>
<div class="paragraph">
<p>ABAP <code>UP TO n ROWS</code></p>
</div>
</div>
</div>
</div>