<div class="paragraph">
<p>Empty <code>case</code> clauses that fall through to the default are useless. Whether or not such a <code>case</code> is present, the <code>default</code> clause will be invoked. Such <code>case</code>s simply clutter the code, and should be removed.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>switch(ch)
{
  case 'a' :
    HandleA();
    break;
  case 'b' :
    HandleB();
    break;
  case 'c' :  // Noncompliant
  default:
    HandleTheRest();
    break;
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>switch(ch)
{
  case 'a' :
    HandleA();
    break;
  case 'b' :
    HandleB();
    break;
  default:
    HandleTheRest();
    break;
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this [empty|redundant] "case" clause.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p><code>case xxx:</code></p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s131">is related to: S131</h3>

</div>
<div class="sect2">
<h3 id="_on_10_dec_2015_091159_tamas_vajk_wrote">on 10 Dec 2015, 09:11:59 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] LGTM</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_11_may_2016_095328_evgeny_mandrikov_wrote">on 11 May 2016, 09:53:28 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~tamas.vajk], [~ann.campbell.2] as of today both SLVS-827 and CPP-1477 cover cases such as</p>
</div>
<div class="paragraph">
<p>{noformat}</p>
</div>
<div class="paragraph">
<p>switch (&#8230;&#8203;) {</p>
</div>
<div class="literalblock">
<div class="content">
<pre>default:</pre>
</div>
</div>
<div class="literalblock">
<div class="content">
<pre>case ...:</pre>
</div>
</div>
<div class="literalblock">
<div class="content">
<pre>...</pre>
</div>
</div>
<div class="paragraph">
<p>}</p>
</div>
<div class="paragraph">
<p>{noformat}</p>
</div>
<div class="paragraph">
<p>in which IMO message <code>Remove this empty "case" clause.</code> is misleading, so for CPP-1477 we changed it on <code>Remove this redundant "case" clause.</code></p>
</div>
</div>
<div class="sect2">
<h3 id="_on_16_may_2016_211532_ann_campbell_wrote">on 16 May 2016, 21:15:32 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Message updated [~evgeny.mandrikov]</p>
</div>
</div>
</div>
</div>