<div class="paragraph">
<p>C&#43;&#43;20 introduces ranges library. A range is a group of items that can be iterated over. It should provide a <code>begin</code> iterator and an <code>end</code> sentinel. As you can guess, all the STL containers are ranges. This makes working with the STL library much more powerful by introducing range adaptors and much less verbose by introducing a constrained version of most algorithms in the namespace <code>std::ranges</code>. Before the ranges library, you had to specify the <code>begin</code> and <code>end</code> iterator when calling the STL algorithms even when you want to iterate over the whole container.</p>
</div>
<div class="paragraph">
<p>This rule focuses on making your code less verbose and more readable by suggesting range-based over iterator-based algorithms when convenient.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>auto printEven = [](auto i) {
  if (i % 2 == 0) {
    std::cout &lt;&lt; i;
  }
};

void f1(const std::vector&lt;int&gt;&amp; v) {
  std::for_each(v.begin(), v.end(), printEven); // Noncompliant
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>auto printEven = [](auto i) {
  if (i % 2 == 0) {
    std::cout &lt;&lt; i;
  }
};

void f2(const std::vector&lt;int&gt;&amp; v) {
  std::ranges::for_each(v, printEven); // Compliant
}</pre>
</div>
</div>
</div>
</div>