<div class="paragraph">
<p>Whenever a CICS command is used with a <code>NOHANDE</code> or <code>RESP</code> option, the default CICS exception handling is disabled. The correct approach is then to make sure that every possible exception is handled properly directly in the code. This is done by examining the <code>RESP</code> value or the <code>EIBRESP</code> field value.</p>
</div>
<div class="paragraph">
<p>It is possible to compare their value to hardcoded numbers corresponding to their binary value, however this makes the code difficult to read and maintain. It is recommended to use instead the DFHRESP built-in translator function which enables the use of equivalent symbolic values.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when the <code>EIBRESP</code> field is compared to a hardcoded-number. This rule does not handle <code>RESP</code> values for now.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>if EIBRESP=36 then /* Noncompliant */
end;

select (EIBRESP);
    when(36) ...;  /* Noncompliant */
end;</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>if EIBRESP=DFHRESP(MAPFAIL) then
end;

select (EIBRESP);
    when(DFHRESP(MAPFAIL)) ...;
end;</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://www.ibm.com/docs/en/cics-ts/5.6?topic=conditions-handling-exception-by-inline-code">Handling exception conditions by in-line code</a></p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace this hardcoded binary value with a call to "DFHRESP".</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Primary: The hardcoded value</p>
</div>
<div class="paragraph">
<p>Secondary: The EIBRESP field</p>
</div>
</div>
</div>
</div>