<div class="paragraph">
<p>"&gt;" and "&lt;" can be used in Java both as comparison operators or when working with generics.  To prevent any confusion, extra whitespaces around these characters should be avoided when defining types.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public class Foo &lt; A &gt; {...}
...
List &lt; String &gt; myList = new ArrayList &lt; String &gt; ();
...
boolean same = Util. &lt; Integer, String &gt; compare(p1, p2);</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public class Foo&lt;A&gt; {...}
...
List&lt;String&gt; myList = new ArrayList&lt;String&gt;();
...
boolean same = Util.&lt;Integer, String&gt;compare(p1, p2);</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove whitespaces [before|after] ["&lt;"|"&gt;"] character</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_7_nov_2013_094730_ann_campbell_wrote">on 7 Nov 2013, 09:47:30 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>The Compliant Solution removes the space between "List" and "&lt;String&gt;".</p>
</div>
<div class="paragraph">
<p>Does that imply that this rule will flag "List &lt;String&gt;", which I find just as readable (perhaps more so) as "List&lt;String&gt;"?</p>
</div>
</div>
</div>
</div>