<div class="paragraph">
<p>Setting the creation of an anonymous class inside a loop means a new instance will be created each time through the loop. But since the anonymous class is likely independent of any values that change within the loop, it&#8217;s more efficient to create it once, outside the loop and use the same instance for each iteration.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>private void setUpListeners(List&lt;GuiObject&gt; list) {
  for (GuiObject gObj : list) {
    gObj.addListener(new Listener() {   // Noncompliant
      public void onChange(Event e) {
        // ...
      }
    });
  }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>private void setUpListeners(List&lt;GuiObject&gt; list) {

  Listener listener = new Listener() {
    public void onChange(Event e) {
      // ...
    }
  }

  for (GuiObject gObj : list) {
    gObj.addListener(listener);
  }
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Move the definition of this class outside the loop.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s1515">relates to: S1515</h3>

</div>
<div class="sect2">
<h3 id="_on_11_jun_2015_182537_ann_campbell_wrote">on 11 Jun 2015, 18:25:37 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>CodePro: Anonymous Class in Loop</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_16_jun_2015_131346_nicolas_peru_wrote">on 16 Jun 2015, 13:13:46 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>Looks good.</p>
</div>
</div>
</div>
</div>