<div class="paragraph">
<p>The contract and expectation of a <code>get</code> function is that it will return something. Failing to return a value on all paths is surely an error.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>var x = {
  _prop : "foo",
  get prop(){  // Noncompliant
    console.log("prop getter");
  }
}

x.prop; // returns undefined</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>var x = {
  _prop : "foo",
  get prop(){
    console.log("prop getter");
    return _prop;
  }
}

x.prop; // returns "foo"</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Return a value from this function.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>signature</p>
</div>
</div>
</div>
</div>