<div class="paragraph">
<p>The <a href="https://docs.python.org/3/reference/compound_stmts.html#the-with-statement"><code>with</code> statement</a> should only be used with <a href="https://docs.python.org/3/reference/datamodel.html#context-managers">context managers</a>. A context manager should have an <code>__enter__</code> and an <code>__exit__</code> method.</p>
</div>
<div class="paragraph">
<p>Note that only <code>async with</code> statements should be used on <a href="https://docs.python.org/3/reference/datamodel.html#async-context-managers">asynchronous context managers</a>.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when a <code>with</code> statement is used on something else than a context manager.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>from contextlib import contextmanager, asynccontextmanager

def generator():
    yield 42

with generator() as e:  # Noncompliant
    print(e)

class Empty:
    pass

with Empty():  # Noncompliant
    pass

@asynccontextmanager
async def async_context_manager():
    yield 42

with async_context_manager() as context:  # Noncompliant Missing "async"
    print(context)</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>from contextlib import contextmanager, asynccontextmanager

@contextmanager
def simple_contextmanager():
    print("enter")
    yield
    print("exit")

with simple_contextmanager() as e:
    print(e)

class SimpleContextManager:
    def __enter__(self):
        print("enter")
    def __exit__(self, type_, value, traceback):
        print("exit")

with SimpleContextManager():
    pass

@asynccontextmanager
async def async_context_manager():
    yield 42

async def call_async_context_manager():
    async with async_context_manager() as context:
        print(context)

asyncio.run(call_async_context_manager())</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>Replace this expression with a context manager.</p>
</li>
<li>
<p>Add "async" before "with"; Expression is an async context manager.</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Primary: the expression used as a context manager</p>
</div>
<div class="paragraph">
<p>Secondary: the "with" keyword</p>
</div>
</div>
</div>
</div>