<div class="paragraph">
<p>C-style arrays (such as <code>int i[10]</code>) are not very convenient to use:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>They are fixed size (even C VLA are not truly variable size, and they are not supported in C&#43;&#43;)</p>
</li>
<li>
<p>If the number of elements in the array can vary, it will lead to manual memory allocation (or people will use fixed-size arrays that "should be large enough", which is both a waste of memory and a limitation of the program)</p>
</li>
<li>
<p>It is very easy to lose the size of the array since an array passed to a function decays into a pointer</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>The C&#43;&#43; standard library proposes two types that are better than C-style arrays and together cover all the use cases of C-style arrays:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>For fixed-size arrays, where the memory is on the stack, use <code>std::array</code>. It is like a C-style array, except that it has a normal argument passing semantic, and the size is always a part of the type. If <code>std::array</code> is not available to you (before C&#43;&#43;11), you can roll your own version.</p>
</li>
<li>
<p>For variable-size arrays, use <code>std::vector</code>. It can be resized and handles memory allocation transparently.</p>
</li>
<li>
<p>For character strings, you should use <code>std::string</code> instead of arrays of characters.</p>
</li>
<li>
<p>For arrays of characters that are not strings (e.g., alphabet, exit codes, keyboard control list) perfer <code>std::array</code> or <code>std::vector</code> as per the first two bullets.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>The rule S945 is related to this rule but focuses on passing arguments of an array type. S5025 will flag the use of dynamic memory allocation that could be replaced by <code>std::vector</code>.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>void f() {
  int a[10]; // Noncompliant
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>void f() {
  std::array&lt;int, 10&gt; a1; // If the size really is a constant
  // Or
  std::vector&lt;int&gt;a2; // For variable size

  auto s = "Hello!"; // Compliant by exception
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_exceptions">Exceptions</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule will not report the use of C-style arrays in <code>extern "C"</code> code (since those arrays are often required here for compatibility with external code) and in the arguments of <code>main</code>.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://github.com/isocpp/CppCoreGuidelines/blob/46dadd0b16b904fe0ff88c1fef9b0702dfd139e2/CppCoreGuidelines.md#slcon1-prefer-using-stl-array-or-vector-instead-of-a-c-array">C&#43;&#43; Core Guidelines SL.con.1</a> - Prefer using STL array or vector instead of a C array</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s945">relates to: S945</h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s5025">relates to: S5025</h3>

</div>
</div>
</div>