<div class="paragraph">
<p>If you&#8217;re using a <code>struct</code>, it is likely because you&#8217;re interested in performance. But by failing to implement <code>IEquatable&lt;T&gt;</code> you&#8217;re loosing performance when comparisons are made because without <code>IEquatable&lt;T&gt;</code>, boxing and reflection are used to make comparisons.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>struct MyStruct  // Noncompliant
{
  private int i;
  public int I
  {
    //...
  }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>struct MyStruct : IEquatable&lt;MyStruct&gt;
{
  private int i;
  public int I
  {
    //...
  }

  public bool Equals(MyStruct other)
  {
    throw new NotImplementedException();
  }
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Implement 'IEquatable&lt;T&gt;' in value type 'XXX'.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>type declaration</p>
</div>
</div>
</div>
</div>