<div class="paragraph">
<p>Taking a <code>shared_ptr</code> by r-value reference is unnecessary. If done on purpose, it might imply that <code>unique_ptr</code> is a better choice since it transfers unique ownership.</p>
</div>
<div class="paragraph">
<p>In general, a function should take:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>A copy of a <code>shared_ptr</code>, if the function takes part in the ownership of the managed object</p>
</li>
<li>
<p>A reference to a <code>shared_ptr</code> if the function plans to modify the shared_ptr itself by calling <code>reset</code> or <code>swap</code></p>
</li>
<li>
<p>A reference to a <code>const</code> <code>shared_ptr</code>, if the function might take part in the ownership by copying the reference to another shared_ptr on at least one of its paths</p>
</li>
<li>
<p>A raw pointer/reference to the object, if the function is only interested in the current value of the managed object</p>
</li>
</ul>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>Class Circle{};

void fn(shared_ptr&lt;Circle&gt;&amp;&amp; circle);  // Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>Class Circle{};

void fn(shared_ptr&lt;Circle&gt; circle);</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_exceptions">Exceptions</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The move constructor of the <code>shared_ptr</code> itself.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://github.com/isocpp/CppCoreGuidelines/blob/c553535fb8dda2839d13ab5f807ffbc66b63d67b/CppCoreGuidelines.md#r34-take-a-shared_ptrwidget-parameter-to-express-that-a-function-is-part-owner">C&#43;&#43; Core Guidelines R.34</a> - Warn if a function takes a Shared_ptr&lt;T&gt; by rvalue reference. Suggesting taking it by value instead.</p>
</li>
</ul>
</div>
</div>
</div>