<div class="paragraph">
<p>Shared naming conventions allow teams to collaborate efficiently. This rule checks that all function names match a provided regular expression.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="paragraph">
<p>With default provided regular expression: <code>^[a-z][a-zA-Z0-9]*$</code>:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>function DoSomething(){...}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>function doSomething(){...}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_exceptions">Exceptions</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Methods with an <code>@inheritdoc</code> annotation, as well as magic methods (<code>__construct()</code>, <code>__destruct()</code>, <code>__call()</code>, <code>__callStatic()</code>, <code>__get()</code>, <code>__set()</code>, <code>__isset()</code>, <code>__unset()</code>, <code>__sleep()</code>, <code>__wakeup()</code>, <code>__toString()</code>, <code>__invoke()</code>, <code>__set_state()</code>, <code>__clone()</code>, <code>__debugInfo()</code>) are ignored.</p>
</div>
<div class="listingblock">
<div class="content">
<pre>function __construct(){...}
function __destruct(){...}

/**
 * {@inheritdoc}
 */
function myFunc(){...}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Rename [method|function|subroutine] "XXXX" to match the regular expression ${format}.</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="ulist">
<ul>
<li>
<p>key = format</p>
</li>
<li>
<p>Description: Regular expression used to check the [method|function|subroutine] names against</p>
</li>
<li>
<p>Default: ^[a-z][a-zA-Z0-9]*$</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s1542">is related to: S1542</h3>

</div>
<div class="sect2">
<h3 id="_on_20_may_2013_142913_fabrice_bellingard_wrote">on 20 May 2013, 14:29:13 Fabrice Bellingard wrote:</h3>
<div class="paragraph">
<p>Implementation: http://jira.codehaus.org/browse/SONARJAVA-118</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_2_aug_2013_090559_freddy_mallet_wrote">on 2 Aug 2013, 09:05:59 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-2776</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_12_feb_2014_180827_freddy_mallet_wrote">on 12 Feb 2014, 18:08:27 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3455 for PHP</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_6_feb_2018_153050_alban_auzeill_wrote">on 6 Feb 2018, 15:30:50 Alban Auzeill wrote:</h3>
<div class="paragraph">
<p>COBOL use [RSPEC-1589] instead.</p>
</div>
<div class="paragraph">
<p>PL/I, PL/SQL, T-SQL, VB6 use [RSPEC-1542] instead.</p>
</div>
</div>
</div>
</div>