<div class="paragraph">
<p>The use of <code>[DefaultValue]</code> with <code>[Optional]</code> has no more effect than <code>[Optional]</code> alone. That&#8217;s because <code>[DefaultValue]</code> doesn&#8217;t actually do anything; it merely indicates the intent for the value. More than likely, <code>[DefaultValue]</code> was used in confusion instead of <code>[DefaultParameterValue]</code>.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class MyClass
{
    public void DoStuff([Optional][DefaultValue(4)]int i, int j = 5)  // Noncompliant
    {
        Console.WriteLine(i);
    }

    public static void Main()
    {
        new MyClass().DoStuff(); // prints 0
    }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class MyClass
{
    public void DoStuff([Optional][DefaultParameterValue(4)]int i, int j = 5)
    {
        Console.WriteLine(i);
    }

    public static void Main()
    {
        new MyClass().DoStuff(); // prints 4
    }
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use "[DefaultParameterValue]" instead.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p><code>[DefaultValue]</code></p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_8_dec_2015_095538_tamas_vajk_wrote">on 8 Dec 2015, 09:55:38 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] LGTM, but I&#8217;m not 100% happy with the title. <code>[DefaultValue]</code> can be used with <code>[Optional]</code>, you might build your own tooling around it, and read the value specified in <code>[DefaultValue]</code>. So we should probably not say that you shouldn&#8217;t use it. But I have no better ideas than the current one.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_8_dec_2015_150514_ann_campbell_wrote">on 8 Dec 2015, 15:05:14 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~tamas.vajk] I struggled with this title myself, and I recognize that <code>[DefaultValue]</code> + <code>[Optional]</code> is a valid usage. But the best alternate title gets into intent, which I don&#8217;t want to do: '[DefaultValue]" should not be used with "[Optional]" when "[DefaultParamterValue]" was intended'</p>
</div>
<div class="paragraph">
<p>And if you&#8217;re building tooling around <code>[DefaultValue]</code> then you&#8217;ll turn this rule off.</p>
</div>
</div>
</div>
</div>