<div class="paragraph">
<p>To prevent portability issues <code>!=</code>, <code>!&gt;</code> and <code>!&lt;</code> operators should be replaced by the ANSI standard operators: <code>&lt;&gt;</code>, <code>&lt;=</code> or <code>&gt;=</code>.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>SELECT C1 FROM S1TESTMD WHERE BIRTHDATE != 2000</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>SELECT C1 FROM S1TESTMD WHERE BIRTHDATE &lt;&gt; 2000</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace this "XX" operator with the ANSI standard equivalent: "YY" operator.</p>
</div>
</div>
</div>
</div>