<div class="paragraph">
<p>PHP lets you make static calls to non-static methods, ex: <code>A::f();</code>. But the fact that you can doesn&#8217;t mean you <em>should</em>. While such calls will work when there&#8217;s no <code>$this</code> reference in the called method, a fatal error will occur when there is. Furthermore, such calls have been deprecated in PHP 7, and their support may be removed from future releases.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class A {
  function f() {
    // ...
  }
}

A::f();      // Noncompliant, "f" is non-static</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class A {
  function f() {
    // ...
  }
}

$a = new A();
$a-&gt;f();</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use an object instance to call this non-static method.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The class name that should be replaced by an object instance variable.</p>
</div>
</div>
</div>
</div>