<div class="paragraph">
<p>Using <code>Thread.sleep</code> in a test is just generally a bad idea. It creates brittle tests that can fail unpredictably depending on environment ("Passes on my machine!") or load. Don&#8217;t rely on timing (use mocks) or use libraries such as <code>Awaitility</code> for asynchroneous testing.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>@Test
public void testDoTheThing(){

  MyClass myClass = new MyClass();
  myClass.doTheThing();

  Thread.sleep(500);  // Noncompliant
  // assertions...
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>@Test
public void testDoTheThing(){

  MyClass myClass = new MyClass();
  myClass.doTheThing();

  await().atMost(2, Duration.SECONDS).until(didTheThing());  // Compliant
  // assertions...
}

private Callable&lt;Boolean&gt; didTheThing() {
  return new Callable&lt;Boolean&gt;() {
    public Boolean call() throws Exception {
      // check the condition that must be fulfilled...
    }
  };
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this use of "Thread.sleep()".</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_12_may_2015_145251_david_gageot_wrote">on 12 May 2015, 14:52:51 David Gageot wrote:</h3>
<div class="paragraph">
<p>lgtm</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_20_may_2015_150839_nicolas_peru_wrote">on 20 May 2015, 15:08:39 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>Ok.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_21_aug_2015_092147_michael_gumowski_wrote">on 21 Aug 2015, 09:21:47 Michael Gumowski wrote:</h3>
<div class="paragraph">
<p>WDYT about changes [~duarte.meneses]?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_21_aug_2015_092958_duarte_meneses_wrote">on 21 Aug 2015, 09:29:58 Duarte Meneses wrote:</h3>
<div class="paragraph">
<p>Looks good.</p>
</div>
<div class="paragraph">
<p>Only suggestion I have is to put the timeout in the solution longer (2/3s?), otherwise it wouldn&#8217;t fix the problem of dependency on the environment.</p>
</div>
<div class="paragraph">
<p>With awaitility we can use larger timeouts but still proceed quickly as soon as the polling conditions is satisfied.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_21_aug_2015_093555_michael_gumowski_wrote">on 21 Aug 2015, 09:35:55 Michael Gumowski wrote:</h3>
<div class="paragraph">
<p>Ok, updated. Thanks for your feedback! Feels free to modify it directly next time!</p>
</div>
</div>
</div>
</div>