<div class="paragraph">
<p>Duplication of selectors might indicate a copy-paste mistake. The rule detects the following kinds of duplications:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>within a list of selectors in a single rule set</p>
</li>
<li>
<p>for duplicated selectors in different rule sets within a single stylesheet.</p>
</li>
</ul>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>.foo, .bar, .foo { ... }  /* Noncompliant */

.class1 { ... }
.class1 { ... }  /* Noncompliant */</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>.foo, .bar { ... }

.class1 { ... }
.class2 { ... }</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this duplicated selector.</p>
</div>
</div>
</div>
</div>