<div class="paragraph">
<p>Overriding <code>operator new</code> typically indicates that custom memory allocation is required for the class. When that&#8217;s the case, it must be balanced with a custom memory deallocation in a matching <code>operator delete</code> method. Otherwise memory leaks or memory corruption will result.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class AirPlane
{
public:
  void* operator new(size_t size);
  void fly();
};</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class AirPlane
{
public:
  void* operator new(size_t size);
  void operator delete(void* deadObject, size_t size);
  void fly();
};</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://wiki.sei.cmu.edu/confluence/x/KX0-BQ">CERT, DCL54-CPP.</a> - Overload allocation and deallocation functions as a pair in the same scope</p>
</li>
<li>
<p><a href="https://github.com/isocpp/CppCoreGuidelines/blob/036324/CppCoreGuidelines.md#r15-always-overload-matched-allocationdeallocation-pairs">C&#43;&#43; Core Guidelines R.15</a> - Always overload matched allocation/deallocation pairs</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add an "operator delete" to this class.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_26_may_2015_182621_evgeny_mandrikov_wrote">on 26 May 2015, 18:26:21 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] word "class" looks strange for me after "operator delete" in description. Is it a typo or just bad knowledge of english by me?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_27_may_2015_140728_ann_campbell_wrote">on 27 May 2015, 14:07:28 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>It&#8217;s not your English [~evgeny.mandrikov], it&#8217;s mine. ;)</p>
</div>
<div class="paragraph">
<p>Check it now.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_27_may_2015_144446_evgeny_mandrikov_wrote">on 27 May 2015, 14:44:46 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>LGTM.</p>
</div>
</div>
</div>
</div>