<div class="paragraph">
<p>Unused parameters are misleading. Whatever the values passed to such parameters, the behavior will be the same.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>func compute(start int) { // Noncompliant; start is not used
	sum := 0
	for i := 0; i &lt; 10; i++ {
		sum += i
	}
	fmt.Println("Result:", sum)
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>func compute() {
	sum := 0
	for i := 0; i &lt; 10; i++ {
		sum += i
	}
	fmt.Println("Result:", sum)
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove the unused function parameter(s) "{}".</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_duplicated_by_s903">is duplicated by: S903</h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s904">is related to: S904</h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s1763">is related to: S1763</h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s901">is related to: S901</h3>

</div>
<div class="sect2">
<h3 id="_on_31_jul_2013_151111_freddy_mallet_wrote">on 31 Jul 2013, 15:11:11 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARJAVA-261</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_31_jul_2013_233140_ann_campbell_wrote">on 31 Jul 2013, 23:31:40 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Will this rule be clever enough to skip methods that are overrides, but aren&#8217;t marked as such?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_1_aug_2013_132248_freddy_mallet_wrote">on 1 Aug 2013, 13:22:48 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>This is the goal Ann otherwise this rule will generate too many false-positives</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_2_dec_2013_165134_freddy_mallet_wrote">on 2 Dec 2013, 16:51:34 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by https://jira.codehaus.org/browse/SONARPLUGINS-3303 for Flex</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_19_feb_2014_150700_freddy_mallet_wrote">on 19 Feb 2014, 15:07:00 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3493 for PHP</p>
</div>
</div>
</div>
</div>