<div class="paragraph">
<p>If an enumerator list is given with no explicit initialization of members, then C/C&#43;&#43; allocates a sequence of integers starting at zero for the first element and increasing by one for each subsequent element.</p>
</div>
<div class="paragraph">
<p>An explicit initialization of the first element, as permitted by this rule, forces the allocation of integers to start at the given value. When adopting this approach it is essential to ensure that the initialization value used is small enough that no subsequent value in the list will exceed the <code>int</code> storage used by enumeration constants.</p>
</div>
<div class="paragraph">
<p>Explicit initialization of all items in the list, which is also permissible, prevents the mixing of automatic and manual allocation, which is error prone.</p>
</div>
<div class="paragraph">
<p>However, it is then the responsibility of the developer to ensure that all values are in the required range, and that values are not unintentionally duplicated.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>enum color { red = 3, blue, green, yellow = 5 }; // Noncompliant; both green and yellow = 5</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>enum color { red = 3, blue = 4, green = 5, yellow = 5 }; // Compliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C:2004, 9.3 - In an enumerator list, the "=" construct shall not be used to explicitly initialize members other than the first, unless all items are explicitly initialized.</p>
</li>
<li>
<p>MISRA C&#43;&#43;:2008, 8-5-3 - In an enumerator list, the = construct shall not be used to explicitly initialize members other than the first, unless all items are explicitly initialized.</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_20_mar_2015_213617_evgeny_mandrikov_wrote">on 20 Mar 2015, 21:36:17 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~massimo.paladin] could you please complete this RSPEC, so that it will match implementation?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_25_mar_2015_093733_massimo_paladin_wrote">on 25 Mar 2015, 09:37:33 Massimo PALADIN wrote:</h3>
<div class="paragraph">
<p>\[~evgeny.mandrikov] done.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_20_jul_2015_115623_ann_campbell_wrote">on 20 Jul 2015, 11:56:23 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>double-check my title update, please [~evgeny.mandrikov]</p>
</div>
</div>
</div>
</div>