<div class="paragraph">
<p>The writer of an annotation can set one of three retention policies for it:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>RetentionPolicy.SOURCE</code> - these annotations are dropped during compilation, E.G. <code>@Override</code>, <code>@SuppressWarnings</code>.</p>
</li>
<li>
<p><code>RetentionPolicy.CLASS</code> - these annotations are present in a compiled class but not loaded into the JVM at runtime. This is the default.</p>
</li>
<li>
<p><code>RetentionPolicy.RUNTIME</code> - these annotations are present in the class file and loaded into the JVM.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Only annotations that have been given a <code>RUNTIME</code> retention policy will be available to reflection. Testing for annotations with any other retention policy is simply an error, since the test will always return false.</p>
</div>
<div class="paragraph">
<p>This rule checks that reflection is not used to detect annotations that do not have <code>RUNTIME</code> retention.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>Method m = String.class.getMethod("getBytes", new Class[] {int.class,
int.class, byte[].class, int.class});
if (m.isAnnotationPresent(Override.class)) {  // Noncompliant; test will always return false, even when @Override is present in the code</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>"@xxx" is not available at runtime and cannot be seen with reflection.</p>
</div>
</div>
</div>
</div>