<div class="paragraph">
<p>Calling <code>GC.Collect</code> is rarely necessary, and can significantly affect application performance. That&#8217;s because it triggers a blocking operation that examines <em>every object in memory</em> for cleanup. Further, you don&#8217;t have control over when this blocking cleanup will actually run.</p>
</div>
<div class="paragraph">
<p>As a general rule, the consequences of calling this method far outweigh the benefits unless perhaps you&#8217;ve just triggered some event that is unique in the run of your program that caused a lot of long-lived objects to die.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when <code>GC.Collect</code> is invoked.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>static void Main(string[] args)
{
  // ...
  GC.Collect(2, GCCollectionMode.Optimized); // Noncompliant
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Don&#8217;t try to be smarter than the JVM, remove this call to run the garbage collector.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_20_aug_2013_152400_freddy_mallet_wrote">on 20 Aug 2013, 15:24:00 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARJAVA-307</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_21_aug_2013_150412_dinesh_bolkensteyn_wrote">on 21 Aug 2013, 15:04:12 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>Nice description!</p>
</div>
</div>
</div>
</div>