<div class="paragraph">
<p>The point of calling <code>notify</code>, or more prefereably <code>notifyAll</code>, is to alert other threads that the state of <code>this</code> object has changed. Doing so without a state change seems pointless, and may indicate a logic error.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public void run(){
  synchronized(this){
    int total = 0;
    for(int i=0; i&lt;100 ; i++){
      total += i;
    }
    notify();  // Noncompliant; only local variables have been modified. What's the point?
  }
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>No heap object visible to multiple threads has been modified.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_11_mar_2015_152233_ann_campbell_wrote">on 11 Mar 2015, 15:22:33 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Per our discussion [~nicolas.peru], I have specified this rule loosely with the understanding that we&#8217;ll clarify &amp; potentially tighten at implementation time.</p>
</div>
</div>
</div>
</div>