<div class="paragraph">
<p>Having a single declaration of a type, object or function allows the compiler to detect incompatible types for the same entity.</p>
</div>
<div class="paragraph">
<p>Normally, this will mean declaring an external identifier in a header file that will be included in any file where the identifier is defined or used.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>// header.hpp
extern int16_t a;

// file1.cpp
#include "header.hpp"

extern int16_t b;

// file2.cpp
#include "header.hpp"
extern int32_t b; // Noncompliant, compiler may not detect the error
int32_t a; // Compliant, compiler will detect the error</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>// header.hpp
extern int16_t a; // Compliant, declared once in a header
extern int32_t b; // Compliant, declared once in a header

// file1.cpp
#include "header.hpp"

// file2.cpp
#include "header.hpp"</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C&#43;&#43;:2008, 3-2-3</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Move the declaration of "xxx" to a header file.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s826">relates to: S826</h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s828">relates to: S828</h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s829">relates to: S829</h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s831">relates to: S831</h3>

</div>
</div>
</div>