<div class="paragraph">
<p>Paragraphs, sections and statements must be correctly indented for better code readability.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>       IDENTIFICATION DIVISION.
       PROGRAM-ID. foo.

       PROCEDURE DIVISION.
           IF "foo" = "bar" THEN
           DISPLAY "foo = bar!"      *&gt; Noncompliant
           ELSE
           DISPLAY "foo &lt;&gt; bar!".      *&gt; Noncompliant
       END PROGRAM foo.</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>       IDENTIFICATION DIVISION.
       PROGRAM-ID. foo.

       PROCEDURE DIVISION.
           IF "foo" = "bar" THEN
              DISPLAY "foo = bar!"
           ELSE
              DISPLAY "foo &lt;&gt; bar!".
       END PROGRAM foo.</pre>
</div>
</div>
</div>
</div>