<div class="paragraph">
<p>When rethrowing an exception, you should do it by simply calling <code>throw;</code> and not <code>throw exc;</code>, because the stack trace is reset with the second syntax, making debugging a lot harder.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>try
{}
catch(ExceptionType1 exc)
{
  Console.WriteLine(exc);
  throw exc; // Noncompliant; stacktrace is reset
}
catch (ExceptionType2 exc)
{
  throw new Exception("My custom message", exc);  // Compliant; stack trace preserved
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>try
{}
catch(ExceptionType1 exc)
{
  Console.WriteLine(exc);
  throw;
}
catch (ExceptionType2 exc)
{
  throw new Exception("My custom message", exc);
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use "throw;" instead of "throw xxx;" to preserve the stack trace.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The whole "throw exc;" statement</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_8_dec_2015_091418_tamas_vajk_wrote">on 8 Dec 2015, 09:14:18 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] LGTM</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_26_jul_2016_175015_tamas_vajk_wrote">on 26 Jul 2016, 17:50:15 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I don&#8217;t think this is a bug.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_27_jul_2016_133000_freddy_mallet_wrote">on 27 Jul 2016, 13:30:00 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Agreed [~tamas.vajk] and this is the limit of the usage of this "Bug" issue type because this is really a "Reliability" issue that might prevent someone in production to understand an unexpected behavior as the root cause of the issue is lost. That&#8217;s why we&#8217;re classifying this issue as a bug. cc [~ann.campbell.2]</p>
</div>
</div>
</div>
</div>