<div class="paragraph">
<p>A serialVersionUID field is required in a <code>Serializable</code> class. In a non-<code>Serializable</code>, it&#8217;s just confusing.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public class MyClass {
  private static final long serialVersionUID = -1L;  // Noncompliant
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this serialVersionUID.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_12_jun_2015_151642_ann_campbell_wrote">on 12 Jun 2015, 15:16:42 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>CodePro: Non-serializable Class Declares serialVersionUID</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_15_jun_2015_202146_nicolas_peru_wrote">on 15 Jun 2015, 20:21:46 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>Looks good</p>
</div>
</div>
</div>
</div>