<div class="paragraph">
<p>The <code>await</code> keyword can only be used on <a href="https://docs.python.org/3/library/asyncio-task.html#awaitables">"Awaitable" objects</a>. Python has three types of awaitables: Future, Task and Coroutines. Calling await on any other object will raise a <code>TypeError</code>.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>import asyncio

def myfunction():
    print("myfunction")

async def otherfunction():
    await myfunction()  # Noncompliant. myfunction is not marked as "async"

asyncio.run(otherfunction())</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>import asyncio

async def myfunction():
    print("myfunction")

async def otherfunction():
    await myfunction()

asyncio.run(otherfunction())</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.python.org/3/library/asyncio-task.html">Python documentation - Coroutines and Tasks</a></p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>change this statement so that it awaits on a Future, a Coroutine or a Task</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Primary: the await keyword</p>
</div>
</div>
</div>
</div>