<div class="paragraph">
<p>A method or class with too many type parameters has likely aggregated too many responsibilities and should be split.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="paragraph">
<p>With the default parameter value of 2:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>&lt;S, T, U, V&gt; void foo() {} // Noncompliant; not really readable
&lt;String, Integer, Object, String&gt;foo(); // especially on invocations</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Reduce the number of type parameters in this [method|class] to no more than the n authorized.</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="ulist">
<ul>
<li>
<p>key = max
description: Maximum authorized number of parameters.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>default: 2</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_20_jan_2015_154140_ann_campbell_wrote">on 20 Jan 2015, 15:41:40 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~nicolas.peru], you&#8217;re going to have to help me out with the whys and wherefores on this one.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_20_jan_2015_160001_freddy_mallet_wrote">on 20 Jan 2015, 16:00:01 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>This rule seems to duplicate RSPEC-107 ???</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_21_jan_2015_173938_nicolas_peru_wrote">on 21 Jan 2015, 17:39:38 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>\[~freddy.mallet] This is about Type parameters not parameters.</p>
</div>
<div class="paragraph">
<p>\[<sub>ann.campbell.2][</sub>freddy.mallet] For me this is the same reasons as for RSPEC-107 : if you have too many type parameters then your class/method is probably doing too many things.</p>
</div>
<div class="paragraph">
<p>Another reason to avoid too many type params is readability :</p>
</div>
<div class="listingblock">
<div class="content">
<pre>&lt;S, T, U, V&gt; void foo() {} //not really readable
&lt;String, Integer, Object, String&gt;foo(); // especially on invocations</pre>
</div>
</div>
</div>
</div>
</div>