<div class="paragraph">
<p>Memory that is allocated with <code>new T[n]</code> <em>must</em> be freed with <code>delete[]</code>. Leave out the <code>[]</code>, and the likely result is heap corruption or, as a best-case scenario, premature program termination.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>char *cp = new char[10];
// ...
delete cp;  // Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>char *cp = new char[10];
// ...
delete[] cp;</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_duplicates_s1232">duplicates: S1232</h3>

</div>
<div class="sect2">
<h3 id="_on_10_mar_2016_222947_evgeny_mandrikov_wrote">on 10 Mar 2016, 22:29:47 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>Looks like duplicate of RSPEC-1232</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_11_mar_2016_083155_ann_campbell_wrote">on 11 Mar 2016, 08:31:55 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Thanks [~evgeny.mandrikov], you&#8217;re right.</p>
</div>
</div>
</div>
</div>