<div class="paragraph">
<p>The PSR2 standard recommends listing modifiers in the following order to improve the readability of PHP source code:</p>
</div>
<div class="olist arabic">
<ol class="arabic">
<li>
<p>final or abstract</p>
</li>
<li>
<p>public or protected or private</p>
</li>
<li>
<p>static</p>
</li>
</ol>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>static protected $foo;
...
public static final function bar(){...}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>protected static $foo;
...
final public static function bar(){...}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Reorder the modifiers to comply with the [Java Language Specification|PSR2 standard].</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_duplicated_by_s1789">is duplicated by: S1789</h3>

</div>
</div>
</div>