<div class="paragraph">
<p>Shared coding conventions allow teams to collaborate efficiently. This rule checks that file-code names conform to a specified regular expression.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Given a pattern of <code>FC-.*</code>:</p>
</div>
<div class="listingblock">
<div class="content">
<pre> INPUT-OUTPUT SECTION.
 FILE-CONTROL.

     SELECT Y27MVTS       ASSIGN  TO     S1        &gt;Noncompliant; S1 doesn't match "FC-.*" pattern
                          FILE STATUS IS S1.</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre> INPUT-OUTPUT SECTION.
 FILE-CONTROL.

     SELECT Y27MVTS     ASSIGN  TO     FC-S1     &gt;OK as FC-S1 matches "FC-.*" pattern
                          FILE STATUS IS FS-S1.</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="ulist">
<ul>
<li>
<p>key = regexPatternString</p>
</li>
<li>
<p>description: A regular expression to specify the File-Code naming convention</p>
</li>
</ul>
</div>
</div>
</div>
</div>