<div class="paragraph">
<p>Control characters aren&#8217;t visible to maintainers, so they should be escaped.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>const char* foo = "A	B";  // Noncompliant, contains a tabulation</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>const char* foo = "A\tB";  // Compliant, use escaped value</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_exceptions">Exceptions</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Raw string literals has no escape character mechanism.</p>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace this character by its escaped value "xxx".</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>control character</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_1_nov_2018_095615_jean_baptiste_lievremont_wrote">on 1 Nov 2018, 09:56:15 Jean-Baptiste Lievremont wrote:</h3>
<div class="paragraph">
<p>Could you please provide a Java-specific description?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_14_nov_2018_115604_alexandre_gigleux_wrote">on 14 Nov 2018, 11:56:04 Alexandre Gigleux wrote:</h3>
<div class="paragraph">
<p>\[~jeanbaptiste.lievremont] Done (/)</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_3_oct_2019_110015_alban_auzeill_wrote">on 3 Oct 2019, 11:00:15 Alban Auzeill wrote:</h3>
<div class="paragraph">
<p>The Unicode characters set <a href="https://en.wikipedia.org/wiki/C0_and_C1_control_codes">C1</a> (from \u0080 to \u009F) is considered by some languages as control characters (e.g.: C# <a href="https://docs.microsoft.com/en-us/dotnet/api/system.char.iscontrol?view=netframework-4.8">Char.IsControl(Char)</a> ). But it&#8217;s not true for all encodings. For example, Windows-1250 code page encodes euro sign (€ unicode \u20AC) by using \u0080, and it&#8217;s not a control character. So if a source code is encoded using UTF-8, we can not guess what means characters between \u0080-\u00FF because it depends of the original code page used to write characters in this range before converting it to UTF-8.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_20_may_2020_111136_damien_urruty_wrote">on 20 May 2020, 11:11:36 Damien Urruty wrote:</h3>
<div class="paragraph">
<p>I think this rule could be extended (or maybe another one) to also check character literals. Examples in Java:</p>
</div>
<div class="literalblock">
<div class="content">
<pre>String tabInside = "A B" + ' ';</pre>
</div>
</div>
<div class="paragraph">
<p>String zeroWidthSpaceInside = "foo​bar" + '​';</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_26_may_2020_151129_alban_auzeill_wrote">on 26 May 2020, 15:11:29 Alban Auzeill wrote:</h3>
<div class="paragraph">
<p>\[~damien.urruty] I agree and I created [SONARJAVA-3433]</p>
</div>
</div>
</div>
</div>