<div class="paragraph">
<p>Taking the address of an object of incomplete type, where the complete type contains a user declared <code>operator &amp;</code> leads to undefined behavior.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>// A.h
class A
{
public:
  A * operator &amp; ( ); // Noncompliant
};

// f1.cc
class A;
void f ( A &amp; a )
{
  &amp;a; // uses built-in operator &amp;
}

// f2.cc
#include "A.h"
void f2 ( A &amp; a )
{
  &amp;a; // use user-defined operator &amp;
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C&#43;&#43; 2008, 5-3-3 - The unary &amp; operator shall not be overloaded.</p>
</li>
<li>
<p><a href="https://github.com/isocpp/CppCoreGuidelines/blob/036324/CppCoreGuidelines.md#Ro-address-of">C&#43;&#43; Core Guidelines C.166</a> - Overload unary <code>&amp;</code> only as part of a system of smart pointers and references</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_31_mar_2015_190452_evgeny_mandrikov_wrote">on 31 Mar 2015, 19:04:52 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] implementation seems more complete (SQALE, description) than this spec.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_13_apr_2015_193309_evgeny_mandrikov_wrote">on 13 Apr 2015, 19:33:09 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I&#8217;m wondering why blocker, but not active by default? Note that in implementation currently major and active.</p>
</div>
</div>
</div>
</div>