<div class="paragraph">
<p><code>private</code> classes that are never used are dead code: unnecessary, inoperative code that should be removed. Cleaning out dead code decreases the size of the maintained codebase, making it easier to understand the program and preventing bugs from being introduced.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public class Foo
{
  ...
  private class MyUnusedPrivateClass {...} // Noncompliant
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this unused private "XXXX" class.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The name of the unused private class</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s1068">relates to: S1068</h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s1144">relates to: S1144</h3>

</div>
<div class="sect2">
<h3 id="_on_21_jul_2017_130722_amaury_levé_wrote">on 21 Jul 2017, 13:07:22 Amaury Levé wrote:</h3>
<div class="paragraph">
<p>Covered by S1144 for C#</p>
</div>
</div>
</div>
</div>