<div class="paragraph">
<p>Type hints can be used to communicate the intended type of a given variable. These are not enforced at runtime and not respecting them might not necessarily lead to runtime errors.</p>
</div>
<div class="paragraph">
<p>It is however confusing and could lead to maintainability issues.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>def my_function():
    my_int: int = "string"  # Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>def my_function():
    my_str: str = "string"</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Primary:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Assign to "XXX" a value of type "YYY" instead of "ZZZ" or update the type hint of "XXX".
Secondary:</p>
</li>
<li>
<p>Type hint.</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>Primary: The assigned value</p>
</li>
<li>
<p>Secondary: The type hint</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_12_may_2021_004518_mike_heyns_wrote">on 12 May 2021, 00:45:18 Mike Heyns wrote:</h3>
<div class="paragraph">
<p>This rule does not appear to be working for <a href="https://docs.python.org/3/library/dataclasses.html#init-only-variables">Dataclass Init-Only variables</a>:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>@dataclass
class Book:
    name: str
    condition: InitVar[str] = ''</pre>
</div>
</div>
<div class="paragraph">
<p><em><em>_Assign to "condition" a value of type "InitVar[str]" instead of "str" or update its type hint.</em>_</em></p>
</div>
<div class="paragraph">
<p>In fact, most examples including the official documentation use <code>None</code> as the default for init-only arguments. Attempting to use <code>InitVar[Optional[str]]</code> has the same effect as above.</p>
</div>
</div>
</div>
</div>