<div class="paragraph">
<p>Class members that are not assigned a default value and are not initialized in a constructor will be set to null by the compiler. Even if code exists to properly set those members, there is a risk that they will be dereferenced before it is called, resulting in a <code>NullPointerException</code>.</p>
</div>
<div class="paragraph">
<p>Because you cannot guarantee that such classes will always be used properly, class members should always be initialized.</p>
</div>
<div class="paragraph">
<p>This rule flags members which have no default value and which are left uninitialized by at least one class constructor, but which are unconditionally dereferenced somewhere in the code.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public class Team {

  int limit = 30;
  List&lt;Player&gt; roster;  // Noncompliant; no default &amp; not initialized by constructor
  Person coach;

  public Team (Person coach) { // roster is left uninitialized
    this.coach = coach;
  }

  public void add(Player p) {
    if (roster == null) {
      roster = new ArrayList&lt;Player&gt;();
    }
    roster.add(p);
  }

  public boolean isFull() {  // NPE if called before add()
    return roster.size() &lt; limit;
  }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public class Team {

  int limit = 30;
  List&lt;Player&gt; roster = new ArrayList&lt;Player&gt;();
  Person coach;

  public Team (Person coach) {
    this.coach = coach;
  }

  public void add(Player p) {
    roster.add(p);
  }

  // ...</pre>
</div>
</div>
<div class="paragraph">
<p>or</p>
</div>
<div class="listingblock">
<div class="content">
<pre>public class Team {

  int limit = 30;
  List&lt;Player&gt; roster;
  Person coach;

  public Team (Person coach) {
    this.coach = coach;
    this.roster = new ArrayList&lt;Player&gt;();
  }

  public void add(Player p) {
    roster.add(p);
  }

  // ...</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>This field has no default and is not initialized in any constructors.</p>
</li>
<li>
<p>This field has no default and is not initialized in n constructors.</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s1258">relates to: S1258</h3>

</div>
</div>
</div>