<div class="paragraph">
<p>PHP doesn&#8217;t fail if two <code>define</code> statements assign a value to the same constant name: in such a case, PHP only issues a warning and ignores the second <code>define</code>. However, that code should be fixed. In the best case, the second <code>define</code> tries to assign the same value to the constant: it&#8217;s just useless code which can be removed. In the worst case, the second <code>define</code> attempts to assign a different value and is followed by code which wrongly assumes the value of the constant: it can be a nasty bug.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>define( 'WP_DEBUG', 1 );
define( 'SCRIPT_DEBUG', 1 );
define( 'WP_DEBUG', 0 ); // Noncompliant, redefines constant defined 2 lines above
echo WP_DEBUG; // output "1"</pre>
</div>
</div>
</div>
</div>