<div class="paragraph">
<p>Any insert which omits a value for a <code>NOT NULL</code> column in a database table will be automatically rejected by the database unless a default value has been specified for the column.</p>
</div>
<div class="paragraph">
<p><strong>Noteworthy</strong></p>
</div>
<div class="paragraph">
<p>This rule raises issues only when a <strong>Database Catalog</strong> is provided during the analysis. See https://docs.sonarqube.org/latest/analysis/languages/cobol/</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="paragraph">
<p>With the table <code>MY_TABLE</code> having a <code>NOT NULL</code> column <code>N2</code> without default value and a <code>NOT NULL</code> column <code>N3</code> with default value:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>EXEC SQL
  INSERT INTO MY_TABLE  *&gt; Noncompliant; N2 value omitted
  (
    N1
  )
  VALUES
  (
    :ITQ1-NUMBER,
  )
END-EXEC.</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>EXEC SQL
  INSERT INTO MY_TABLE  *&gt; Compliant even though N3 value not supplied
  (
    N1,
    N2
  )
  VALUES
  (
    :ITQ1-NUMBER,
    :ITQ2-NUMBER,
  )
END-EXEC.</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Include a value for "xxx" in this insert.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>table name</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s3484">is related to: S3484</h3>

</div>
</div>
</div>