<div class="paragraph">
<p>Looking at the set of methods and fields in a <code>class</code> and finding two that differ only by capitalization is confusing to users of the class.</p>
</div>
<div class="paragraph">
<p>This situation may simply indicate poor naming. Method names should be action-oriented, and thus contain a verb, which is unlikely in the case where both a method and a field have the same name (with or without capitalization differences). However, renaming a public method could be disruptive to callers. Therefore renaming the member is the recommended action.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class SomeClass:
    lookUp = false
    def lookup():       # Non-compliant; method name differs from field name only by capitalization
        pass</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class SomeClass:
    lookUp = false
    def getLookUp():
        pass</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Rename [method|field] "XXXX" to prevent any misunderstanding/clash with [method|field] "YYYY" defined on line n.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_deprecates_s1701">deprecates: S1701</h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s4025">relates to: S4025</h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s1221">relates to: S1221</h3>

</div>
<div class="sect2">
<h3 id="_on_11_mar_2015_142322_ann_campbell_wrote">on 11 Mar 2015, 14:23:22 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Please double-check me &amp; make sure I didn&#8217;t miss any cases [~elena.vilchik]</p>
</div>
</div>
</div>
</div>