<div class="paragraph">
<p><code>UPDATE</code> and <code>DELETE</code> statements should contain <code>WHERE</code> clauses to keep the modification of records under control. Otherwise unexpected data loss could result.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>DELETE FROM countries
UPDATE employee SET status = 'retired' FROM table1 AS employee</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>TRUNCATE TABLE countries
DELETE FROM countries WHERE CODE = @country_code
UPDATE employee SET status = 'retired' FROM table1 AS employee WHERE age &gt; @maxAge</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_exceptions">Exceptions</h2>
<div class="sectionbody">
<div class="paragraph">
<p>No issue is reported for statements made on temporary tables.</p>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add a "WHERE" clause to this "xxx" statement.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_deprecates_s2987">deprecates: S2987</h3>

</div>
<div class="sect2">
<h3 id="_deprecates_s2988">deprecates: S2988</h3>

</div>
</div>
</div>