<div class="paragraph">
<p>Never use <code>with</code> statements, since they decrease readability. When you do not specify a variable&#8217;s scope, you do not always know where you are setting properties, so your code can be confusing.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>with (foo) { // Noncompliant
  return x;  // is it a property of foo or local variable ?
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>return foo.x;</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this use of "with".</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_5_jan_2015_184302_ann_campbell_wrote">on 5 Jan 2015, 18:43:02 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Note that this subtask variation of SQALE is an exception to the rule - both JavaScript and Flex plugins had implemented the rule with the same legacy key, but with different SQALE settings.</p>
</div>
</div>
</div>
</div>