<div class="paragraph">
<p>When a closure contains only a <code>return</code> statement, the <code>return</code> itself can be omitted.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>someList.sort { a, b in
  return a &gt; b
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>someList.sort { a, b in a &gt; b }</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove the "return" keyword.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_20_may_2015_181345_ann_campbell_wrote">on 20 May 2015, 18:13:45 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~elena.vilchik] I suspect a standard "for greater readability" argument is expected here, but it&#8217;s hard for me to make it. Feel free to add it yourself. :-)</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_21_may_2015_071319_elena_vilchik_wrote">on 21 May 2015, 07:13:19 Elena Vilchik wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] what do you mean "argument"? Anyway I looks good to me now :)</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_3_feb_2016_171153_ann_campbell_wrote">on 3 Feb 2016, 17:11:53 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~elena.vilchik] I just now saw your response (doh!). In this case "argument" == "reason".</p>
</div>
</div>
</div>
</div>