<div class="paragraph">
<p>Sharing some naming conventions is a key point to make it possible for a team to efficiently collaborate.</p>
</div>
<div class="paragraph">
<p>This rule allows to check that all class names (along with Objective-C interface, protocol and implementation names) match a provided regular expression.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="paragraph">
<p>With default provided regular expression <code>^[A-Z][a-zA-Z0-9]*$</code>:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>// C++
class foo // Noncompliant
{
};

// Objective-C
@interface nonCompliant : NSObject
@end</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>// C++
class Foo // Compliant
{
};

// Objective-C
@interface Compliant : NSObject
@end</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Rename class "XXXX" to match the regular expression ${format}.</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="ulist">
<ul>
<li>
<p>key = format</p>
</li>
<li>
<p>description = Regular expression used to check the class names against.</p>
</li>
<li>
<p>default = ^[A-Z][a-zA-Z0-9]*$</p>
</li>
<li>
<p>type = STRING</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_20_may_2013_121647_fabrice_bellingard_wrote">on 20 May 2013, 12:16:47 Fabrice Bellingard wrote:</h3>
<div class="paragraph">
<p>Implementation: http://jira.codehaus.org/browse/SONARJAVA-119</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_31_oct_2013_160900_freddy_mallet_wrote">on 31 Oct 2013, 16:09:00 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3228 for Flex</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_19_feb_2014_154602_freddy_mallet_wrote">on 19 Feb 2014, 15:46:02 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3495 for PHP</p>
</div>
</div>
</div>
</div>