<div class="paragraph">
<p>Shared naming conventions allow teams to collaborate efficiently. This rule checks whether or not type names are using PascalCase. To reduce noise, two consecutive upper case characters are allowed unless they form the whole type name. So, <code>MyXClass</code> is compliant, but <code>XC</code> on its own is not.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class my_class {...}
class SOMEName42 {...}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class MyClass {...}
class SomeName42 {...}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_exceptions">Exceptions</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>The rule ignores types that are marked with <code>ComImportAttribute</code> or <code>InterfaceTypeAttribute</code>.</p>
</li>
<li>
<p>The rule allows for two-letter acronyms in which both letters are capitalized, as shown in the following identifier: <code>IOStream</code></p>
</li>
<li>
<p>The rule allows having <code>'_'</code> characters in class names inside test projects.</p>
</li>
</ul>
</div>
<div class="listingblock">
<div class="content">
<pre>class Some_Name___42 {...} // valid in test
class Some_name___42 {...} // still not valid
class Some_Name_XC {...} // invalid because of XC, should be Some_Name_Xc</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/capitalization-conventions">Microsoft Capitalization Conventions</a></p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>Rename [class|interface|struct] "XXX" to match PascalCase naming rules, consider using "Xxx".</p>
</li>
<li>
<p>Rename [class|interface|struct] "XXX" to match PascalCase naming rules, trim underscores from the name.</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_20_may_2013_121647_fabrice_bellingard_wrote">on 20 May 2013, 12:16:47 Fabrice Bellingard wrote:</h3>
<div class="paragraph">
<p>Implementation: http://jira.codehaus.org/browse/SONARJAVA-119</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_31_oct_2013_160900_freddy_mallet_wrote">on 31 Oct 2013, 16:09:00 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3228 for Flex</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_19_feb_2014_154602_freddy_mallet_wrote">on 19 Feb 2014, 15:46:02 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3495 for PHP</p>
</div>
</div>
</div>
</div>