<div class="paragraph">
<p>The use of a <code>StringBuilder</code> or <code>StringBuffer</code> is supposed to make <code>String</code> assembly more efficient than plain concatenation. So don&#8217;t ruin the effect by concatenating the arguments to <code>append</code>.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>StringBuilder sb = new StringBuilder();
sb.append("foo is: " + getFoo()); // Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>StringBuilder sb = new StringBuilder();
sb.append("foo is: ").append(getFoo());</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>"append" each concatenated value separately.</p>
</div>
</div>
</div>
</div>