<div class="paragraph">
<p>An inherited member function can be hidden in a derived class and that creates a class that behaves differently depending on which interface is used to manipulate it.</p>
</div>
<div class="paragraph">
<p>Overriding happens when the inherited method is virtual and a method declared in the derived class uses the same identifier as well as the same signature (the return types can be different, as long as they are covariant). However, if the inherited method is non-virtual or if the two declarations of the method do not share the same signature, the method of the base class will be hidden.</p>
</div>
<div class="paragraph">
<p>Such a class increases the inheritance complexity, and confuses consumers with its non-polymorphic behavior, which can lead to errors.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class Base {
public:
  void shutdown();
  virtual void log(int a);
};

class Derived : public Base {
public:
  void shutdown(); //Noncompliant
  void log(float a); //Noncompliant
};

void stopServer(Base *obj, Derived *obj2) {
  obj-&gt;shutdown(); // always calls Base::shutdown even if the given object's type is Derived
  obj-&gt;log(2); // calls Base::log(int) even if the given object's type is Derived
  obj2-&gt;shutdown(); // calls Derived::shutdown
  obj2-&gt;log(2); // calls Derived::log(float), even if this requires a conversion int-&gt;float.
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class Base {
public:
  void shutdown();
  virtual void log(int a);
};

class Derived : public Base {
public:
  void shutdownAndUpdate(); // Define a method with a different name
  void log(int a) override; // Or make the method a proper override
};

void stopServer(Base *obj) {
  obj-&gt;shutdown(); // calls Base::shutdown and there is no confusion
  obj-&gt;log(2); // calls Derived::log(int) if the given object's type is Derived
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://github.com/isocpp/CppCoreGuidelines/blob/036324/CppCoreGuidelines.md#c138-create-an-overload-set-for-a-derived-class-and-its-bases-with-using">C&#43;&#43; Core Guidelines C.138</a> - Create an overload set for a derived class and its bases with using</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Rename this member function so that it doesn&#8217;t hide an inherited non-virtual function, or make it virtual in the base class "XXX".</p>
</div>
<div class="paragraph">
<p>"XXX" hides overloaded virtual functions</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Function name</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_2_may_2016_163204_ann_campbell_wrote">on 2 May 2016, 16:32:04 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>I&#8217;ve made some small edits [~alban.auzeill], but basically this looks good.</p>
</div>
</div>
</div>
</div>