<div class="paragraph">
<p>JSP expressions (using <code>&lt;%= ... %&gt;</code>) have been deprecated because they:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Are not unit testable.</p>
</li>
<li>
<p>Are not reusable.</p>
</li>
<li>
<p>Cannot make use of object oriented concepts such as inheritence.</p>
</li>
<li>
<p>Have poor error handling capabilities: if an exception is thrown, an empty page is rended.</p>
</li>
<li>
<p>Mix the business and presentation logic.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>JSP Standard Tag Library (JSTL) and Expression Language should be used instead, enabiling the adoption of the model-view-controller (MVC) design pattern which reduces the coupling between the presentation tier and the business logic.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>&lt;input type="text" name="foo" value="&lt;%= request.getParameter("foo") %&gt;" /&gt;</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>&lt;input type="text" name="foo" value="${fn:escapeXml(param.foo)}" /&gt;</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace this JSP expression using tag libraries and expression language.</p>
</div>
</div>
</div>
</div>