<div class="paragraph">
<p>When verifying that code raises an exception, a good practice is to avoid having multiple method calls inside the tested code, to be explicit about what is exactly tested.</p>
</div>
<div class="paragraph">
<p>When two of the methods can raise the same exception, not respecting this good practice is a bug, since it is not possible to know what is really tested.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public function testSomething()
{
  try {
    g(y(1)); // Noncompliant
    $this-&gt;fail('RuntimeException is not thrown');
  } catch (RuntimeException $e) {}
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public function testSomething()
{
  $y = y(1);
  try {
    g($y);
    $this-&gt;fail('RuntimeException is not thrown by g()');
  } catch (RuntimeException $e) {}
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Refactor {the body of this try/catch|the code of this assertThrows} to have only one invocation throwing an exception.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>Primary: assertThrows/try keyword</p>
</li>
<li>
<p>Secondaries: Methods calls</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s5778">relates to: S5778</h3>

</div>
</div>
</div>