<div class="paragraph">
<p>The <code>/EJECT</code> compiler directive adds a page break when the code is printed, and should be used at the end of each <code>F</code>, <code>D</code>, and <code>C</code> specification section to make the structure of the code clearer and to enhance overall readability.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>F* Conversion Job Activity File (CV)
F*
FCVJBAVP   IF   E           K DISK
F* Conversion Job Dependency File (CZ)
F*
FCVJBDPP   IF   E           K DISK
F*
DP0RTCD           S              1
DP0JBNM           S             10A
DP0ERMG           S            100A
DK0JBNM           S                   LIKE(CVJBNM)
DK1JBNM           S                   LIKE(CVJBNM)
DJST              S             10A   DIM(10)
DI                S              3  0 INZ
C
C                   EXSR      SR999
C
C                   EXSR      SR100</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>F* Conversion Job Activity File (CV)
F*
FCVJBAVP   IF   E           K DISK
F* Conversion Job Dependency File (CZ)
F*
FCVJBDPP   IF   E           K DISK
F*
F/EJECT
DP0RTCD           S              1
DP0JBNM           S             10A
DP0ERMG           S            100A
DK0JBNM           S                   LIKE(CVJBNM)
DK1JBNM           S                   LIKE(CVJBNM)
DJST              S             10A   DIM(10)
DI                S              3  0 INZ
D/EJECT
C
C                   EXSR      SR999
C
C                   EXSR      SR100
C/EJECT</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add an "/EJECT" statement at the end of this "X" specification section</p>
</div>
</div>
</div>
</div>