<div class="paragraph">
<p>Default interceptors, such as application security interceptors, must be listed in the <code>ejb-jar.xml</code> file, or they will not be treated as default.</p>
</div>
<div class="paragraph">
<p>This rule applies to projects that contain JEE Beans (any one of <code>javax.ejb.Singleton</code>, <code>MessageDriven</code>, <code>Stateless</code> or <code>Stateful</code>).</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>// file: ejb-interceptors.xml
&lt;assembly-descriptor&gt;
 &lt;interceptor-binding&gt; &lt;!-- should be declared in ejb-jar.xml --&gt;
      &lt;ejb-name&gt;*&lt;/ejb-name&gt;
      &lt;interceptor-class&gt;com.myco.ImportantInterceptor&lt;/interceptor-class&gt; &lt;!-- Noncompliant; will NOT be treated as default --&gt;
   &lt;/interceptor-binding&gt;
&lt;/assembly-descriptor&gt;</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>// file: ejb-jar.xml
&lt;assembly-descriptor&gt;
 &lt;interceptor-binding&gt;
      &lt;ejb-name&gt;*&lt;/ejb-name&gt;
      &lt;interceptor-class&gt;com.myco.ImportantInterceptor&lt;/interceptor-class&gt;
   &lt;/interceptor-binding&gt;
&lt;/assembly-descriptor&gt;</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://owasp.org/Top10/A05_2021-Security_Misconfiguration/">OWASP Top 10 2021 Category A5</a> - Security Misconfiguration</p>
</li>
<li>
<p><a href="https://www.owasp.org/index.php/Top_10-2017_A6-Security_Misconfiguration">OWASP Top 10 2017 Category A6</a> - Security Misconfiguration</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Move this default interceptor to "ejb-jar.xml"</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_23_jul_2015_130201_ann_campbell_wrote">on 23 Jul 2015, 13:02:01 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Rule origin: https://groups.google.com/forum/#!topic/sonarqube/cYQdBhf00eo</p>
</div>
<div class="paragraph">
<p>Project is EJB if it contains JEE Beans (any one of javax.ejb.Singleton, MessageDriven, Stateless or Stateful) (@Local/@Remote interfaces are not mandatory)</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_30_jul_2015_130452_nicolas_peru_wrote">on 30 Jul 2015, 13:04:52 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>Your comment should probably be part of the rule description. Otherwise looks good.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_30_jul_2015_140139_ann_campbell_wrote">on 30 Jul 2015, 14:01:39 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>done [~nicolas.peru]</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_19_mar_2018_095411_sébastien_gioria_appsecfr_wrote">on 19 Mar 2018, 09:54:11 Sébastien GIORIA - AppSecFR wrote:</h3>
<div class="paragraph">
<p>Could be tagged  OWASP A6:2017</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_20_mar_2018_072420_freddy_mallet_wrote">on 20 Mar 2018, 07:24:20 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Done [~SPoint], thanks !</p>
</div>
</div>
</div>
</div>