<div class="paragraph">
<p>The <code>PartCreationPolicyAttribute</code> attribute, which is part of the Managed Extensibility Framework (MEF), is used to specify how the exported object will be created. Therefore it doesn&#8217;t make sense not to export this a class with this attribute using the <code>ExportAttribute</code> attribute.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when a class is marked as shared with a <code>PartCreationPolicyAttribute</code> but lacks a <code>ExportAttribute</code>.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>&lt;PartCreationPolicy(CreationPolicy.Any)&gt; ' Noncompliant
Public Class FooBar
    Inherits IFooBar
End Class</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>&lt;Export(GetType(IFooBar))&gt;
&lt;PartCreationPolicy(CreationPolicy.Any)&gt;
Public Class FooBar
    Inherits IFooBar
End Class</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add the "ExportAttribute" or remove "PartCreationPolicyAttribute" to/from this type definition.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>PartCreationPolicyAttribute attribute</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_23_jan_2018_163856_amaury_levé_wrote">on 23 Jan 2018, 16:38:56 Amaury Levé wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Could you review this draft? I put it as a <code>Bug Detection</code> as the class is never going to work as expected but I am not sure about the severity.</p>
</div>
<div class="paragraph">
<p>If everything is find could you change the completeness?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_23_jan_2018_171215_duncan_pocklington_wrote">on 23 Jan 2018, 17:12:15 Duncan Pocklington wrote:</h3>
<div class="paragraph">
<p>\[~amaury.leve] looks ok to me.</p>
</div>
<div class="paragraph">
<p>A couple of comments:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>a class can be exported using <em>Export</em> or <em>InheritedExport</em>, or a custom attribute that inherits from one of those two attributes.</p>
</li>
<li>
<p>there are other MEF attributes like <em>Import</em> and <em>ImportingConstructor</em> that don&#8217;t make sense unless they are on an exported class. Should we have those as separate rules, or make this rule more generic e.g. "MEF attribute should only be used on exported types"</p>
</li>
</ul>
</div>
</div>
</div>
</div>