<div class="paragraph">
<p><code>C&#43;&#43;17</code> introduced <code>as_const</code>: a helper function that converts a value to its corresponding const value in a succinct and more explicit way.</p>
</div>
<div class="paragraph">
<p>This is usually done to force an overloaded function call on a non-const object to resolve to the const alternative. This also can be used to instantiate a template with a const type rather than the original non-const one.</p>
</div>
<div class="paragraph">
<p>This rule detects casts that can be replaced by <code>as_const</code>.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>void fn(std::vector&lt;int&gt;&amp; vec);
void fn(const std::vector&lt;int&gt;&amp; vec);

void constFCaller() {
  std::vector&lt;int&gt; vec;
  // Set the content of vec
  // ...
  fn(static_cast&lt;const std::vector&lt;int&gt;&amp;&gt;(vec)); // Noncompliant
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>void fn(std::vector&lt;int&gt;&amp; vec);
void fn(const std::vector&lt;int&gt;&amp; vec);

void constFCaller() {
  std::vector&lt;int&gt; vec;
  // Set the content of vec
  // ...
  fn(as_const(vec)); // Compliant
}</pre>
</div>
</div>
</div>
</div>