<div class="paragraph">
<p>Unlike strongly typed languages, JavaScript does not enforce a return type on a function. This means that different paths through a function can return different types of values, which can be very confusing to the user and significantly harder to maintain.</p>
</div>
<div class="paragraph">
<p>In particular a function, in JavaScript, will return <code>undefined</code> in any of the following cases:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>It exits without a <code>return</code> statement.</p>
</li>
<li>
<p>It executes a <code>return</code> with no value.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>This rule verifies that return values are either always or never specified for each path through a function.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>function foo(a) { // Noncompliant, function exits without "return"
  if (a == 1) {
    return true;
  }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>function foo(a) {
  if (a == 1) {
    return true;
  }
  return false;
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Refactor this function to use "return" consistently.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>Primary: function name.</p>
</li>
<li>
<p>Secondary: return statements.
message should be one of</p>
<div class="ulist">
<ul>
<li>
<p>'Return with value.'</p>
</li>
<li>
<p>'Return without value.'</p>
</li>
<li>
<p>'Implicit return without value.' (when the language allows it)</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s935">relates to: S935</h3>

</div>
</div>
</div>