<div class="paragraph">
<p>If an <code>IN</code> or <code>EXISTS</code> subquery pulls data from the same table(s) involved in the outer SQL statement, there&#8217;s usually little justification for not simply moving the subquery conditions up into the main statement. Similarly subquery conditions that reference only columns from the outer query should also be "promoted". Doing so will improve readability and likely performance as well, depending on the optimizer.</p>
</div>
<div class="paragraph">
<p>In very complex situations this may not be possible or might make the SQL statement unnecessarily complicated. But in general, this should be preferred.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>SELECT name
FROM product
WHERE id IN (SELECT id FROM product WHERE discount IS NULL)  -- Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>SELECT name
FROM product
WHERE discount IS NULL</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Merge this subquery up into the main query.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p><code>[(NOT) IN | EXISTS] (subquery)</code></p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_17_jun_2016_103824_ann_campbell_wrote">on 17 Jun 2016, 10:38:24 Ann Campbell wrote:</h3>
<div class="paragraph">
<p><a href="https://www.researchgate.net/publication/222432151_Semantic_errors_in_SQL_queries_A_quite_complete_list" class="bare">https://www.researchgate.net/publication/222432151_Semantic_errors_in_SQL_queries_A_quite_complete_list</a></p>
</div>
</div>
</div>
</div>