<div class="paragraph">
<p>Running finalizers on JVM exit is disabled by default. It can be enabled with <code>System.runFinalizersOnExit</code> and <code>Runtime.runFinalizersOnExit</code>, but both methods are deprecated because they are inherently unsafe.</p>
</div>
<div class="paragraph">
<p>According to the Oracle Javadoc:</p>
</div>
<div class="quoteblock">
<blockquote>
<div class="paragraph">
<p>It may result in finalizers being called on live objects while other threads are concurrently manipulating those objects, resulting in erratic behavior or deadlock.</p>
</div>
</blockquote>
</div>
<div class="paragraph">
<p>If you really want to execute something when the virtual machine begins its shutdown sequence, you should attach a shutdown hook.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public static void main(String [] args) {
  ...
  System.runFinalizersOnExit(true);  // Noncompliant
  ...
}

protected void finalize(){
  doSomething();
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public static void main(String [] args) {
  Runtime.addShutdownHook(new Runnable() {
    public void run(){
      doSomething();
    }
  });
  //...</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://wiki.sei.cmu.edu/confluence/x/4jZGBQ">CERT, MET12-J.</a> - Do not use finalizers</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this call to "XXX.runFinalizersOnExit()".</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_25_nov_2014_164538_freddy_mallet_wrote">on 25 Nov 2014, 16:45:38 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>@Ann, according to my compliant solution I would increase the remediation cost to 20 minutes.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_16_jan_2015_094023_sébastien_gioria_wrote">on 16 Jan 2015, 09:40:23 Sébastien Gioria wrote:</h3>
<div class="paragraph">
<p>Could be tag security and cwe :</p>
</div>
<div class="paragraph">
<p>CERT Secure Coding reference : https://www.securecoding.cert.org/confluence/display/java/MET12-J.+Do+not+use+finalizers</p>
</div>
<div class="paragraph">
<p>CWE : http://cwe.mitre.org/data/definitions/586.html</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_19_jan_2015_083906_ann_campbell_wrote">on 19 Jan 2015, 08:39:06 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~sebastien.gioria], I&#8217;ve added the CERT reference, but not the one for CWE, which seems to have a pretty narrow scope.</p>
</div>
</div>
</div>
</div>