<div class="paragraph">
<p><em>Mutexes</em> are synchronization primitives that allow to manage concurrency.</p>
</div>
<div class="paragraph">
<p>Their use requires following a well-defined life-cycle.</p>
</div>
<div class="ulist">
<ul>
<li>
<p><em>Mutexes</em> need to be initialized (<code>pthread_mutex_init</code>) before being used. Once it is initialized, a <em>mutex</em> is in an <em>unlocked</em> state.</p>
</li>
<li>
<p><em>Mutexes</em> need to be destroyed (<code>pthread_mutex_destroy</code>) to free the associated internal resources. Only <em>unlocked</em> <em>mutexes</em> can be safely destroyed.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Before initialization or after destruction, a mutex is in an uninitialized state.</p>
</div>
<div class="paragraph">
<p>About this life-cycle, the following patterns should be avoided as they result in an undefined behavior:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>trying to initialize an initialized <em>mutex</em></p>
</li>
<li>
<p>trying to destroy an initialized <em>mutex</em> that is in a <em>locked</em> state</p>
</li>
<li>
<p>trying to destroy an uninitialized <em>mutex</em></p>
</li>
<li>
<p>trying to lock an uninitialized <em>mutex</em></p>
</li>
<li>
<p>trying to unlock an uninitialized <em>mutex</em></p>
</li>
</ul>
</div>
<div class="paragraph">
<p>In C&#43;&#43;, it is recommended to wrap mutex creation/destruction in a RAII class, as well as mutex lock/unlock. Those RAII classes will perform the right operations, even in presence of exceptions.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>pthread_mutex_t mtx1;

void bad1(void)
{
  pthread_mutex_init(&amp;mtx1);
  pthread_mutex_init(&amp;mtx1);
}

void bad2(void)
{
  pthread_mutex_init(&amp;mtx1);
  pthread_mutex_lock(&amp;mtx1);
  pthread_mutex_destroy(&amp;mtx1);
}

void bad3(void)
{
  pthread_mutex_init(&amp;mtx1);
  pthread_mutex_destroy(&amp;mtx1);
  pthread_mutex_destroy(&amp;mtx1);
}

void bad4(void)
{
  pthread_mutex_init(&amp;mtx1);
  pthread_mutex_destroy(&amp;mtx1);
  pthread_mutex_lock(&amp;mtx1);
}

void bad5(void)
{
  pthread_mutex_init(&amp;mtx1);
  pthread_mutex_destroy(&amp;mtx1);
  pthread_mutex_unlock(&amp;mtx1);
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>pthread_mutex_t mtx1;

void ok1(void)
{
  pthread_mutex_init(&amp;mtx1);
  pthread_mutex_destroy(&amp;mtx1);
}

void ok2(void)
{
  pthread_mutex_init(&amp;mtx1);
  pthread_mutex_lock(&amp;mtx1);
  pthread_mutex_unlock(&amp;mtx1);
  pthread_mutex_destroy(&amp;mtx1);
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://pubs.opengroup.org/onlinepubs/009695399/functions/pthread_mutex_destroy.html">The Open Group</a> pthread_mutex_init, pthread_mutex_destroy</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s5486">relates to: S5486</h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s5489">relates to: S5489</h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s5486">is related to: S5486</h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s5489">is related to: S5489</h3>

</div>
<div class="sect2">
<h3 id="_on_6_nov_2019_233311_loïc_joly_wrote">on 6 Nov 2019, 23:33:11 Loïc Joly wrote:</h3>
<div class="paragraph">
<p>\[~geoffray.adde] Can you please review my changes?</p>
</div>
</div>
</div>
</div>