<div class="paragraph">
<p>Use either the <code>// ...</code> or <code>/* ... */</code> comment syntax, but be consistent and do not mix them within the same file.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>/* Noncompliant; both comment syntaxes are used in the same file */
// Foo
/* Bar */</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>// Compliant; uniform comment syntax
// Foo
// Bar</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://wiki.sei.cmu.edu/confluence/x/0zVGBQ">CERT, MSC55-J.</a> - Use comments consistently and in a readable fashion</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use either // or /* &#8230;&#8203; */ comments, but do not mix them (see lines X and Y).</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s1103">relates to: S1103</h3>

</div>
<div class="sect2">
<h3 id="_on_15_feb_2018_150543_alban_auzeill_wrote">on 15 Feb 2018, 15:05:43 Alban Auzeill wrote:</h3>
<div class="paragraph">
<p>Irrelevant because:</p>
</div>
<div class="paragraph">
<p>ABAP has <code>*</code> and <code>"</code> comments, but for two different locations, not exchangeable.</p>
</div>
<div class="paragraph">
<p>Java, JavaScript, Flex, PHP, Swift, TypeScript have <code>/*</code> and <code>//</code>, but <code>/**</code> is used for api doc and <code>//</code> in the code.</p>
</div>
<div class="paragraph">
<p>Python has <code>#</code> and <code>"""</code> comments, but for two different usages, piece of code and docstrings.</p>
</div>
<div class="paragraph">
<p>C#, PL/SQL, T-SQL the rule [RSPEC-2544] is more appropriate.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_7_mar_2018_171720_janos_gyerik_wrote">on 7 Mar 2018, 17:17:20 Janos Gyerik wrote:</h3>
<div class="paragraph">
<p>Irrelevant for Go, there is a more specific, different recommendation for comments, explained in <a href="https://golang.org/doc/effective_go.html#commentary">Effective Go</a>, that probably deserves it&#8217;s own rule.</p>
</div>
</div>
</div>
</div>