<div class="paragraph">
<p>Looking for a given substring starting from a specified offset can be achieved by such code: <code>str.Substring(startIndex).IndexOf(char1)</code>. This works well, but it creates a new <code>string</code> for each call to the <code>Substring</code> method. When this is done in a loop, a lot of <code>strings</code> are created for nothing, which can lead to performance problems if <code>str</code> is large.</p>
</div>
<div class="paragraph">
<p>To avoid performance problems, <code>string.Substring(startIndex)</code> should not be chained with the following methods:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>IndexOf</code></p>
</li>
<li>
<p><code>IndexOfAny</code></p>
</li>
<li>
<p><code>LastIndexOf</code></p>
</li>
<li>
<p><code>LastIndexOfAny</code></p>
</li>
</ul>
</div>
<div class="paragraph">
<p>For each of these methods, another method with an additional parameter is available to specify an offset.</p>
</div>
<div class="paragraph">
<p>Using these methods gives the same result while avoiding the creation of additional <code>String</code> instances.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>str.Substring(StartIndex).IndexOf(char1); // Noncompliant; a new string is going to be created by "Substring"</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>str.IndexOf(char1, startIndex) - startIndex;</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace "XXX" with the overload that accepts an offset parameter.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>substring(beginIndex).indexOf(char1)</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_8_oct_2019_181815_sammy_baraka_wrote">on 8 Oct 2019, 18:18:15 Sammy Baraka wrote:</h3>
<div class="paragraph">
<p>I can take this issue if its still available.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_8_oct_2019_194424_ann_campbell_wrote">on 8 Oct 2019, 19:44:24 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Thanks for the offer [~sbaraka]! C# is listed in the targeted languages, so that&#8217;s still open. Before you get started though it would be best to open a new thread in the <a href="https://community.sonarsource.com">community</a></p>
</div>
</div>
</div>
</div>