<div class="paragraph">
<p>Multi-line string literals representing a big chunk of text should not be used directly in complex expressions, as it decreases the code&#8217;s readability. It is preferred to extract the string literal and store it in a variable or a field that can be accessed in the context of the complex expression.</p>
</div>
<div class="paragraph">
<p>This rule reports an issue when a multi-line string literal consists of more lines than specified in the rule&#8217;s parameter and is directly used within a lambda literal.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>listOfString
    .map { str -&gt;
        str != """
        &lt;project&gt;
          &lt;modelVersion&gt;4.0.0&lt;/modelVersion&gt;
          &lt;parent&gt;
            &lt;groupId&gt;com.mycompany.app&lt;/groupId&gt;
            &lt;artifactId&gt;my-app&lt;/artifactId&gt;
            &lt;version&gt;1&lt;/version&gt;
          &lt;/parent&gt;

          &lt;groupId&gt;com.mycompany.app&lt;/groupId&gt;
          &lt;artifactId&gt;my-module&lt;/artifactId&gt;
          &lt;version&gt;1&lt;/version&gt;
        &lt;/project&gt;
        """
    }</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>val myTextBlock = """
    &lt;project&gt;
      &lt;modelVersion&gt;4.0.0&lt;/modelVersion&gt;
      &lt;parent&gt;
        &lt;groupId&gt;com.mycompany.app&lt;/groupId&gt;
        &lt;artifactId&gt;my-app&lt;/artifactId&gt;
        &lt;version&gt;1&lt;/version&gt;
      &lt;/parent&gt;

      &lt;groupId&gt;com.mycompany.app&lt;/groupId&gt;
      &lt;artifactId&gt;my-module&lt;/artifactId&gt;
      &lt;version&gt;1&lt;/version&gt;
    &lt;/project&gt;
    """

listOfString.map { str -&gt; str != myTextBlock }</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://kotlinlang.org/spec/expressions.html#string-interpolation-expressions">8.2 String interpolation expressions</a> in the Kotlin language specification</p>
</li>
</ul>
</div>
</div>
</div>