<div class="paragraph">
<p>There is no point in creating a <code>const</code> reference to a literal numeric value. Most likely the intent was not to create a reference, but a constant value.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>const int &amp; weekdayCount = 7;  // Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>const int weekdayCount = 7;</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Make this reference a constant value by removing the "&amp;".</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_8_apr_2015_190031_evgeny_mandrikov_wrote">on 8 Apr 2015, 19:00:31 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] actually</p>
</div>
<div class="paragraph">
<p>{noformat}</p>
</div>
<div class="paragraph">
<p>int &amp; weekdayCount = 7;</p>
</div>
<div class="paragraph">
<p>{noformat}</p>
</div>
<div class="paragraph">
<p>not allowed per standard, so I deleted part about removal of "const" qualifier from message.</p>
</div>
</div>
</div>
</div>