<div class="paragraph">
<p>Every paragraph should be commented to explain its goal and how it works. This comment can be placed either just before or just after the paragraph label. Moreover paragraphs used to close a module can be left uncommented.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>PROCEDURE DIVISION.

* Some comments
    CORRECTLY-COMMENTED-PARAGRAPH1.             *&gt;  Compliant, the comment is just before

    ANOTHER-CORRECTLY-COMMENTED-PARAGRAPH1.     *&gt;  Compliant, the comment is just after
* Some comments
    ...

    UNCORRECTLY-COMMENTED-PARAGRAPH1.           *&gt; Not Compliant
    ...

*-------
    UNCORRECTLY-COMMENTED-PARAGRAPH2.           *&gt; Not Compliant as the comment is empty
    ...

      PERFORM P1 THRU P2.
    ...

*Some comments                                  *&gt; Compliant
    P1.
      ....


    P2.                                         *&gt; No violation as the this P2 paragraph close a module
       MOVE A TO B.
       ...
       EXIT.</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s1305">is related to: S1305</h3>

</div>
</div>
</div>