<div class="paragraph">
<p>Deserialization takes a stream of bits and turns it into an object. If the stream contains the type of object you expect, all is well. But if you&#8217;re deserializing untrusted input, and an attacker has inserted some other type of object, you&#8217;re in trouble. Why? There are a few different attack scenarios, but one widely-documented one goes like this: Deserialization first instantiates an <code>Object</code>, then uses the  <code>readObject</code> method to populate it. If the attacker has overridden <code>readObject</code> then he is entirely in control of what code executes during that process. It is only after <code>readObject</code> has completed that your newly-minted <code>Object</code> can be cast to the type you expected. A <code>ClassCastException</code> or <code>ClassNotFoundException</code> will be thrown, but at that point it&#8217;s too late.</p>
</div>
<div class="paragraph">
<p>To prevent this, you should either use look-ahead deserialization (pre-Java 9) or filtering to make sure you&#8217;re dealing with the correct type of object <em>before</em> you act on it.</p>
</div>
<div class="paragraph">
<p>Several third-party libraries offer look-ahead deserialization, including:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>ikkisoft&#8217;s <code>SerialKiller</code></p>
</li>
<li>
<p>Apache Commons Class IO&#8217;s <code>ValidatingObjectInputStream</code></p>
</li>
<li>
<p>contrast-rO0&#8217;s <code>SafeObjectInputStream</code></p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Note that it is possible to set a deserialization filter at the level of the JVM, but relying on that requires that your environment be configured perfectly. Every time. Additionally, such a filter may have unwanted impacts on other applications in the environment. On the other hand, setting a filter as close as possible to the deserialization that uses it allows you to specify a very narrow, focused filter.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>FileInputStream in = new FileInputStream("obj");
ObjectInputStream ois = new ObjectInputStream(in);  // Noncompliant
Foo reconstitutedFoo = (foo)ois.readObject();</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>FileInputStream in = new FileInputStream("obj");
ObjectInputStream ois = new SerialKiller(is, "/etc/serialkiller.conf");
String msg = (String) ois.readObject();</pre>
</div>
</div>
<div class="paragraph">
<p>Or for Java 9</p>
</div>
<div class="listingblock">
<div class="content">
<pre>class MyFilter implements ObjectInputFilter {
  @Override
  public Status checkInput(Filterinfo filterInfo) { ...}
}

//...
    FileInputStream in = new FileInputStream("obj");
    ObjectInputStream ois = new ObjectInputStream(in);
    ois.setObjectInputFilter(new MyFilter());  //
    Foo reconstitutedFoo = (foo)ois.readObject();</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://owasp.org/Top10/A08_2021-Software_and_Data_Integrity_Failures/">OWASP Top 10 2021 Category A8</a> - Software and Data Integrity Failures</p>
</li>
<li>
<p><a href="https://www.securecoding.cert.org/confluence/x/ZwBzCg">CERT, SER12-J.</a> - Prevent deserialization of untrusted data</p>
</li>
<li>
<p>OWASP Top 10 2017 Category A8 - Insecure Deserialization</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_duplicates_s4508">duplicates: S4508</h3>

</div>
</div>
</div>