<div class="paragraph">
<p>Using a "bald" function name is likely a bug. Rather than testing the return value of a function with a <code>void</code> parameter list, it implicitly retrieves the address of that function in memory. If that&#8217;s truly what&#8217;s intended, then it should be made explicit with the use of the <code>&amp;</code> (address-of) operator. If it&#8217;s not, then a parameter list (even an empty one) should be added after the function name.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>int func(void) {
  // ...
}

void f2(int a, int b) {
  // ...
  if (func) {  // Noncompliant - tests that the memory address of func() is non-null
    //...
  }
  // ...
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>void f2(int a, int b) {
  // ...
  if (func()) {  // tests that the return value of func() &gt; 0
    //...
  }
  // ...
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_exceptions">Exceptions</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Callback functions are a common occurrence and are usually not passed with a preceding &amp;. There is however little ambiguity so this rule ignores function identifiers when used as a parameter of a function call.</p>
</div>
<div class="listingblock">
<div class="content">
<pre>void foo() {
  // ...
}

registerEvent(AnEvent, foo);</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C:2004, 16.9 - A function identifier shall only be used with either a preceding &amp;, or with a parenthesized parameter list, which may be empty.</p>
</li>
<li>
<p>MISRA C&#43;&#43;:2008, 8-4-4 - A function identifier shall only be used to call the function or it shall be preceded by &amp;.</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Either add a parameter list or the "&amp;" operator to this use of "xxx".</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_19_sep_2014_161220_freddy_mallet_wrote">on 19 Sep 2014, 16:12:20 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>@Ann, here I would have reused exactly the MISRA title, description and code snippets which are really meaningful for this rule.</p>
</div>
</div>
</div>
</div>