<div class="paragraph">
<p>Shared naming conventions allow teams to collaborate efficiently. This rule checks that all [report/]program names match a provided regular expression.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="paragraph">
<p>With the regular expression <code>^([A-Z0-9_]*|[a-z0-9_]*)$</code>:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>    IDENTIFICATION DIVISION.
       PROGRAM-ID. MY-PROGRAM.      *&gt; Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>    IDENTIFICATION DIVISION.
       PROGRAM-ID. MY_PROGRAM.</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Rename this 'XXXX' report/program to match the regular expression ${format}</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="ulist">
<ul>
<li>
<p>key = format</p>
</li>
<li>
<p>Description: Regular expression used to check the report/program names against.</p>
</li>
<li>
<p>Default value:  ^(<strong class="A-Z0-9_">|[a-z0-9_]</strong>)$</p>
</li>
</ul>
</div>
</div>
</div>
</div>