<div class="paragraph">
<p>Cumulatively, saved keystrokes can add up to a lot of saved time. Or at least it can feel that way. So the difference between using <code>L#macro_arg</code> and <code>L###macro_arg</code> to create a wide string literal may seem perfectly justified because MSVC yields the same result for each. But if you ever need to switch to another compiler, that saved time - and then some - will be lost. So it&#8217;s best to use the cross-compiler standard from the start.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>#define MY_WIDE_CONSTANT_MAP(x) my_wide_constant_string_map[x] = L#x;</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>#define MY_WIDE_CONSTANT_MAP(x) my_wide_constant_string_map[x] = L###x;</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use "L###xxx" instead.</p>
</div>
</div>
</div>
</div>