<div class="paragraph">
<p>Sharing some naming conventions is a key point to make it possible for a team to efficiently collaborate. This rule allows to check that field names match a provided regular expression.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="paragraph">
<p>With the default regular expression <code>^[a-z][a-zA-Z0-9]*$</code>:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>class MyClass {
   private int my_field;
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class MyClass {
   private int myField;
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Rename this field "XXX" to match the regular expression ${format}.</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="ulist">
<ul>
<li>
<p>key = format
Description: Regular expression used to check the field names against.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Default Value: ^[a-z][a-zA-Z0-9]*$</p>
</div>
<div class="paragraph">
<p>Default Value for Flex: ^[_a-z][a-zA-Z0-9]*$</p>
</div>
<div class="paragraph">
<p>Default Value for Python: ^[_a-z][_a-z0-9]*$</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s117">is related to: S117</h3>

</div>
<div class="sect2">
<h3 id="_on_20_may_2013_123248_fabrice_bellingard_wrote">on 20 May 2013, 12:32:48 Fabrice Bellingard wrote:</h3>
<div class="paragraph">
<p>Implementation: http://jira.codehaus.org/browse/SONARJAVA-145</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_30_oct_2013_112757_freddy_mallet_wrote">on 30 Oct 2013, 11:27:57 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3189 for Flex</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_19_feb_2014_155138_freddy_mallet_wrote">on 19 Feb 2014, 15:51:38 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3496 for PHP</p>
</div>
</div>
</div>
</div>