<div class="paragraph">
<p>Variables can be declared with or without types. Variables declared without a type will be implicitly typed if the declaration includes an initialization, and compiler type checking will be automatically applied to any typed variable. But if you declare a variable with the <code>any</code> "type" then you&#8217;ve explicitly told the compiler not to do any type checking, which is risky.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>let a = 42;  // implicitly typed to number
let b: number = 42;  // explicitly typed to number
let c: any = 42;  // Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>let a = 42;
let b: number = 42;
let c: number = 42;</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>Remove this use of "any" so that implicit typing will be used.</p>
</li>
<li>
<p>Replace "any" with a specific type.</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p><code>any</code></p>
</div>
</div>
</div>
</div>