<div class="paragraph">
<p><code>private</code> methods that are never executed are dead code: unnecessary, inoperative code that should be removed. Cleaning out dead code decreases the size of the maintained codebase, making it easier to understand the program and preventing bugs from being introduced.</p>
</div>
<div class="paragraph">
<p>Note that this rule does not take reflection into account, which means that issues will be raised on <code>private</code> methods that are only accessed using the reflection API.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public class Foo implements Serializable
{
  private Foo(){}     //Compliant, private empty constructor intentionally used to prevent any direct instantiation of a class.
  public static void doSomething(){
    Foo foo = new Foo();
    ...
  }
  private void unusedPrivateMethod(){...}
  private void writeObject(ObjectOutputStream s){...}  //Compliant, relates to the java serialization mechanism
  private void readObject(ObjectInputStream in){...}  //Compliant, relates to the java serialization mechanism
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public class Foo implements Serializable
{
  private Foo(){}     //Compliant, private empty constructor intentionally used to prevent any direct instantiation of a class.
  public static void doSomething(){
    Foo foo = new Foo();
    ...
  }

  private void writeObject(ObjectOutputStream s){...}  //Compliant, relates to the java serialization mechanism

  private void readObject(ObjectInputStream in){...}  //Compliant, relates to the java serialization mechanism
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_exceptions">Exceptions</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule doesn&#8217;t raise any issue on annotated methods.</p>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this unused private "XXXX" [method|function|constructor|member].</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_duplicated_by_s902">is duplicated by: S902</h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s1816">relates to: S1816</h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s3241">relates to: S3241</h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s902">is related to: S902</h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s1068">is related to: S1068</h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s3985">is related to: S3985</h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s4487">is related to: S4487</h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s5536">is related to: S5536</h3>

</div>
<div class="sect2">
<h3 id="_on_22_oct_2013_144313_freddy_mallet_wrote">on 22 Oct 2013, 14:43:13 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3193 for Flex</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_15_feb_2018_115723_elena_vilchik_wrote">on 15 Feb 2018, 11:57:23 Elena Vilchik wrote:</h3>
<div class="paragraph">
<p>For TypeScript unused private methods are covered by RSPEC-1481</p>
</div>
</div>
</div>
</div>