<div class="paragraph">
<p>This rule checks that the specified child tag does not appear as a direct child of the specified parent.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Assuming a parent/child combination of <code>head</code>/<code>body</code>:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>&lt;head&gt;
  ...
  &lt;body&gt;  &lt;!-- Noncompliant --&gt;
    ...
  &lt;/body&gt;
&lt;/head&gt;</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>&lt;head&gt;
  ...
&lt;/head&gt;</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this "XXX" tag; it is not a legal child of "YYY".</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="ulist">
<ul>
<li>
<p>key = child</p>
<div class="ulist">
<ul>
<li>
<p>description: Name of the child element</p>
</li>
</ul>
</div>
</li>
<li>
<p>key: parent</p>
<div class="ulist">
<ul>
<li>
<p>description: Name of the parent element</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_deprecates_s3734">deprecates: S3734</h3>

</div>
<div class="sect2">
<h3 id="_on_25_aug_2014_064441_freddy_mallet_wrote">on 25 Aug 2014, 06:44:41 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Hi @Ann, is there any default values for the properties of this rule ? Thanks</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_25_aug_2014_201750_ann_campbell_wrote">on 25 Aug 2014, 20:17:50 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>This is the documentation (&amp; cleanup of the format) of an existing Web rule.</p>
</div>
</div>
</div>
</div>