<div class="paragraph">
<p>The use of <code>strlen</code> to determine the length of a string to which you are trying to append a null character is an anti-pattern. <code>strlen</code> requires as input an already null-terminated string;Â the result of passing a non-null-terminated string as an input to <code>strlen</code> is undefined. It may even lead to a memory access violation.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>void f() {
  char *myStr;
  // ...
  myStr[strlen(myStr)] = '\0'; // Noncompliant, strlen requires an already null-terminated string
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Review this suspicious assignment.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_3_sep_2020_202121_abbas_sabra_wrote">on 3 Sep 2020, 20:21:21 Abbas Sabra wrote:</h3>
<div class="paragraph">
<p>\[~massimo.paladin] This looks more like a Bug to me. Update if you agree, let&#8217;s discuss if you don&#8217;t</p>
</div>
</div>
</div>
</div>