<div class="paragraph">
<p>Certain prefixes are widely recognized and expected to correspond to specific namespaces. For instance, most users will expect "xs" to correspond to "http://www.w3.org/2001/XMLSchema". If you subvert that expected correspondence, you run the risk of severely confusing the users of your XSD.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>&lt;xs:schema targetNamespace="http://www.codeSamples.com/fruit"
           elementFormDefault="qualified" version="1"
           xmlns:xs="http://www.w3.org/2001/XMLSchema"
           xmlns:wsdl="http://CodeSamples.com/animal&gt;       &lt;!-- Noncompliant --&gt;
...</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>&lt;xs:schema targetNamespace="http://www.codeSamples.com/fruit"
           elementFormDefault="qualified" version="1"
           xmlns:xs="http://www.w3.org/2001/XMLSchema"
           xmlns:animal="http://CodeSamples.com/animal&gt;
...</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use a different prefix for this namespace.</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="ulist">
<ul>
<li>
<p>key = reservedPrefixes
descriptions: Comma-delimited list of namespace prefixes that should not be used</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>default value: xml,xmlns,xs,xsd,xsi,wsdl</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p><code>xmlns:xxx</code></p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_18_jan_2017_084423_ann_campbell_wrote">on 18 Jan 2017, 08:44:23 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~yves.duboispelerin] I almost want to rename this: Standard namespace prefixes should not be hijacked</p>
</div>
<div class="paragraph">
<p>WDYT?</p>
</div>
</div>
</div>
</div>