<div class="paragraph">
<p>Assembling a <code>StringBuilder</code> or <code>StringBuffer</code> into a <code>String</code> merely to see if it&#8217;s empty is a waste of cycles. Instead, jump right to the heart of the matter and get its <code>.length()</code> instead.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>StringBuilder sb = new StringBuilder();
// ...
if ("".equals(sb.toString()) { // Noncompliant
  // ...
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>StringBuilder sb = new StringBuilder();
// ...
if (sb.length() == 0) {
  // ...
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Test "xxx.length()" against 0 to see if "xxx" is empty.</p>
</div>
</div>
</div>
</div>