<div class="paragraph">
<p>Defining a function inside of a loop can yield unexpected results. Such a function keeps references to the variables which are defined in outer scopes. All function instances created inside the loop therefore see the same values for these variables, which is probably not expected.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>var funs = [];
for (var i = 0; i &lt; 13; i++) {
  funs[i] = function() { // Non-Compliant
    return i;
  };
}
console.log(funs[0]()); // 13 instead of 0
console.log(funs[1]()); // 13 instead of 1
console.log(funs[2]()); // 13 instead of 2
console.log(funs[3]()); // 13 instead of 3
...</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Make sure this function is not called after the loop completes.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>primary location: function declaration</p>
</div>
<div class="paragraph">
<p>secondary: loop</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_23_mar_2020_185747_elena_vilchik_wrote">on 23 Mar 2020, 18:57:47 Elena Vilchik wrote:</h3>
<div class="paragraph">
<p>legacy key FunctionDefinitionInsideLoop</p>
</div>
</div>
<div class="sect2">
<h3 id="_is_related_to_s3044">is related to: S3044</h3>

</div>
<div class="sect2">
<h3 id="_on_21_apr_2017_144945_elena_vilchik_wrote">on 21 Apr 2017, 14:49:45 Elena Vilchik wrote:</h3>
<div class="paragraph">
<p>\[~jeanchristophe.collet] Could you please update message for this rule? Something like "Review this function declated inside loop and make sure it is not called after loop ends and it doesn&#8217;t depend on variables updated with each iteration." (https://github.com/SonarSource/sonar-javascript/issues/573)</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_16_feb_2018_104258_elena_vilchik_wrote">on 16 Feb 2018, 10:42:58 Elena Vilchik wrote:</h3>
<div class="paragraph">
<p>Function declared inside is a problem when using <code>var</code> declaration for counter. This is very unlikely that in TS somebody will use <code>var</code>. So this rule will have very low ROI. So we will not implement it for TS.</p>
</div>
</div>
</div>
</div>