<div class="paragraph">
<p>Utility classes, which are collections of <code>static</code> members, are not meant to be instantiated. Even abstract utility classes, which can be extended, should not have public constructors.</p>
</div>
<div class="paragraph">
<p>Java adds an implicit public constructor to every class which does not define at least one explicitly. Hence, at least one non-public constructor should be defined.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class StringUtils { // Noncompliant

  public static String concatenate(String s1, String s2) {
    return s1 + s2;
  }

}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class StringUtils { // Compliant

  private StringUtils() {
    throw new IllegalStateException("Utility class");
  }

  public static String concatenate(String s1, String s2) {
    return s1 + s2;
  }

}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_exceptions">Exceptions</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When class contains <code>public static void main(String[] args)</code> method it is not considered as utility class and will be ignored by this rule.</p>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Hide this public constructor.</p>
</div>
<div class="paragraph">
<p>Add a private constructor to hide the implicit public one.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_23_jul_2013_143150_dinesh_bolkensteyn_wrote">on 23 Jul 2013, 14:31:50 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>Implemented by http://jira.codehaus.org/browse/SONARJAVA-223</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_11_apr_2017_162329_tibor_blenessy_wrote">on 11 Apr 2017, 16:23:29 Tibor Blenessy wrote:</h3>
<div class="paragraph">
<p>I changed compliant example to use <code>IllegalStateException</code> rather than <code>IllegalAccessError</code> as errors should be raised only by <code>javac</code> and not at runtime.</p>
</div>
</div>
</div>
</div>