<div class="paragraph">
<p>Lambdas can use variables from their enclosing scope (called "capture") either by reference or by value. Since lambdas may run asynchronously, reference capture should be used with caution because by the time the lambda runs, the referenced variable may be out of scope, resulting in an access violation at run time.</p>
</div>
<div class="paragraph">
<p>You can specify default capture by reference (<code>[&amp;]</code>), or by value (<code>[=]</code>). Clearly default reference capture can cause scope issues, but default value capture can also lead to problems. That&#8217;s because both forms of default capture implicitly also capture <code>*this</code>, which would automatically be used if for example you referenced a method from the enclosing scope.</p>
</div>
<div class="paragraph">
<p>If the lambda is used immediately (for instance, passed as an argument to <code>std::sort</code>), there is no risk of dangling reference. For those lambdas, it is safe to pass everything through a default capture by reference. See also S5495.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when default capture is used, unless the lambda is immediately executed.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>void fun() {
  Foo foo;
  ...
  executor-&gt;Schedule([&amp;] {  // Noncompliant
    maybeMember(foo);  // implicit use of *this reference if maybeMember is a member function. foo and maybeMember may both be gone by the time this is invoked
  });
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>void fun() {
  Foo foo;
  ...
  executor-&gt;Schedule([&amp;foo] { // it is clear that foo is captured by reference and compilation is going to fail if maybeMember is a member function
    maybeMember(foo);
  });
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Explicitly capture the required scope variables.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p><code>[&amp;|=]</code></p>
</div>
</div>
</div>
</div>