<div class="paragraph">
<p>The use of <code>ref</code> or <code>out</code> in combination with <code>[Optional]</code> is both confusing and contradictory. <code>[Optional]</code> indicates that the parameter doesn&#8217;t have to be provided, while <code>out</code> and <code>ref</code> mean that the parameter will be used to return data to the caller (<code>ref</code> additionally indicates that the parameter may also be used to pass data into the method).</p>
</div>
<div class="paragraph">
<p>Thus, making it <code>[Optional]</code> to provide the parameter in which you will be passing back the method results doesn&#8217;t make sense. In fact, the compiler will raise an error on such code. Unfortunately, it raises the error on method calls where the <code>[Optional]</code> parameter has been omitted, not the source of the problem, the method declaration.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class MyClass
{
    public void DoStuff([Optional] ref int i) // Noncompliant
    {
        Console.WriteLine(i);
    }

    public static void Main()
    {
        new MyClass().DoStuff(); // This doesn't compile, CS7036 shows
    }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class MyClass
{
  public void DoStuff(ref int i)
  {
    Console.WriteLine(i);
  }

  public static void Main()
  {
    var i = 42;
    new MyClass().DoStuff(ref i);
  }
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove the "Optional" attribute, it cannot be used with "[ref|out]".</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>"[Optional]" attribute</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_8_dec_2015_092445_tamas_vajk_wrote">on 8 Dec 2015, 09:24:45 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I changed this rule to suggest removing the <code>[Optional]</code> and not the <code>ref</code>/<code>out</code>.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_8_dec_2015_150951_ann_campbell_wrote">on 8 Dec 2015, 15:09:51 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>okay [~tamas.vajk]</p>
</div>
</div>
</div>
</div>