<div class="paragraph">
<p>Naming conventions are an important tool in efficient team collaboration. This rule checks that all form names match a regular expression naming convention.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="paragraph">
<p>With the default regular expression:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>FORM MyForm.
...
ENDFORM.</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>FORM MY_FORM.
...
ENDFORM.</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Rename form "XXXX" to match the regular expression ${format}</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="ulist">
<ul>
<li>
<p>key = format
Description : Regular expression used to check the form names against.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Default for ABAP : ^(<strong class="A-Z0-9_">|[a-z0-9_]</strong>)$</p>
</div>
</div>
</div>
</div>