<div class="paragraph">
<p>Explicit specializations of function templates are not considered in overload resolution, only the main template. As a consequence, the function that will be selected might very well be different from what seems natural to the developer, leading to hard to understand bugs. Moreover, function templates don&#8217;t allow partial specialization.</p>
</div>
<div class="paragraph">
<p>Instead of specializing a function template, you may choose to overload it with another template or non template function, since a more specialized overload will be preferred to a generic overload.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>template &lt;typename T&gt; void f ( T );
template &lt;&gt; void f&lt;char*&gt; ( char * ); // explicit specialization, noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>template &lt;typename T&gt; void f ( T );
void f( char * ); // overload, compliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_exceptions">Exceptions</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule ignores cases where none of the main function template arguments depend on a template parameter: Even if the code could still be written without function template specialization (by deferring the real work to a class template, and offering specializations of this class template as customization point to the user), there is no risk of confusion for overload resolution in these cases.</p>
</div>
<div class="listingblock">
<div class="content">
<pre>// For real code, use std::numeric_limits instead...
template &lt;class T&gt; T max();
template &lt;&gt; float max&lt;float&gt;() { return FLT_MAX; }  // Ignored

template&lt;class T&gt;
bool isMax(T t){
    return t == max&lt;T&gt;();
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C&#43;&#43;:2008, 14-8-1</p>
</li>
<li>
<p><a href="https://github.com/isocpp/CppCoreGuidelines/blob/036324/CppCoreGuidelines.md#t144-dont-specialize-function-templates">C&#43;&#43; Core Guidelines T.144</a> - Donâ€™t specialize function templates</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use overloading instead of specializing this function template.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_21_oct_2014_190922_ann_campbell_wrote">on 21 Oct 2014, 19:09:22 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~samuel.mercier] the text under the Compliant Solution heading should be incorporated in the description. Also, this is likely a "bug" or "pitfall".</p>
</div>
</div>
</div>
</div>