<div class="paragraph">
<p>Any class that has memory to manage should provide all the methods necessary to properly manage that memory, including a copy constructor and an override of <code>operator=</code>. Without those methods, you&#8217;re likely to end up with memory leaks and multiple class instances pointing at the same segments of memory for their members.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class MyClass // Noncompliant
{
  private:
    char* cpData;
  public
    MyClass(const char* value);
    ~MyClass();
}

MyClass a = new MyClass("The quick red fox");
MyClass b = new MyClass("How now brown cow");

b = a;  // cpData pointer, not value copied. Also b's old value not deleted: Memory leak.</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://www.securecoding.cert.org/confluence/x/SAAV">CERT, OOP-06-CPP.</a> - Create a private copy constructor and assignment operator for non copyable objects</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_duplicates_s3624">duplicates: S3624</h3>

</div>
</div>
</div>