<div class="paragraph">
<p>A loop with at most one iteration is equivalent to the use of an <code>IF</code> statement to conditionally execute one piece of code. No developer expects to find such usage of a loop statement. If the initial intention of the author was really to conditionally execute one piece of code, an <code>IF</code> statement should be used in place.</p>
</div>
<div class="paragraph">
<p>At worst that was not the initial intention of the author and so the body of the loop should be fixed to use the nested <code>RETURN</code>, <code>EXIT</code>, <code>RAISE</code> or <code>GOTO</code> statements in a more appropriate way.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>LOOP
  counter := counter + 1;
  dbms_output.put_line(counter);
  EXIT;   -- Noncompliant
END LOOP;</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>LOOP
  counter := counter + 1;
  IF counter &gt; 10 THEN
    EXIT;
  ELSE
    dbms_output.put_line(counter);
  END IF;
END LOOP;</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Refactor this loop to do more than one iteration.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>primary: first line of the loop statement</p>
</li>
<li>
<p>secondary: all the nested control flow statements <code>return</code>, <code>break</code>, <code>throw</code> or <code>goto</code> leading to have at most one iteration
message: 'The loop stops here.'</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_duplicated_by_s3554">is duplicated by: S3554</h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s3626">relates to: S3626</h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s1763">is related to: S1763</h3>

</div>
</div>
</div>