<div class="paragraph">
<p>There should not be any statements after <code>EXIT PROGRAM</code>. Such statements cannot be reached, and are therefore dead code. Dead code makes a program more complex and therefore more difficult to maintain.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>  PROCEDURE DIVISION.
    PARAGRAPH1.
      MOVE A TO B.
      EXIT PROGRAM.   &gt;NOK as the following "MOVE B TO C" statement will never be called
      MOVE B TO C.</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>  PROCEDURE DIVISION.
    PARAGRAPH1.
      MOVE A TO B.
      EXIT PROGRAM.</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s1475">is related to: S1475</h3>

</div>
</div>
</div>