<div class="paragraph">
<p>Code annotated as deprecated should not be used since it will be removed sooner or later.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>@Deprecated("")
interface Old

class Example : Old // Noncompliant</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>Remove this call to a deprecated method.</p>
</li>
<li>
<p>Remove this use of a deprecated [class|field].</p>
</li>
<li>
<p>Remove this use of "xxx"; it is deprecated.</p>
</li>
<li>
<p>Don&#8217;t override a deprecated method or explicitly mark it as "@Deprecated".</p>
</li>
</ul>
</div>
</div>
</div>
</div>