<div class="paragraph">
<p>Using the standard <code>getClassLoader()</code> may not return the <em>right</em> class loader in a JEE context. Instead, go through the <code>currentThread</code>.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>ClassLoader cl = this.getClass().getClassLoader();  // Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>ClassLoader cl = Thread.currentThread().getContextClassLoader();</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use "Thread.currentThread().getContextClassLoader()" instead.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_15_oct_2018_111925_massimo_paladin_wrote">on 15 Oct 2018, 11:19:25 Massimo PALADIN wrote:</h3>
<div class="paragraph">
<p>Rule removed from "Sonar Way", we cannot detect a JEE context and would generate too much noise to raise an issue for each <code>getClassLoader()</code> method invocation.</p>
</div>
</div>
</div>
</div>