<div class="paragraph">
<p>There is no good excuse for an empty class. If it&#8217;s being used simply as a common extension point, it should be replaced with an <code>interface</code>. If it was stubbed in as a placeholder for future development it should be fleshed-out. In any other case, it should be eliminated.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public class Nothing {  // Noncompliant
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public interface Nothing {
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_exceptions">Exceptions</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Empty classes can be used as marker types (for Spring for instance), therefore empty classes that are annotated will be ignored.</p>
</div>
<div class="listingblock">
<div class="content">
<pre>@Configuration
@EnableWebMvc
public final class ApplicationConfiguration {

}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this empty class, write its code or make it an "interface".</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_19_jul_2016_160658_evgeny_mandrikov_wrote">on 19 Jul 2016, 16:06:58 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>Note to myself: study http://stackoverflow.com/questions/4035180/advantages-of-an-empty-class-in-c</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_18_jan_2019_172020_loïc_joly_wrote">on 18 Jan 2019, 17:20:20 Loïc Joly wrote:</h3>
<div class="paragraph">
<p>Empty classes are used in some C&#43;&#43; idioms (mostly as tags), as described in the linked shared by Evgeny. Removing C&#43;&#43; from the list of targeted languages.</p>
</div>
</div>
</div>
</div>