<div class="paragraph">
<p>Because the <code>equals</code> method takes a generic <code>Object</code> as a parameter, any type of object may be passed to it. The method should not assume it will only be used to test objects of its class type. It must instead check the parameter&#8217;s type.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public boolean equals(Object obj) {
  MyClass mc = (MyClass)obj;  // Noncompliant
  // ...
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public boolean equals(Object obj) {
  if (obj == null)
    return false;

  if (this.getClass() != obj.getClass())
    return false;

  MyClass mc = (MyClass)obj;
  // ...
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add a type test to this method.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_3_oct_2014_200910_ann_campbell_wrote">on 3 Oct 2014, 20:09:10 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~nicolas.peru] this will be covered by RSPEC-1944 if it includes downcasts.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_12_oct_2014_190005_freddy_mallet_wrote">on 12 Oct 2014, 19:00:05 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>This rule should be linked to Findbugs rule BC_EQUALS_METHOD_SHOULD_WORK_FOR_ALL_OBJECTS</p>
</div>
</div>
</div>
</div>