<div class="paragraph">
<p>Every filter defined in <code>web.xml</code> file should be used in a <code>&lt;filter-mapping&gt;</code> element. Otherwise such filters are not invoked.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>  &lt;filter&gt;
     &lt;filter-name&gt;DefinedNotUsed&lt;/filter-name&gt;
     &lt;filter-class&gt;com.myco.servlet.ValidationFilter&lt;/filter-class&gt;
  &lt;/filter&gt;</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>  &lt;filter&gt;
     &lt;filter-name&gt;ValidationFilter&lt;/filter-name&gt;
     &lt;filter-class&gt;com.myco.servlet.ValidationFilter&lt;/filter-class&gt;
  &lt;/filter&gt;

  &lt;filter-mapping&gt;
     &lt;filter-name&gt;ValidationFilter&lt;/filter-name&gt;
     &lt;url-pattern&gt;/*&lt;/url-pattern&gt;
  &lt;/filter-mapping&gt;</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://owasp.org/Top10/A05_2021-Security_Misconfiguration/">OWASP Top 10 2021 Category A5</a> - Security Misconfiguration</p>
</li>
<li>
<p><a href="https://www.owasp.org/index.php/Top_10-2017_A6-Security_Misconfiguration">OWASP Top 10 2017 Category A6</a> - Security Misconfiguration</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>"xxx" filter should have a mapping.</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_2_oct_2015_191358_ann_campbell_wrote">on 2 Oct 2015, 19:13:58 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~nicolas.peru], I know we&#8217;re not ready to check <code>web.xml</code> yet, but when we are&#8230;&#8203;</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_4_mar_2016_123216_sytze_van_koningsveld_wrote">on 4 Mar 2016, 12:32:16 sytze van koningsveld wrote:</h3>
<div class="paragraph">
<p>scrubbing sounds like blacklisting, which is different from true validation (white listing), so maybe a misnomer. The "validation filter" should not replace true valdation, typically done in controllers.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_19_mar_2018_095633_sébastien_gioria_appsecfr_wrote">on 19 Mar 2018, 09:56:33 Sébastien GIORIA - AppSecFR wrote:</h3>
<div class="paragraph">
<p>I might tag this OWASP A6:2017 and not A1:2017. This is not a injection, more a configuration problem</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_20_mar_2018_072240_freddy_mallet_wrote">on 20 Mar 2018, 07:22:40 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>I tend to agree with [~SPoint]. [~alexandre.gigleux] would you be confortable with this change ? Thanks</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_20_mar_2018_081615_alexandre_gigleux_wrote">on 20 Mar 2018, 08:16:15 Alexandre Gigleux wrote:</h3>
<div class="paragraph">
<p>I agree [~freddy.mallet] / [~SPoint] and I applied the suggested change.</p>
</div>
</div>
</div>
</div>