<div class="paragraph">
<p>An empty multi-line comment is likely to be a mistake and doesn&#8217;t help to improve the readability of the code. For these reasons, it should be removed.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>=begin
=end

=begin


=end</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this comment, it is empty.</p>
</div>
</div>
</div>
</div>