<div class="paragraph">
<p>There&#8217;s no point in chaining multiple <code>OrderBy</code> calls in a LINQ; only the last one will be reflected in the result because each subsequent call  completely reorders the list. Thus, calling <code>OrderBy</code> multiple times is a performance issue as well, because all of the sorting will be executed, but only the result of the last sort will be kept.</p>
</div>
<div class="paragraph">
<p>Instead, use <code>ThenBy</code> for each call after the first.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>var x = personList
  .OrderBy(person =&gt; person.Age)
  .OrderBy(person =&gt; person.Name)  // Noncompliant
  .ToList();  // x is sorted by Name, not sub-sorted</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>var x = personList
  .OrderBy(person =&gt; person.Age)
  .ThenBy(person =&gt; person.Name)
  .ToList();</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use "ThenBy" instead.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_30_jun_2015_135154_ann_campbell_wrote">on 30 Jun 2015, 13:51:54 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~tamas.vajk] I wonder if this is also an efficiency issue. The end result is a list that is sorted only by the last OrderBy argument, but doe all the previous OrderBy&#8217;s take place, then get thrown away? If so, this would probably be worth adding to the description.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_30_jun_2015_135528_tamas_vajk_wrote">on 30 Jun 2015, 13:55:28 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I added a performance related sentence.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_30_jun_2015_145239_ann_campbell_wrote">on 30 Jun 2015, 14:52:39 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>I shuffled the text some, [~tamas.vajk]</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_1_jul_2015_064010_tamas_vajk_wrote">on 1 Jul 2015, 06:40:10 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Shouldn&#8217;t this issue have some performance related label as well?</p>
</div>
<div class="paragraph">
<p>I simplified the message as the ordering might not happen by some property, but by some complex logic, and in this case we can&#8217;t display the whole expression and <code>Comparer</code> in the message.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_1_jul_2015_112648_ann_campbell_wrote">on 1 Jul 2015, 11:26:48 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>added [~tamas.vajk]</p>
</div>
</div>
</div>
</div>