<div class="paragraph">
<p>Many consider <code>clone</code> and <code>Cloneable</code> broken in Java, largely because the rules for overriding <code>clone</code> are tricky and difficult to get right, according to Joshua Bloch:</p>
</div>
<div class="quoteblock">
<blockquote>
<div class="paragraph">
<p>Object&#8217;s clone method is very tricky. It&#8217;s based on field copies, and it&#8217;s "extra-linguistic." It creates an object without calling a constructor. There are no guarantees that it preserves the invariants established by the constructors. There have been lots of bugs over the years, both in and outside Sun, stemming from the fact that if you just call super.clone repeatedly up the chain until you have cloned an object, you have a shallow copy of the object.  The clone generally shares state with the object being cloned. If that state is mutable, you don&#8217;t have two independent objects. If you modify one, the other changes as well. And all of a sudden, you get random behavior.</p>
</div>
</blockquote>
</div>
<div class="paragraph">
<p>A copy constructor or copy factory should be used instead.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when <code>clone</code> is overridden, whether or not <code>Cloneable</code> is implemented.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public class MyClass {
  // ...

  public Object clone() { // Noncompliant
    //...
  }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public class MyClass {
  // ...

  MyClass (MyClass source) {
    //...
  }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://www.artima.com/intv/bloch13.html">Copy Constructor versus Cloning</a></p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_see_also">See Also</h3>
<div class="ulist">
<ul>
<li>
<p>S2157 - "Cloneables" should implement "clone"</p>
</li>
<li>
<p>S1182 - Classes that override "clone" should be "Cloneable" and call "super.clone()"</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this "clone" implementation; use a copy constructor or copy factory instead.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s1182">relates to: S1182</h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s2157">relates to: S2157</h3>

</div>
<div class="sect2">
<h3 id="_on_22_may_2015_194848_ann_campbell_wrote">on 22 May 2015, 19:48:48 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>consulted http://stackoverflow.com/questions/2326758/how-to-properly-override-clone-method</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_16_jun_2015_170147_nicolas_peru_wrote">on 16 Jun 2015, 17:01:47 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>Looks good.</p>
</div>
</div>
</div>
</div>