<div class="paragraph">
<p>Classes should only hold <code>weak</code> references to delegate fields with <code>class</code> type. Otherwise, the owning class will have a <code>strong</code> reference to its delegate, and vice versa, and the OS won&#8217;t be able to deallocate either of them..</p>
</div>
<div class="paragraph">
<p>Note that this only applies to non-computed delegate fields in classes, and not to fields in <code>struct</code>s and <code>enum</code>s.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class MyClass {
  var delegate: ConventionDelegate?  // Noncompliant
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class MyClass {
  weak var delegate: ConventionDelegate?
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Mark this variable as "weak" to avoid reference cycles.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>variable declaration</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_5_sep_2017_175758_elena_vilchik_wrote">on 5 Sep 2017, 17:57:58 Elena Vilchik wrote:</h3>
<div class="paragraph">
<p>Problem of reference cycles is an important one in Swift (see this <a href="https://developer.apple.com/library/content/documentation/Swift/Conceptual/Swift_Programming_Language/AutomaticReferenceCounting.html">doc</a>), but in order to track this pattern we need a semantic information, which we don&#8217;t have.</p>
</div>
</div>
</div>
</div>