<div class="paragraph">
<p>By convention, the name of a file should match the name of the top-level class it contains. Anything else may confuse maintainers.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>// file path: Gui/Interface.cs  -- Noncompliant
namespace Gui
{
  class Screen
  {
  }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>// file path: Gui/Screen.cs
namespace Gui
{
  class Screen
  {
  }
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Rename this file to "xxx".</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s3259">relates to: S3259</h3>

</div>
<div class="sect2">
<h3 id="_on_14_jul_2016_160925_ann_campbell_wrote">on 14 Jul 2016, 16:09:25 Ann Campbell wrote:</h3>
<div class="paragraph">
<p><a href="https://github.com/google/error-prone/blob/master/docs/bugpattern/ClassName.md" class="bare">https://github.com/google/error-prone/blob/master/docs/bugpattern/ClassName.md</a></p>
</div>
</div>
</div>
</div>