<div class="paragraph">
<p>There is no good reason to create a new object to not do anything with it. Most of the time, this is due to a missing piece of code and so could lead to an unexpected behavior in production.</p>
</div>
<div class="paragraph">
<p>If it was done on purpose because the constructor has side-effects, then that side-effect code should be moved into a separate, static method and called directly.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>if ($x &lt; 0) {
  new foo;  // Noncompliant
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>$var = NULL;
if ($x &lt; 0) {
  $var = new foo;
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Either remove this useless object instantiation of class "XXXX" or use it</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s905">is related to: S905</h3>

</div>
<div class="sect2">
<h3 id="_on_21_aug_2015_152029_freddy_mallet_wrote">on 21 Aug 2015, 15:20:29 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>This RSPEC should be part of RSPEC-905 but as this seems to be impossible to not generate some false-positives, we prefer to keep it isolated.</p>
</div>
</div>
</div>
</div>