<div class="paragraph">
<p>The standard assertions library methods such as <code>AreEqual</code> and <code>AreSame</code> in <strong>MSTest</strong> and <strong>NUnit</strong>, or <code>Equal</code> and <code>Same</code> in <strong>XUnit</strong>, expect the first argument to be the expected value and the second argument to be the actual value. Swap them, and your test will still have the same outcome (succeed/fail when it should) but the error messages will be confusing.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when the second argument to an assertions library method is a hard-coded value and the first argument is not.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>Assert.AreEqual(runner.ExitCode, 0, "Unexpected exit code"); // Noncompliant; Yields error message like: Expected:&lt;-1&gt;. Actual:&lt;0&gt;.</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>Assert.AreEqual(0, runner.ExitCode, "Unexpected exit code");</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Swap these 2 arguments so they are in the correct order: expected value, actual value.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>Primary: second argument to swap</p>
</li>
<li>
<p>Secondary: first argument to swap</p>
<div class="ulist">
<ul>
<li>
<p>message: 'Other argument to swap.'</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_24_nov_2015_154114_tamas_vajk_wrote">on 24 Nov 2015, 15:41:14 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] LGTM</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_24_nov_2015_161708_freddy_mallet_wrote">on 24 Nov 2015, 16:17:08 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Hi [~ann.campbell.2], I would make this rule also targeting Java and would associate the tag 'java-top' to it. Thanks</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_24_nov_2015_212340_ann_campbell_wrote">on 24 Nov 2015, 21:23:40 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Java targeted &amp; java-top added [~freddy.mallet].</p>
</div>
<div class="paragraph">
<p>CC [~nicolas.peru]</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_25_nov_2015_105059_nicolas_peru_wrote">on 25 Nov 2015, 10:50:59 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] No problem for java and java-top but this would need a dedicated example for java. And, IMO, a clear definition of what is a "hardcoded" value : constant ? literal ? both ?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_30_nov_2015_170455_ann_campbell_wrote">on 30 Nov 2015, 17:04:55 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~nicolas.peru] I&#8217;d start with literals and locally-defined (class or method level) vars.</p>
</div>
</div>
</div>
</div>