<div class="paragraph">
<p>Shared coding conventions allow teams to collaborate efficiently. This rule checks that paragraph names match a provided regular expression.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="paragraph">
<p>With a regular expression of <code>[A-Z-]+</code>:</p>
</div>
<div class="listingblock">
<div class="content">
<pre> PROCEDURE DIVISION.

    Do_The_Thing.           *&gt; Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre> PROCEDURE DIVISION.

    DO-THE-THING           *&gt; Noncompliant</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="ulist">
<ul>
<li>
<p>key = regexPatternString</p>
</li>
<li>
<p>description: A regular expression to specify the paragraph naming convention</p>
</li>
</ul>
</div>
</div>
</div>
</div>