<div class="paragraph">
<p>For maximum reusability, methods should accept parameters with as little specialization as possible. So unless specific features from a child class are required by a method, a type higher up the class hierarchy should be used instead.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>interface IVehicle { void Go(); }
public class Submarine : IVehicle
{
  public void Go() {  /* ... */ }
  public void Submerge(int depth) { /* ... */ }
}

// ...
public void Travel(Submarine transport) // Noncompliant; no class-specific features used
{
  transport.Go();
}

public void ExploreDepths(Submarine transport) // Compliant; class-specific feature used
{
  var depth = 0;
  // ...
  transport.Submerge(depth);
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>interface IVehicle { void Go(); }
public class Submarine : IVehicle
{
  public void Go() {  /* ... */ }
  public void Submerge(int depth) { /* ... */ }
}

// ...
public void Travel(IVehicle transport)
{
  transport.Go();
}

public void ExploreDepths(Submarine transport)
{
  var depth = 0;
  // ...
  transport.Submerge(depth);
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use '{0}' here; it is a more general type than '{1}'.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_8_jul_2015_085153_tamas_vajk_wrote">on 8 Jul 2015, 08:51:53 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Looks good</p>
</div>
</div>
</div>
</div>