<div class="paragraph">
<p>The scope of objects shall be restricted to functions where possible. File scope shall only be used where objects need to have either internal or external linkage. Where objects are declared at file scope MISRA C 2004 Rule 8.10 applies. It is considered good practice to avoid making identifiers global except where necessary.</p>
</div>
<div class="paragraph">
<p>Whether objects are declared at the outermost or innermost block is largely a matter of style.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>int temp;
int function() {
  temp = someFunction1();
  temp += someFunction2();
  // ...
  return temp;
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>int function() {
  int temp;
  temp = someFunction1();
  temp += someFunction2();
  // ...
  return temp;
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C:2004, 8.7 - Objects shall be defined at block scope if they are only accessed from within a single function.</p>
</li>
<li>
<p>MISRA C:2012, 8.9 - An object should be defined at block scope if its identifier only appears in a single function</p>
</li>
<li>
<p><a href="https://wiki.sei.cmu.edu/confluence/x/XTZGBQ">CERT, DCL53-J.</a> - Minimize the scope of variables</p>
</li>
<li>
<p><a href="https://wiki.sei.cmu.edu/confluence/x/z9YxBQ">CERT, DCL19-C.</a> - Minimize the scope of variables and functions</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Move this declaration inside the body of function "xxx"</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s806">relates to: S806</h3>

</div>
<div class="sect2">
<h3 id="_on_17_oct_2014_133501_ann_campbell_wrote">on 17 Oct 2014, 13:35:01 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~freddy.mallet] this rule overlaps the C&#43;&#43; version: RSPEC-806</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_17_oct_2014_133548_ann_campbell_wrote">on 17 Oct 2014, 13:35:48 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Nicely done [~samuel.mercier]. Good SQALE choice too.</p>
</div>
</div>
</div>
</div>