<div class="paragraph">
<p>Flag variables are boolean parameters used to control the behavior of a method. Specifying such a parameter may seem innocuous, particularly if it&#8217;s well named.</p>
</div>
<div class="paragraph">
<p>Unfortunately, the maintainers of the code calling the method won&#8217;t see the parameter name, only its value. They&#8217;ll be forced either to guess at the meaning or to take extra time to look the method up.</p>
</div>
<div class="paragraph">
<p>Instead, separate methods should be written.</p>
</div>
<div class="paragraph">
<p>This rule flags methods where a boolean parameter is used to control the majority of the code in the method.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public void getDressed(boolean isCold) {

  if (isCold()) {
    turtleneck();
    sweater();
    thickSocks();
    heavyPants();
    boots();
  } else {
    tshirt();
    shorts();
    sandals();
  }
}

getDressed(false); // does this mean go naked?</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public void getDressedColdWeather() {

  turtleneck();
  sweater();
  thickSocks();
  heavyPants();
  boots();
}

public void getDressedWarmWeather() {
  tshirt();
  shorts();
  sandals();
}

getDressedWarmWeather();</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_duplicates_s2301">duplicates: S2301</h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s2494">is related to: S2494</h3>

</div>
</div>
</div>