<div class="paragraph">
<p>In a CREATE PACKAGE BODY, the purpose of the initialization section is to set the initial values of the package&#8217;s global variables. It is therefore surprising to find a RETURN statement there, as all its following statements will be unreachable.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>SET SERVEROUTPUT ON

CREATE OR REPLACE PACKAGE foo AS
  FUNCTION getBar RETURN PLS_INTEGER;
  bar PLS_INTEGER;
END;
/

CREATE OR REPLACE PACKAGE BODY foo AS
  FUNCTION getBar RETURN PLS_INTEGER AS
  BEGIN
    RETURN bar; -- Compliant
  END;
BEGIN
  bar := 42;
  DBMS_OUTPUT.PUT_LINE('package loaded');
  RETURN; -- Noncompliant
  DBMS_OUTPUT.PUT_LINE('this is unreachable code');
END;
/

DROP PACKAGE BODY foo;

DROP PACKAGE foo;</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this "RETURN" statement.</p>
</div>
</div>
</div>
</div>