<div class="paragraph">
<p>Combine inheritance and array covariance, and the results can potentially get nasty. Together, they allow you to write array stores that will compile, but cause errors at runtime.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public class Bowl {

  abstract class Fruit { }
  class Apple : Fruit { }
  class Orange : Fruit { }

  Fruit [] fruits;

  public void fillBowl(){
    fruits = new Apple[30];
    fruits[0] = new Orange();  // Noncompliant; this compiles but raises exception at runtime
  }
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this store of "xxx" into "yyy"; "yyy" is a "yyy-type[]".</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s2330">is related to: S2330</h3>

</div>
<div class="sect2">
<h3 id="_on_15_jun_2015_145618_ann_campbell_wrote">on 15 Jun 2015, 14:56:18 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~tamas.vajk] I&#8217;m afraid the description is a little lame, but since this rule applies to both C# and Java, I didn&#8217;t want to use any wording that was specific to one language or the other.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_16_jun_2015_095027_tamas_vajk_wrote">on 16 Jun 2015, 09:50:27 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I modified the title. WDYT?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_16_jun_2015_115710_ann_campbell_wrote">on 16 Jun 2015, 11:57:10 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>I&#8217;m afraid it&#8217;s too broad [~tamas.vajk]. The problem is writing the wrong stuff to a covariant array, not writing at all.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_16_jun_2015_121016_tamas_vajk_wrote">on 16 Jun 2015, 12:10:16 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] You are right, I modified again the title, still not the best, but getting there&#8230;&#8203;</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_16_jun_2015_135241_ann_campbell_wrote">on 16 Jun 2015, 13:52:41 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~tamas.vajk] I&#8217;ve come back to "bad stores" because <em>some</em> derived elements are okay. :-/</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_29_jun_2015_141331_tamas_vajk_wrote">on 29 Jun 2015, 14:13:31 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] okay, let&#8217;s keep it this way.</p>
</div>
</div>
</div>
</div>