<div class="paragraph">
<p>Member functions can only be used with an instance of a class. But <code>friend</code> functions can be used with an implicitly converted type. So loosening access privileges to <code>friend</code> on overloaded binary operators makes them more flexible. Specifically, with a <code>friend</code> function, the class instance can be on either the right or the left of the operator, but with a member function, it can only be on the left.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue for all non-<code>friend</code> overloaded binary operators except:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>"<code>=</code>", "<code>[ ]</code>", and "<code>-&gt;</code>", which cannot be overloaded as <code>friend</code> functions.</p>
</li>
<li>
<p>"<code>+=</code>", "<code>-=</code>", "<code>*=</code>", "<code>/=</code>", "<code>%=</code>", "<code>^=</code>", "<code>&amp;=</code>", "<code>|=</code>", "<code>&lt;&lt;=</code>", and "<code>&gt;&gt;=</code>", which are not symmetric operators.</p>
</li>
</ul>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>bool operator==(const MyClass &amp;RHS);  // Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>friend bool operator==(const MyClass &amp;LHS, const MyClass &amp;RHS);</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://github.com/isocpp/CppCoreGuidelines/blob/036324/CppCoreGuidelines.md#c161-use-nonmember-functions-for-symmetric-operators">C&#43;&#43; Core Guidelines C.161</a> - Use nonmember functions for symmetric operators</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Make this a free function, potentially "friend".</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_17_nov_2015_090134_evgeny_mandrikov_wrote">on 17 Nov 2015, 09:01:34 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] could you please help me update this RSPEC - operators <code>=</code>, <code>[]</code> and <code>-&gt;</code> can&#8217;t be overloaded as friend functions.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_17_nov_2015_140138_ann_campbell_wrote">on 17 Nov 2015, 14:01:38 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>See what you think [~evgeny.mandrikov]</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_17_nov_2015_203854_evgeny_mandrikov_wrote">on 17 Nov 2015, 20:38:54 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] LGTM</p>
</div>
</div>
</div>
</div>