<div class="paragraph">
<p>According to <a href="http://docs.oracle.com/javase/specs/jls/se7/html/jls-8.html#jls-8.9">the Java Language Specification-8.9</a>:</p>
</div>
<div class="quoteblock">
<blockquote>
<div class="paragraph">
<p>Nested enum types are implicitly <code>static</code>.</p>
</div>
</blockquote>
</div>
<div class="paragraph">
<p>So there&#8217;s no need to declare them <code>static</code> explicitly.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public class Flower {
  static enum Color { // Noncompliant; static is redundant here
    RED, YELLOW, BLUE, ORANGE
  }

  // ...
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public class Flower {
  enum Color { // Compliant
    RED, YELLOW, BLUE, ORANGE
  }

  // ...
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this redundant "static" qualifier; nested enum types are implicitly static.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Primary Location: the 'static' keyword</p>
</div>
<div class="paragraph">
<p>Secondary Location: the 'enum' keyword</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_31_mar_2015_150956_nicolas_peru_wrote">on 31 Mar 2015, 15:09:56 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2]could be nice to add a link or a quote from the jls : http://docs.oracle.com/javase/specs/jls/se7/html/jls-8.html#jls-8.9</p>
</div>
<div class="quoteblock">
<blockquote>
<div class="paragraph">
<p>Nested enum types are implicitly static. It is permissible to explicitly declare a nested enum type to be static.</p>
</div>
</blockquote>
</div>
</div>
<div class="sect2">
<h3 id="_on_31_mar_2015_174606_ann_campbell_wrote">on 31 Mar 2015, 17:46:06 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~nicolas.peru] why would we quote the docs as saying it&#8217;s permissible to do exactly what this rule says not to do?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_1_apr_2015_084230_nicolas_peru_wrote">on 1 Apr 2015, 08:42:30 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>Then let&#8217;s just keep the first part of the sentence :</p>
</div>
<div class="quoteblock">
<blockquote>
<div class="literalblock">
<div class="content">
<pre>Nested enum types are implicitly static.</pre>
</div>
</div>
</blockquote>
</div>
<div class="paragraph">
<p>point is to refer to the correct section of the JLS to let user know we did not invent this.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_1_apr_2015_170642_ann_campbell_wrote">on 1 Apr 2015, 17:06:42 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Done [~nicolas.peru]</p>
</div>
</div>
</div>
</div>