<div class="paragraph">
<p>The <code>hasNext</code> method of an <code>Iterator</code> should only report on the state of the iterator, not change it. Making a change to an iterator in its <code>hasNext</code> method violates all expectations of what the method will do, and almost guarantees bad results when the iterator class is used.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public class MyItr implements Iterator&lt;MyClass&gt; {
  //...

  public boolean hasNext() {
    if (next() != null) {  // Noncompliant
      return true;
    }
    return false;
  }</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public class MyItr implements Iterator&lt;MyClass&gt; {
  private List&lt;MyClass&gt; list;
  private int index = 0;

  //...

  public boolean hasNext() {
    return index &lt; list.size();
  }</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_duplicates_s1849">duplicates: S1849</h3>

</div>
<div class="sect2">
<h3 id="_on_10_oct_2014_153037_freddy_mallet_wrote">on 10 Oct 2014, 15:30:37 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>I would prefer the following rule title @Ann:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>"Iterator.next()" method should never be called in the "Iterator.hasNext()" method</p>
</li>
</ul>
</div>
</div>
</div>
</div>