<div class="paragraph">
<p>The special method <a href="https://docs.python.org/3/reference/datamodel.html?highlight=__exit__%20special#object.__exit__"><code>__exit__</code></a> should only raise an exception when it fails. It should never raise the provided exception, it is the caller&#8217;s responsibility.</p>
</div>
<div class="paragraph">
<p>Raising this exception will make the stack trace difficult to understand.</p>
</div>
<div class="paragraph">
<p>The <code>__exit__</code> method can filter passed-in exceptions by simply returning True or False.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>an <code>__exit__</code> method has a bare <code>raise</code> outside of an <code>except</code> block.</p>
</li>
<li>
<p>an <code>__exit__</code> method raises the exception provided as parameter.</p>
</li>
</ul>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class MyContextManager:
    def __enter__(self):
        return self
    def __exit__(self, *args):
        raise  # Noncompliant
        raise args[2]  # Noncompliant

class MyContextManager:
    def __enter__(self):
        return self
    def __exit__(self, exc_type, exc_value, traceback):
        raise exc_value # Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class MyContextManager:
    def __enter__(self):
        return self
    def __exit__(self, exc_type, exc_value, traceback):
        # by default the function will return None, which is always False, and the exc_value will naturally raise.
        pass

class MyContextManager:
    def __enter__(self, stop_exceptions):
        return self
    def __exit__(self, *args):
        try:
            print("42")
        except:
            print("exception")
            raise  # No issue when raising another exception. The __exit__ method can fail and raise an exception
        raise MemoryError("No more memory")  # This is ok too.</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>Python documentation – <a href="https://docs.python.org/3/reference/datamodel.html?highlight=__exit__%20special#object.__exit__">The <code>__exit__</code> special method</a></p>
</li>
<li>
<p>PEP 343 – <a href="https://www.python.org/dev/peps/pep-0343/">The "with" Statement</a></p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>remove this "raise" statement and return "False" instead.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The "raise" statement.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s5747">is related to: S5747</h3>

</div>
</div>
</div>