<div class="paragraph">
<p>Using an <code>ORDER BY</code> clause in a <code>SELECT</code> helps ensure runtime reproducibility. Without one, <em>some</em> databases <em>may</em> return results in a consistent order, but it is likely to be one that appears random to users and is not optimal for processing the data.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>SELECT model   -- Noncompliant
FROM car
WHERE wheel_count = 4</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>SELECT model
FROM car
WHERE wheel_count = 4
ORDER BY year, make</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add an "ORDER BY" clause to this select.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s1592">is related to: S1592</h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s2241">is related to: S2241</h3>

</div>
<div class="sect2">
<h3 id="_on_14_jul_2015_145527_pierre_yves_nicolas_wrote">on 14 Jul 2015, 14:55:27 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>Looks good.</p>
</div>
</div>
</div>
</div>