<div class="paragraph">
<p>Be careful about your use of Oracle-specific data types like <code>ROWID</code> and <code>UROWID</code>. They might offer a slight improvement in performance over other means of identifying a single row (primary key or unique index value), but that is by no means guaranteed.</p>
</div>
<div class="paragraph">
<p>On the other hand, the use of <code>ROWID</code> or <code>UROWID</code> means that your SQL statement will not be portable to other SQL databases. Further, many developers are not familiar with these data types, which can make the code harder to maintain.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>SET SERVEROUTPUT ON

DECLARE
  id rowid; -- Non-Compliant
  universeId urowid; -- Non-Compliant
BEGIN
  SELECT rowid INTO id FROM DUAL;
  SELECT rowid INTO universeId FROM DUAL;

  DBMS_OUTPUT.PUT_LINE('id = ' || id);
  DBMS_OUTPUT.PUT_LINE('universe id = ' || universeId);
END;
/</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use a different data type.</p>
</div>
</div>
</div>
</div>