<div class="paragraph">
<p>When a loop variable is not used outside of a loop, it should be declared inside the loop declaration:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>It improves readability, the scope of the variable is clearly defined</p>
</li>
<li>
<p>It reduces the number of mistakes, the variable can&#8217;t be accidentally misused outside of the loop</p>
</li>
<li>
<p>Resources are not retained longer than necessary</p>
</li>
</ul>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>using namespace std;

void f() {
  int i = 0; // Noncompliant: i is not used outside of the loop
  for (i = 0; i &lt; 10; ++i) {
    cout &lt;&lt; i &lt;&lt; endl;
  }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>using namespace std;

void f() {
  for (int i = 0; i &lt; 10; ++i) {
    cout &lt;&lt; i &lt;&lt; endl;
  }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://github.com/isocpp/CppCoreGuidelines/blob/c553535fb8dda2839d13ab5f807ffbc66b63d67b/CppCoreGuidelines.md#es5-keep-scopes-small">C&#43;&#43; Core Guidelines ES.5</a> - Keep scopes small</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Declare the variable XXX inside the loop.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s806">is related to: S806</h3>

</div>
</div>
</div>