<div class="paragraph">
<p>Web applications can be made available in multiple languages through the use of internationalization. This allows the server to plug in the correct version of a piece of text based on the language chosen, but it requires that internationalization messages be used instead of hard-coded text.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>  &lt;form method="post"&gt;
    &lt;label for="username"&gt;Username:&lt;/label&gt;  &lt;!-- Noncompliant --&gt;
    &lt;input type="text" id="username" name="username"&gt;
    &lt;br&gt;
    &lt;label for="password"&gt;Password:&lt;/label&gt;  &lt;!-- Noncompliant --&gt;
    &lt;input type="password" id="password" name="password"&gt;
    &lt;br&gt;
    &lt;input type="submit" name="submit" value="${buttonValue}"&gt;
  &lt;/form&gt;</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>  &lt;form method="post"&gt;
    &lt;label for="username"&gt;&lt;fmt:message key="login.label.username" /&gt;:&lt;/label&gt;
    &lt;input type="text" id="username" name="username"&gt;
    &lt;br&gt;
    &lt;label for="password"&gt;&lt;fmt:message key="login.label.password" /&gt;:&lt;/label&gt;
    &lt;input type="password" id="password" name="password"&gt;
    &lt;br&gt;
    &lt;input type="submit" name="submit" value="${buttonValue}"&gt;
  &lt;/form&gt;</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Define this label in the resource bundle.</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="ulist">
<ul>
<li>
<p>key = attributes</p>
<div class="ulist">
<ul>
<li>
<p>description: Attributes</p>
</li>
<li>
<p>default: outputLabel.value, outputText.value</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
</div>
</div>
</div>