<div class="paragraph">
<p>A dynamic class defines an object that can be altered at run time by adding or changing properties and methods. This extremely powerful mechanism should be used very carefully, and only in very limited use cases.</p>
</div>
<div class="paragraph">
<p>Indeed, by definition dynamic classes make refactoring difficult and prevent the compiler from raising potential errors at compile time.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>dynamic public class DynamicFoo
{...}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public class Foo  //Note that the class has been renamed to avoid confusion
{...}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Make this "XXXX" class non-dynamic</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_31_oct_2013_153924_freddy_mallet_wrote">on 31 Oct 2013, 15:39:24 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3225 for Flex</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_3_nov_2013_093858_ann_campbell_wrote">on 3 Nov 2013, 09:38:58 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Double-check my Compliant Solution</p>
</div>
</div>
</div>
</div>