<div class="paragraph">
<p>Shared coding conventions allow teams to collaborate efficiently. This rule checks that first level data item names match a provided regular expression.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Given an regular expression of <code>WS-.*</code>:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>       WORKING-STORAGE SECTION.
            01 WRONG.                                       &gt; Noncompliant; name doesn't match the pattern "WS-.*"
              02  LINK.                                     &gt; Compliant; this is not first level

       LINKAGE SECTION.
             01     DFHCOMMAREA PIC X(1500).                &gt; Compliant; the data item is defined in the LINKAGE SECTION</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>       WORKING-STORAGE SECTION.
            01 WS-LINK.
              02  LINK.

       LINKAGE SECTION.
             01     DFHCOMMAREA PIC X(1500).</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="ulist">
<ul>
<li>
<p>key = regexPatternString</p>
</li>
<li>
<p>description: A regular expression to specify the data item naming convention</p>
</li>
</ul>
</div>
</div>
</div>
</div>