<div class="paragraph">
<p>Loggers should be:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>private</code>: not accessible outside of their parent classes. If another class needs to log something, it should instantiate its own logger.</p>
</li>
<li>
<p><code>static</code>: not dependent on an instance of a class (an object). When logging something, contextual information can of course be provided in the messages but the logger should be created at class level to prevent creating a logger along with each object.</p>
</li>
<li>
<p><code>const</code>: created once and only once per class.</p>
</li>
</ul>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="paragraph">
<p>With the default regular expression <code>LOG(?:GER)?</code>:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>public const logger:ILogger = LogUtil.getLogger(MyClass);</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>private static const LOG:ILogger = LogUtil.getLogger(MyClass);</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Make the logger "XXXX" private static [final|const] and rename it to comply with the format "{}".</p>
</div>
<div class="paragraph">
<p>Make the logger "XXXX" private static [final|const].</p>
</div>
<div class="paragraph">
<p>Rename the "XXX" logger to comply with the format "{}".</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="ulist">
<ul>
<li>
<p>key = format</p>
</li>
<li>
<p>Default value : LOG(?:GER)?</p>
</li>
<li>
<p>Description: Regular expression used to check the logger names against.</p>
</li>
<li>
<p>type: String</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_22_nov_2013_125331_freddy_mallet_wrote">on 22 Nov 2013, 12:53:31 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3280 for Flex</p>
</div>
</div>
</div>
</div>