<div class="paragraph">
<p>When using the Backbone.js framework, the internal hash containing the model&#8217;s state, <code>model.attributes</code>, should not be accessed directly. Doing so bypasses the listeners set on the model and could put your application in a bad state.</p>
</div>
<div class="paragraph">
<p>Instead, you should use <code>set</code> to update, <code>get</code> to read, and <code>_.clone(model.attributes)</code> to obtain a copy.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>myModel.attributes.someAttribute = 1;</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>myModel.set({ someAttribute: 1 });</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use "(set|get)" instead.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_20_may_2015_123322_linda_martin_wrote">on 20 May 2015, 12:33:22 Linda Martin wrote:</h3>
<div class="paragraph">
<p>ok!</p>
</div>
</div>
</div>
</div>