<div class="paragraph">
<p><code>sizeof()</code> is an alias of <code>count()</code> so the two offer the same results, but <code>sizeof()</code> has a different meaning in other languages and may confuse developers who are new to PHP.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>$fruit=("apple","pear","raspberry","gooseberry","red currant");
echo sizeof($fruit);  // Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>$fruit=("apple","pear","raspberry","gooseberry","red currant");
echo count($fruit);</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use "count()" instead.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_20_jan_2015_075955_ann_campbell_wrote">on 20 Jan 2015, 07:59:55 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Source: https://twitter.com/declaassen/status/555301362846937088</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_19_may_2015_155710_linda_martin_wrote">on 19 May 2015, 15:57:10 Linda Martin wrote:</h3>
<div class="paragraph">
<p>OK!</p>
</div>
</div>
</div>
</div>