<div class="paragraph">
<p>The standard order for <code>using</code> directives is alphabetic with the exception of <code>System</code> directives, which come first for higher visibility. Using a different order may cause maintainers to overlook a directive or misunderstand what&#8217;s being used.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>using C;  // Noncompliant
using System.A.A;
using A;
using D;
using B;
using System;
using System.A;
using System.B;</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>using System;
using System.A;
using System.A.A;
using System.B;
using A;
using C;
using B;
using D;</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Reorder these "using" directives to put "System" directives first, then sub-sort alphabetically.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>first out-of-order directive</p>
</div>
</div>
</div>
</div>