<div class="paragraph">
<p>Views should not be responsible for directly triggering coroutines. Hence, <code>ViewModel</code> classes should prefer creating coroutines instead of exposing suspending functions to perform some piece of business logic. This allows for easier testing of your application, as <code>ViewModel</code> classes can be unit tested, whereas views require instrumentation tests.</p>
</div>
<div class="paragraph">
<p>Please refer to the <a href="https://developer.android.com/kotlin/coroutines/coroutines-best-practices#viewmodel-coroutines">Android docs</a> for more advanced examples and mechanisms of updating the views with data generated asynchronously.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when suspending functions are exposed by classes inheriting from <code>ViewModel</code>.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class MyViewModel : ViewModel() {
    suspend fun performAction() = suspendingWorker()
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class MyViewModel : ViewModel() {
    fun performAction() =
        viewModelScope.launch {
            suspendingWorker()
        }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://developer.android.com/kotlin/coroutines/coroutines-best-practices#viewmodel-coroutines">The ViewModel should create coroutines</a> (Android coroutines best practices)</p>
</li>
</ul>
</div>
</div>
</div>