<div class="paragraph">
<p>Jump statements, such as <code>return</code>, <code>break</code> and <code>continue</code> let you change the default flow of program execution, but jump statements that direct the control flow to the original direction are just a waste of keystrokes.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>function redundantJump(x) {
  if (x == 1) {
    console.log("x == 1");
    return; // Noncompliant
  }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>function redundantJump(x) {
  if (x == 1) {
    console.log("x == 1");
  }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_exceptions">Exceptions</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>break</code> and <code>return</code> inside <code>switch</code> statement are ignored, because they are often used for consistency. <code>continue</code> with label is also ignored, because label is usually used for clarity. Also a jump statement being a single statement in a block is ignored.</p>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this redundant jump.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>statement</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s1751">is related to: S1751</h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s3031">is related to: S3031</h3>

</div>
<div class="sect2">
<h3 id="_on_2_jun_2016_134753_tamas_vajk_wrote">on 2 Jun 2016, 13:47:53 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I drafted the rule, let me know what you think.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_2_jun_2016_140507_ann_campbell_wrote">on 2 Jun 2016, 14:05:07 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Note that there is a potential overlap of this rule with RSPEC-1751: unconditional <code>continue</code> at the end of a loop.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_2_jun_2016_140522_ann_campbell_wrote">on 2 Jun 2016, 14:05:22 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>looks fine [~tamas.vajk]</p>
</div>
</div>
</div>
</div>