<div class="paragraph">
<p>Jump statements, such as <code>return</code>, <code>break</code> and <code>continue</code> let you change the default flow of program execution, but jump statements that direct the control flow to the original direction are just a waste of keystrokes.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>def redundant_jump(x):
    if x == 1:
        print(True)
        return  # NonCompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>def redundant_jump(x):
    if x == 1:
        print(True)</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_exceptions">Exceptions</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>No issue is raised if the jump statement is the only statement of a statement suite:</p>
</li>
</ul>
</div>
<div class="listingblock">
<div class="content">
<pre>def my_function(x):
    if x &gt; 5:
        do_something()
    elif x == 0:
        return # ok even it could be changed to "pass"
    else:
        do_something_else()</pre>
</div>
</div>
<div class="ulist">
<ul>
<li>
<p>No issue is raised for <code>return None</code> because this was certainly done on purpose to be explicit that a function is really returning <code>None</code>.</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this redundant jump.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>statement</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s1751">is related to: S1751</h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s3031">is related to: S3031</h3>

</div>
<div class="sect2">
<h3 id="_on_2_jun_2016_134753_tamas_vajk_wrote">on 2 Jun 2016, 13:47:53 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I drafted the rule, let me know what you think.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_2_jun_2016_140507_ann_campbell_wrote">on 2 Jun 2016, 14:05:07 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Note that there is a potential overlap of this rule with RSPEC-1751: unconditional <code>continue</code> at the end of a loop.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_2_jun_2016_140522_ann_campbell_wrote">on 2 Jun 2016, 14:05:22 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>looks fine [~tamas.vajk]</p>
</div>
</div>
</div>
</div>