<div class="paragraph">
<p><code>COPY ... SUPPRESS</code> suppresses the inclusion of the copybook contents from the source listing, making it very difficult to gain a complete understanding of what&#8217;s happening in the code. This could hinder both maintenance and debugging.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>COPY XX001234 SUPPRESS.  &lt;* Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>COPY XX001234.</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Don&#8217;t "SUPPRESS" inclusion of this library in the code listing.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p><code>SUPPRESS</code></p>
</div>
</div>
</div>
</div>