<div class="paragraph">
<p>Methods with a <code>void</code> return type that neither alter their objects' state nor the states of their arguments are effectively useless. They should either be eliminated, or - more likely - corrected to include the missing side effects.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public void whatsThePoint(List&lt;Integer&gt; ints) {  // Noncompliant
  int sum = 0;
  for (Integer i : ints) {
    sum += i.intValue();
  }
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this useless method or refactor it to add the missing side effects.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_18_jun_2015_183615_nicolas_peru_wrote">on 18 Jun 2015, 18:36:15 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>updated title</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_18_jun_2015_183721_ann_campbell_wrote">on 18 Jun 2015, 18:37:21 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>looks fine [~nicolas.peru]</p>
</div>
</div>
</div>
</div>