<div class="paragraph">
<p>Changing the configuration of database options <code>ANSI_NULLS</code>, <code>ANSI_PADDING</code> and <code>CONCAT_NULL_YIELDS_NULL</code> is deprecated. The future versions of SQL Server will only support the <code>ON</code> value, and the <code>SET</code> statement for those options to <code>OFF</code> will eventually generate an error.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>SET ANSI_NULLS OFF -- Noncompliant
SELECT column1 FROM table1 WHERE id = NULL

SET ANSI_PADDING OFF -- Noncompliant
SET CONCAT_NULL_YIELDS_NULL ON -- Noncompliant

SET ANSI_PADDING ON -- "ON" is ignored</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>SELECT column1 FROM table1 WHERE id IS NULL</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this use of "[ANSI_NULLS|ANSI_PADDING|CONCAT_NULL_YIELDS_NULL]".</p>
</div>
</div>
</div>
</div>