<div class="paragraph">
<p><em>Mutexes</em> are synchronization primitives that allow to manage concurrency.</p>
</div>
<div class="ulist">
<ul>
<li>
<p><em>non recursive mutexes</em> are targeted by this rule. They can be locked/unlocked only once. Any locking/unlocking sequence that contains two consecutive identical operations leads to an undefined behaviour.</p>
</li>
<li>
<p><em>recursive mutexes</em> are not target by this rule.  They can be locked several times and unlocked several times as long as the number of locks/unlocks is the same.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>This rule raises an issue when a <code>pthread_mutex_t</code> is locked or unlocked several times in a row. We assume that all <code>pthread_mutex_t</code> are non-recursive (this is the most common case).</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>pthread_mutex_t mtx1;

void bad1(void)
{
  pthread_mutex_lock(&amp;mtx1);
  pthread_mutex_lock(&amp;mtx1);
}

void bad2(void)
{
  pthread_mutex_unlock(&amp;mtx1);
  pthread_mutex_unlock(&amp;mtx1);
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>pthread_mutex_t mtx1;

void ok(void)
{
  pthread_mutex_lock(&amp;mtx1);
  pthread_mutex_unlock(&amp;mtx1);
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://pubs.opengroup.org/onlinepubs/009695399/functions/pthread_mutex_destroy.html">The Open Group</a> pthread_mutex_init, pthread_mutex_destroy</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s5487">relates to: S5487</h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s5489">relates to: S5489</h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s5487">is related to: S5487</h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s5489">is related to: S5489</h3>

</div>
<div class="sect2">
<h3 id="_on_6_nov_2019_233248_loïc_joly_wrote">on 6 Nov 2019, 23:32:48 Loïc Joly wrote:</h3>
<div class="paragraph">
<p>\[~geoffray.adde]Can you please review my changes?</p>
</div>
</div>
</div>
</div>