<div class="paragraph">
<p>A generator without a <code>yield</code> statement is at best confusing, and at worst a bug in your code, since the iterator produced by your code will always be empty.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>function* myGen(a, b) {  // Noncompliant
  let answer = 0;
  answer += a * b;
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>function* myGen(a, b) {
  let answer = 0;
  while (answer &lt; 42) {
    answer += a * b;
    yield answer;
  }
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add a "yield" statement to this generator.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p><code>function* xxx</code></p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_5_feb_2016_155249_elena_vilchik_wrote">on 5 Feb 2016, 15:52:49 Elena Vilchik wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Could you add to description smth like "It even could be a bug in your problem as you your iterator produced by this generator function is always empty"?</p>
</div>
<div class="paragraph">
<p>And I would like to remove highlighting and specific message for return, I don&#8217;t think it&#8217;s worth that. Are you ok?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_5_feb_2016_160615_ann_campbell_wrote">on 5 Feb 2016, 16:06:15 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>done [~elena.vilchik]</p>
</div>
</div>
</div>
</div>