<div class="paragraph">
<p>The use of unnecessary types makes the eye stumble, and inhibits the smooth reading of code.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public void doSomething() {

  &lt;String&gt;foo("blah");  // Noncompliant; &lt;String&gt; is inferred
}
public void &lt;T&gt; foo(T t) {
  // ...
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public void doSomething() {

  foo("blah");
}
public void &lt;T&gt; foo(T t) {
  // ...
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this unneeded "xxx" parameter.</p>
</div>
</div>
</div>
</div>