<div class="paragraph">
<p>OS/VS COBOL accepted the <code>NOTE</code> statement, but IBM Enterprise COBOL does not. Therefore all <code>NOTE</code> statements should be replaced by standard comment lines.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>       IDENTIFICATION DIVISION.
       PROGRAM-ID. foo.

       PROCEDURE DIVISION.
      * Non-Compliant
         NOTE This is a comment.

      * This is a compliant comment.
       END PROGRAM foo.</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>       IDENTIFICATION DIVISION.
       PROGRAM-ID. foo.

       PROCEDURE DIVISION.
      * Compliant
      * This is a comment.
       END PROGRAM foo.</pre>
</div>
</div>
</div>
</div>