<div class="paragraph">
<p>Omitting a join condition is a subtle, yet classic blunder. Do so, and you&#8217;re guaranteed not to get the results you intended.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when a table is included in the <code>FROM</code> list but not in the <code>WHERE</code> list in relation to other tables, and when only part of a compound key is used in the conditions.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>SELECT p.name, f.name
FROM product p,
  product_feature f  -- Noncompliant
WHERE p.id=?

SELECT p.name, ck.date
FROM product p
JOIN compound_key_table ck
  ON p.manufacturer_product_id=ck.manufacturer_product_id  -- Noncompliant
WHERE ...</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>SELECT p.name, f.name
FROM product p,
  product_feature f
WHERE p.id=f.product_id AND p.id=?

SELECT p.name, ck.date
FROM product p
JOIN compound_key_table ck
  ON p.manufacturer_product_id=ck.manufacturer_product_id AND p.manufacturer_id=ck.manufacturer_id
WHERE ...</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>"ttt" has a composite key, but not all components are included in this "JOIN".</p>
</li>
<li>
<p>"ttt" is included in the table list, but no join conditions are used.</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>name of table with missing/incomplete condition</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_17_jun_2016_113340_ann_campbell_wrote">on 17 Jun 2016, 11:33:40 Ann Campbell wrote:</h3>
<div class="paragraph">
<p><a href="https://www.researchgate.net/publication/222432151_Semantic_errors_in_SQL_queries_A_quite_complete_list" class="bare">https://www.researchgate.net/publication/222432151_Semantic_errors_in_SQL_queries_A_quite_complete_list</a></p>
</div>
</div>
<div class="sect2">
<h3 id="_on_17_jun_2016_150344_ann_campbell_wrote">on 17 Jun 2016, 15:03:44 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>from https://www.researchgate.net/publication/222432151_Semantic_errors_in_SQL_queries_A_quite_complete_list:</p>
</div>
<div class="paragraph">
<p>Error 27: Uncorrelated EXISTS-subqueries.If an EXISTS-subquery makes no reference to a tuple variable from theouter query, is is either globally true or globally false. This is a very unusualbehaviour. Actually, uncorrelated EXISTS-subqueries are simply missing joinconditions (possibly for anti-joins).</p>
</div>
<div class="paragraph">
<p>Not sure if we want to handle this here or in RSPEC-3651</p>
</div>
</div>
</div>
</div>