<div class="paragraph">
<p>Using the <code>assert</code> statement on a tuple literal will always fail if the tuple is empty, and always succeed otherwise.</p>
</div>
<div class="paragraph">
<p>The <code>assert</code> statement does not have parentheses around its parameters. Calling <code>assert(x, y)</code> will test if the tuple <code>(x, y)</code> is True, which is always the case.</p>
</div>
<div class="paragraph">
<p>There are two possible fixes:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>If your intention is to test the first value of the tuple and use the second value as a message, simply remove the parentheses.</p>
</li>
<li>
<p>If your intention is to check that every element of the tuple is <code>True</code>, test each value separately.</p>
</li>
</ul>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>def test_values(a, b):
    assert (a, b)  # Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>def test_values(a, b):
    # If you mean to test "a" and use "b" as an error message
    assert a, b

    # If you mean to test the values of "a" and "b"
    assert a and b</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.python.org/3/reference/simple_stmts.html#the-assert-statement">Python documentation - The <code>assert</code> statement</a></p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Fix this assertion on a tuple literal; did you mean "assert A, B".</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The tuple parameter</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_deprecates_s1721">deprecates: S1721</h3>

</div>
</div>
</div>