<div class="paragraph">
<p>There is no need to declare a type parameter when naming a type constraint is not required. Using wildcards makes it easier to read.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>&lt;T extends MyClass&gt; void foo(List&lt;T&gt; list) {  // Noncompliant, T is used only once
  for (MyClass myObj : list) {
    doSomething(myObj);
  }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>void foo(List&lt;? extends MyClass&gt; list) {
  for (MyClass myObj : list) {
    doSomething(myObj);
  }
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this type parameter and use a wildcard instead</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Type parameter declaration</p>
</div>
</div>
</div>
</div>