<div class="paragraph">
<p>String literals written on multiple lines should not be used, especially since the new line characters are <strong>not</strong> part of the string anyway. The concatenation operator <code>||</code> should be used instead. Moreover, if trailing comments are present, they can be automatically removed by text editors and lead to unexpected results.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>foo: proc options(main);
  put list ('Hello,      /* Noncompliant; trailing space is not readable and may be stripped */
world');
end;</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>foo: proc options(main);
  put list ('Hello, ' ||
'world');
end;</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Consolidate this string declaration to one line or use the concatenation operator.</p>
</div>
</div>
</div>
</div>