<div class="paragraph">
<p>WordPress has a database repair and optimization mode that can be activated by setting <code>WP_ALLOW_REPAIR</code> to <code>true</code> in the configuration.</p>
</div>
<div class="paragraph">
<p>If activated, the repair page can be accessed by any user, authenticated or not. This makes sense because if the database is corrupted, the authentication mechanism might not work.</p>
</div>
<div class="paragraph">
<p>Malicious users could trigger this potentially costly operation repeatadly slowing down the website, and making it unavailable.</p>
</div>
<div class="sect1">
<h2 id="_ask_yourself_whether">Ask Yourself Whether</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>The database is not currently corrupted.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>There is a risk if you answered yes to this question.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_recommended_secure_coding_practices">Recommended Secure Coding Practices</h2>
<div class="sectionbody">
<div class="paragraph">
<p>It&#8217;s recommended to enable automatic database repair mode only in case of database corruption. This feature should be deactivated again when the database issue is resolved.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>define( 'WP_ALLOW_REPAIR', true ); // Sensitive</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>// The default value is false, so the value does not have to be expilicitly set.
define( 'WP_ALLOW_REPAIR', false );</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://owasp.org/Top10/A05_2021-Security_Misconfiguration/">OWASP Top 10 2021 Category A5</a> - Security Misconfiguration</p>
</li>
<li>
<p><a href="https://owasp.org/Top10/A07_2021-Identification_and_Authentication_Failures/">OWASP Top 10 2021 Category A7</a> - Identification and Authentication Failures</p>
</li>
<li>
<p><a href="https://wordpress.org/support/article/editing-wp-config-php/#automatic-database-optimizing">wordpress.org</a> - Automatic Database Optimizing</p>
</li>
<li>
<p><a href="https://owasp.org/www-project-top-ten/2017/A6_2017-Security_Misconfiguration.html">OWASP Top 10 2017 Category A6</a> - Security Misconfiguration</p>
</li>
</ul>
</div>
</div>
</div>