<div class="paragraph">
<p>When constraints have not been applied to restrict a generic type parameter to be a reference type, then a value type, such as a <code>struct</code>, could also be passed. In such cases, comparing the type parameter to <code>null</code> would always be false, because a <code>struct</code> can be empty, but never <code>null</code>. If a value type is truly what&#8217;s expected, then the comparison should use <code>default()</code>. If it&#8217;s not, then constraints should be added so that no value type can be passed.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>private bool IsDefault&lt;T&gt;(T value)
{
  if (value == null) // Noncompliant
  {
    // ...
  }
  // ...
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>private bool IsDefault&lt;T&gt;(T value)
{
  if(object.Equals(value, default(T)))
  {
    // ...
  }
  // ...
}</pre>
</div>
</div>
<div class="paragraph">
<p>or</p>
</div>
<div class="listingblock">
<div class="content">
<pre>private bool IsDefault&lt;T&gt;(T value) where T : class
{
  if (value == null)
  {
    // ...
  }
  // ...
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use a comparison to "default(xxx)" instead or add a constraint to "xxx" so that it can&#8217;t be a value type.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_22_may_2015_100425_tamas_vajk_wrote">on 22 May 2015, 10:04:25 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>Fixed some minor wording issues, and the sample</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_22_may_2015_120732_ann_campbell_wrote">on 22 May 2015, 12:07:32 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Thanks [~tamas.vajk]. Looks good.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_29_may_2015_125040_tamas_vajk_wrote">on 29 May 2015, 12:50:40 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Could you run through the description? I&#8217;ve change the wording "false negative" because it sounded strange.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_29_may_2015_145044_ann_campbell_wrote">on 29 May 2015, 14:50:44 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>looks good [~tamas.vajk]</p>
</div>
</div>
</div>
</div>