<div class="paragraph">
<p>If a <code>RuntimeException</code> is thrown while code is unit tested, it&#8217;s likely to be thrown again in production. Therefore, test methods should be allowed to ``throw RuntimeException``s so you&#8217;re alerted early to problems in the code.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>@Test
public void testTheThing() {
  try {
    // ...
  } catch (RuntimeException e) {  // Noncompliant
    // ...
  }
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_11_dec_2015_204742_ann_campbell_wrote">on 11 Dec 2015, 20:47:42 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>fyi [~tamas.vajk], I&#8217;ve targeted this to C# too.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_19_nov_2020_140438_nicolas_harraudeau_wrote">on 19 Nov 2020, 14:04:38 Nicolas Harraudeau wrote:</h3>
<div class="paragraph">
<p>It&#8217;s not clear what this rule tries to detect. It could be either detecting when RuntimeExceptions are ignored, or detecting when a more specific exception type should be caught. Rule RSPEC-2486 covers the first case, and rule RSPEC-5958 covers the second case. Thus we don&#8217;t seem to need RSPEC-3477.</p>
</div>
</div>
</div>
</div>