<div class="paragraph">
<p>Java 7 introduced the try-with-resources statement, which guarantees that the resource in question will be closed. Since the new syntax is closer to bullet-proof, it should be preferred over the older <code>try</code>/<code>catch</code>/<code>finally</code> version.</p>
</div>
<div class="paragraph">
<p>This rule checks that <code>close</code>-able resources are opened in a try-with-resources statement.</p>
</div>
<div class="paragraph">
<p><strong>Note</strong> that this rule is automatically disabled when the project&#8217;s <code>sonar.java.source</code> is lower than <code>7</code>.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>FileReader fr = null;
BufferedReader br = null;
try {
  fr = new FileReader(fileName);
  br = new BufferedReader(fr);
  return br.readLine();
} catch (...) {
} finally {
  if (br != null) {
    try {
      br.close();
    } catch(IOException e){...}
  }
  if (fr != null ) {
    try {
      br.close();
    } catch(IOException e){...}
  }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>try (
    FileReader fr = new FileReader(fileName);
    BufferedReader br = new BufferedReader(fr)
  ) {
  return br.readLine();
}
catch (...) {}</pre>
</div>
</div>
<div class="paragraph">
<p>or</p>
</div>
<div class="listingblock">
<div class="content">
<pre>try (BufferedReader br =
        new BufferedReader(new FileReader(fileName))) { // no need to name intermediate resources if you don't want to
  return br.readLine();
}
catch (...) {}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://wiki.sei.cmu.edu/confluence/x/6DZGBQ">CERT, ERR54-J.</a> - Use a try-with-resources statement to safely handle closeable resources</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Change this "try" to a try-with-resources.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_12_oct_2014_182626_freddy_mallet_wrote">on 12 Oct 2014, 18:26:26 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Minor point @Ann but I would associate the tag 'bug' to this rule.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_12_oct_2014_221837_ann_campbell_wrote">on 12 Oct 2014, 22:18:37 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>I disagree [~freddy.mallet]. Properly written (and we have other rules to catch if it&#8217;s not) there&#8217;s no bug.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_17_oct_2014_101750_freddy_mallet_wrote">on 17 Oct 2014, 10:17:50 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Ok [~ann.campbell.2]</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_6_jan_2016_101615_nicolas_peru_wrote">on 6 Jan 2016, 10:16:15 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>Adding to default profile as we can rely on detection of java version.</p>
</div>
</div>
</div>
</div>