<div class="paragraph">
<p>Developers can use type hints to specify which type a function is expected to return. These annotations are not enforced at runtime and returning a different type might not fail. It is however likely to be unintended and will lead to maintainability issues, if not bugs.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when a function or method returns a value that contradicts its type hint.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>def hello() -&gt; str:
    return 42  # Noncompliant. Function's type hint asks for a string return value</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>def hello() -&gt; str:
    return "Hello"</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.python.org/3/library/typing.html">Python documentation - Support for type hints</a></p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>*Return a "XXX" instead of a "YYY" or update function "ZZZ" type hint.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>If the function returns the wrong type:</p>
<div class="ulist">
<ul>
<li>
<p>Primary: The return statement</p>
</li>
<li>
<p>Â Secondaries: 1. the function name, 2. The type hint</p>
</li>
</ul>
</div>
</li>
<li>
<p>If the function might terminate without reaching a return statement:</p>
<div class="ulist">
<ul>
<li>
<p>Primary Location: The function name</p>
</li>
<li>
<p>Secondary: the type hint</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
</div>
</div>
</div>