<div class="paragraph">
<p><code>StringBuilder</code> is more efficient than string concatenation, especially when the operator is repeated over and over as in loops.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>string str = "";
for (int i = 0; i &lt; arrayOfStrings.Length ; ++i)
{
  str = str + arrayOfStrings[i];
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>StringBuilder bld = new StringBuilder();
for (int i = 0; i &lt; arrayOfStrings.Length; ++i)
{
  bld.Append(arrayOfStrings[i]);
}
string str = bld.ToString();</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use a StringBuilder instead.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_4_jun_2015_142927_ann_campbell_wrote">on 4 Jun 2015, 14:29:27 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~tamas.vajk], startlingly the name of the preferred replacement class is the same in Java and C#, so I simply removed "in Java" from the description because I&#8217;m guessing the same code samples would work too&#8230;&#8203;?</p>
</div>
<div class="paragraph">
<p>If it doesn&#8217;t work, or if the bit about performance degrading as the string gets longer is not true in C#, let me know and I&#8217;ll create a sub-task.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_29_jun_2015_141555_tamas_vajk_wrote">on 29 Jun 2015, 14:15:55 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Do we care about letter casing mismatch? In C# <code>.toString()</code> is <code>.ToString()</code>, <code>.append()</code> is <code>.Append()</code>.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_30_jun_2015_123634_ann_campbell_wrote">on 30 Jun 2015, 12:36:34 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~tamas.vajk], we do care. I&#8217;ve created a subtask</p>
</div>
</div>
</div>
</div>