<div class="paragraph">
<p>It is possible in the same statement, to declare a user-defined type (<code>class</code>, <code>struct</code>, <code>union</code> or <code>enum</code>) followed by variable declarations of this type. But mixing more than one concern in a single statement is confusing for maintainers.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when a variable is declared at the end of a user-defined type declaration statement.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>struct Container { int size; } container; // Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>struct Container { int size; };
Container container;</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://github.com/isocpp/CppCoreGuidelines/blob/036324/CppCoreGuidelines.md#Rc-standalone">C&#43;&#43; Core Guidelines C.7</a> - Don&#8217;t define a class or enum and declare a variable of its type in the same statement</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Declare this variable in a separate statement.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>variable name</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_17_jun_2016_164347_ann_campbell_wrote">on 17 Jun 2016, 16:43:47 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~alban.auzeill] I&#8217;ve simplified the code samples. Please double-check me.</p>
</div>
</div>
</div>
</div>