<div class="paragraph">
<p><code>std::string</code> and <code>std::string_view</code> find-related methods return <code>npos</code> if the substring was not found. Therefore, their returned values should never be treated as boolean. This rule raises an issue when objects returned from the following methods are treated as if they were of boolean type:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>find</code></p>
</li>
<li>
<p><code>rfind</code></p>
</li>
<li>
<p><code>find_first_of</code></p>
</li>
<li>
<p><code>find_last_of</code></p>
</li>
<li>
<p><code>find_first_not_of</code></p>
</li>
<li>
<p><code>find_last_not_of</code></p>
</li>
</ul>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>void f() {
  std::string s = "";
  if (s.find("42")) { // Noncompliant
    // this branch is taken even if "s" doesn't contain "42"
  }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>void f() {
  std::string s = "";
  if (s.find("42") != std::string::npos) {
    // this branch is correctly not taken
  }
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Compare the method result with std::(string|string_view)::npos.</p>
</div>
</div>
</div>
</div>