<div class="paragraph">
<p>Sharing some naming conventions is a key point to make it possible for a team to efficiently collaborate. This rule allows to check that all variable names match a provided regular expression.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>foo: proc options(main);
  declare bar# fixed decimal; /* Noncompliant */
end;</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>foo: proc options(main);
  declare bar fixed decimal;  /* Compliant */
end;</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Rename this variable "XXXX" to match the regular expression ${format}</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="ulist">
<ul>
<li>
<p>key = regularExpression</p>
</li>
<li>
<p>description = Regular expression used to check the names against.</p>
</li>
<li>
<p>default = [a-zA-Z0-9_]&#43;&#43;</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s117">relates to: S117</h3>

</div>
<div class="sect2">
<h3 id="_on_18_jan_2019_153336_alexandre_gigleux_wrote">on 18 Jan 2019, 15:33:36 Alexandre Gigleux wrote:</h3>
<div class="paragraph">
<p>For PL/SQL, this one is already covered by RSPEC-117</p>
</div>
</div>
</div>
</div>