<div class="paragraph">
<p>There&#8217;s no need to end a <code>void</code> method with a <code>return</code> statement. Since nothing is actually being returned, the <code>return</code> can be omitted; <code>void</code> will be returned implicitly.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public void incrementCount() {
  this.count++;
  return;  // Noncompliant
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public void incrementCount() {
  this.count++;
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this useless "return" statement.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s3626">relates to: S3626</h3>

</div>
<div class="sect2">
<h3 id="_on_13_oct_2015_073909_nicolas_peru_wrote">on 13 Oct 2015, 07:39:09 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>fine by me.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_9_jan_2018_162937_amaury_levé_wrote">on 9 Jan 2018, 16:29:37 Amaury Levé wrote:</h3>
<div class="paragraph">
<p>Implemented in RSPEC-3626 for C#</p>
</div>
</div>
</div>
</div>