<div class="paragraph">
<p>Queries with contradictory <code>WHERE</code> clauses will always return empty result sets. This is clearly a bug.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>SELECT *
FROM fruit
WHERE type='apple' AND type='orange'  -- Noncompliant</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>"XXX" cannot be both "yyyy" and "zzzz".</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>*primary: <code>WHERE</code></p>
</div>
<div class="ulist">
<ul>
<li>
<p>secondary: each instance of "XXX"</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s2583">relates to: S2583</h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s3651">is related to: S3651</h3>

</div>
<div class="sect2">
<h3 id="_on_15_jun_2016_101656_ann_campbell_wrote">on 15 Jun 2016, 10:16:56 Ann Campbell wrote:</h3>
<div class="paragraph">
<p><a href="http://users.informatik.uni-halle.de/~brass/sqllint/semerr2.pdf" class="bare">http://users.informatik.uni-halle.de/~brass/sqllint/semerr2.pdf</a></p>
</div>
</div>
</div>
</div>