<div class="paragraph">
<p>In addition to being obtuse from a syntax perspective, function constructors are also dangerous: their execution evaluates the constructor&#8217;s string arguments similar to the way <code>eval</code> works, which could expose your program to random, unintended code which can be both slow and a security risk.</p>
</div>
<div class="paragraph">
<p>In general it is better to avoid it altogether, particularly when used to parse JSON data. You should use ECMAScript 5&#8217;s built-in JSON functions or a dedicated library.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>var obj =  new Function("return " + data)();  // Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>var obj = JSON.parse(data);</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_exceptions">Exceptions</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Function calls where the argument is a string literal (e.g. <code>(Function('return this'))()</code>) are ignored.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>OWASP Top 10 2017 Category A1 - Injection</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Review this "Function" call and make sure its arguments are properly validated.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>primary: <code>new Function</code></p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_31_aug_2018_153545_nicolas_harraudeau_wrote">on 31 Aug 2018, 15:35:45 Nicolas Harraudeau wrote:</h3>
<div class="paragraph">
<p>Detecting <code>new Function("...")</code> should be added to the Hotspot rule RSPEC-1523</p>
</div>
</div>
</div>
</div>