<div class="paragraph">
<p>It is far faster to compare the length of a string to 0 than it is to compare the string itself to empty string. Further, the <code>LenB</code> implementation is faster than the <code>Len</code> implementation, and is therefore preferred.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>If Text = "" Then
If Text &lt;&gt; "" Then</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>If LenB(Text) = 0 Then
If LenB(Text) &lt;&gt; 0 Then</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Test the length of "xxx" against 0</p>
</div>
</div>
</div>
</div>