<div class="paragraph">
<p>In a Zen-like manner, "NULL" is never equal to anything, even itself. Therefore comparisons using equality operators will always return <code>False</code>, even when the value actually <code>IS NULL</code>.</p>
</div>
<div class="paragraph">
<p>For that reason, comparison operators should never be used to make comparisons with <code>NULL</code>; <code>IS NULL</code> and <code>IS NOT NULL</code> should be used instead.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>UPDATE books
SET title = 'unknown'
WHERE title = NULL -- Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>UPDATE books
SET title = 'unknown'
WHERE title IS NULL</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use "[IS NULL/IS NOT NULL]" instead.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_15_nov_2016_160730_pierre_yves_nicolas_wrote">on 15 Nov 2016, 16:07:30 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>Oracle seems to treat an empty string as equivalent to NULL but that&#8217;s not the case for DB2 (see http://www-01.ibm.com/support/docview.wss?uid=swg21181029).</p>
</div>
</div>
</div>
</div>