<div class="paragraph">
<p>Shared coding conventions allow teams to collaborate efficiently. This rule checks that all non-private <code>Const</code> field names comply with the provided regular expression.</p>
</div>
<div class="paragraph">
<p>The default configuration is the one recommended by Microsoft:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Pascal casing, starting with an upper case character, e.g. BackColor</p>
</li>
<li>
<p>Short abbreviations of 2 letters can be capitalized, e.g. GetID</p>
</li>
<li>
<p>Longer abbreviations need to be lower cased, e.g. GetHtml</p>
</li>
</ul>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="paragraph">
<p>With the default regular expression <code>^([A-Z]{1,3}[a-z0-9]+)*([A-Z]{2})?$</code>:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>Module Module1
    Public Const foo = 0  ' Noncompliant
End Module</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>Module Module1
    Public Const Foo = 0  ' Compliant
End Module</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Rename "yyy" to match the regular expression: "xxx".</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="ulist">
<ul>
<li>
<p>key = format
description: Regular expression used to check the non-private constant names against.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>default:  ^([A-Z]{1,3}[a-z0-9]+)*([A-Z]{2})?$</p>
</div>
</div>
</div>
</div>