<div class="paragraph">
<p>The <a href="https://docs.python.org/3/library/stdtypes.html#iterator-types">iterator protocol</a> specifies that the <code>__iter__</code> method can only return an iterator, i.e. an object with a <code>__next__</code> and an <code>__iter__</code> method (because iterators are also iterable). Returning anything else than an iterator will raise a <code>TypeError</code> as soon as the iteration begins.</p>
</div>
<div class="paragraph">
<p>Note that <a href="https://docs.python.org/3/tutorial/classes.html#generators">generators</a> and <a href="https://docs.python.org/3/tutorial/classes.html#generator-expressions">generator expressions</a> have both <code>__next__</code> and <code>__iter__</code> methods generated automatically.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when the object returned by an <code>__iter__</code> method does not have a method <code>__next__</code>.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class MyIterable:
    def __init__(self, values):
        self._values = values

    def __iter__(self):
        return None  # Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class MyIterable:
    def __init__(self, values):
        self._values = values

    def __iter__(self):
        return MyIterator(self._values)

class MyIterator:
    def __init__(self, values):
        self._values = values
        self._index = 0

    def __next__(self):
        if self._index &gt;= len(self._values):
            raise StopIteration()
        value = self._values[self._index]
        self._index += 1
        return value

    def __iter__(self):
        return self</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://www.python.org/dev/peps/pep-0234/#python-api-specification">PEP 234 - Iterators</a></p>
</li>
<li>
<p><a href="https://docs.python.org/3/library/stdtypes.html#iterator-types">Python documentation - Iterator Types</a></p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>Return an object complying with iterator protocol.</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s5625">is related to: S5625</h3>

</div>
<div class="sect2">
<h3 id="_on_28_apr_2015_142143_ann_campbell_wrote">on 28 Apr 2015, 14:21:43 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>back to you [~elena.vilchik] for a last check.</p>
</div>
</div>
</div>
</div>