<div class="paragraph">
<p>Shared coding conventions allow teams to collaborate efficiently. This rule checks that type specifiers always appear in the following order:</p>
</div>
<div class="olist arabic">
<ol class="arabic">
<li>
<p><code>typedef</code></p>
</li>
<li>
<p>type name, spelling of built-in types with more than one type-specifier:</p>
<div class="olist loweralpha">
<ol class="loweralpha" type="a">
<li>
<p>signedness - <code>signed</code> or <code>unsigned</code></p>
</li>
<li>
<p>last single type-specifier or</p>
<div class="ulist">
<ul>
<li>
<p><code>short int</code></p>
</li>
<li>
<p><code>long int</code></p>
</li>
<li>
<p><code>long long int</code></p>
</li>
<li>
<p><code>long double</code></p>
</li>
</ul>
</div>
</li>
</ol>
</div>
</li>
</ol>
</div>
<div class="paragraph">
<p>Since the positioning of the <code>const</code> keyword is controversial, this rule does not check it.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>int typedef T;

double long d;
char unsigned ch;
long signed int i;</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>typedef int T;

long double d;
unsigned char ch;
signed long int i;</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://github.com/isocpp/CppCoreGuidelines/blob/036324/CppCoreGuidelines.md#Rl-const">C&#43;&#43; Core Guidelines NL.26</a> - Use conventional <code>const</code> notation</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Reorder these specifiers to: "xxx yyy zzz&#8230;&#8203;"</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_14_may_2014_160915_ann_campbell_wrote">on 14 May 2014, 16:09:15 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>look this over &amp; see if you&#8217;d like anything changed, please</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_26_may_2015_182936_evgeny_mandrikov_wrote">on 26 May 2015, 18:29:36 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] LGTM.</p>
</div>
</div>
</div>
</div>