<div class="paragraph">
<p>Overriding a variable declared in an outer scope can strongly impact the readability, and therefore the maintainability, of a piece of code. Further, it could lead maintainers to introduce bugs because they think they&#8217;re using one variable but are really using another.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>SET SERVEROUTPUT ON

DECLARE
  foo VARCHAR2(42) := 'foo';
BEGIN
  DECLARE
    foo VARCHAR2(42) := 'bar'; -- Noncompliant - this variable hides the one above and should be renamed
  BEGIN
    DBMS_OUTPUT.PUT_LINE(foo); -- Displays "bar", which is confusing
  END;

  DBMS_OUTPUT.PUT_LINE(foo); -- Displays "foo"
END;
/</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>SET SERVEROUTPUT ON

DECLARE
  foo VARCHAR2(42) := 'foo';
BEGIN
  DECLARE
    bar VARCHAR2(42) := 'bar'; -- Compliant
  BEGIN
    DBMS_OUTPUT.PUT_LINE(bar); -- Displays "bar"
  END;

  DBMS_OUTPUT.PUT_LINE(foo); -- Displays "foo"
END;
/</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>"{0}" hides or potentially hides a variable declared in an outer scope at line n.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_duplicated_by_s801">is duplicated by: S801</h3>

</div>
<div class="sect2">
<h3 id="_is_duplicated_by_s1063">is duplicated by: S1063</h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s801">is related to: S801</h3>

</div>
<div class="sect2">
<h3 id="_on_8_jan_2016_102334_freddy_mallet_wrote">on 8 Jan 2016, 10:23:34 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>hi [~ann.campbell.2], don&#8217;t you think that we should merge this ticket with RSPEC-801 ? Thanks</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_11_jan_2016_182128_ann_campbell_wrote">on 11 Jan 2016, 18:21:28 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>done [~freddy.mallet]</p>
</div>
</div>
</div>
</div>