<div class="paragraph">
<p>When the <code>FILE STATUS</code> is not specified on a file, any read operations on the file should handle the "AT END" or "INVALID KEY" conditions.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>SELECT MY_FILE
  ASSIGN TO 'foobar.txt'
  ORGANIZATION IS SEQUENTIAL.
...
READ MY_FILE
  NOT AT END PERFORM COMPUTE_LINE
END-READ.</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>SELECT MY_FILE
  ASSIGN TO 'foobar.txt'
  ORGANIZATION IS SEQUENTIAL.
...
READ MY_FILE
  NOT AT END PERFORM COMPUTE_LINE
  AT END MOVE 'Y' TO EOF-FLAG
END-READ.</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Specify the "AT END" or "INVALID KEY" clauses of the READ statement.</p>
</div>
</div>
</div>
</div>