<div class="paragraph">
<p><code>IMPORT</code>ed values should be treated as read-only. Doing otherwise could result in unpleasant surprises down the line.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>D VAR2             S              5  0 IMPORT
 * Noncompliant
C           MOVE       MYVAR     VAR2

C/FREE
  // Noncompliant
  VAR2=MYVAR;
/END-FREE</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>"xxx" was imported and should not be updated</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_2_apr_2015_192250_ann_campbell_wrote">on 2 Apr 2015, 19:22:50 Ann Campbell wrote:</h3>
<div class="paragraph">
<p><a href="http://www.bmeyers.net/faqs/14-tips/32-rpg-iv-style?start=2" class="bare">http://www.bmeyers.net/faqs/14-tips/32-rpg-iv-style?start=2</a></p>
</div>
</div>
</div>
</div>