<div class="paragraph">
<p>Failure to specify a locale for <code>String</code> case conversions means the system default encoding will be used, possibly creating problems with international characters. Such code may work fine in its "home" environment, but break in ways that are extremely difficult to diagnose for customers who use different encodings. Such bugs can be nearly, if not completely, impossible to reproduce when it&#8217;s time to fix them.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>String myStr = "José is a nice guy.";
String myUcString = myStr.toUpperCase(); // Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>String myStr = "José is a nice guy.";
String myUcString = myStr.toUpperCase(Locale.US);</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_duplicates_s1449">duplicates: S1449</h3>

</div>
<div class="sect2">
<h3 id="_on_10_oct_2014_114549_freddy_mallet_wrote">on 10 Oct 2014, 11:45:49 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>@Ann, this ticket duplicates RSPEC-1449, either delete RSPEC-1449 or this ticket. Moreover, I would mention the fact that this is useful only for few kind of locales.</p>
</div>
</div>
</div>
</div>