<div class="paragraph">
<p>The <code>suspend</code> function modifier is used to mark a function which might take some time to execute and could suspend the caller coroutine. The location where such function is called is a "suspension point". Functions marked as <code>suspend</code> ("suspending functions") should themselves contain at least one suspension point, otherwise it makes no sense to mark it as <code>suspend</code></p>
</div>
<div class="paragraph">
<p>This rule reports an issue if a function with <code>suspend</code> modifier has no calls to other <code>suspend</code> functions inside its body.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>suspend fun function() { // Noncompliant, redundant 'suspend' modifier
    println("Hello!")
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>fun function() {
    println("Hello!")
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://kotlinlang.org/docs/coroutines-guide.html">Coroutines guide</a></p>
</li>
</ul>
</div>
</div>
</div>