<div class="paragraph">
<p>ECMAScript 2015 introduced the <code>let</code> and <code>const</code> keywords for block-scope variable declaration. Using <code>const</code> creates a read-only (constant) variable.</p>
</div>
<div class="paragraph">
<p>The distinction between the variable types created by <code>var</code> and by <code>let</code> is significant, and a switch to <code>let</code> will help alleviate many of the variable scope issues which have caused confusion in the past.</p>
</div>
<div class="paragraph">
<p>Because these new keywords create more precise variable types, they are preferred in environments that support ECMAScript 2015. However, some refactoring may be required by the switch from <code>var</code> to <code>let</code>, and you should be aware that they raise <code>SyntaxError</code>s in pre-ECMAScript 2015 environments.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when <code>var</code> is used instead of <code>const</code> or <code>let</code>.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>var color = "blue";
var size = 4;</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>const color = "blue";
let size = 4;</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace "var" with "let" or "const"</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>"var"</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_duplicated_by_s1252">is duplicated by: S1252</h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s994">is related to: S994</h3>

</div>
<div class="sect2">
<h3 id="_on_19_jan_2016_180626_elena_vilchik_wrote">on 19 Jan 2016, 18:06:26 Elena Vilchik wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Could you add to the description that <code>let</code> statement declares a <strong>block scope</strong> local variable?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_19_jan_2016_180834_elena_vilchik_wrote">on 19 Jan 2016, 18:08:34 Elena Vilchik wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] WDYT about "creates a read-only variable" instead of "creates a true, unupdatable constant"?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_19_jan_2016_182032_elena_vilchik_wrote">on 19 Jan 2016, 18:20:32 Elena Vilchik wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I would change severity on <code>major</code> and replace <code>confusing</code> on <code>bad-practice</code> or even <code>obsolete</code>.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_20_jan_2016_093028_elena_vilchik_wrote">on 20 Jan 2016, 09:30:28 Elena Vilchik wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] looks good! But IMO code sample in description is excessive, and what you can understand from this code sample is that "let" fails with error but "var" is save and works smoothly :)</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_20_jan_2016_130243_ann_campbell_wrote">on 20 Jan 2016, 13:02:43 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>okay [~elena.vilchik]</p>
</div>
</div>
</div>
</div>