<div class="paragraph">
<p>As stated in the Glibc documentation:</p>
</div>
<div class="quoteblock">
<blockquote>
<div class="paragraph">
<p>The behavior of the 's[n]printf()' functions are undefined if copying takes place between objects that overlap—for example, if s is also given as an argument to be printed under control of the ‘%s’ conversion</p>
</div>
</blockquote>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>sprintf(buf, "%s Entry: %d\n", buf, i); //Noncompliant</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use a new, dedicated buffer to store the result of this convertion</p>
</div>
</div>
</div>
</div>