<div class="paragraph">
<p>Assistive technologies, such as screen readers, use <code>&lt;th&gt;</code> headers to provide some context when users navigates a table. Without it the user gets rapidly lost in the flow of data.</p>
</div>
<div class="paragraph">
<p>Headers should be properly associated with the corresponding <code>&lt;td&gt;</code> cells by using either a <code>scope</code> attribute or <code>headers</code> and <code>id</code> attributes. See https://www.w3.org/WAI/tutorials/tables/tips/[W3C WAI Web Accessibility Tutorials] for more information.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue whenever a <code>&lt;table&gt;</code> does not contain any <code>&lt;th&gt;</code> elements.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>&lt;table&gt; &lt;!-- Noncompliant --&gt;
  &lt;tr&gt;
    &lt;td&gt;Name&lt;/td&gt;
    &lt;td&gt;Age&lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
    &lt;td&gt;John Doe&lt;/td&gt;
    &lt;td&gt;24&lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
    &lt;td&gt;Alice Doe&lt;/td&gt;
    &lt;td&gt;54&lt;/td&gt;
  &lt;/tr&gt;
&lt;/table&gt;</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>&lt;table&gt;
  &lt;tr&gt;
    &lt;th scope="col"&gt;Name&lt;/th&gt;
    &lt;th scope="col"&gt;Age&lt;/th&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
    &lt;td&gt;John Doe&lt;/td&gt;
    &lt;td&gt;24&lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
    &lt;td&gt;Alice Doe&lt;/td&gt;
    &lt;td&gt;54&lt;/td&gt;
  &lt;/tr&gt;
&lt;/table&gt;</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_exceptions">Exceptions</h2>
<div class="sectionbody">
<div class="paragraph">
<p>No issue will be raised on &lt;table&gt; used for layout purpose, i.e. when it contains a <code>role</code> attribute set to <code>"presentation"</code> or <code>"none"</code>. Note that <a href="https://www.w3schools.com/html/html_layout.asp">using &lt;table&gt; for layout purpose is a bad practice</a>.</p>
</div>
<div class="paragraph">
<p>No issue will be raised on &lt;table&gt; containing an <code>aria-hidden</code> attribute set to <code>"true"</code>.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://www.w3.org/WAI/WCAG21/quickref/?versions=2.0#qr-content-structure-separation-programmatic">WCAG2, 1.3.1</a> - Info and Relationships</p>
</li>
<li>
<p><a href="https://www.w3.org/TR/WCAG20-TECHS/H51">WCAG2, H51</a> - Using table markup to present tabular information</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add "&lt;th&gt;" headers to this "&lt;table&gt;"</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The opening &lt;table&gt; tag, without its content.</p>
</div>
</div>
</div>
</div>