<div class="paragraph">
<p>A dictionary cannot have two values with the same key. When a key is repeated in a dictionary literal, only the last occurence will remain. Thus duplicate keys should be either modified or removed.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when the same value is used multiple times as a key in a dictionary literal.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>{"one": 1, "two": 2, "one": 3}  # Noncompliant

def func(a1, a2, a3):
    {a1: 1, a2: 2, a1: 3}  # Noncompliant.</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>{"one": 1, "two": 2, "three": 3}

def func(a1, a2, a3):
    {a1: 1, a2: 2, a3: 3}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.python.org/3/reference/expressions.html#dictionary-displays">Python documentation - Dictionary displays</a></p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Change or remove duplicates of this key.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Primary: the first occurence of a duplicated key.</p>
</div>
<div class="paragraph">
<p>Secondary:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>location: other occurences of the key.</p>
</li>
<li>
<p>message: "Duplicate key"</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s4143">relates to: S4143</h3>

</div>
<div class="sect2">
<h3 id="_on_16_apr_2020_221623_nicolas_harraudeau_wrote">on 16 Apr 2020, 22:16:23 Nicolas Harraudeau wrote:</h3>
<div class="paragraph">
<p>Similar to RSPEC-4143. The difference is that RSPEC-4143 focuses on key replacement in already existing maps, whereas RSPEC-5780 focuses duplicate keys in Map creation.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_22_jun_2020_163029_alexandre_gigleux_wrote">on 22 Jun 2020, 16:30:29 Alexandre Gigleux wrote:</h3>
<div class="paragraph">
<p>I changed from Bug to Code Smell because from what I can see on Peach, majority of the issues are related to duplication of the key AND the value. It means there is no impact/bug, it&#8217;s a left over copy/paste. The only case where the behavior can not be the expected one is when the duplicated key is initialized with 2 different values.</p>
</div>
</div>
</div>
</div>