<div class="paragraph">
<p>Serialization event handlers that don&#8217;t have the correct signature will simply not be called, thus bypassing any attempts to augment the automated de/serialization.</p>
</div>
<div class="paragraph">
<p>This rule raises issue when a method marked with one of the following attributes is <code>Public</code>, <code>Shared</code>, is a <code>Function</code> not a <code>Sub</code>, has type parameters, or does not have a single parameter of type <code>System.Runtime.Serialization.StreamingContext</code>:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>System.Runtime.Serialization.OnSerializingAttribute</code></p>
</li>
<li>
<p><code>System.Runtime.Serialization.OnSerializedAttribute</code></p>
</li>
<li>
<p><code>System.Runtime.Serialization.OnDeserializingAttribute</code></p>
</li>
<li>
<p><code>System.Runtime.Serialization.OnDeserializedAttribute</code></p>
</li>
</ul>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>&lt;Serializable&gt;
Public Class Foo
    &lt;OnSerializing&gt;
    Public Sub OnSerializing(ByVal context As StreamingContext) ' Noncompliant should be private
    End Sub

    &lt;OnSerialized&gt;
    Private Function OnSerialized(ByVal context As StreamingContext) As Integer '  Noncompliant should return void
    End Function

    &lt;OnDeserializing&gt;
    Private Sub OnDeserializing() ' Noncompliant should have a single parameter of type StreamingContext
    End Sub

    &lt;OnSerializing&gt;
    Public Sub OnSerializing2(Of T)(ByVal context As StreamingContext) ' Noncompliant should have no type parameters
    End Sub

    &lt;OnDeserialized&gt;
    Private Sub OnDeserialized(ByVal context As StreamingContext, ByVal str As String) ' Noncompliant should have a single parameter of type StreamingContext
    End Sub
End Class</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>&lt;Serializable&gt;
Public Class Foo
    &lt;OnSerializing&gt;
    Private Sub OnSerializing(ByVal context As StreamingContext)
    End Sub

    &lt;OnSerialized&gt;
    Private Sub OnSerialized(ByVal context As StreamingContext)
    End Sub

    &lt;OnDeserializing&gt;
    Private Sub OnDeserializing(ByVal context As StreamingContext)
    End Sub

    &lt;OnDeserialized&gt;
    Private Sub OnDeserialized(ByVal context As StreamingContext)
    End Sub
End Class</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Make this method [non-public | non-shared | a 'Sub' not a 'Function' | have no type parameters | have a single parameter of type 'StreamingContext'].</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Method identifier.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_20_mar_2017_120901_ann_campbell_wrote">on 20 Mar 2017, 12:09:01 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>I&#8217;ve edited. Double-check me [~amaury.leve]</p>
</div>
</div>
</div>
</div>