<div class="paragraph">
<p>Parentheses are not required after the <code>assert</code>, <code>del</code>, <code>elif</code>, <code>except</code>, <code>for</code>, <code>if</code>, <code>in</code>, <code>not</code>, <code>raise</code>, <code>return</code>, <code>while</code>, and <code>yield</code> keywords, and using them unnecessarily impairs readability. They should therefore be omitted.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>x = 1
while (x &lt; 10):
    print "x is now %d" % (x)
    x += 1</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>x = 1
while x &lt; 10:
    print "x is now %d" % (x)
    x += 1</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove the parentheses after this "XXX" keyword</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_5_may_2017_155908_pierre_yves_nicolas_wrote">on 5 May 2017, 15:59:08 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] This rule is deprecated in favor of RSPEC-1110. However, the scope of RSPEC-1110 was reduced. Should we still deprecate this rule?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_9_may_2017_091500_freddy_mallet_wrote">on 9 May 2017, 09:15:00 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Indeed [~pierre-yves.nicolas] and [~ann.campbell.2], I don&#8217;t think there is any remaining overlap between this rule and RSPEC-1110.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_23_jul_2020_111539_nicolas_harraudeau_wrote">on 23 Jul 2020, 11:15:39 Nicolas Harraudeau wrote:</h3>
<div class="paragraph">
<p>This rule is deprecated because it raises mostly false positives on tuples and unpacking. Ex: <code>return (a,b)</code> or <code>for (a,b) in x:</code></p>
</div>
<div class="paragraph">
<p>Also it provides very little value as there can be cases when parentheses improve readability.</p>
</div>
<div class="paragraph">
<p>Some cases deserve dedicated rules:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>S5905 "Assert should not be called on a tuple literal": it is a common bug, not a code smell.</p>
</li>
<li>
<p>A rule should be created later to detect when single element tuple was intended. ex: <code>a = (b)</code> when developer probably wanted <code>a = (b,)</code></p>
</li>
</ul>
</div>
</div>
</div>
</div>