<div class="paragraph">
<p>A common idiom in JavaScript to differentiate between two possible types is to check for the presence in the object of a member of the desired type. Usually, to simplify the code, a boolean function is created to check the type.</p>
</div>
<div class="paragraph">
<p>Typescript provides user defined type guard functions. These are just functions with a return type of <code>argumentName is SomeType</code>. Such functions return <code>true</code> if the argument is of the specified type. One of the advantages of using such a function is that in a conditional block where the condition is a type guard, the compiler automatically performs the appropriate casts, so explicit casting becomes unnecessary.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when a boolean function checking for the type of its only argument can be replaced with a user-defined type guard function.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>function isSomething(x: BaseType) : boolean { // Noncompliant
  return (&lt;Something&gt;x).foo !== undefined;
}

if (isSomething(v)) {
  (&lt;Something&gt;v).foo();
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>function isSomething(x: BaseType) : x is Something {
  return (&lt;Something&gt;x).foo !== undefined;
}

if (isSomething(v)) {
  v.foo();
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="paragraph">
<p><a href="https://www.typescriptlang.org/docs/handbook/advanced-types.html">TypeScript advanced types</a></p>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Change this boolean return type into a type predicate</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The function declaration</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_14_nov_2017_212312_ann_campbell_wrote">on 14 Nov 2017, 21:23:12 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~jeanchristophe.collet] from this description I don&#8217;t understand what type guards do, and from the examples, I have no idea how they work.</p>
</div>
</div>
</div>
</div>