<div class="paragraph">
<p>Shared coding conventions allow teams to collaborate effectively. This rule checks that labels match a provided regular expression.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="paragraph">
<p>With default provided regular expression <code>^[A-Z][A-Z0-9_]*$</code>:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>exit:  // Noncompliant
  doCleanup();</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>EXIT:  // Compliant
  doCleanup();</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Rename label "xxx" to match the regular expression: xxx</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="ulist">
<ul>
<li>
<p>key = format</p>
</li>
<li>
<p>description: Regular expression used to check the label names against.</p>
</li>
<li>
<p>default: ^[A-Z][A-Z0-9_]*$</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_2_jul_2015_134304_ann_campbell_wrote">on 2 Jul 2015, 13:43:04 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Rule created [~evgeny.mandrikov], but I Google didn&#8217;t give me much guidance on what the default regex should be (discussions were about content not casing) so I went with all upper case.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_2_jul_2015_154742_evgeny_mandrikov_wrote">on 2 Jul 2015, 15:47:42 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] LGTM.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_3_jul_2015_131148_massimo_paladin_wrote">on 3 Jul 2015, 13:11:48 Massimo PALADIN wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] why sqale is 10min when RSPEC-2343 has 5min?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_6_jul_2015_141646_ann_campbell_wrote">on 6 Jul 2015, 14:16:46 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~massimo.paladin] my assumption was that IDE&#8217;s are sophisticated enough to auto-handle renaming things like vars, enums, &amp;etc, but that label renaming would likely have to be done manually.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_7_jul_2015_091339_massimo_paladin_wrote">on 7 Jul 2015, 09:13:39 Massimo PALADIN wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] changing it to 5min as IDE supports renaming of labels (at least Clion does).</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_7_jul_2015_114345_ann_campbell_wrote">on 7 Jul 2015, 11:43:45 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~evgeny.mandrikov] why did you remove the default regex from the description? I&#8217;ve got a ticket in my backlog to add them to such tickets, and as was made clear on the mailing list, if the rule is activated with a different regex, it&#8217;s not really clear to users what standard the Noncompliant Example is held to.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_7_jul_2015_154351_evgeny_mandrikov_wrote">on 7 Jul 2015, 15:43:51 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I did this for consistency with RSPEC, which I was implementing at the same time - RSPEC-3084.</p>
</div>
</div>
</div>
</div>