<div class="paragraph">
<p>When specifying base classes for a class, the access level is private by default. While private inheritance has its uses, it is far less used than public inheritance. Therefore:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Either you want public inheritance, and you have to specify it</p>
</li>
<li>
<p>Or you want private inheritance, and you should be explicit about this design decision, by specifying it too.</p>
</li>
</ul>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class B {
};

class C : B {
};</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class B {
};

class C :  private B { // Or public, if it was private by mistake
};</pre>
</div>
</div>
<div class="sect2">
<h3 id="_see_also">See Also</h3>
<div class="ulist">
<ul>
<li>
<p>S5966 is a similar rule that deals with base visibility in structs</p>
</li>
<li>
<p>S3540 is another rule that favors a different coding style for base classes. It should not be activated at the same time as this rule.</p>
</li>
</ul>
</div>
</div>
</div>
</div>