<div class="paragraph">
<p>You&#8217;re allowed to make function calls with positionally-identified parameters or with named parameters. In long parameter lists, explicitly naming your parameters both eliminates the risk of swapping parameter values and makes the call clearer to maintainers.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="paragraph">
<p>With the default threshold of 3:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>def coordinateSurpriseParty(self, hostName, guestOfHonor, caterer, invitees, peopleNotToInvite):
  # ...

host="John"
guestOfHonor="Fred"
caterer="Barry"
invitees="Thea, Will, Mark, Mary"
peopleToExclude="Susan, Alberta, Frank, Larry"

coordinateSurpriseParty(caterer, host, guestOfHonor, peopleToExclude, invitees)  # Noncompliant; this party will be a train wreck!</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>def coordinateSurpriseParty(self, hostName, guestOfHonor, caterer, invitees, peopleNotToInvite):
  # ...

host="John"
guestOfHonor="Fred"
caterer="Barry"
invitees="Thea, Will, Mark, Mary"
peopleToExclude="Susan, Alberta, Frank, Larry"

coordinateSurpriseParty(caterer=caterer, hostName=host, geustOfHonor=guestOfHonor, peopleNotToInvite=peopleToExclude, invitees=invitees)</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Specify the argument names for these n unnamed parameters</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="ulist">
<ul>
<li>
<p>key = threshold</p>
</li>
<li>
<p>description: Maximum number of unnamed arguments</p>
</li>
<li>
<p>default: 3</p>
</li>
</ul>
</div>
</div>
</div>
</div>