<div class="paragraph">
<p>The bitwise operators <code>&amp;</code>, <code>|</code> can be mistaken for the boolean operators <code>&amp;&amp;</code> and <code>||</code>.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when <code>&amp;</code> or <code>|</code> is used in a boolean context.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>if (a &amp; b) { ... } // Noncompliant; &amp; used in error</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>if (a &amp;&amp; b) { ... }</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_exceptions">Exceptions</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When a file contains other bitwise operations, (<code>^</code>, <code>&lt;&lt;</code>, <code>&gt;&gt;&gt;</code>, <code>&gt;&gt;</code>, <code>~</code>, <code>&amp;=</code>, <code>^=</code>, <code>|=</code>, <code>&lt;&lt;=</code>, <code>&gt;&gt;=</code>, <code>&gt;&gt;&gt;=</code> and <code>&amp;</code> or <code>|</code> used with a numeric literal as the right operand) all issues in the file are ignored, because it is evidence that bitwise operations are truly intended in the file.</p>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Review this use of bitwise "x" operator; conditional "xx" might have been intended.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>the operator</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_17_mar_2017_095225_elena_vilchik_wrote">on 17 Mar 2017, 09:52:25 Elena Vilchik wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Could you update this RSPEC corresponding to SONARJS-928? I&#8217;ve explained in comment the scope of implementation.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_17_mar_2017_150208_ann_campbell_wrote">on 17 Mar 2017, 15:02:08 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Done.</p>
</div>
</div>
</div>
</div>