<div class="paragraph">
<p>When specifying the class of objects in a catch clause it is important to make sure that the class exists.</p>
</div>
<div class="paragraph">
<p>Since no PHP error will be raised if the class does not exist, this can lead to difficult to debug problems as the catch clause will have no effect and the reason might not be obvious.</p>
</div>
<div class="paragraph">
<p>This mistake often occurs when being in a namespace and catching PHP built-in exception classes without escaping to the global namespace or importing the classes.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when, being in a namespace, an undefined class belonging to that namespace is caught.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>namespace Foo\Bar;

try {
    doSomething();
} catch (Exception $e) { // Noncompliant - Exception will never be caught because the class Exception does not exist in the namespace
    echo $e-&gt;message;
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>namespace Foo\Bar;

try {
    doSomething();
} catch (\Exception $e) { // Compliant used by global namespace
    echo $e-&gt;message;
}

// or

namespace Foo\Bar;

use Exception;

try {
    doSomething();
} catch (Exception $e) { // Compliant imported by use statement
    echo $e-&gt;message;
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Create class {Exception} or check correct import of class</p>
</div>
</div>
</div>
</div>