<div class="paragraph">
<p>A <code>NullPointerException</code> should indicate that a <code>null</code> value was unexpectedly encountered. Good programming practice dictates that code is structured to avoid NPE&#8217;s.</p>
</div>
<div class="paragraph">
<p>Explicitly throwing <code>NullPointerException</code> forces a method&#8217;s callers to explicitly catch it, rather than coding to avoid it. Further, it makes it difficult to distinguish between the unexpectedly-encountered <code>null</code> value and the condition which causes the method to purposely throw an NPE.</p>
</div>
<div class="paragraph">
<p>If an NPE is being thrown to indicate that a parameter to the method should not have been null, use the <code>@NotNull</code> annotation instead.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public void doSomething (String aString) throws NullPointerException {
     throw new NullPointerException();
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public void doSomething (@NotNull String aString) {
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Throw some other exception here, such as "IllegalArgumentException".</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_28_apr_2014_183022_evgeny_mandrikov_wrote">on 28 Apr 2014, 18:30:22 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] similar to RSPEC-1696 - I don&#8217;t know how to "explicitly throw NPE" in C&#43;&#43;, so I doubt that this ticket applicable for C&#43;&#43;.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_28_apr_2014_185855_evgeny_mandrikov_wrote">on 28 Apr 2014, 18:58:55 Evgeny Mandrikov wrote:</h3>
<div class="paragraph">
<p>If in case of C&#43;&#43; this is about NULL macro, then there is another ticket coming from MISRA C&#43;&#43; - see RSPEC-1038.</p>
</div>
</div>
</div>
</div>