<div class="paragraph">
<p>Since C&#43;&#43;11, declaring a variable, class, or function in an unnamed <code>namespace</code> gives it internal linkage. Similarly, marking a declaration <code>static</code> also gives it internal linkage. Because both mechanisms have the same effect (although <code>static</code> has a narrower application) using them together is clearly redundant.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>namespace {
  static int i = 3;  // Noncompliant
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>namespace {
  int i = 3;
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>ISO/IEC 14882:2011 ยง3.5 paragraph 4</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this redundant use of "static".</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p><code>static</code></p>
</div>
</div>
</div>
</div>