<div class="paragraph">
<p>Shared coding conventions allow teams to collaborate efficiently. This rule checks that all <code>Private Shared ReadOnly</code> field names comply with the provided regular expression.</p>
</div>
<div class="paragraph">
<p>The default configuration is:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Optionally, can start with an underscore character or "s_", e.g. <code>_foo</code>, <code>s_foo</code></p>
</li>
<li>
<p>Camel casing, starting with a lower case character, e.g. backColor</p>
</li>
<li>
<p>Short abbreviations of 2 letters can be capitalized only when not at the beginning, e.g. "id" in productID</p>
</li>
<li>
<p>Longer abbreviations need to be lower cased, e.g. html</p>
</li>
</ul>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="paragraph">
<p>With the default regular expression <code>^(s_|_)?[a-z][a-z0-9]*([A-Z]{1,3}[a-z0-9]+)*([A-Z]{2})?$</code>:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>Class Foo
    Private Shared ReadOnly Foo As Integer  ' Noncompliant
End Class</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>Class Foo
    Private Shared ReadOnly foo As Integer  ' Compliant
End Class</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Rename "xxx" to match the regular expression: "yyy".</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="ulist">
<ul>
<li>
<p>key = format
description: Regular expression used to check the "Private Shared ReadOnly" field names against.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>default: ^(s_|<em>)?</em>?[a-z]<strong class="a-z0-9">([A-Z]{1,3}[a-z0-9]+)</strong>([A-Z]{2})?$</p>
</div>
</div>
</div>
</div>