<div class="paragraph">
<p>Method/constructor references are commonly agreed to be, most of the time, more compact and readable than using lambdas, and are therefore preferred.</p>
</div>
<div class="paragraph">
<p>In some rare cases, when it is not clear from the context what kind of function is being described and reference would not increase the clarity, it might be fine to keep the lambda.</p>
</div>
<div class="paragraph">
<p>Similarly, <code>null</code> checks can be replaced with references to the <code>Objects::isNull</code> and <code>Objects::nonNull</code> methods, <code>casts</code> can be replaced with <code>SomeClass.class::cast</code> and <code>instanceof</code> can be replaced with <code>SomeClass.class::isInstance</code>.</p>
</div>
<div class="paragraph">
<p><strong>Note</strong> that this rule is automatically disabled when the project&#8217;s <code>sonar.java.source</code> is lower than <code>8</code>.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class A {
  void process(List&lt;A&gt; list) {
    list.stream()
      .filter(a -&gt; a instanceof B)
      .map(a -&gt; (B) a)
      .map(b -&gt; b.&lt;String&gt;getObject())
      .forEach(b -&gt; { System.out.println(b); });
  }
}

class B extends A {
  &lt;T&gt; T getObject() {
    return null;
  }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class A {
  void process(List&lt;A&gt; list) {
    list.stream()
      .filter(B.class::isInstance)
      .map(B.class::cast)
      .map(B::&lt;String&gt;getObject)
      .forEach(System.out::println);
  }
}

class B extends A {
  &lt;T&gt; T getObject() {
    return null;
  }
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace this lambda with a method reference. [(sonar.java.source not set. Assuming 8 or greater.)]</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_duplicated_by_s2212">is duplicated by: S2212</h3>

</div>
<div class="sect2">
<h3 id="_on_26_feb_2014_121941_freddy_mallet_wrote">on 26 Feb 2014, 12:19:41 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARJAVA-465</p>
</div>
</div>
</div>
</div>