<div class="paragraph">
<p>When two methods have the same implementation, either it was a mistake - something else was intended - or the duplication was intentional, but may be confusing to maintainers. In the latter case, one implementation should invoke the other. Numerical and string literals are not taken into account.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>private String code = 'bounteous';

public String calculateCode() {
  doTheThing();
  return code;
}

public String getName() {  // Noncompliant
  doTheThing();
  return code;
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>private String code = 'bounteous';

public String getCode() {
  doTheThing();
  return code;
}

public String getName() {
  return getCode();
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_exceptions">Exceptions</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Methods that are not accessors (getters and setters), with fewer than 2 statements are ignored.</p>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Update this method so that its implementation is not identical to "xxx".</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>primary: method signature</p>
</li>
<li>
<p>secondary: original method</p>
<div class="ulist">
<ul>
<li>
<p>message: "Original implementation."</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_6_oct_2017_144358_jean_christophe_collet_wrote">on 6 Oct 2017, 14:43:58 Jean-Christophe Collet wrote:</h3>
<div class="paragraph">
<p><a href="http://www.viva64.com/en/b/0363/#ID0EVHBI" class="bare">http://www.viva64.com/en/b/0363/#ID0EVHBI</a></p>
</div>
</div>
</div>
</div>