<div class="paragraph">
<p>The comma operator takes two expressions, executes them from left to right and returns the result of the second one. Use of this operator is generally detrimental to the readability and reliability of code, and the same effect can be achieved by other means.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>i = a += 2, a + b;  // What's the value of i ?</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>a +=  2;
i = a + b;</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_exceptions">Exceptions</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Use of comma operator is tolerated:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>in initialization and increment expressions of <code>for</code> loops.</p>
</li>
</ul>
</div>
<div class="listingblock">
<div class="content">
<pre>for(i = 0, j = 5; i &lt; 6; i++, j++) { ... }</pre>
</div>
</div>
<div class="ulist">
<ul>
<li>
<p>If the expression sequence is explicitly wrapped in parentheses.</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove use of this comma operator.</p>
</div>
<div class="paragraph">
<p>Remove use of all comma operators in this expression.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s6175">is related to: S6175</h3>

</div>
<div class="sect2">
<h3 id="_on_6_aug_2013_231735_ann_campbell_wrote">on 6 Aug 2013, 23:17:35 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>The format of this rule title doesn&#8217;t follow the norm. Is that allowed in this case, or should I correct it?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_30_aug_2013_173239_dinesh_bolkensteyn_wrote">on 30 Aug 2013, 17:32:39 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>Why should this be blocker?</p>
</div>
<div class="paragraph">
<p>I think it should be Major at most</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_7_jan_2014_100611_freddy_mallet_wrote">on 7 Jan 2014, 10:06:11 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3358 in JavaScript</p>
</div>
</div>
</div>
</div>