<div class="paragraph">
<p>There are many ways to implement the Singleton pattern in Java, but none of them is as clean, compact and close to fool-proof as using an <code>enum</code>. Without an <code>enum</code>, the implementer must take care to properly handle thread-safety, serialization, and classloaders, but those things come for free with an <code>enum</code>.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public class Highlander implements Serializable { // Serializable makes Singleton tricky to get right
  private static final Highlander INSTANCE;

  public static synchronized Highlander getInstance() {
    if(INSTANCE == null) {
      INSTANCE = new Highlander();
    }
    return INSTANCE;
  }

  private Highlander () {}

  private final String [] rivals = {"The Kurgan", "Ramirez"}; // oops, not serializable now

  private Object readResolve() {
    return INSTANCE;
  }

  ...
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public enum Highlander {
  INSTANCE;

  private final String [] rivals = {"The Kurgan", "Ramirez"};

  ...
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://wiki.sei.cmu.edu/confluence/x/_zZGBQ">CERT MSC07-J.</a> - Prevent multiple instantiations of singleton objects</p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Make this singleton an "enum".</p>
</div>
</div>
</div>
</div>