<div class="paragraph">
<p>Shared naming conventions allow teams to  collaborate efficiently. This rule checks that static non-final field names match a provided regular expression.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="paragraph">
<p>With the default regular expression <code>^[a-z][a-zA-Z0-9]*$</code>:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>public final class MyClass {
   private static String foo_bar;
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class MyClass {
   private static String fooBar;
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Rename this field "XXX" to match the regular expression ${format}.</p>
</div>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="ulist">
<ul>
<li>
<p>key = format
Description: Regular expression used to check the field names against.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Default Value for Java : ^[a-z][a-zA-Z0-9]*$</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_9_jun_2015_134521_ann_campbell_wrote">on 9 Jun 2015, 13:45:21 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>FYI [~alexandre.gigleux] I&#8217;ve made a few updates.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_9_jun_2015_134813_ann_campbell_wrote">on 9 Jun 2015, 13:48:13 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Note for Java: same as RSPEC-116 but for static non-final fields</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_12_jun_2015_160459_ann_campbell_wrote">on 12 Jun 2015, 16:04:59 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>CodePro: Static Field Naming Convention</p>
</div>
</div>
</div>
</div>