<div class="paragraph">
<p>Calling an overridable method from a constructor could result in failures or strange behaviors when instantiating a subclass which overrides the method.</p>
</div>
<div class="paragraph">
<p>For example:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>The subclass class constructor starts by calling the parent class constructor.</p>
</li>
<li>
<p>The parent class constructor calls the method, which has been overridden in the child class.</p>
</li>
<li>
<p>If the behavior of the child class method depends on fields that are initialized in the child class constructor, unexpected behavior (like a <code>NullReferenceException</code>) can result, because the fields aren&#8217;t initialized yet.</p>
</li>
</ul>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public class Parent
{
  public Parent()
  {
    DoSomething();  // Noncompliant
  }

  public virtual void DoSomething() // can be overridden
  {
    ...
  }
}

public class Child : Parent
{
  private string foo;

  public Child(string foo) // leads to call DoSomething() in Parent constructor which triggers a NullReferenceException as foo has not yet been initialized
  {
    this.foo = foo;
  }

  public override void DoSomething()
  {
    Console.WriteLine(this.foo.Length);
  }
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this call from a constructor to the overridable "XXXX" method.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_23_jun_2015_074721_tamas_vajk_wrote">on 23 Jun 2015, 07:47:21 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] Other then changing the code sample and removing the "by contract" from the first bullet point, I&#8217;ve added this rule to the SonarQube Way profile. I think this is a really useful rule, which should be enabled by default.</p>
</div>
<div class="paragraph">
<p>The main difference between C# and Java in this case is that you need to opt in for overridable methods. So you explicitly need to say that <code>DoSomething</code> is <code>virtual</code>. So in case of Java you might have just forget to add the <code>final</code> to the method, but in C# you explicitly stated that you are calling and overridable method from the constructor.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_23_jun_2015_115147_ann_campbell_wrote">on 23 Jun 2015, 11:51:47 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>looks good to me [~tamas.vajk]</p>
</div>
</div>
</div>
</div>