<div class="paragraph">
<p>Methods that override parent class implementations should end with calls to the <code>super</code> implementations. Otherwise, you risk leaving the job half done.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when the following <code>Activity</code> methods do not end with a call to <code>super</code>:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>onDestroy</code></p>
</li>
<li>
<p><code>onPause</code></p>
</li>
<li>
<p><code>onSaveInstanceState</code></p>
</li>
<li>
<p><code>onStop</code></p>
</li>
<li>
<p><code>onTerminate</code></p>
</li>
</ul>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>//TODO</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>//TODO</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add a call to "super.xxx" at the end of this method.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_18_feb_2016_150842_nicolas_peru_wrote">on 18 Feb 2016, 15:08:42 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>This rule seems really broad. I don&#8217;t see how it can bring value without way too much noise (FP).</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_18_feb_2016_192801_ann_campbell_wrote">on 18 Feb 2016, 19:28:01 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>I&#8217;ve added a sentence to make it explicit that the rule is about the cited methods.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_18_feb_2016_205623_nicolas_peru_wrote">on 18 Feb 2016, 20:56:23 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] in that case I guess the example have to be updated ?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_18_feb_2016_210834_ann_campbell_wrote">on 18 Feb 2016, 21:08:34 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Okay. Yup.</p>
</div>
</div>
</div>
</div>