<div class="paragraph">
<p>For better readability, it is preferable to use <code>DECLARE</code> instead of the unpronounceable <code>DCL</code> abbreviation.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>foo: proc options(main);
  dcl i fixed decimal init (0); /* Noncompliant */
end;</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>foo: proc options(main);
  declare i fixed decimal init (0); /* Compliant */
end;</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this use of "DCL"</p>
</div>
</div>
</div>
</div>