<div class="paragraph">
<p>Postfix increment and decrement typically involves making a copy of the object being incremented or decremented, whereas its prefix form does not. Therefore the prefix form is usually the more efficient form, and should be preferred.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue if a postfix increment or decrement operator is used, but its return value is not read.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>void myFunc(int lim)
{
  int i;
  for (i = 0; i &lt; lim; i++)
  {
    // do something
  }
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>void myFunc(int lim)
{
  int i;
  for (i = 0; i &lt; lim; ++i)
  {
    // do something
  }
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use prefix (increment|decrement)</p>
</div>
</div>
</div>
</div>