<div class="paragraph">
<p>Having all branches in a <code>CASE</code> or <code>IF</code> chain with the same implementation is an error. Either a copy-paste error was made and something different should be executed, or there shouldn&#8217;t be a <code>CASE</code>/<code>IF</code> chain at all.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>IF a &gt;0.
    doSomething.
ELSE IF b&gt; 0.
    doSomething.
ELSE.
    doSomething.
ENDIF.

CASE i.
  WHEN 1 OR 3.
    doSomething.
  WHEN 2.
    doSomething.
  WHEN OTHERS.
        doSomething.
ENDCASE.</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_exceptions">Exceptions</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Note that this rule does not apply to <code>IF</code> chains without final <code>ELSE</code>-s, or to <code>CASE</code>-s without <code>WHEN OTHERS</code> clauses.</p>
</div>
<div class="listingblock">
<div class="content">
<pre>IF a &gt;0.   "no issue, this could have been done on purpose to make the code more readable
    doSomething.
ELSEIF b&gt; 0.
    doSomething.
ENDIF.</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this conditional structure or edit its code blocks so that they&#8217;re not all the same.</p>
</div>
<div class="paragraph">
<p>This conditional operation returns the same value whether the condition is "true" or "false".</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p><code>if</code> or <code>switch</code>.</p>
</li>
<li>
<p>For a ternary, condition and <code>?</code></p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_deprecates_s2758">deprecates: S2758</h3>

</div>
</div>
</div>