<div class="paragraph">
<p>Having all branches in a <code>CASE</code> or <code>IF</code>/<code>ELSIF</code> chain with the same implementation is an error. Either a copy-paste error was made and something different should be executed, or there shouldn&#8217;t be a <code>CASE</code>/<code>IF</code>/<code>ELSIF</code> chain at all.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>IF param = 1 THEN
  result := 'A';
ELSIF param = 2 THEN
  result := 'A';
ELSE
  result := 'A';
END IF;

result := CASE param
   WHEN 1 THEN 'A'
   WHEN 2 THEN 'A'
   ELSE 'A'
END;</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>IF param = 1 THEN
  result := 'A';
ELSIF param = 2 THEN
  result := 'B';
ELSE
  result := 'C';
END IF;

result := CASE param
   WHEN 1 THEN 'A'
   WHEN 2 THEN 'B'
   ELSE 'C'
END;</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_exceptions">Exceptions</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule does not apply to <code>IF</code>/<code>CASE</code> chains without <code>ELSE</code> clauses.</p>
</div>
<div class="listingblock">
<div class="content">
<pre>IF param = 1 THEN   -- no issue, this could have been done on purpose to make the code more readable
  result := 'A';
ELSIF param = 2 THEN
  result := 'A';
END IF;</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this conditional structure or edit its code blocks so that they&#8217;re not all the same.</p>
</div>
<div class="paragraph">
<p>This conditional operation returns the same value whether the condition is "true" or "false".</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p><code>if</code> or <code>switch</code>.</p>
</li>
<li>
<p>For a ternary, condition and <code>?</code></p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_deprecates_s2758">deprecates: S2758</h3>

</div>
</div>
</div>