<div class="paragraph">
<p>Having the same implementation in every branch of an <code>if</code> chain is an error. Either a copy-paste error was made and something different should be executed, or there shouldn&#8217;t be an <code>if</code> chain at all.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>if b == 0:  # Noncompliant
    doOneMoreThing()
elif b == 1:
    doOneMoreThing()
else:
    doOneMoreThing()

b = 4 if a &gt; 12 else 4  # Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_exceptions">Exceptions</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This rule does not apply to <code>if</code> chains without <code>else</code>-s.</p>
</div>
<div class="listingblock">
<div class="content">
<pre>if b == 0:  # no issue, this could have been done on purpose to make the code more readable
    doOneMoreThing()
elif b == 1:
    doOneMoreThing()</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this conditional structure or edit its code blocks so that they&#8217;re not all the same.</p>
</div>
<div class="paragraph">
<p>This conditional operation returns the same value whether the condition is "true" or "false".</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>primary: the top "if" keyword</p>
</div>
<div class="paragraph">
<p>secondary: each identical body of the "if-elif-else"</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_deprecates_s2758">deprecates: S2758</h3>

</div>
</div>
</div>