<div class="paragraph">
<p>The first argument to <code>super</code> must be the name of the class making the call. If it&#8217;s not, the result will be a runtime error.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class Person(object):
  #...

class PoliceOfficer(Person):
  def __init__(self, name):
    super().__init__(name)  // Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class Person(object):
  #...

class PoliceOfficer(Person):
  def __init__(self, name):
    super(PoliceOfficer).__init__(name)</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>"X" should be the first argument to this method.</p>
</div>
</div>
</div>
</div>