<div class="paragraph">
<p>Making a public constant just <code>const</code> as opposed to <code>static const</code> leads to duplicating its value for every instance of the class, uselessly increasing the amount of memory required to execute the application.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public class Myclass
{
  public const THRESHOLD:int = 3;
}</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>public class Myclass
{
  public static const THRESHOLD:int = 3;
}</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Make this [final|const] field "static" too.</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s1444">is related to: S1444</h3>

</div>
<div class="sect2">
<h3 id="_on_31_jul_2013_130602_freddy_mallet_wrote">on 31 Jul 2013, 13:06:02 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARJAVA-259</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_2_aug_2013_070311_dinesh_bolkensteyn_wrote">on 2 Aug 2013, 07:03:11 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>What do we use to decide whether or not a field should be static?</p>
</div>
<div class="paragraph">
<p>Do we only verify that an integer literal is assigned to the field?</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_2_aug_2013_074152_dinesh_bolkensteyn_wrote">on 2 Aug 2013, 07:41:52 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>All but method calls in the initializer</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_2_aug_2013_095758_dinesh_bolkensteyn_wrote">on 2 Aug 2013, 09:57:58 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>Should not apply to interface fields which are implicitely already static and final.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_22_nov_2013_093827_freddy_mallet_wrote">on 22 Nov 2013, 09:38:27 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by http://jira.codehaus.org/browse/SONARPLUGINS-3274 for Flex</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_16_jun_2015_183717_ann_campbell_wrote">on 16 Jun 2015, 18:37:17 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Code Pro: Non-blank Final Instance Field</p>
</div>
</div>
</div>
</div>