<div class="paragraph">
<p>Exception chaining enables users to see if an exception was triggered by another exception (see <a href="https://www.python.org/dev/peps/pep-3134/">PEP-3134</a>). Exceptions are chained using either of the following syntax:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>raise NewException() from chained_exception</code></p>
</li>
<li>
<p><code>new_exception.__cause__ = chained_exception</code></p>
</li>
</ul>
</div>
<div class="paragraph">
<p>It is also possible to erase a chaining by setting <code>new_exception.__cause__ = None</code> or using <code>except ... from None</code> (see <a href="https://www.python.org/dev/peps/pep-0409/">PEP-409</a>).</p>
</div>
<div class="paragraph">
<p>Chaining will fail and raise a <code>TypeError</code> if something else than <code>None</code> or a valid exception, i.e. an instance of <code>BaseException</code> or of a subclass, is provided.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>class A:
    pass

try:
    raise ValueError("orig")
except ValueError as e:
    new_exc = TypeError("new")
    new_exc.__cause__ = A()  # Noncompliant
    raise new_exc

try:
    raise ValueError("orig")
except ValueError as e:
    raise TypeError("new") from "test"  # Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>try:
    raise ValueError("orig")
except ValueError as e:
    new_exc = TypeError("new")
    new_exc.__cause__ = None  # Ok
    raise new_exc

try:
    raise ValueError("orig")
except ValueError as e:
    new_exc = TypeError("new")
    new_exc.__cause__ = e  # Ok
    raise new_exc

try:
    raise ValueError("orig")
except ValueError as e:
    raise TypeError("new") from None  # Ok

try:
    raise ValueError("orig")
except ValueError as e:
    raise TypeError("new") from e  # Ok</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>PEP 3134 – <a href="https://www.python.org/dev/peps/pep-3134/">Exception Chaining and Embedded Tracebacks</a></p>
</li>
<li>
<p>PEP 409 – <a href="https://www.python.org/dev/peps/pep-0409/">Suppressing exception context</a></p>
</li>
<li>
<p>PEP 352 - <a href="https://www.python.org/dev/peps/pep-0352/#exception-hierarchy-changes">Required Superclass for Exceptions</a></p>
</li>
<li>
<p>Python documentation - <a href="https://docs.python.org/3/library/exceptions.html">Built-in Exceptions</a></p>
</li>
</ul>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace this expression of type X with an exception or None</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>In a "raise X from Y" statement:</p>
<div class="ulist">
<ul>
<li>
<p>highlight Y</p>
</li>
</ul>
</div>
</li>
<li>
<p>In an "myexception.<em>cause</em> = Y" statement:</p>
<div class="ulist">
<ul>
<li>
<p>highlight Y</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
</div>
</div>
</div>