<div class="paragraph">
<p>While a C-style <code>enum</code> will work in Objective-C, the newer <code>NS_ENUM</code> and <code>NS_OPTIONS</code> macros offer a simple way to define enumerations and options, specifying the type and size for each, and improving code completion in Xcode.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>typedef enum Letter {  // Noncompliant
  A,
  B,
  C
} Letter;</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>typedef NS_ENUM (NSInteger, Letter) {
  A,
  B,
  C
} Letter;</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use "NS_ENUM" or "NS_OPTIONS" instead</p>
</div>
</div>
</div>
</div>