<div class="paragraph">
<p>C&#43;&#43;11 improves the specification of the parser so that at the end of a template, multiple <code>&gt;</code>s will be interpreted as closing the template argument list rather than as the right shift operator or stream extraction operator. It is no longer required to place a space between the two <code>&gt;</code>s.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when such useless spaces exist.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>vector&lt;pair&lt;int,int&gt; &gt; s; // Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>vector&lt;pair&lt;int,int&gt;&gt; s;</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove the space between the two right angle brackets.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>space(s)</p>
</div>
<hr/>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_2_mar_2016_173443_ann_campbell_wrote">on 2 Mar 2016, 17:34:43 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Double-check my changes please [~alban.auzeill]</p>
</div>
</div>
</div>
</div>