<div class="paragraph">
<p>Shared coding conventions allow teams to collaborate effectively. This rule checks that all C&#43;&#43; concept names match a provided regular expression.</p>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="paragraph">
<p>With default provided regular expression <code>^[A-Z][a-zA-Z0-9]*$</code>:</p>
</div>
<div class="listingblock">
<div class="content">
<pre>template &lt;class T&gt;
concept integral = std::is_integral_v&lt;T&gt;; // Noncompliant</pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre>template &lt;class T&gt;
concept Integral = std::is_integral_v&lt;T&gt;; // Compliant</pre>
</div>
</div>
<hr/>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="ulist">
<ul>
<li>
<p>key = format</p>
</li>
<li>
<p>description = Regular expression used to check the class names against.</p>
</li>
<li>
<p>default = ^[A-Z][a-zA-Z0-9]*$</p>
</li>
<li>
<p>type = STRING</p>
</li>
</ul>
</div>
</div>
</div>
</div>